{
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "# **SaprotHub: Making Protein Modeling Accessible to All Biologists**\n",
        "\n",
        "<a href=\"https://www.biorxiv.org/content/10.1101/2024.05.24.595648v3\"><img src=\"https://img.shields.io/badge/Paper-bioRxiv-green\" style=\"max-width: 100%;\"></a>\n",
        "<a href=\"https://huggingface.co/SaProtHub\"><img src=\"https://img.shields.io/badge/%F0%9F%A4%97%20Hugging%20Face-red?label=SaprotHub\" style=\"max-width: 100%;\"></a>\n",
        "<a href=\"https://github.com/westlake-repl/SaprotHub\"><img src=\"https://img.shields.io/badge/Github-black?logo=github\" style=\"max-width: 100%;\"></a>\n",
        "<a href=\"https://theopmc.github.io/\"><img src=\"https://img.shields.io/badge/Website-OPMC-yellow\" style=\"max-width: 100%;\"></a>\n",
        "<a href=\"https://cbirt.net/no-coding-required-saprothub-brings-protein-modeling-to-every-biologist/\" alt=\"blog\"><img src=\"https://img.shields.io/badge/Blog-Medium-purple\" /></a>\n",
        "<a href=\"https://x.com/sokrypton/status/1795525127653986415\"><img src=\"https://img.shields.io/badge/Twitter-blue?logo=twitter\" style=\"max-width: 100%;\"></a>\n",
        "\n",
        "\n",
        "This is **ColabSaprot**, the Colab version of [SaProt](https://github.com/westlake-repl/SaProt), a pre-trained protein language model designed for various downstream protein tasks.\n",
        "\n",
        "**ColabSaprot** is a platform where **Protein Language Models(PLMs)** are more accessible and user-friendly for biologists, enabling effortless model training and sharing within the scientific community.\n",
        "\n",
        "We've established the **SaprotHub**([website](https://huggingface.co/SaProtHub), [paper](https://www.biorxiv.org/content/10.1101/2024.05.24.595648v3)) for storing and sharing models and datasets, where you can explore extensive collections for specific protein prediction tasks.\n",
        "\n",
        "We hope ColabSaprot and SaprotHub can contribute to advancing biological research, fostering collaboration, and accelerating discoveries in the field. You can access [our paper](https://www.biorxiv.org/content/10.1101/2024.05.24.595648v3) for further details.\n",
        "\n",
        "Check these videos ([training](https://www.youtube.com/watch?v=r42z1hvYKfw), [predicting](https://www.youtube.com/watch?v=N5VMBwM_ukQ)) to see how to use  ColabSaprot.\n",
        "\n",
        "Joining [**OPMC**](https://theopmc.github.io/) as an author of SaprotHub.\n",
        "\n",
        "ColabSaprot supports hundreds of [protein prediction tasks](https://github.com/westlake-repl/SaProtHub/blob/main/task_list.md).\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "\n"
      ],
      "metadata": {
        "id": "i7atSR7woQ-b"
      },
      "id": "i7atSR7woQ-b"
    },
    {
      "cell_type": "code",
      "source": [
        "#@title **First, click the run button to install SaProt**\n",
        "\n",
        "#@markdown (Please waiting for 2-8 minutes to install...)\n",
        "################################################################################\n",
        "########################### install saprot #####################################\n",
        "################################################################################\n",
        "%load_ext autoreload\n",
        "%autoreload 2\n",
        "\n",
        "import os\n",
        "# Check whether the server is local or from google cloud\n",
        "root_dir = os.getcwd()\n",
        "\n",
        "from google.colab import output\n",
        "# output.enable_custom_widget_manager()\n",
        "\n",
        "try:\n",
        "  import sys\n",
        "  sys.path.append(f\"{root_dir}/SaprotHub\")\n",
        "  import saprot\n",
        "  print(\"SaProt is installed successfully!\")\n",
        "  os.system(f\"chmod +x {root_dir}/SaprotHub/bin/*\")\n",
        "\n",
        "except ImportError:\n",
        "  print(\"Installing SaProt...\")\n",
        "  os.system(f\"rm -rf {root_dir}/SaprotHub\")\n",
        "  # !rm -rf /content/SaprotHub/\n",
        "\n",
        "  !git clone https://github.com/westlake-repl/SaprotHub.git\n",
        "\n",
        "  # !pip install /content/SaprotHub/saprot-0.4.7-py3-none-any.whl\n",
        "  os.system(f\"pip install -r {root_dir}/SaprotHub/requirements.txt\")\n",
        "  # !pip install -r /content/SaprotHub/requirements.txt\n",
        "\n",
        "  os.system(f\"pip install {root_dir}/SaprotHub\")\n",
        "\n",
        "\n",
        "  os.system(f\"mkdir -p {root_dir}/SaprotHub/LMDB\")\n",
        "  os.system(f\"mkdir -p {root_dir}/SaprotHub/bin\")\n",
        "  os.system(f\"mkdir -p {root_dir}/SaprotHub/output\")\n",
        "  os.system(f\"mkdir -p {root_dir}/SaprotHub/datasets\")\n",
        "  os.system(f\"mkdir -p {root_dir}/SaprotHub/adapters/classification/Local\")\n",
        "  os.system(f\"mkdir -p {root_dir}/SaprotHub/adapters/regression/Local\")\n",
        "  os.system(f\"mkdir -p {root_dir}/SaprotHub/adapters/token_classification/Local\")\n",
        "  os.system(f\"mkdir -p {root_dir}/SaprotHub/adapters/pair_classification/Local\")\n",
        "  os.system(f\"mkdir -p {root_dir}/SaprotHub/adapters/pair_regression/Local\")\n",
        "  os.system(f\"mkdir -p {root_dir}/SaprotHub/structures\")\n",
        "  # !mkdir -p /content/SaprotHub/LMDB\n",
        "  # !mkdir -p /content/SaprotHub/bin\n",
        "  # !mkdir -p /content/SaprotHub/output\n",
        "  # !mkdir -p /content/SaprotHub/datasets\n",
        "  # !mkdir -p /content/SaprotHub/adapters/classification/Local\n",
        "  # !mkdir -p /content/SaprotHub/adapters/regression/Local\n",
        "  # !mkdir -p /content/SaprotHub/adapters/token_classification/Local\n",
        "  # !mkdir -p /content/SaprotHub/adapters/pair_classification/Local\n",
        "  # !mkdir -p /content/SaprotHub/adapters/pair_regression/Local\n",
        "  # !mkdir -p /content/SaprotHub/structures\n",
        "\n",
        "  # !pip install gdown==v4.6.3 --force-reinstall --quiet\n",
        "  # os.system(\n",
        "  #   f\"wget 'https://drive.usercontent.google.com/download?id=1B_9t3n_nlj8Y3Kpc_mMjtMdY0OPYa7Re&export=download&authuser=0' -O {root_dir}/SaprotHub/bin/foldseek\"\n",
        "  # )\n",
        "\n",
        "  os.system(f\"chmod +x {root_dir}/SaprotHub/bin/*\")\n",
        "  # !chmod +x /content/SaprotHub/bin/foldseek\n",
        "  import sys\n",
        "  sys.path.append(f\"{root_dir}/SaprotHub\")\n",
        "\n",
        "  # !mv /content/SaprotHub/ColabSaprotSetup/foldseek /content/SaprotHub/bin/\n",
        "\n",
        "################################################################################\n",
        "################################################################################\n",
        "################################## global ######################################\n",
        "################################################################################\n",
        "################################################################################\n",
        "\n",
        "import ipywidgets\n",
        "import pandas as pd\n",
        "import torch\n",
        "import numpy as np\n",
        "import lmdb\n",
        "import base64\n",
        "import copy\n",
        "import os\n",
        "import json\n",
        "import zipfile\n",
        "import yaml\n",
        "import argparse\n",
        "import pprint\n",
        "import subprocess\n",
        "import py3Dmol\n",
        "import matplotlib.pyplot as plt\n",
        "import shutil\n",
        "import torch.nn.functional as F\n",
        "import warnings\n",
        "warnings.filterwarnings(\"ignore\", category=FutureWarning)\n",
        "\n",
        "from loguru import logger\n",
        "from easydict import EasyDict\n",
        "from colorama import init, Fore, Back, Style\n",
        "from IPython.display import clear_output\n",
        "from huggingface_hub import snapshot_download\n",
        "from ipywidgets import HTML\n",
        "from IPython.display import display\n",
        "from google.colab import widgets\n",
        "from google.colab import files\n",
        "from pathlib import Path\n",
        "from tqdm import tqdm\n",
        "from datetime import datetime\n",
        "from transformers import AutoTokenizer, EsmForProteinFolding, EsmTokenizer\n",
        "from transformers.models.esm.openfold_utils.protein import to_pdb, Protein as OFProtein\n",
        "from transformers.models.esm.openfold_utils.feats import atom14_to_atom37\n",
        "from string import ascii_uppercase,ascii_lowercase\n",
        "from saprot.utils.mpr import MultipleProcessRunnerSimplifier\n",
        "from saprot.data.parse import get_chain_ids\n",
        "from saprot.scripts.training import my_load_model\n",
        "from safetensors import safe_open\n",
        "\n",
        "DATASET_HOME = Path(f'{root_dir}/SaprotHub/datasets')\n",
        "ADAPTER_HOME = Path(f'{root_dir}/SaprotHub/adapters')\n",
        "STRUCTURE_HOME = Path(f\"{root_dir}/SaprotHub/structures\")\n",
        "LMDB_HOME = Path(f'{root_dir}/SaprotHub/LMDB')\n",
        "OUTPUT_HOME = Path(f'{root_dir}/SaprotHub/output')\n",
        "UPLOAD_FILE_HOME = Path(f'{root_dir}/SaprotHub/upload_files')\n",
        "FOLDSEEK_PATH = Path(f\"{root_dir}/SaprotHub/bin/foldseek\")\n",
        "aa_set = {\"A\", \"C\", \"D\", \"E\", \"F\", \"G\", \"H\", \"I\", \"K\", \"L\", \"M\", \"N\", \"P\", \"Q\", \"R\", \"S\", \"T\", \"V\", \"W\", \"Y\"}\n",
        "foldseek_struc_vocab = \"pynwrqhgdlvtmfsaeikc#\"\n",
        "\n",
        "data_type_list = [\"Single AA Sequence\",\n",
        "                  \"Single SA Sequence\",\n",
        "                  \"Single UniProt ID\",\n",
        "                  \"Single PDB/CIF Structure\",\n",
        "                  \"Multiple AA Sequences\",\n",
        "                  \"Multiple SA Sequences\",\n",
        "                  \"Multiple UniProt IDs\",\n",
        "                  \"Multiple PDB/CIF Structures\",\n",
        "                  \"SaprotHub Dataset\",\n",
        "                  \"A pair of AA Sequences\",\n",
        "                  \"A pair of SA Sequences\",\n",
        "                  \"A pair of UniProt IDs\",\n",
        "                  \"A pair of PDB/CIF Structures\",\n",
        "                  \"Multiple pairs of AA Sequences\",\n",
        "                  \"Multiple pairs of SA Sequences\",\n",
        "                  \"Multiple pairs of UniProt IDs\",\n",
        "                  \"Multiple pairs of PDB/CIF Structures\",]\n",
        "\n",
        "data_type_list_single = [\n",
        "    \"Single AA Sequence\",\n",
        "    \"Single SA Sequence\",\n",
        "    \"Single UniProt ID\",\n",
        "    \"Single PDB/CIF Structure\",\n",
        "    \"A pair of AA Sequences\",\n",
        "    \"A pair of SA Sequences\",\n",
        "    \"A pair of UniProt IDs\",\n",
        "    \"A pair of PDB/CIF Structures\",]\n",
        "\n",
        "data_type_list_multiple = [\n",
        "    \"Multiple AA Sequences\",\n",
        "    \"Multiple SA Sequences\",\n",
        "    \"Multiple UniProt IDs\",\n",
        "    \"Multiple PDB/CIF Structures\",\n",
        "    \"Multiple pairs of AA Sequences\",\n",
        "    \"Multiple pairs of SA Sequences\",\n",
        "    \"Multiple pairs of UniProt IDs\",\n",
        "    \"Multiple pairs of PDB/CIF Structures\",]\n",
        "\n",
        "task_type_dict = {\n",
        "  \"Protein-level Classification\": \"classification\",\n",
        "  \"Residue-level Classification\" : \"token_classification\",\n",
        "  \"Protein-level Regression\" : \"regression\",\n",
        "  \"Protein-protein Classification\": \"pair_classification\",\n",
        "  \"Protein-protein Regression\": \"pair_regression\",\n",
        "}\n",
        "model_type_dict = {\n",
        "  \"classification\" : \"saprot/saprot_classification_model\",\n",
        "  \"token_classification\" : \"saprot/saprot_token_classification_model\",\n",
        "  \"regression\" : \"saprot/saprot_regression_model\",\n",
        "  \"pair_classification\" : \"saprot/saprot_pair_classification_model\",\n",
        "  \"pair_regression\" : \"saprot/saprot_pair_regression_model\",\n",
        "}\n",
        "dataset_type_dict = {\n",
        "  \"classification\": \"saprot/saprot_classification_dataset\",\n",
        "  \"token_classification\" : \"saprot/saprot_token_classification_dataset\",\n",
        "  \"regression\": \"saprot/saprot_regression_dataset\",\n",
        "  \"pair_classification\" : \"saprot/saprot_pair_classification_dataset\",\n",
        "  \"pair_regression\" : \"saprot/saprot_pair_regression_dataset\",\n",
        "}\n",
        "training_data_type_dict = {\n",
        "  \"Single AA Sequence\": \"AA\",\n",
        "  \"Single SA Sequence\": \"SA\",\n",
        "  \"Single UniProt ID\": \"SA\",\n",
        "  \"Single PDB/CIF Structure\": \"SA\",\n",
        "  \"Multiple AA Sequences\": \"AA\",\n",
        "  \"Multiple SA Sequences\": \"SA\",\n",
        "  \"Multiple UniProt IDs\": \"SA\",\n",
        "  \"Multiple PDB/CIF Structures\": \"SA\",\n",
        "  \"SaprotHub Dataset\": \"SA\",\n",
        "  \"A pair of AA Sequences\": \"AA\",\n",
        "  \"A pair of SA Sequences\": \"SA\",\n",
        "  \"A pair of UniProt IDs\": \"SA\",\n",
        "  \"A pair of PDB/CIF Structures\": \"SA\",\n",
        "  \"Multiple pairs of AA Sequences\": \"AA\",\n",
        "  \"Multiple pairs of SA Sequences\": \"SA\",\n",
        "  \"Multiple pairs of UniProt IDs\": \"SA\",\n",
        "  \"Multiple pairs of PDB/CIF Structures\": \"SA\",\n",
        "}\n",
        "\n",
        "\n",
        "class font:\n",
        "    RED = '\\033[91m'\n",
        "    GREEN = '\\033[92m'\n",
        "    YELLOW = '\\033[93m'\n",
        "    BLUE = '\\033[94m'\n",
        "\n",
        "    BOLD = '\\033[1m'\n",
        "    UNDERLINE = '\\033[4m'\n",
        "\n",
        "    RESET = '\\033[0m'\n",
        "\n",
        "\n",
        "################################################################################\n",
        "############################### adapters #######################################\n",
        "################################################################################\n",
        "def get_adapters_list(task_type=None):\n",
        "\n",
        "    adapters_list = []\n",
        "\n",
        "    if task_type:\n",
        "      for file_path in (ADAPTER_HOME / task_type).glob('**/adapter_config.json'):\n",
        "        adapters_list.append(file_path.relative_to(ADAPTER_HOME / task_type).parent)\n",
        "    else:\n",
        "      for file_path in ADAPTER_HOME.glob('**/adapter_config.json'):\n",
        "        adapters_list.append(file_path.relative_to(ADAPTER_HOME).parent)\n",
        "\n",
        "    return adapters_list\n",
        "\n",
        "def adapters_text(adapters_list):\n",
        "  input = ipywidgets.Text(\n",
        "    value=None,\n",
        "    placeholder='Enter SaprotHub Model ID',\n",
        "    # description='Selected:',\n",
        "    disabled=False)\n",
        "  input.layout.width = '500px'\n",
        "  display(input)\n",
        "\n",
        "  return input\n",
        "\n",
        "def adapters_dropdown(adapters_list):\n",
        "  dropdown = ipywidgets.Dropdown(\n",
        "    # options=[f\"{adapter_path.parent.stem}/{adapter_path.stem}\" for index, adapter_path in enumerate(adapters_list)],\n",
        "    options=adapters_list,\n",
        "    value=None,\n",
        "    placeholder='Select a Local Model here',\n",
        "    # description='Selected:',\n",
        "    disabled=False)\n",
        "  dropdown.layout.width = '500px'\n",
        "  display(dropdown)\n",
        "\n",
        "  return dropdown\n",
        "\n",
        "def adapters_combobox(adapters_list):\n",
        "  combobox = ipywidgets.Combobox(\n",
        "    options=[f\"{adapter_path.parent.stem}/{adapter_path.stem}\" for index, adapter_path in enumerate(adapters_list)],\n",
        "    value=None,\n",
        "    placeholder='Enter SaprotHub Model repository id or select a Local Model here',\n",
        "    # description='Selected:',\n",
        "    disabled=False)\n",
        "  combobox.layout.width = '500px'\n",
        "  display(combobox)\n",
        "\n",
        "  return combobox\n",
        "\n",
        "def adapters_selectmultiple(adapters_list):\n",
        "  selectmulitiple = ipywidgets.SelectMultiple(\n",
        "  # options=[f\"{adapter_path.parent.stem}/{adapter_path.stem}\" for index, adapter_path in enumerate(adapters_list)],\n",
        "  options=adapters_list,\n",
        "  value=[],\n",
        "  #rows=10,\n",
        "  placeholder='Select multiple models',\n",
        "  # description='Fruits',\n",
        "  disabled=False,\n",
        "  layout={'width': '500px'})\n",
        "  display(selectmulitiple)\n",
        "\n",
        "  return selectmulitiple\n",
        "\n",
        "def adapters_textmultiple(adapters_list):\n",
        "  textmultiple = ipywidgets.Text(\n",
        "  value=None,\n",
        "  placeholder='Enter multiple SaprotHub Model IDs, separated by commas.',\n",
        "  # description='Fruits',\n",
        "  disabled=False,\n",
        "  layout={'width': '500px'})\n",
        "  display(textmultiple)\n",
        "\n",
        "  return textmultiple\n",
        "\n",
        "\n",
        "def select_adapter_from(task_type, use_model_from):\n",
        "  adapters_list = get_adapters_list(task_type)\n",
        "\n",
        "  if use_model_from == 'Trained by yourself on ColabSaprot':\n",
        "    print(Fore.BLUE+f\"Local Model ({task_type}):\"+Style.RESET_ALL)\n",
        "    return adapters_dropdown(adapters_list)\n",
        "\n",
        "  elif use_model_from == 'Shared by peers on SaprotHub':\n",
        "    print(Fore.BLUE+\"SaprotHub Model:\"+Style.RESET_ALL)\n",
        "    return adapters_text(adapters_list)\n",
        "\n",
        "  elif use_model_from == \"Saved in your local computer\":\n",
        "    print(Fore.BLUE+\"Click the button to upload the \\\"Model-<task_name>-<model_size>.zip\\\" file of your Model:\"+Style.RESET_ALL)\n",
        "    # 1. upload model.zip\n",
        "    if task_type:\n",
        "      adapter_upload_path = ADAPTER_HOME / task_type / \"Local\"\n",
        "    else:\n",
        "      adapter_upload_path = ADAPTER_HOME / \"Local\"\n",
        "\n",
        "    adapter_zip_path = upload_file(adapter_upload_path)\n",
        "    adapter_path = adapter_upload_path / adapter_zip_path.stem\n",
        "    # 2. unzip model.zip\n",
        "    with zipfile.ZipFile(adapter_zip_path, 'r') as zip_ref:\n",
        "        zip_ref.extractall(adapter_path)\n",
        "    os.remove(adapter_zip_path)\n",
        "    # 3. check adapter_config.json\n",
        "    adapter_config_path = adapter_path / \"adapter_config.json\"\n",
        "    assert adapter_config_path.exists(), f\"Can't find {adapter_config_path}\"\n",
        "\n",
        "    # # 4. move to correct folder\n",
        "    # num_labels, task_type = get_num_labels_and_task_type_by_adapter(adapter_path)\n",
        "    # shutil.move(adapter_path, ADAPTER_HOME / task_type)\n",
        "\n",
        "    return EasyDict({\"value\":  f\"Local/{adapter_zip_path.stem}\"})\n",
        "\n",
        "  elif use_model_from == \"Multi-models on ColabSaprot\":\n",
        "    # 1. select the list of adapters\n",
        "    print(Fore.BLUE+f\"Local Model ({task_type}):\"+Style.RESET_ALL)\n",
        "    print(Fore.BLUE+f\"Multiple values can be selected with \\\"shift\\\" and/or \\\"ctrl\\\" (or \\\"command\\\") pressed and mouse clicks or arrow keys.\"+Style.RESET_ALL)\n",
        "    return adapters_selectmultiple(adapters_list)\n",
        "\n",
        "  elif use_model_from == \"Multi-models on SaprotHub\":\n",
        "    # 1. enter the list of adapters\n",
        "    print(Fore.BLUE+f\"SaprotHub Model IDs, separated by commas ({task_type}):\"+Style.RESET_ALL)\n",
        "    return adapters_textmultiple(adapters_list)\n",
        "\n",
        "\n",
        "\n",
        "################################################################################\n",
        "########################### download dataset ###################################\n",
        "################################################################################\n",
        "def download_dataset(task_name):\n",
        "  import gdown\n",
        "  import tarfile\n",
        "\n",
        "  filepath = LMDB_HOME / f\"{task_name}.tar.gz\"\n",
        "  download_links = {\n",
        "    \"ClinVar\" : \"https://drive.google.com/uc?id=1Le6-v8ddXa1eLJZFo7HPij7NhaBmNUbo\",\n",
        "    \"DeepLoc_cls2\" : \"https://drive.google.com/uc?id=1dGlojkCt1DwUXWiUk4kXRGRNu5sz2uxf\",\n",
        "    \"DeepLoc_cls10\" : \"https://drive.google.com/uc?id=1dGlojkCt1DwUXWiUk4kXRGRNu5sz2uxf\",\n",
        "    \"EC\" : \"https://drive.google.com/uc?id=1VFLFA-jK1tkTZBVbMw8YSsjZqAqlVQVQ\",\n",
        "    \"GO_BP\" : \"https://drive.google.com/uc?id=1DGiGErWbRnEK8jmE2Jpb996By8KVDBfF\",\n",
        "    \"GO_CC\" : \"https://drive.google.com/uc?id=1DGiGErWbRnEK8jmE2Jpb996By8KVDBfF\",\n",
        "    \"GO_MF\" : \"https://drive.google.com/uc?id=1DGiGErWbRnEK8jmE2Jpb996By8KVDBfF\",\n",
        "    \"HumanPPI\" : \"https://drive.google.com/uc?id=1ahgj-IQTtv3Ib5iaiXO_ASh2hskEsvoX\",\n",
        "    \"MetalIonBinding\" : \"https://drive.google.com/uc?id=1rwknPWIHrXKQoiYvgQy4Jd-efspY16x3\",\n",
        "    \"ProteinGym\" : \"https://drive.google.com/uc?id=1L-ODrhfeSjDom-kQ2JNDa2nDEpS8EGfD\",\n",
        "    \"Thermostability\" : \"https://drive.google.com/uc?id=1I9GR1stFDHc8W3FCsiykyrkNprDyUzSz\",\n",
        "  }\n",
        "\n",
        "  try:\n",
        "    gdown.download(download_links[task_name], str(filepath), quiet=False)\n",
        "    with tarfile.open(filepath, 'r:gz') as tar:\n",
        "      tar.extractall(path=str(LMDB_HOME))\n",
        "      print(f\"Extracted: {filepath}\")\n",
        "  except Exception as e:\n",
        "    raise RuntimeError(\"The dataset has not prepared.\")\n",
        "\n",
        "################################################################################\n",
        "############################# upload file ######################################\n",
        "################################################################################\n",
        "def upload_file(upload_path):\n",
        "  upload_path = Path(upload_path)\n",
        "  upload_path.mkdir(parents=True, exist_ok=True)\n",
        "  basepath = Path().resolve()\n",
        "  try:\n",
        "    uploaded = files.upload()\n",
        "    filenames = []\n",
        "    for filename in uploaded.keys():\n",
        "      filenames.append(filename)\n",
        "      shutil.move(basepath / filename, upload_path / filename)\n",
        "    if len(filenames) == 0:\n",
        "      logger.info(\"The uploading process has been interrupted by the user.\")\n",
        "      raise RuntimeError(\"The uploading process has been interrupted by the user.\")\n",
        "  except Exception as e:\n",
        "    logger.error(\"Upload file fail! Please click the button to run again.\")\n",
        "    raise(e)\n",
        "\n",
        "  return upload_path / filenames[0]\n",
        "\n",
        "################################################################################\n",
        "############################ upload dataset ####################################\n",
        "################################################################################\n",
        "\n",
        "def read_csv_dataset(uploaded_csv_path):\n",
        "  df = pd.read_csv(uploaded_csv_path)\n",
        "  df.columns = df.columns.str.lower()\n",
        "  return df\n",
        "\n",
        "def check_column_label_and_stage(csv_dataset_path):\n",
        "  df = read_csv_dataset(csv_dataset_path)\n",
        "  assert {'label', 'stage'}.issubset(df.columns), f\"Make sure your CSV dataset includes both `label` and `stage` columns!\\nCurrent columns: {df.columns}\"\n",
        "  column_values = set(df['stage'].unique())\n",
        "  assert all(value in column_values for value in ['train', 'valid', 'test']), f\"Ensure your dataset includes samples for all three stages: `train`, `valid` and `test`.\\nCurrent columns: {df.columns}\"\n",
        "\n",
        "def get_data_type(csv_dataset_path):\n",
        "  # AA, SA, Pair AA, Pair SA\n",
        "  df = read_csv_dataset(csv_dataset_path)\n",
        "\n",
        "  # AA, SA\n",
        "  if 'sequence' in df.columns:\n",
        "    second_token = df.loc[0, 'sequence'][1]\n",
        "    if second_token in aa_set:\n",
        "      return \"Multiple AA Sequences\"\n",
        "    elif second_token in foldseek_struc_vocab:\n",
        "      return \"Multiple SA Sequences\"\n",
        "    else:\n",
        "      raise RuntimeError(f\"The sequence in the dataset({csv_dataset_path}) are neither SA Sequences nor AA Sequences. Please check carefully.\")\n",
        "\n",
        "  # Pair AA, Pair SA\n",
        "  elif 'sequence_1' in df.columns and 'sequence_2' in df.columns:\n",
        "    second_token = df.loc[0, 'sequence_1'][1]\n",
        "    if second_token in aa_set:\n",
        "      return \"Multiple pairs of AA Sequences\"\n",
        "    elif second_token in foldseek_struc_vocab:\n",
        "      return \"Multiple pairs of SA Sequences\"\n",
        "    else:\n",
        "      raise RuntimeError(f\"The sequence in the dataset({csv_dataset_path}) are neither SA Sequences nor AA Sequences. Please check carefully.\")\n",
        "\n",
        "  else:\n",
        "      raise RuntimeError(f\"The data type of the dataset({csv_dataset_path}) should be one of the following types: Multiple AA Sequences, Multiple SA Sequences, Multiple pairs of AA Sequences, Multiple pairs of SA Sequences\")\n",
        "\n",
        "def check_task_type_and_data_type(original_task_type, data_type):\n",
        "  if \"Protein-protein\" in original_task_type:\n",
        "    assert data_type == \"SaprotHub Dataset\" or \"pair\" in data_type, f\"The current `data_type`({data_type}) is incompatible with the current `task_type`({original_task_type}). Please use Pair Sequence Datset for {original_task_type} task!\"\n",
        "  else:\n",
        "    assert \"pair\" not in data_type, f\"The current `data_type`({data_type}) is incompatible with the current `task_type`({original_task_type}). Please avoid using the Pair Sequence Dataset({data_type}) for the {original_task_type} task!\"\n",
        "\n",
        "def input_raw_data_by_data_type(data_type):\n",
        "\n",
        "  # 0-2. 0. Single AA Sequence, 1. Single SA Sequence, 2. Single UniProt ID\n",
        "  if data_type in data_type_list[:3]:\n",
        "    input_seq = ipywidgets.Text(\n",
        "      value=None,\n",
        "      placeholder=f'Enter {data_type} here',\n",
        "      disabled=False)\n",
        "    input_seq.layout.width = '500px'\n",
        "    print(Fore.BLUE+f\"{data_type}\"+Style.RESET_ALL)\n",
        "    display(input_seq)\n",
        "    return input_seq\n",
        "\n",
        "  # 3. Single PDB/CIF Structure\n",
        "  elif data_type == 'Single PDB/CIF Structure':\n",
        "    input_chain = ipywidgets.Text(\n",
        "      value=\"A\",\n",
        "      placeholder=f'Enter the name of chain here',\n",
        "      disabled=False)\n",
        "    input_chain.layout.width = '500px'\n",
        "    print(Fore.BLUE+\"Chain (to be extracted from the structure):\"+Style.RESET_ALL)\n",
        "    display(input_chain)\n",
        "\n",
        "    print(Fore.BLUE+\"Click to upload a .pdb/.cif file\"+Style.RESET_ALL)\n",
        "    pdb_file_path = upload_file(STRUCTURE_HOME)\n",
        "    print(input_chain)\n",
        "    return pdb_file_path.stem, input_chain\n",
        "\n",
        "  # 4-7 & 13-16. Multiple Sequences\n",
        "  elif data_type in data_type_list_multiple:\n",
        "    print(Fore.BLUE+f\"Please upload the .csv file which contains {data_type}\"+Style.RESET_ALL)\n",
        "    uploaded_csv_path = upload_file(UPLOAD_FILE_HOME)\n",
        "    print(Fore.BLUE+\"Successfully upload your .csv file!\"+Style.RESET_ALL)\n",
        "    print(\"=\"*100)\n",
        "\n",
        "    if data_type in ['Multiple PDB/CIF Structures', 'Multiple pairs of PDB/CIF Structures']:\n",
        "      # upload and unzip PDB files\n",
        "      print(Fore.BLUE+f\"Please upload your .zip file which contains {data_type} files\"+Style.RESET_ALL)\n",
        "      pdb_zip_path = upload_file(UPLOAD_FILE_HOME)\n",
        "      if pdb_zip_path.suffix != \".zip\":\n",
        "        logger.error(\"The data type does not match. Please click the run button again to upload a .zip file!\")\n",
        "        raise RuntimeError(\"The data type does not match.\")\n",
        "      print(Fore.BLUE+\"Successfully upload your .zip file!\"+Style.RESET_ALL)\n",
        "      print(\"=\"*100)\n",
        "\n",
        "      import zipfile\n",
        "      with zipfile.ZipFile(pdb_zip_path, 'r') as zip_ref:\n",
        "        zip_ref.extractall(STRUCTURE_HOME)\n",
        "\n",
        "    return uploaded_csv_path\n",
        "\n",
        "  # 8. SaprotHub Dataset\n",
        "  elif data_type == \"SaprotHub Dataset\":\n",
        "    input_repo_id = ipywidgets.Text(\n",
        "      value=None,\n",
        "      placeholder=f'Copy and paste the SaprotHub Dataset ID here',\n",
        "      disabled=False)\n",
        "    input_repo_id.layout.width = '500px'\n",
        "    print(Fore.BLUE+f\"{data_type}\"+Style.RESET_ALL)\n",
        "    display(input_repo_id)\n",
        "    return input_repo_id\n",
        "\n",
        "  # 9-11. A pair of seq\n",
        "  elif data_type in [\"A pair of AA Sequences\", \"A pair of SA Sequences\", \"A pair of UniProt IDs\"]:\n",
        "    print()\n",
        "\n",
        "    seq_type = data_type[len(\"A pair of \"):-1]\n",
        "\n",
        "    input_seq1 = ipywidgets.Text(\n",
        "      value=None,\n",
        "      placeholder=f'Enter the {seq_type} of Sequence 1 here',\n",
        "      disabled=False)\n",
        "    input_seq1.layout.width = '500px'\n",
        "    print(Fore.BLUE+f\"Sequence 1:\"+Style.RESET_ALL)\n",
        "    display(input_seq1)\n",
        "\n",
        "    input_seq2 = ipywidgets.Text(\n",
        "      value=None,\n",
        "      placeholder=f'Enter the {seq_type} of Sequence 2 here',\n",
        "      disabled=False)\n",
        "    input_seq2.layout.width = '500px'\n",
        "    print(Fore.BLUE+f\"Sequence 2:\"+Style.RESET_ALL)\n",
        "    display(input_seq2)\n",
        "\n",
        "    return (input_seq1, input_seq2)\n",
        "\n",
        "  # 12. Pair Single PDB/CIF Structure\n",
        "  elif data_type == 'A pair of PDB/CIF Structures':\n",
        "    print(\"Please provide the structure type, chain and your structure file.\")\n",
        "\n",
        "    dropdown_type1 = ipywidgets.Dropdown(\n",
        "      value=\"PDB\",\n",
        "      options=[\"PDB\", \"AF2\"],\n",
        "      disabled=False)\n",
        "    dropdown_type1.layout.width = '500px'\n",
        "    print(Fore.BLUE+\"The first structure type:\"+Style.RESET_ALL)\n",
        "    display(dropdown_type1)\n",
        "\n",
        "    input_chain1 = ipywidgets.Text(\n",
        "      value=\"A\",\n",
        "      placeholder=f'Enter the name of chain of the first structure here',\n",
        "      disabled=False)\n",
        "    input_chain1.layout.width = '500px'\n",
        "    print(Fore.BLUE+\"Chain of the first structure:\"+Style.RESET_ALL)\n",
        "    display(input_chain1)\n",
        "\n",
        "    print(Fore.BLUE+\"Please upload a .pdb/.cif file\"+Style.RESET_ALL)\n",
        "    pdb_file_path1 = upload_file(STRUCTURE_HOME)\n",
        "\n",
        "\n",
        "    dropdown_type2 = ipywidgets.Dropdown(\n",
        "      value=\"PDB\",\n",
        "      options=[\"PDB\", \"AF2\"],\n",
        "      disabled=False)\n",
        "    dropdown_type2.layout.width = '500px'\n",
        "    print(Fore.BLUE+\"The second structure type:\"+Style.RESET_ALL)\n",
        "    display(dropdown_type2)\n",
        "\n",
        "    input_chain2 = ipywidgets.Text(\n",
        "      value=\"A\",\n",
        "      placeholder=f'Enter the name of chain of the second structure here',\n",
        "      disabled=False)\n",
        "    input_chain2.layout.width = '500px'\n",
        "    print(Fore.BLUE+\"Chain of the second structure:\"+Style.RESET_ALL)\n",
        "    display(input_chain2)\n",
        "\n",
        "    print(Fore.BLUE+\"Please upload a .pdb/.cif file\"+Style.RESET_ALL)\n",
        "    pdb_file_path2 = upload_file(STRUCTURE_HOME)\n",
        "    return (pdb_file_path1.stem, dropdown_type1, input_chain1, pdb_file_path2.stem, dropdown_type2, input_chain2)\n",
        "\n",
        "def get_SA_sequence_by_data_type(data_type, raw_data):\n",
        "\n",
        "  # Multiple sequences\n",
        "  # raw_data = upload_files/xxx.csv\n",
        "\n",
        "  # 8. SaprotHub Dataset\n",
        "  if data_type == \"SaprotHub Dataset\":\n",
        "    input_repo_id = raw_data\n",
        "    REPO_ID = input_repo_id.value\n",
        "\n",
        "    if REPO_ID.startswith('/'):\n",
        "      return Path(REPO_ID)\n",
        "\n",
        "    snapshot_download(repo_id=REPO_ID, repo_type=\"dataset\", local_dir=DATASET_HOME / REPO_ID)\n",
        "    csv_dataset_path = DATASET_HOME / REPO_ID / 'dataset.csv'\n",
        "    assert csv_dataset_path.exists(), f\"Can't find {csv_dataset_path}\"\n",
        "    protein_df = read_csv_dataset(csv_dataset_path)\n",
        "\n",
        "    data_type = get_data_type(csv_dataset_path)\n",
        "\n",
        "    return get_SA_sequence_by_data_type(data_type, csv_dataset_path)\n",
        "\n",
        "    # # AA, SA\n",
        "    # if data_type == \"Multiple AA Sequences\":\n",
        "    #   for index, value in protein_df['sequence'].items():\n",
        "    #     sa_seq = ''\n",
        "    #     for aa in value:\n",
        "    #       sa_seq += aa + '#'\n",
        "    #     protein_df.at[index, 'sequence'] = sa_seq\n",
        "\n",
        "    # # Pair AA, Pair SA\n",
        "    # elif data_type in [\"Multiple pairs of AA Sequences\", \"Multiple pairs of SA Sequences\"]:\n",
        "    #   for i in ['1', '2']:\n",
        "    #     if data_type == \"Multiple pairs of AA Sequences\":\n",
        "    #       for index, value in protein_df[f'sequence_{i}'].items():\n",
        "    #         sa_seq = ''\n",
        "    #         for aa in value:\n",
        "    #           sa_seq += aa + '#'\n",
        "    #         protein_df.at[index, f'sequence_{i}'] = sa_seq\n",
        "\n",
        "    #     protein_df[f'name_{i}'] = f'name_{i}'\n",
        "    #     protein_df[f'chain_{i}'] = 'A'\n",
        "\n",
        "    # protein_df.to_csv(csv_dataset_path, index=None)\n",
        "\n",
        "    # return csv_dataset_path\n",
        "\n",
        "  elif data_type in data_type_list_multiple:\n",
        "    uploaded_csv_path = raw_data\n",
        "    csv_dataset_path = DATASET_HOME / uploaded_csv_path.name\n",
        "    protein_df = read_csv_dataset(uploaded_csv_path)\n",
        "\n",
        "    if 'pair' in data_type:\n",
        "      assert {'sequence_1', 'sequence_2'}.issubset(protein_df.columns), f\"The CSV dataset ({uploaded_csv_path}) must contain `sequence_1` and `sequence_2` columns. \\n Current columns:{protein_df.columns}\"\n",
        "    else:\n",
        "      assert 'sequence' in protein_df.columns, f\"The CSV Dataset({uploaded_csv_path}) must contain a `sequence` column. \\n Current columns:{protein_df.columns}\"\n",
        "\n",
        "    # 4. Multiple AA Sequences\n",
        "    if data_type == 'Multiple AA Sequences':\n",
        "      for index, value in protein_df['sequence'].items():\n",
        "        sa_seq = ''\n",
        "        for aa in value:\n",
        "          sa_seq += aa + '#'\n",
        "        protein_df.at[index, 'sequence'] = sa_seq\n",
        "\n",
        "      protein_df.to_csv(csv_dataset_path, index=None)\n",
        "      return csv_dataset_path\n",
        "\n",
        "    # 5. Multiple SA Sequences\n",
        "    elif data_type == 'Multiple SA Sequences':\n",
        "      protein_df.to_csv(csv_dataset_path, index=None)\n",
        "      return csv_dataset_path\n",
        "\n",
        "    # 6. Multiple UniProt IDs\n",
        "    elif data_type == 'Multiple UniProt IDs':\n",
        "      protein_list = protein_df.loc[:, 'sequence'].tolist()\n",
        "      uniprot2pdb(protein_list)\n",
        "      protein_list = [(uniprot_id, \"AF2\", \"A\") for uniprot_id in protein_list]\n",
        "      mprs = MultipleProcessRunnerSimplifier(protein_list, pdb2sequence, n_process=2, return_results=True)\n",
        "      outputs = mprs.run()\n",
        "\n",
        "      protein_df['sequence'] = [output.split(\"\\t\")[1] for output in outputs]\n",
        "      protein_df.to_csv(csv_dataset_path, index=None)\n",
        "      return csv_dataset_path\n",
        "\n",
        "    # 7. Multiple PDB/CIF Structures\n",
        "    elif data_type == 'Multiple PDB/CIF Structures':\n",
        "      # protein_list = [(uniprot_id, type, chain), ...]\n",
        "      # protein_list = [item.split('.')[0] for item in protein_df.iloc[:, 0].tolist()]\n",
        "      # uniprot2pdb(protein_list)\n",
        "      protein_list = []\n",
        "      for row_tuple in protein_df.itertuples(index=False):\n",
        "        assert row_tuple.type in ['PDB', 'AF2'],  \"The type of structure must be either \\\"PDB\\\" or \\\"AF2\\\"!\"\n",
        "        protein_list.append(row_tuple)\n",
        "      mprs = MultipleProcessRunnerSimplifier(protein_list, pdb2sequence, n_process=2, return_results=True)\n",
        "      outputs = mprs.run()\n",
        "\n",
        "      protein_df['sequence'] = [output.split(\"\\t\")[1] for output in outputs]\n",
        "      protein_df.to_csv(csv_dataset_path, index=None)\n",
        "      return csv_dataset_path\n",
        "\n",
        "    # 13. Pair Multiple AA Sequences\n",
        "    elif data_type == \"Multiple pairs of AA Sequences\":\n",
        "      for i in ['1', '2']:\n",
        "        for index, value in protein_df[f'sequence_{i}'].items():\n",
        "          sa_seq = ''\n",
        "          for aa in value:\n",
        "            sa_seq += aa + '#'\n",
        "          protein_df.at[index, f'sequence_{i}'] = sa_seq\n",
        "\n",
        "        protein_df[f'name_{i}'] = f'name_{i}'\n",
        "        protein_df[f'chain_{i}'] = 'A'\n",
        "\n",
        "      protein_df.to_csv(csv_dataset_path, index=None)\n",
        "      return csv_dataset_path\n",
        "\n",
        "    # 14. Pair Multiple SA Sequences\n",
        "    elif data_type == \"Multiple pairs of SA Sequences\":\n",
        "      for i in ['1', '2']:\n",
        "        protein_df[f'name_{i}'] = f'name_{i}'\n",
        "        protein_df[f'chain_{i}'] = 'A'\n",
        "\n",
        "      protein_df.to_csv(csv_dataset_path, index=None)\n",
        "      return csv_dataset_path\n",
        "\n",
        "    # 15. Pair Multiple UniProt IDs\n",
        "    elif data_type == \"Multiple pairs of UniProt IDs\":\n",
        "      for i in ['1', '2']:\n",
        "        protein_list = protein_df.loc[:, f'sequence_{i}'].tolist()\n",
        "        uniprot2pdb(protein_list)\n",
        "        protein_df[f'name_{i}'] = protein_list\n",
        "        protein_list = [(uniprot_id, \"AF2\", \"A\") for uniprot_id in protein_list]\n",
        "        mprs = MultipleProcessRunnerSimplifier(protein_list, pdb2sequence, n_process=2, return_results=True)\n",
        "        outputs = mprs.run()\n",
        "\n",
        "        protein_df[f'sequence_{i}'] = [output.split(\"\\t\")[1] for output in outputs]\n",
        "        protein_df[f'chain_{i}'] = 'A'\n",
        "\n",
        "      protein_df.to_csv(csv_dataset_path, index=None)\n",
        "      return csv_dataset_path\n",
        "\n",
        "    elif data_type ==  \"Multiple pairs of PDB/CIF Structures\":\n",
        "      # columns: sequence_1, sequence_2, type_1, type_2, chain_1, chain_2, label, stage\n",
        "\n",
        "      # protein_list = [(uniprot_id, type, chain), ...]\n",
        "      # protein_list = [item.split('.')[0] for item in protein_df.iloc[:, 0].tolist()]\n",
        "      # uniprot2pdb(protein_list)\n",
        "\n",
        "      for i in ['1', '2']:\n",
        "        protein_list = []\n",
        "        for index, row in protein_df.iterrows():\n",
        "          assert row[f\"type_{i}\"] in ['PDB', 'AF2'],  \"The type of structure must be either \\\"PDB\\\" or \\\"AF2\\\"!\"\n",
        "          row_tuple = (row[f\"sequence_{i}\"], row[f\"type_{i}\"], row[f\"chain_{i}\"])\n",
        "          protein_list.append(row_tuple)\n",
        "        mprs = MultipleProcessRunnerSimplifier(protein_list, pdb2sequence, n_process=2, return_results=True)\n",
        "        outputs = mprs.run()\n",
        "\n",
        "        # add name column, del type column\n",
        "        protein_df[f'name_{i}'] = protein_df[f'sequence_{i}'].apply(lambda x: x.split('.')[0])\n",
        "        protein_df.drop(f\"type_{i}\", axis=1, inplace=True)\n",
        "        protein_df[f'sequence_{i}'] = [output.split(\"\\t\")[1] for output in outputs]\n",
        "\n",
        "      # columns: name_1, name_2, chain_1, chain_2, sequence_1, sequence_2, label, stage\n",
        "      protein_df.to_csv(csv_dataset_path, index=None)\n",
        "      return csv_dataset_path\n",
        "\n",
        "  else:\n",
        "    # 0. Single AA Sequence\n",
        "    if data_type == 'Single AA Sequence':\n",
        "      input_seq = raw_data\n",
        "      aa_seq = input_seq.value\n",
        "\n",
        "      sa_seq = ''\n",
        "      for aa in aa_seq:\n",
        "          sa_seq += aa + '#'\n",
        "      return sa_seq\n",
        "\n",
        "    # 1. Single SA Sequence\n",
        "    elif data_type == 'Single SA Sequence':\n",
        "      input_seq = raw_data\n",
        "      sa_seq = input_seq.value\n",
        "\n",
        "      return sa_seq\n",
        "\n",
        "    # 2. Single UniProt ID\n",
        "    elif data_type == 'Single UniProt ID':\n",
        "      input_seq = raw_data\n",
        "      uniprot_id = input_seq.value\n",
        "\n",
        "\n",
        "      protein_list = [(uniprot_id, \"AF2\", \"A\")]\n",
        "      uniprot2pdb([protein_list[0][0]])\n",
        "      mprs = MultipleProcessRunnerSimplifier(protein_list, pdb2sequence, n_process=2, return_results=True)\n",
        "      seqs = mprs.run()\n",
        "      sa_seq = seqs[0].split('\\t')[1]\n",
        "      return sa_seq\n",
        "\n",
        "    # 3. Single PDB/CIF Structure\n",
        "    elif data_type == 'Single PDB/CIF Structure':\n",
        "      uniprot_id = raw_data[0]\n",
        "      chain = raw_data[1].value\n",
        "\n",
        "      protein_list = [(uniprot_id, chain)]\n",
        "      mprs = MultipleProcessRunnerSimplifier(protein_list, pdb2sequence, n_process=2, return_results=True)\n",
        "      seqs = mprs.run()\n",
        "      assert len(seqs)>0, \"Unable to convert to SA sequence. Please check the `type`, `chain`, and `.pdb/.cif file`.\"\n",
        "      sa_seq = seqs[0].split('\\t')[1]\n",
        "      return sa_seq\n",
        "\n",
        "    # 9. Pair Single AA Sequences\n",
        "    elif data_type == \"A pair of AA Sequences\":\n",
        "      input_seq_1, input_seq_2 = raw_data\n",
        "      sa_seq1 = get_SA_sequence_by_data_type('Single AA Sequence', input_seq_1)\n",
        "      sa_seq2 = get_SA_sequence_by_data_type('Single AA Sequence', input_seq_2)\n",
        "\n",
        "      return (sa_seq1, sa_seq2)\n",
        "\n",
        "    # 10. Pair Single SA Sequences\n",
        "    elif data_type ==  \"A pair of SA Sequences\":\n",
        "      input_seq_1, input_seq_2 = raw_data\n",
        "      sa_seq1 = get_SA_sequence_by_data_type('Single SA Sequence', input_seq_1)\n",
        "      sa_seq2 = get_SA_sequence_by_data_type('Single SA Sequence', input_seq_2)\n",
        "\n",
        "      return (sa_seq1, sa_seq2)\n",
        "\n",
        "    # 11. Pair Single UniProt IDs\n",
        "    elif data_type ==  \"A pair of UniProt IDs\":\n",
        "      input_seq_1, input_seq_2 = raw_data\n",
        "      sa_seq1 = get_SA_sequence_by_data_type('Single UniProt ID', input_seq_1)\n",
        "      sa_seq2 = get_SA_sequence_by_data_type('Single UniProt ID', input_seq_2)\n",
        "\n",
        "      return (sa_seq1, sa_seq2)\n",
        "\n",
        "    # 12. Pair Single PDB/CIF Structure\n",
        "    elif data_type == \"A pair of PDB/CIF Structures\":\n",
        "      uniprot_id1 = raw_data[0]\n",
        "      struc_type1 = raw_data[1].value\n",
        "      chain1 = raw_data[2].value\n",
        "\n",
        "      protein_list1 = [(uniprot_id1, struc_type1, chain1)]\n",
        "      mprs1 = MultipleProcessRunnerSimplifier(protein_list1, pdb2sequence, n_process=2, return_results=True)\n",
        "      seqs1 = mprs1.run()\n",
        "      sa_seq1 = seqs1[0].split('\\t')[1]\n",
        "\n",
        "      uniprot_id2 = raw_data[3]\n",
        "      struc_type2 = raw_data[4].value\n",
        "      chain2 = raw_data[5].value\n",
        "\n",
        "      protein_list2 = [(uniprot_id2, struc_type2, chain2)]\n",
        "      mprs2 = MultipleProcessRunnerSimplifier(protein_list2, pdb2sequence, n_process=2, return_results=True)\n",
        "      seqs2 = mprs2.run()\n",
        "      sa_seq2 = seqs2[0].split('\\t')[1]\n",
        "      return sa_seq1, sa_seq2\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "################################################################################\n",
        "########################## Download predicted structures #######################\n",
        "################################################################################\n",
        "def uniprot2pdb(uniprot_ids, nprocess=20):\n",
        "  from saprot.utils.downloader import AlphaDBDownloader\n",
        "\n",
        "  os.makedirs(STRUCTURE_HOME, exist_ok=True)\n",
        "  af2_downloader = AlphaDBDownloader(uniprot_ids, \"pdb\", save_dir=STRUCTURE_HOME, n_process=20)\n",
        "  af2_downloader.run()\n",
        "\n",
        "\n",
        "\n",
        "################################################################################\n",
        "############### Form foldseek sequences by multiple processes ##################\n",
        "################################################################################\n",
        "# def pdb2sequence(process_id, idx, uniprot_id, writer):\n",
        "#   from saprot.utils.foldseek_util import get_struc_seq\n",
        "\n",
        "#   try:\n",
        "#     pdb_path = f\"{STRUCTURE_HOME}/{uniprot_id}.pdb\"\n",
        "#     cif_path = f\"{STRUCTURE_HOME}/{uniprot_id}.cif\"\n",
        "#     if Path(pdb_path).exists():\n",
        "#       seq = get_struc_seq(FOLDSEEK_PATH, pdb_path, [\"A\"], process_id=process_id)[\"A\"][-1]\n",
        "#     if Path(cif_path).exists():\n",
        "#       seq = get_struc_seq(FOLDSEEK_PATH, cif_path, [\"A\"], process_id=process_id)[\"A\"][-1]\n",
        "\n",
        "#     writer.write(f\"{uniprot_id}\\t{seq}\\n\")\n",
        "#   except Exception as e:\n",
        "#     print(f\"Error: {uniprot_id}, {e}\")\n",
        "\n",
        "# clear_output(wait=True)\n",
        "# print(\"Installation finished!\")\n",
        "\n",
        "def pdb2sequence(process_id, idx, row_tuple, writer):\n",
        "\n",
        "  # print(\"=\"*100)\n",
        "  # print(row_tuple)\n",
        "  # print(\"=\"*100)\n",
        "  uniprot_id = row_tuple[0].split('.')[0]     #\n",
        "  chain = row_tuple[1]\n",
        "  plddt_mask= True\n",
        "\n",
        "  from saprot.utils.foldseek_util import get_struc_seq\n",
        "  try:\n",
        "    pdb_path = f\"{STRUCTURE_HOME}/{uniprot_id}.pdb\"\n",
        "    cif_path = f\"{STRUCTURE_HOME}/{uniprot_id}.cif\"\n",
        "    if Path(pdb_path).exists():\n",
        "      seq = get_struc_seq(FOLDSEEK_PATH, pdb_path, [chain], process_id=process_id, plddt_mask=plddt_mask)[chain][-1]\n",
        "    elif Path(cif_path).exists():\n",
        "      seq = get_struc_seq(FOLDSEEK_PATH, cif_path, [chain], process_id=process_id, plddt_mask=plddt_mask)[chain][-1]\n",
        "    else:\n",
        "      raise BaseException(f\"The {uniprot_id}.pdb/{uniprot_id}.cif file doesn't exists!\")\n",
        "    writer.write(f\"{uniprot_id}\\t{seq}\\n\")\n",
        "\n",
        "  except Exception as e:\n",
        "    print(f\"Error: {uniprot_id}, {e}\")\n",
        "\n",
        "\n",
        "pymol_color_list = [\"#33ff33\",\"#00ffff\",\"#ff33cc\",\"#ffff00\",\"#ff9999\",\"#e5e5e5\",\"#7f7fff\",\"#ff7f00\",\n",
        "          \"#7fff7f\",\"#199999\",\"#ff007f\",\"#ffdd5e\",\"#8c3f99\",\"#b2b2b2\",\"#007fff\",\"#c4b200\",\n",
        "          \"#8cb266\",\"#00bfbf\",\"#b27f7f\",\"#fcd1a5\",\"#ff7f7f\",\"#ffbfdd\",\"#7fffff\",\"#ffff7f\",\n",
        "          \"#00ff7f\",\"#337fcc\",\"#d8337f\",\"#bfff3f\",\"#ff7fff\",\"#d8d8ff\",\"#3fffbf\",\"#b78c4c\",\n",
        "          \"#339933\",\"#66b2b2\",\"#ba8c84\",\"#84bf00\",\"#b24c66\",\"#7f7f7f\",\"#3f3fa5\",\"#a5512b\"]\n",
        "\n",
        "alphabet_list = list(ascii_uppercase+ascii_lowercase)\n",
        "\n",
        "\n",
        "def convert_outputs_to_pdb(outputs):\n",
        "\tfinal_atom_positions = atom14_to_atom37(outputs[\"positions\"][-1], outputs)\n",
        "\toutputs = {k: v.to(\"cpu\").numpy() for k, v in outputs.items()}\n",
        "\tfinal_atom_positions = final_atom_positions.cpu().numpy()\n",
        "\tfinal_atom_mask = outputs[\"atom37_atom_exists\"]\n",
        "\tpdbs = []\n",
        "\toutputs[\"plddt\"] *= 100\n",
        "\n",
        "\tfor i in range(outputs[\"aatype\"].shape[0]):\n",
        "\t\taa = outputs[\"aatype\"][i]\n",
        "\t\tpred_pos = final_atom_positions[i]\n",
        "\t\tmask = final_atom_mask[i]\n",
        "\t\tresid = outputs[\"residue_index\"][i] + 1\n",
        "\t\tpred = OFProtein(\n",
        "\t\t    aatype=aa,\n",
        "\t\t    atom_positions=pred_pos,\n",
        "\t\t    atom_mask=mask,\n",
        "\t\t    residue_index=resid,\n",
        "\t\t    b_factors=outputs[\"plddt\"][i],\n",
        "\t\t    chain_index=outputs[\"chain_index\"][i] if \"chain_index\" in outputs else None,\n",
        "\t\t)\n",
        "\t\tpdbs.append(to_pdb(pred))\n",
        "\treturn pdbs\n",
        "\n",
        "\n",
        "# This function is copied from ColabFold!\n",
        "def show_pdb(path, show_sidechains=False, show_mainchains=False, color=\"lddt\"):\n",
        "  file_type = str(path).split(\".\")[-1]\n",
        "  if file_type == \"cif\":\n",
        "    file_type == \"mmcif\"\n",
        "\n",
        "  view = py3Dmol.view(js='https://3dmol.org/build/3Dmol.js',)\n",
        "  view.addModel(open(path,'r').read(),file_type)\n",
        "\n",
        "  if color == \"lDDT\":\n",
        "    view.setStyle({'cartoon': {'colorscheme': {'prop':'b','gradient': 'roygb','min':50,'max':90}}})\n",
        "  elif color == \"rainbow\":\n",
        "    view.setStyle({'cartoon': {'color':'spectrum'}})\n",
        "  elif color == \"chain\":\n",
        "    chains = len(get_chain_ids(path))\n",
        "    for n,chain,color in zip(range(chains),alphabet_list,pymol_color_list):\n",
        "       view.setStyle({'chain':chain},{'cartoon': {'color':color}})\n",
        "\n",
        "  if show_sidechains:\n",
        "    BB = ['C','O','N']\n",
        "    view.addStyle({'and':[{'resn':[\"GLY\",\"PRO\"],'invert':True},{'atom':BB,'invert':True}]},\n",
        "                        {'stick':{'colorscheme':f\"WhiteCarbon\",'radius':0.3}})\n",
        "    view.addStyle({'and':[{'resn':\"GLY\"},{'atom':'CA'}]},\n",
        "                        {'sphere':{'colorscheme':f\"WhiteCarbon\",'radius':0.3}})\n",
        "    view.addStyle({'and':[{'resn':\"PRO\"},{'atom':['C','O'],'invert':True}]},\n",
        "                        {'stick':{'colorscheme':f\"WhiteCarbon\",'radius':0.3}})\n",
        "  if show_mainchains:\n",
        "    BB = ['C','O','N','CA']\n",
        "    view.addStyle({'atom':BB},{'stick':{'colorscheme':f\"WhiteCarbon\",'radius':0.3}})\n",
        "\n",
        "  view.zoomTo()\n",
        "  return view\n",
        "\n",
        "\n",
        "def plot_plddt_legend(dpi=100):\n",
        "  thresh = ['plDDT:','Very low (<50)','Low (60)','OK (70)','Confident (80)','Very high (>90)']\n",
        "  plt.figure(figsize=(1,0.1),dpi=dpi)\n",
        "  ########################################\n",
        "  for c in [\"#FFFFFF\",\"#FF0000\",\"#FFFF00\",\"#00FF00\",\"#00FFFF\",\"#0000FF\"]:\n",
        "    plt.bar(0, 0, color=c)\n",
        "  plt.legend(thresh, frameon=False,\n",
        "             loc='center', ncol=6,\n",
        "             handletextpad=1,\n",
        "             columnspacing=1,\n",
        "             markerscale=0.5,)\n",
        "  plt.axis(False)\n",
        "  return plt\n",
        "\n",
        "\n",
        "################################################################################\n",
        "###############   Download file to local computer   ##################\n",
        "################################################################################\n",
        "def file_download(path: str):\n",
        "  with open(path, \"rb\") as r:\n",
        "    res = r.read()\n",
        "\n",
        "  #FILE\n",
        "  filename = os.path.basename(path)\n",
        "  b64 = base64.b64encode(res)\n",
        "  payload = b64.decode()\n",
        "\n",
        "  #BUTTONS\n",
        "  html_buttons = '''<html>\n",
        "  <head>\n",
        "  <meta name=\"viewport\" content=\"width=device-width, initial-scale=1\">\n",
        "  </head>\n",
        "  <body>\n",
        "  <a download=\"{filename}\" href=\"data:text/csv;base64,{payload}\" download>\n",
        "  <button class=\"p-Widget jupyter-widgets jupyter-button widget-button mod-warning\">Download File</button>\n",
        "  </a>\n",
        "  </body>\n",
        "  </html>\n",
        "  '''\n",
        "\n",
        "  html_button = html_buttons.format(payload=payload,filename=filename)\n",
        "  display(HTML(html_button))\n",
        "\n",
        "  # Automatically download file if the server is from google cloud.\n",
        "  if root_dir == \"/content\":\n",
        "    files.download(path)\n",
        "\n",
        "################################################################################\n",
        "############################ MODEL INFO #######################################\n",
        "################################################################################\n",
        "def get_base_model(adapter_path):\n",
        "  adapter_config = Path(adapter_path) / \"adapter_config.json\"\n",
        "  with open(adapter_config, 'r') as f:\n",
        "    adapter_config_dict = json.load(f)\n",
        "    base_model = adapter_config_dict['base_model_name_or_path']\n",
        "    if 'SaProt_650M_AF2' in base_model:\n",
        "      base_model = \"westlake-repl/SaProt_650M_AF2\"\n",
        "    elif 'SaProt_35M_AF2' in base_model:\n",
        "      base_model = \"westlake-repl/SaProt_35M_AF2\"\n",
        "    else:\n",
        "      raise RuntimeError(\"Please ensure the base model is \\\"SaProt_650M_AF2\\\" or \\\"SaProt_35M_AF2\\\"\")\n",
        "  return base_model\n",
        "\n",
        "def check_training_data_type(adapter_path, data_type):\n",
        "  metadata_path = Path(adapter_path) / \"metadata.json\"\n",
        "  if metadata_path.exists():\n",
        "    with open(metadata_path, 'r') as f:\n",
        "      metadata = json.load(f)\n",
        "      required_training_data_type = metadata['training_data_type']\n",
        "  else:\n",
        "    required_training_data_type = \"SA\"\n",
        "\n",
        "  if (required_training_data_type == \"AA\") and (\"AA\" not in data_type):\n",
        "    print(Fore.RED+f\"This model ({adapter_path}) is trained on {required_training_data_type} sequences, and predictions work better with AA sequences.\"+Style.RESET_ALL)\n",
        "    print(Fore.RED+f\"The current data type ({data_type}) includes structural information, which will not be used for predictions.\"+Style.RESET_ALL)\n",
        "    print()\n",
        "    print('='*100)\n",
        "  elif (required_training_data_type == \"SA\") and (\"AA\" in data_type):\n",
        "    print(Fore.RED+f\"This model ({adapter_path}) is trained on {required_training_data_type} sequences, and predictions work better with SA sequences.\"+Style.RESET_ALL)\n",
        "    print(Fore.RED+f\"The current data type ({data_type}) does not include structural information, which may lead to weak prediction performance.\"+Style.RESET_ALL)\n",
        "    print(Fore.RED+f\"If you only have the amino acid sequence, we strongly recommend using AF2 to predict the structure and generate a PDB file before prediction.\"+Style.RESET_ALL)\n",
        "    print()\n",
        "    print('='*100)\n",
        "\n",
        "  return required_training_data_type\n",
        "\n",
        "def mask_struc_token(sequence):\n",
        "    return ''.join('#' if i % 2 == 1 and char.islower() else char for i, char in enumerate(sequence))\n",
        "\n",
        "def get_num_labels_by_adapter(adapter_path):\n",
        "    adapter_path = Path(adapter_path)\n",
        "\n",
        "    if (adapter_path / 'adapter_model.safetensors').exists():\n",
        "        file_path = adapter_path / 'adapter_model.safetensors'\n",
        "        with safe_open(file_path, framework=\"pt\") as f:\n",
        "          if 'base_model.model.classifier.out_proj.bias' in f.keys():\n",
        "              tensor = f.get_tensor('base_model.model.classifier.out_proj.bias')\n",
        "          elif 'base_model.model.classifier.bias' in f.keys():\n",
        "              tensor = f.get_tensor('base_model.model.classifier.bias')\n",
        "          else:\n",
        "              raise KeyError(f\"Neither 'base_model.model.classifier.out_proj.bias' nor 'base_model.model.classifier.bias' found in the file({file_path}).\")\n",
        "\n",
        "    elif (adapter_path / 'adapter_model.bin').exists():\n",
        "      file_path = adapter_path / 'adapter_model.bin'\n",
        "      state_dict = torch.load(file_path)\n",
        "      if 'base_model.model.classifier.out_proj.bias' in state_dict.keys():\n",
        "        tensor = state_dict['base_model.model.classifier.out_proj.bias']\n",
        "      elif 'base_model.model.classifier.bias' in f.keys():\n",
        "        tensor = state_dict['base_model.model.classifier.bias']\n",
        "      else:\n",
        "        raise KeyError(f\"Neither 'base_model.model.classifier.out_proj.bias' nor 'base_model.model.classifier.bias' found in the file({file_path}).\")\n",
        "\n",
        "    else:\n",
        "        raise FileNotFoundError(f\"Neither 'adapter_model.safetensors' nor 'adapter_model.bin' found in the provided path({adapter_path}).\")\n",
        "\n",
        "    num_labels = list(tensor.shape)[0]\n",
        "    return num_labels\n",
        "\n",
        "def get_num_labels_and_task_type_by_adapter(adapter_path):\n",
        "    adapter_path = Path(adapter_path)\n",
        "\n",
        "    task_type = None\n",
        "    if (adapter_path / 'adapter_model.safetensors').exists():\n",
        "      file_path = adapter_path / 'adapter_model.safetensors'\n",
        "      with safe_open(file_path, framework=\"pt\") as f:\n",
        "        if 'base_model.model.classifier.out_proj.bias' in f.keys():\n",
        "          tensor = f.get_tensor('base_model.model.classifier.out_proj.bias')\n",
        "        elif 'base_model.model.classifier.bias' in f.keys():\n",
        "          task_type = 'token_classification'\n",
        "          tensor = f.get_tensor('base_model.model.classifier.bias')\n",
        "        else:\n",
        "          raise KeyError(f\"Neither 'base_model.model.classifier.out_proj.bias' nor 'base_model.model.classifier.bias' found in the file({file_path}).\")\n",
        "\n",
        "    elif (adapter_path / 'adapter_model.bin').exists():\n",
        "      file_path = adapter_path / 'adapter_model.bin'\n",
        "      state_dict = torch.load(file_path)\n",
        "      if 'base_model.model.classifier.out_proj.bias' in state_dict.keys():\n",
        "        tensor = state_dict['base_model.model.classifier.out_proj.bias']\n",
        "      elif 'base_model.model.classifier.bias' in f.keys():\n",
        "        task_type = 'token_classification'\n",
        "        tensor = state_dict['base_model.model.classifier.bias']\n",
        "      else:\n",
        "        raise KeyError(f\"Neither 'base_model.model.classifier.out_proj.bias' nor 'base_model.model.classifier.bias' found in the file({file_path}).\")\n",
        "\n",
        "    else:\n",
        "        raise FileNotFoundError(f\"Neither 'adapter_model.safetensors' nor 'adapter_model.bin' found in the provided path({adapter_path}).\")\n",
        "\n",
        "    num_labels = list(tensor.shape)[0]\n",
        "    if task_type != 'token_classification':\n",
        "      if num_labels > 1:\n",
        "        task_type = 'classification'\n",
        "      elif num_labels == 1:\n",
        "        task_type = 'regression'\n",
        "\n",
        "    return num_labels, task_type\n",
        "\n",
        "################################################################################\n",
        "############################ INFO ##############################################\n",
        "################################################################################\n",
        "clear_output(wait=True)\n",
        "print(\"Installation finished!\")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "cellView": "form",
        "id": "LHKi8xI-aU4E",
        "outputId": "11c791c4-4b65-410a-a184-c67119e45ea8"
      },
      "id": "LHKi8xI-aU4E",
      "execution_count": 1,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Installation finished!\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#@title **Now click the run button to start using ColabSaprot**\n",
        "import markdown\n",
        "import ipywidgets\n",
        "\n",
        "from ipywidgets import Button, Label, HTML, Layout\n",
        "from IPython.display import display, clear_output\n",
        "from functools import partial\n",
        "\n",
        "\n",
        "# HINT = Fore.RED + \"\\nNote: At any time you can click the run button again to start restart.\" + Style.RESET_ALL\n",
        "HINT = HTML(markdown.markdown(\"\\n\\n<font color=red>Note: At any time you can click the run button <img src='https://github.com/westlake-repl/SaProtHub/blob/dev/Figure/run_button.png?raw=true' height='25px' width='25px' align='center'> again to restart.</font>\"))\n",
        "\n",
        "\n",
        "# Jump to next page\n",
        "def jump(button, next):\n",
        "  clear_output()\n",
        "  next()\n",
        "  display(HINT)\n",
        "\n",
        "\n",
        "def custom_display(*args, **kwargs):\n",
        "  clear_output()\n",
        "  display(*args, **kwargs)\n",
        "  display(HINT)\n",
        "\n",
        "\n",
        "######################################################################\n",
        "#            Backend functions             #\n",
        "######################################################################\n",
        "def generate_download_btn(path: str):\n",
        "  with open(path, \"rb\") as r:\n",
        "    res = r.read()\n",
        "\n",
        "  #FILE\n",
        "  filename = os.path.basename(path)\n",
        "  b64 = base64.b64encode(res)\n",
        "  payload = b64.decode()\n",
        "\n",
        "  #BUTTONS\n",
        "  html_buttons = '''<html>\n",
        "  <head>\n",
        "  <meta name=\"viewport\" content=\"width=device-width, initial-scale=1\">\n",
        "  </head>\n",
        "  <body>\n",
        "  <a download=\"{filename}\" href=\"data:text/csv;base64,{payload}\" download>\n",
        "  <button class=\"p-Widget jupyter-widgets jupyter-button widget-button mod-warning\">Download File</button>\n",
        "  </a>\n",
        "  </body>\n",
        "  </html>\n",
        "  '''\n",
        "\n",
        "  html_button = html_buttons.format(payload=payload,filename=filename)\n",
        "  return HTML(html_button)\n",
        "\n",
        "\n",
        "def load_zeroshot_model():\n",
        "  try:\n",
        "    zero_shot_model\n",
        "  except Exception:\n",
        "    from saprot.model.saprot.saprot_foldseek_mutation_model import SaprotFoldseekMutationModel\n",
        "    base_model = \"westlake-repl/SaProt_650M_AF2\"\n",
        "    config = {\n",
        "      \"foldseek_path\": None,\n",
        "      \"config_path\": base_model,\n",
        "      \"load_pretrained\": True,\n",
        "    }\n",
        "\n",
        "    zero_shot_model = SaprotFoldseekMutationModel(**config)\n",
        "    device = \"cuda\" if torch.cuda.is_available() else \"cpu\"\n",
        "    zero_shot_model.to(device)\n",
        "\n",
        "  return zero_shot_model\n",
        "\n",
        "\n",
        "# Zero-shot prediction\n",
        "def predict_mut(sa_seq, mut_info):\n",
        "  zero_shot_model = load_zeroshot_model()\n",
        "  score = zero_shot_model.predict_mut(sa_seq, mut_info)\n",
        "  return score\n",
        "\n",
        "\n",
        "# Zero-shot prediction for single-site saturation mutagenesis\n",
        "def predict_all_mut(sa_seq):\n",
        "  zero_shot_model = load_zeroshot_model()\n",
        "\n",
        "  timestamp = datetime.now().strftime(\"%y%m%d%H%M%S\")\n",
        "  output_path = OUTPUT_HOME / f'{timestamp}_prediction_output.csv'\n",
        "\n",
        "  mut_dicts = []\n",
        "  aa_seq = sa_seq[0::2]\n",
        "  for i in tqdm(range(len(aa_seq)), leave=False, desc=f\"Predicting\"):\n",
        "    mut_dict = zero_shot_model.predict_pos_mut(sa_seq, i+1)\n",
        "    mut_dicts.append(mut_dict)\n",
        "\n",
        "  mut_list = [{'mutation': key, 'score': value} for d in mut_dicts for key, value in d.items()]\n",
        "  df = pd.DataFrame(mut_list)\n",
        "  df.to_csv(output_path, index=None)\n",
        "  return output_path\n",
        "\n",
        "# Perform protein sequence design\n",
        "def inverse_folding(aa_seq, struc_seq, method, num_samples):\n",
        "  try:\n",
        "    saprot_if_model\n",
        "\n",
        "  except Exception:\n",
        "    from saprot.model.saprot.saprot_if_model import SaProtIFModel\n",
        "\n",
        "    base_model = \"westlake-repl/SaProt_650M_AF2_inverse_folding\"\n",
        "    config = {\n",
        "        \"config_path\": base_model,\n",
        "        \"load_pretrained\": True,\n",
        "    }\n",
        "\n",
        "    saprot_if_model = SaProtIFModel(**config)\n",
        "    tokenizer = saprot_if_model.tokenizer\n",
        "    device = \"cuda\" if torch.cuda.is_available() else \"cpu\"\n",
        "    saprot_if_model.to(device)\n",
        "\n",
        "  pred_aa_seqs = saprot_if_model.predict(aa_seq, struc_seq, method=method, num_samples=num_samples)\n",
        "  return pred_aa_seqs\n",
        "\n",
        "\n",
        "######################################################################\n",
        "#          Training or Prediction            #\n",
        "######################################################################\n",
        "def train_or_pred():\n",
        "  question = HTML(markdown.markdown(\"## Do you want to train your own model or you only want to use existing models to make prediction?\"))\n",
        "  train_btn = Button(description='I want to train my own model', layout=Layout(width='400px', height='30px'))\n",
        "  pred_btn = Button(description='I want to use existing models to make prediction', layout=Layout(width='400px', height='30px'))\n",
        "\n",
        "  items = [question, train_btn, pred_btn]\n",
        "\n",
        "  # Set click events\n",
        "  train_btn.on_click(partial(jump, next=choose_training_task))\n",
        "  pred_btn.on_click(partial(jump, next=choose_pred_task))\n",
        "\n",
        "  display(*items)\n",
        "\n",
        "\n",
        "######################################################################\n",
        "#          Choose training task             #\n",
        "######################################################################\n",
        "def choose_training_task():\n",
        "  title = HTML(markdown.markdown(\"## Please finish the setting of your training task\"))\n",
        "  WIDTH = \"500px\"\n",
        "  HEIGHT= \"30px\"\n",
        "\n",
        "  task_hint = HTML(markdown.markdown(\"### Task setting:\"))\n",
        "  task_name = ipywidgets.Text(\n",
        "              value=\"Demo\",\n",
        "              placeholder=f'Enter the task name',\n",
        "              disabled=False,\n",
        "              description=\"Name your task:\",\n",
        "              layout=Layout(width=WIDTH, height=HEIGHT),\n",
        "              style={'description_width': 'initial'},\n",
        "              )\n",
        "  task_type = ipywidgets.Dropdown(\n",
        "            options=['Protein-level Classification', 'Protein-level Regression', 'Residue-level Classification', \"Protein-protein Classification\", \"Protein-protein Regression\"],\n",
        "            value='Protein-level Classification',\n",
        "            description='Task type:',\n",
        "            disabled=False,\n",
        "            layout=Layout(width=WIDTH, height=HEIGHT)\n",
        "          )\n",
        "\n",
        "  num_label = ipywidgets.BoundedIntText(\n",
        "        value=2,\n",
        "        min=2,\n",
        "        max=100,\n",
        "        step=1,\n",
        "        description='Number of classes:',\n",
        "        disabled=False,\n",
        "        style={'description_width': 'initial'},\n",
        "        layout=Layout(width=WIDTH, height=HEIGHT)\n",
        "      )\n",
        "\n",
        "  model_hint = HTML(markdown.markdown(\"### Model setting:\"))\n",
        "  model_type = ipywidgets.Dropdown(\n",
        "            options=['Official SaProt (35M)', \"Official SaProt (650M)\", \"Trained by yourself on ColabSaprot\", \"Shared by peers on SaprotHub\"],\n",
        "            value='Official SaProt (35M)',\n",
        "            description='Base model:',\n",
        "            disabled=False,\n",
        "            layout=Layout(width=WIDTH, height=HEIGHT)\n",
        "          )\n",
        "\n",
        "  dataset_hint = HTML(markdown.markdown(\"### Dataset setting:\"))\n",
        "  data_type_hint = HTML(markdown.markdown(\"**Your training data type:**\"))\n",
        "  data_type = ipywidgets.RadioButtons(\n",
        "      options=['protein sequence', 'protein structure'],\n",
        "      disabled=False,\n",
        "      style={'description_width': 'initial'},\n",
        "      )\n",
        "\n",
        "  data_src_hint = HTML(markdown.markdown(\"**Where is your data from:**\"))\n",
        "  data_src_type = ipywidgets.RadioButtons(\n",
        "      options=['SaprotHub in HuggingFace', 'Upload mannually'],\n",
        "      layout={'width': 'max-content'}, # If the items' names are long\n",
        "      disabled=False,\n",
        "      style={'description_width': 'initial'},\n",
        "      )\n",
        "\n",
        "  data_split_hint = HTML(markdown.markdown(\"**Have you already split your data into training, validation and test set?**\"))\n",
        "  data_split_type = ipywidgets.RadioButtons(\n",
        "              options=['Yes', 'No (We will randomly split your data with 8:1:1 for train:valid:test)'],\n",
        "            layout={'width': 'max-content'}, # If the items' names are long\n",
        "              disabled=False,\n",
        "              style={'description_width': 'initial'},\n",
        "          )\n",
        "\n",
        "  saprothub_data_id = ipywidgets.Text(\n",
        "              value=None,\n",
        "              placeholder=f'Enter SaprotHub dataset id in HuggingFace',\n",
        "              disabled=False,\n",
        "              description=\"Dataset id:\",\n",
        "              layout=Layout(width=WIDTH, height=HEIGHT),\n",
        "              )\n",
        "  saprothub_data_id_hint = HTML(markdown.markdown(\"**<font color=red>Please make sure that the SaprotHub dataset is consistent with the task type above</font>**\"))\n",
        "\n",
        "  hyperparameter_hint = HTML(markdown.markdown(\"### Training hyper-parameters:\"))\n",
        "  batch_size = ipywidgets.Dropdown(\n",
        "            options=[\"Adaptive\", \"1\", \"2\", \"4\", \"8\", \"16\", \"32\", \"64\", \"128\", \"256\"],\n",
        "            value='Adaptive',\n",
        "            description='Batch size:',\n",
        "            disabled=False,\n",
        "            layout=Layout(width=WIDTH, height=HEIGHT)\n",
        "            )\n",
        "\n",
        "  epoch = ipywidgets.BoundedIntText(\n",
        "        value=5,\n",
        "        min=1,\n",
        "        max=100,\n",
        "        step=1,\n",
        "        description='Epoch:',\n",
        "        disabled=False,\n",
        "        style={'description_width': 'initial'},\n",
        "        layout=Layout(width=WIDTH, height=HEIGHT)\n",
        "      )\n",
        "\n",
        "  lr = ipywidgets.FloatText(\n",
        "      value=5e-4,\n",
        "      description='Learning rate:',\n",
        "      disabled=False,\n",
        "      style={'description_width': 'initial'},\n",
        "      layout=Layout(width=WIDTH, height=HEIGHT)\n",
        "  )\n",
        "\n",
        "  advanced_setting = Button(description='Show advanced setting', layout=Layout(width=WIDTH, height=HEIGHT))\n",
        "  lora_hint = HTML(markdown.markdown(f\"### Lora config\"))\n",
        "  lora_r = ipywidgets.BoundedIntText(\n",
        "        value=8,\n",
        "        min=1,\n",
        "        max=128,\n",
        "        step=1,\n",
        "        description='r:',\n",
        "        disabled=False,\n",
        "        style={'description_width': 'initial'},\n",
        "        layout=Layout(width=WIDTH, height=HEIGHT)\n",
        "      )\n",
        "  lora_dropout_box = ipywidgets.FloatText(\n",
        "      value=0.0,\n",
        "      description='Lora dropout:',\n",
        "      disabled=False,\n",
        "      style={'description_width': 'initial'},\n",
        "      layout=Layout(width=WIDTH, height=HEIGHT)\n",
        "  )\n",
        "  lora_alpha_box = ipywidgets.BoundedIntText(\n",
        "        value=16,\n",
        "        min=1,\n",
        "        max=128,\n",
        "        step=1,\n",
        "        description='Lora alpha:',\n",
        "        disabled=False,\n",
        "        style={'description_width': 'initial'},\n",
        "        layout=Layout(width=WIDTH, height=HEIGHT)\n",
        "      )\n",
        "\n",
        "  trainer_hint = HTML(markdown.markdown(f\"### PyTorch Lightning trainer config\"))\n",
        "  val_check_interval_box = ipywidgets.FloatText(\n",
        "      value=0.5,\n",
        "      description='val_check_interval:',\n",
        "      disabled=False,\n",
        "      style={'description_width': 'initial'},\n",
        "      layout=Layout(width=WIDTH, height=HEIGHT)\n",
        "  )\n",
        "  limit_train_batches_box = ipywidgets.FloatText(\n",
        "      value=1.0,\n",
        "      description='limit_train_batches:',\n",
        "      disabled=False,\n",
        "      style={'description_width': 'initial'},\n",
        "      layout=Layout(width=WIDTH, height=HEIGHT)\n",
        "  )\n",
        "  limit_val_batches_box = ipywidgets.FloatText(\n",
        "      value=1.0,\n",
        "      description='limit_val_batches:',\n",
        "      disabled=False,\n",
        "      style={'description_width': 'initial'},\n",
        "      layout=Layout(width=WIDTH, height=HEIGHT)\n",
        "  )\n",
        "  limit_test_batches_box = ipywidgets.FloatText(\n",
        "      value=1.0,\n",
        "      description='limit_test_batches:',\n",
        "      disabled=False,\n",
        "      style={'description_width': 'initial'},\n",
        "      layout=Layout(width=WIDTH, height=HEIGHT)\n",
        "  )\n",
        "\n",
        "  start_hint = HTML(markdown.markdown(f\"### {'-'*75}\"))\n",
        "  start_btn = Button(description='Start training', layout=Layout(width=WIDTH, height=HEIGHT), button_style=\"info\")\n",
        "\n",
        "\n",
        "  items = [\n",
        "      title,\n",
        "      task_hint,\n",
        "      task_name,\n",
        "      task_type,\n",
        "      num_label,\n",
        "      model_hint,\n",
        "      model_type,\n",
        "      dataset_hint,\n",
        "      data_src_hint,\n",
        "      data_src_type,\n",
        "      data_type_hint,\n",
        "      data_type,\n",
        "      data_split_hint,\n",
        "      data_split_type,\n",
        "      saprothub_data_id,\n",
        "      saprothub_data_id_hint,\n",
        "      hyperparameter_hint,\n",
        "      batch_size,\n",
        "      epoch,\n",
        "      lr,\n",
        "      advanced_setting,\n",
        "      lora_hint,\n",
        "      lora_r,\n",
        "      lora_dropout_box,\n",
        "      lora_alpha_box,\n",
        "      trainer_hint,\n",
        "      limit_train_batches_box,\n",
        "      limit_val_batches_box,\n",
        "      limit_test_batches_box,\n",
        "      start_hint,\n",
        "      start_btn\n",
        "      ]\n",
        "\n",
        "  # Set click events\n",
        "  def change_task_type(change):\n",
        "    now_type = change[\"new\"]\n",
        "    if \"Classification\" in now_type:\n",
        "      num_label.layout.display = None\n",
        "    else:\n",
        "      num_label.layout.display = \"none\"\n",
        "\n",
        "  def change_model_type(change):\n",
        "    model_type_value = change[\"new\"]\n",
        "    task_type_value = task_type_dict[task_type.value]\n",
        "    if model_type_value == \"Trained by yourself on ColabSaprot\":\n",
        "      adapter_combobox = select_adapter_from(task_type_value, use_model_from=model_type_value)\n",
        "      adapter_combobox.layout.width = WIDTH\n",
        "      adapter_combobox.description = \"Select your local model:\"\n",
        "      adapter_combobox.style = {'description_width': 'initial'}\n",
        "      model_type_idx = items.index(model_type)\n",
        "      new_items = items[:model_type_idx+1] + [adapter_combobox] + items[model_type_idx+1:]\n",
        "      custom_display(*new_items)\n",
        "\n",
        "    elif model_type_value == \"Shared by peers on SaprotHub\":\n",
        "      adapter_combobox = select_adapter_from(task_type_value, use_model_from=model_type_value)\n",
        "      adapter_combobox.layout.width = WIDTH\n",
        "      adapter_combobox.placeholder = \"Enter SaprotHub model id in HuggingFace\"\n",
        "      adapter_combobox.description = \"Model id:\"\n",
        "      model_type_idx = items.index(model_type)\n",
        "      new_items = items[:model_type_idx+1] + [adapter_combobox] + items[model_type_idx+1:]\n",
        "      custom_display(*new_items)\n",
        "\n",
        "    else:\n",
        "      custom_display(*items)\n",
        "\n",
        "  def change_data_src_type(change):\n",
        "    data_src_type_value = change[\"new\"]\n",
        "    if data_src_type_value == \"SaprotHub in HuggingFace\":\n",
        "      saprothub_data_id.layout.display = None\n",
        "      saprothub_data_id_hint.layout.display = None\n",
        "      for item in [data_type_hint, data_type, data_split_hint, data_split_type]:\n",
        "        item.layout.display = \"none\"\n",
        "\n",
        "    else:\n",
        "      saprothub_data_id.layout.display = \"none\"\n",
        "      saprothub_data_id_hint.layout.display = \"none\"\n",
        "      for item in [data_type_hint, data_type, data_split_hint, data_split_type]:\n",
        "        item.layout.display = None\n",
        "\n",
        "  def click_advanced_setting(button):\n",
        "    if button.description == \"Show advanced setting\":\n",
        "      button.description = \"Hide advanced setting\"\n",
        "      for item in [lora_hint, lora_r, lora_dropout_box, lora_alpha_box, trainer_hint, limit_train_batches_box, limit_val_batches_box, limit_test_batches_box,]:\n",
        "        item.layout.display = None\n",
        "    else:\n",
        "      button.description = \"Show advanced setting\"\n",
        "      for item in [lora_hint, lora_r, lora_dropout_box, lora_alpha_box, trainer_hint, limit_train_batches_box, limit_val_batches_box, limit_test_batches_box,]:\n",
        "        item.layout.display = \"none\"\n",
        "\n",
        "\n",
        "  def start_training(button):\n",
        "    button.disabled = True\n",
        "    button.description = \"Training...\"\n",
        "\n",
        "    ######################################################################\n",
        "    #            Start training              #\n",
        "    ######################################################################\n",
        "    from saprot.config.config_dict import Default_config\n",
        "    config = copy.deepcopy(Default_config)\n",
        "\n",
        "    # training config\n",
        "    GPU_batch_size = 0\n",
        "    accumulate_grad_batches = 0\n",
        "    num_workers = 2\n",
        "    seed = 20000812\n",
        "\n",
        "    # lora config\n",
        "    r = lora_r.value\n",
        "    lora_dropout = lora_dropout_box.value\n",
        "    lora_alpha = lora_alpha_box.value\n",
        "\n",
        "    # dataset config\n",
        "    val_check_interval = val_check_interval_box.value\n",
        "    limit_train_batches = limit_train_batches_box.value\n",
        "    limit_val_batches = limit_val_batches_box.value\n",
        "    limit_test_batches = limit_test_batches_box.value\n",
        "\n",
        "    mask_struc_ratio=None\n",
        "    if torch.cuda.is_available() is False:\n",
        "      raise BaseException(\"Please switch your Runtime to a GPU!\")\n",
        "\n",
        "    # Task config\n",
        "    task_name_value = task_name.value\n",
        "    task_type_value = task_type.value\n",
        "    original_task_type = task_type_value\n",
        "    task_type_value = task_type_dict[task_type_value]\n",
        "\n",
        "    #####################################################################\n",
        "    #            Model config               #\n",
        "    #####################################################################\n",
        "    base_model = model_type.value\n",
        "    # continue learning\n",
        "    if base_model in [\"Trained by yourself on ColabSaprot\", \"Shared by peers on SaprotHub\"]:\n",
        "      continue_learning = True\n",
        "    else:\n",
        "      continue_learning = False\n",
        "\n",
        "    # base_model\n",
        "    if continue_learning:\n",
        "      adapter_path = ADAPTER_HOME / task_name_value / adapter_combobox.value\n",
        "      print(f\"Training on an existing model: {adapter_path}\")\n",
        "\n",
        "      if base_model == \"Shared by peers on SaprotHub\":\n",
        "        if not adapter_path.exists():\n",
        "          snapshot_download(repo_id=adapter_combobox.value, repo_type=\"model\", local_dir=adapter_path)\n",
        "\n",
        "      adapter_config_path = Path(adapter_path) / \"adapter_config.json\"\n",
        "      assert adapter_config_path.exists(), f\"Can't find {adapter_config_path}\"\n",
        "      with open(adapter_config_path, 'r') as f:\n",
        "        adapter_config = json.load(f)\n",
        "        base_model = adapter_config['base_model_name_or_path']\n",
        "\n",
        "    elif base_model == \"Official SaProt (35M)\":\n",
        "      base_model = \"westlake-repl/SaProt_35M_AF2\"\n",
        "\n",
        "    elif base_model == \"Official SaProt (650M)\":\n",
        "      base_model = \"westlake-repl/SaProt_650M_AF2\"\n",
        "\n",
        "    # model size and model name\n",
        "    if base_model == \"westlake-repl/SaProt_650M_AF2\":\n",
        "      model_size = \"650M\"\n",
        "      model_name = f\"Model-{task_name_value}-{model_size}\"\n",
        "    elif base_model == \"westlake-repl/SaProt_35M_AF2\":\n",
        "      model_size = \"35M\"\n",
        "      model_name = f\"Model-{task_name_value}-{model_size}\"\n",
        "\n",
        "    config.setting.run_mode = \"train\"\n",
        "    config.setting.seed = seed\n",
        "\n",
        "    if task_type_value in [\"classification\", \"token_classification\", \"pair_classification\"]:\n",
        "      config.model.kwargs.num_labels = num_label.value\n",
        "\n",
        "    config.model.model_py_path = model_type_dict[task_type_value]\n",
        "    config.model.kwargs.config_path = base_model\n",
        "    config.dataset.kwargs.tokenizer = base_model\n",
        "\n",
        "    config.model.save_path = str(ADAPTER_HOME / f\"{task_type_value}\" / \"Local\" / model_name)\n",
        "\n",
        "    if task_type_value in [\"regression\", \"pair_regression\"]:\n",
        "      config.model.kwargs.extra_config = {}\n",
        "      config.model.kwargs.extra_config.attention_probs_dropout_prob=0\n",
        "      config.model.kwargs.extra_config.hidden_dropout_prob=0\n",
        "\n",
        "    config.model.kwargs.lora_kwargs = EasyDict({\n",
        "      \"is_trainable\": True,\n",
        "      \"num_lora\": 1,\n",
        "      \"r\": r,\n",
        "      \"lora_dropout\": lora_dropout,\n",
        "      \"lora_alpha\": lora_alpha,\n",
        "      \"config_list\": []})\n",
        "    if continue_learning:\n",
        "      config.model.kwargs.lora_kwargs.config_list.append({\"lora_config_path\": adapter_path})\n",
        "\n",
        "    #####################################################################\n",
        "    #            Dataset config              #\n",
        "    #####################################################################\n",
        "    if data_src_type.value == \"SaprotHub in HuggingFace\":\n",
        "      data_src_type_value = \"SaprotHub Dataset\"\n",
        "      raw_data = saprothub_data_id\n",
        "      csv_dataset_path = get_SA_sequence_by_data_type(data_src_type_value, raw_data)\n",
        "\n",
        "      # Get the meta-data of the dataset\n",
        "      metadata_path = DATASET_HOME / saprothub_data_id.value / 'metadata.json'\n",
        "      metadata = json.load(open(metadata_path, \"r\"))\n",
        "\n",
        "      assert metadata[\"training_task_type\"] == task_type.value, f\"This dataset is for '{metadata['training_task_type']}', which is not suitable for your task '{task_type.value}'\"\n",
        "\n",
        "    check_column_label_and_stage(csv_dataset_path)\n",
        "    from saprot.utils.construct_lmdb import construct_lmdb\n",
        "    construct_lmdb(csv_dataset_path, LMDB_HOME, task_name_value, task_type_value)\n",
        "    lmdb_dataset_path = LMDB_HOME / task_name_value\n",
        "\n",
        "    config.dataset.dataset_py_path = dataset_type_dict[task_type_value]\n",
        "\n",
        "    config.dataset.train_lmdb = str(lmdb_dataset_path / \"train\")\n",
        "    config.dataset.valid_lmdb = str(lmdb_dataset_path / \"valid\")\n",
        "    config.dataset.test_lmdb = str(lmdb_dataset_path / \"test\")\n",
        "\n",
        "    # num_workers\n",
        "    config.dataset.dataloader_kwargs.num_workers = num_workers\n",
        "\n",
        "    def get_accumulate_grad_samples(num_samples):\n",
        "        if num_samples > 3200:\n",
        "            return 64\n",
        "        elif 1600 < num_samples <= 3200:\n",
        "            return 32\n",
        "        elif 800 < num_samples <= 1600:\n",
        "            return 16\n",
        "        elif 400 < num_samples <= 800:\n",
        "            return 8\n",
        "        elif 200 < num_samples <= 400:\n",
        "            return 4\n",
        "        elif 100 < num_samples <= 200:\n",
        "            return 2\n",
        "        else:\n",
        "            return 1\n",
        "\n",
        "    # advanced config\n",
        "    if (GPU_batch_size > 0) and (accumulate_grad_batches > 0):\n",
        "      config.dataset.dataloader_kwargs.batch_size = GPU_batch_size\n",
        "      config.Trainer.accumulate_grad_batches= accumulate_grad_batches\n",
        "\n",
        "    elif (GPU_batch_size == 0) and (accumulate_grad_batches == 0):\n",
        "\n",
        "      # batch_size\n",
        "      if base_model == \"westlake-repl/SaProt_650M_AF2\" and root_dir == \"/content\":\n",
        "        GPU_batch_size = 1\n",
        "      else:\n",
        "        GPU_batch_size_dict = {\n",
        "          \"Tesla T4\": 2,\n",
        "          \"NVIDIA L4\": 2,\n",
        "          \"NVIDIA A100-SXM4-40GB\": 4,\n",
        "          }\n",
        "        GPU_name = torch.cuda.get_device_name(0)\n",
        "        GPU_batch_size = GPU_batch_size_dict[GPU_name] if GPU_name in GPU_batch_size_dict else 2\n",
        "\n",
        "        if task_type_value in [\"pair_classification\", \"pair_regression\"]:\n",
        "          GPU_batch_size = int(max(GPU_batch_size / 2, 1))\n",
        "\n",
        "      config.dataset.dataloader_kwargs.batch_size = GPU_batch_size\n",
        "\n",
        "      # accumulate_grad_batches\n",
        "      if batch_size.value == \"Adaptive\":\n",
        "\n",
        "        env = lmdb.open(config.dataset.train_lmdb, readonly=True)\n",
        "\n",
        "        with env.begin() as txn:\n",
        "          stat = txn.stat()\n",
        "          num_samples = stat['entries']\n",
        "\n",
        "        accumulate_grad_samples = get_accumulate_grad_samples(num_samples)\n",
        "\n",
        "      else:\n",
        "        accumulate_grad_samples = int(batch_size_value)\n",
        "\n",
        "      accumulate_grad_batches = max(int(accumulate_grad_samples / GPU_batch_size), 1)\n",
        "\n",
        "      config.Trainer.accumulate_grad_batches= accumulate_grad_batches\n",
        "\n",
        "    else:\n",
        "      raise BaseException(f\"Please make sure `GPU_batch_size`({GPU_batch_size}) and `accumulate_grad_batches`({accumulate_grad_batches}) are both greater than zero!\")\n",
        "\n",
        "    #####################################################################\n",
        "    #          Hyper-parameter config            #\n",
        "    #####################################################################\n",
        "    epoch_value = epoch.value\n",
        "    lr_value = lr.value\n",
        "\n",
        "    #####################################################################\n",
        "    #               Trainer               #\n",
        "    #####################################################################\n",
        "    config.Trainer.accelerator = \"gpu\" if torch.cuda.is_available() else \"cpu\"\n",
        "\n",
        "    # epoch\n",
        "    config.Trainer.max_epochs = epoch_value\n",
        "    # test only: load the existing model\n",
        "    if config.Trainer.max_epochs == 0 and continue_learning:\n",
        "      config.model.save_path = config.model.kwargs.lora_kwargs.config_list[0]['lora_config_path']\n",
        "\n",
        "    # learning rate\n",
        "    config.model.lr_scheduler_kwargs.init_lr = lr_value\n",
        "\n",
        "    # trainer\n",
        "    config.Trainer.limit_train_batches=limit_train_batches\n",
        "    config.Trainer.limit_val_batches=limit_val_batches\n",
        "    config.Trainer.limit_test_batches=limit_test_batches\n",
        "    config.Trainer.val_check_interval=val_check_interval\n",
        "\n",
        "    # strategy\n",
        "    strategy = {\n",
        "        # - deepspeed\n",
        "        # 'class': 'DeepSpeedStrategy',\n",
        "        # 'stage': 2\n",
        "\n",
        "        # - None\n",
        "        # 'class': None,\n",
        "\n",
        "        # - DP\n",
        "        # 'class': 'DataParallelStrategy',\n",
        "\n",
        "        # - DDP\n",
        "        # 'class': 'DDPStrategy',\n",
        "        # 'find_unused_parameter': True\n",
        "    }\n",
        "    config.Trainer.strategy = strategy\n",
        "\n",
        "    ####################################################################\n",
        "    #          Run the training task           #\n",
        "    ####################################################################\n",
        "    print('='*100)\n",
        "    print(Fore.BLUE+f\"Training task type: {task_type_value}\"+Style.RESET_ALL)\n",
        "    print(Fore.BLUE+f\"Dataset: {lmdb_dataset_path}\"+Style.RESET_ALL)\n",
        "    print(Fore.BLUE+f\"Base Model: {config.model.kwargs.config_path}\"+Style.RESET_ALL)\n",
        "    if continue_learning:\n",
        "      print(Fore.BLUE+f\"Existing model: {config.model.kwargs.lora_kwargs.config_list[0]['lora_config_path']}\"+Style.RESET_ALL)\n",
        "    print('='*100)\n",
        "    pprint.pprint(config)\n",
        "    print('='*100)\n",
        "\n",
        "    from saprot.scripts.training import finetune\n",
        "    finetune(config)\n",
        "\n",
        "    ####################################################################\n",
        "    #            Save the adapter            #\n",
        "    ####################################################################\n",
        "    def add_training_data_type_to_config(metadata_path, metadata):\n",
        "      with open(metadata_path, 'w') as file:\n",
        "          json.dump(metadata, file, indent=4)\n",
        "\n",
        "    metadata_path = Path(config.model.save_path) / \"metadata.json\"\n",
        "    add_training_data_type_to_config(metadata_path, metadata)\n",
        "\n",
        "    print(Fore.BLUE)\n",
        "    print(f\"Model is saved to \\\"{config.model.save_path}\\\" on Colab Server\")\n",
        "    print(Style.RESET_ALL)\n",
        "\n",
        "\n",
        "    adapter_zip = Path(config.model.save_path) / f\"{model_name}.zip\"\n",
        "    !cd $config.model.save_path && zip -r $adapter_zip \"adapter_config.json\" \"adapter_model.safetensors\" \"README.md\" \"metadata.json\"\n",
        "    # !cd $config.model.save_path && zip -r $adapter_zip \"adapter_config.json\" \"adapter_model.safetensors\" \"adapter_model.bin\" \"README.md\" \"metadata.json\"\n",
        "    print(\"Click to download the model to your local computer\")\n",
        "    if adapter_zip.exists():\n",
        "      file_download(adapter_zip)\n",
        "\n",
        "\n",
        "\n",
        "  # Set click events\n",
        "  task_type.observe(change_task_type, names='value')\n",
        "  model_type.observe(change_model_type, names='value')\n",
        "  data_src_type.observe(change_data_src_type, names='value')\n",
        "  advanced_setting.on_click(click_advanced_setting)\n",
        "  start_btn.on_click(start_training)\n",
        "\n",
        "  # Set default state\n",
        "  for item in [data_type_hint, data_type, data_split_hint, data_split_type]:\n",
        "    item.layout.display  = \"none\"\n",
        "\n",
        "  for item in [lora_hint, lora_r, lora_dropout_box, lora_alpha_box, trainer_hint, limit_train_batches_box, limit_val_batches_box, limit_test_batches_box,]:\n",
        "    item.layout.display = \"none\"\n",
        "\n",
        "  display(*items)\n",
        "\n",
        "\n",
        "######################################################################\n",
        "#          Choose prediction task            #\n",
        "######################################################################\n",
        "def choose_pred_task():\n",
        "  question = HTML(markdown.markdown(\"## ColabSaprot supports several prediction tasks, which one would you like to choose?\"))\n",
        "  normal_pred = Button(description='Protein property prediction', layout=Layout(width='500px', height='30px'))\n",
        "  zeroshot_pred = Button(description='Mutational effect prediction', layout=Layout(width='500px', height='30px'))\n",
        "  design_pred = Button(description='Protein sequence design', layout=Layout(width='500px', height='30px'))\n",
        "  repr_pred = Button(description='Obtain protein-level embeddings', layout=Layout(width='500px', height='30px'))\n",
        "  back_btn = Button(description='Go back', layout=Layout(width='500px', height='30px'))\n",
        "\n",
        "  items = [\n",
        "      question,\n",
        "      normal_pred,\n",
        "      zeroshot_pred,\n",
        "      design_pred,\n",
        "      repr_pred,\n",
        "      # back_btn\n",
        "      ]\n",
        "\n",
        "  # Set click events\n",
        "  zeroshot_pred.on_click(partial(jump, next=start_mut_pred))\n",
        "  back_btn.on_click(partial(jump, next=train_or_pred))\n",
        "\n",
        "  display(*items)\n",
        "\n",
        "\n",
        "######################################################################\n",
        "#       Start mutational effect predction         #\n",
        "######################################################################\n",
        "def start_mut_pred():\n",
        "  question = HTML(markdown.markdown(\"## Please choose the mutation task you want to perform\"))\n",
        "  single_btn = Button(description='Single-site or Multi-site mutagenesis', layout=Layout(width='500px', height='30px'))\n",
        "  all_btn = Button(description='Single-site saturation mutagenesis', layout=Layout(width='500px', height='30px'))\n",
        "\n",
        "  items = [\n",
        "      question,\n",
        "      single_btn,\n",
        "      all_btn,\n",
        "  ]\n",
        "\n",
        "  # Set click events\n",
        "  single_btn.on_click(partial(jump, next=single_mut_pred))\n",
        "  all_btn.on_click(partial(jump, next=saturatopm_mut_pred))\n",
        "\n",
        "  display(*items)\n",
        "\n",
        "\n",
        "######################################################################\n",
        "#      Single-site or Multi-site mutagenesis        #\n",
        "######################################################################\n",
        "def single_mut_pred():\n",
        "  hint = HTML(markdown.markdown(\"# Single-site or Multi-site mutagenesis\\n\\n## Please upload the protein structure\\n If you only have protein sequence, you could use [ColabFold](https://colab.research.google.com/github/sokrypton/ColabFold/blob/main/AlphaFold2.ipynb) to predict its structure and upload it here.\"))\n",
        "\n",
        "  chain_hint = HTML(markdown.markdown(\"Chain (to be extracted from the structure):\"))\n",
        "  input_chain = ipywidgets.Text(value=\"A\",placeholder=f'Enter the name of chain here', layout=Layout(width='500px', height='30px'))\n",
        "  upload_hint = HTML(markdown.markdown(\"Upload the protein structure:\"))\n",
        "  upload_btn = ipywidgets.FileUpload(accept='',multiple=False)\n",
        "\n",
        "  items = [\n",
        "      hint,\n",
        "      chain_hint,\n",
        "      input_chain,\n",
        "      upload_hint,\n",
        "      upload_btn,\n",
        "  ]\n",
        "\n",
        "  # Set click events\n",
        "  def on_upload_file(change):\n",
        "    upload_path = Path(STRUCTURE_HOME)\n",
        "    upload_path.mkdir(parents=True, exist_ok=True)\n",
        "    basepath = Path().resolve()\n",
        "\n",
        "    # Write to specific path\n",
        "    name = list(upload_btn.value.keys())[0]\n",
        "    content = upload_btn.value[name][\"content\"]\n",
        "    save_path = upload_path / name\n",
        "    with open(save_path, \"wb\") as wb:\n",
        "      wb.write(content)\n",
        "\n",
        "    chain = input_chain.value\n",
        "    print(chain)\n",
        "    protein_list = [(name, chain)]\n",
        "    mprs = MultipleProcessRunnerSimplifier(protein_list, pdb2sequence, n_process=2, return_results=True, verbose=False)\n",
        "    seqs = mprs.run()\n",
        "\n",
        "    sa_seq = seqs[0].split(\"\\t\")[-1]\n",
        "    aa_seq = sa_seq[0::2]\n",
        "    struc_seq = sa_seq[1::2].replace(\"#\", \"\\#\")\n",
        "\n",
        "    seq_info = HTML(markdown.markdown(f\"**{name}**\\n\\n**3Di sequence:**\\n\\n{struc_seq}\\n\\n**Amino acid sequence:**\\n\\n{aa_seq}\"))\n",
        "\n",
        "    # Mutation information box\n",
        "    mut_hint = HTML(markdown.markdown(\n",
        "        \"**Please input the mutation information:**\\n\\n\"\n",
        "        \"**For single-site mutation, e.g. M1E means mutating the amino acid M to E at first position. For multi-site mutation, you are expected to separate each position by ':', e.g. M1E:P2V.**\"\n",
        "        )\n",
        "    )\n",
        "    input_mut = ipywidgets.Text(placeholder='Enter Mutation Information here', layout=Layout(width='1000px', height='30px'))\n",
        "    submit_btn = Button(description='Calculate mutation score', layout=Layout(width='250px', height='30px'))\n",
        "\n",
        "\n",
        "    new_items = items + [seq_info, mut_hint, input_mut, submit_btn]\n",
        "\n",
        "    # Click to calculate mutation score\n",
        "    def calc_mut_score(button):\n",
        "      mut_info = input_mut.value\n",
        "      print(f\"Predict the mutation score for {mut_info}...\")\n",
        "      score = predict_mut(sa_seq, mut_info)\n",
        "      score_hint = HTML(markdown.markdown(f\"The score for {mut_info} is <font color=red>{score.item()}</font>. A positive score means the mutation is better than the wild type from evolution perspective (the larger the better).\"))\n",
        "      new_items = items + [seq_info, mut_hint, input_mut, submit_btn, score_hint]\n",
        "      custom_display(*new_items)\n",
        "\n",
        "    submit_btn.on_click(calc_mut_score)\n",
        "    custom_display(*new_items)\n",
        "\n",
        "  upload_btn.observe(on_upload_file, names='value')\n",
        "\n",
        "  display(*items)\n",
        "\n",
        "\n",
        "######################################################################\n",
        "#       Single-site saturation mutagenesis          #\n",
        "######################################################################\n",
        "def saturatopm_mut_pred():\n",
        "  hint = HTML(markdown.markdown(\"# Single-site saturation mutagenesis\\n\\n## Please upload the protein structure\\n If you only have protein sequence, you could use [ColabFold](https://colab.research.google.com/github/sokrypton/ColabFold/blob/main/AlphaFold2.ipynb) to predict its structure and upload it here.\"))\n",
        "\n",
        "  chain_hint = HTML(markdown.markdown(\"Chain (to be extracted from the structure):\"))\n",
        "  input_chain = ipywidgets.Text(value=\"A\",placeholder=f'Enter the name of chain here', layout=Layout(width='500px', height='30px'))\n",
        "  upload_hint = HTML(markdown.markdown(\"Upload the protein structure:\"))\n",
        "  upload_btn = ipywidgets.FileUpload(accept='',multiple=False)\n",
        "\n",
        "  items = [\n",
        "      hint,\n",
        "      chain_hint,\n",
        "      input_chain,\n",
        "      upload_hint,\n",
        "      upload_btn,\n",
        "  ]\n",
        "\n",
        "  # Set click events\n",
        "  def on_upload_file(change):\n",
        "    upload_path = Path(STRUCTURE_HOME)\n",
        "    upload_path.mkdir(parents=True, exist_ok=True)\n",
        "    basepath = Path().resolve()\n",
        "\n",
        "    # Write to specific path\n",
        "    name = list(upload_btn.value.keys())[0]\n",
        "    content = upload_btn.value[name][\"content\"]\n",
        "    save_path = upload_path / name\n",
        "    with open(save_path, \"wb\") as wb:\n",
        "      wb.write(content)\n",
        "\n",
        "    chain = input_chain.value\n",
        "    print(chain)\n",
        "    protein_list = [(name, chain)]\n",
        "    mprs = MultipleProcessRunnerSimplifier(protein_list, pdb2sequence, n_process=2, return_results=True, verbose=False)\n",
        "    seqs = mprs.run()\n",
        "\n",
        "    sa_seq = seqs[0].split(\"\\t\")[-1]\n",
        "    aa_seq = sa_seq[0::2]\n",
        "    struc_seq = sa_seq[1::2].replace(\"#\", \"\\#\")\n",
        "\n",
        "    seq_info = HTML(markdown.markdown(f\"**{name}**\\n\\n**3Di sequence:**\\n\\n{struc_seq}\\n\\n**Amino acid sequence:**\\n\\n{aa_seq}\"))\n",
        "    submit_btn = Button(description='Calculate mutation score for all single-site mutations', layout=Layout(width='500px', height='30px'))\n",
        "\n",
        "\n",
        "    new_items = items + [seq_info, submit_btn]\n",
        "\n",
        "    # Click to calculate mutation score\n",
        "    def calc_mut_score(button):\n",
        "      print(f\"Predicting mutation scores for all single-site mutations...\")\n",
        "      save_path = predict_all_mut(sa_seq)\n",
        "      score_hint = HTML(markdown.markdown(f\"The result has been saved to {save_path}. You can click to download the file.\\n\\n\"\n",
        "      \"**For mutation score, A positive score means the mutation is better than the wild type from evolution perspective (the larger the better).**\"))\n",
        "      download_btn = generate_download_btn(save_path)\n",
        "\n",
        "      new_items = items + [seq_info, submit_btn, score_hint, download_btn]\n",
        "      custom_display(*new_items)\n",
        "\n",
        "    submit_btn.on_click(calc_mut_score)\n",
        "    custom_display(*new_items)\n",
        "\n",
        "  upload_btn.observe(on_upload_file, names='value')\n",
        "\n",
        "  display(*items)\n",
        "\n",
        "\n",
        "train_or_pred()\n",
        "display(HINT)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 181,
          "referenced_widgets": [
            "e9acbb732b244f549a20473a27945043",
            "ded12d17c69841acbb4a9cc94e4281e9",
            "62ed23d0ef564e5e96ea089c6185b8d7",
            "326e7f7cb2da4f8287e31dacde987c38",
            "517461bb890441549ac514632f85e0e8",
            "6eccf22bf3ce418aafe68803aa15bb54",
            "31b1215d2aa64e0aaeec12cf5adba750",
            "d4b03491384b4b1d854c05a9aa1e427c",
            "9f5c2c18d8b046299f1a59cb5f20e93f",
            "77b44d6bb79d42229f01c2078cb8a58b",
            "ac513893c04945a59f3338269b07c62d",
            "5d2a9091f7c24f4b95bdd3c2c2cf0c70"
          ]
        },
        "id": "8K-R9TBloWGc",
        "outputId": "d11c91f6-957a-4564-f23c-b03307dfba22",
        "cellView": "form"
      },
      "id": "8K-R9TBloWGc",
      "execution_count": 74,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "HTML(value='<h2>Do you want to train your own model or you only want to use existing models to make prediction"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "e9acbb732b244f549a20473a27945043"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "Button(description='I want to train my own model', layout=Layout(height='30px', width='400px'), style=ButtonSt"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "326e7f7cb2da4f8287e31dacde987c38"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "Button(description='I want to use existing models to make prediction', layout=Layout(height='30px', width='400"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "31b1215d2aa64e0aaeec12cf5adba750"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "HTML(value=\"<p><font color=red>Note: At any time you can click the run button <img src='https://github.com/wes"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "77b44d6bb79d42229f01c2078cb8a58b"
            }
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "title = HTML(markdown.markdown(\"## Protein sequence design\"))\n",
        "WIDTH = \"500px\"\n",
        "HEIGHT= \"30px\"\n",
        "\n",
        "task_hint = HTML(markdown.markdown(\"### Please upload the structure backbone:\"))\n",
        "chain_hint = HTML(markdown.markdown(\"Chain (to be extracted from the structure):\"))\n",
        "input_chain = ipywidgets.Text(value=\"A\",placeholder=f'Enter the name of chain here', layout=Layout(width=WIDTH, height=HEIGHT))\n",
        "upload_hint = HTML(markdown.markdown(\"Upload the protein structure:\"))\n",
        "upload_btn = ipywidgets.FileUpload(accept='',multiple=False)\n",
        "\n",
        "items = [\n",
        "    title,\n",
        "    task_hint,\n",
        "    chain_hint,\n",
        "    input_chain,\n",
        "    upload_hint,\n",
        "    upload_btn,\n",
        "]\n",
        "\n",
        "# Set click events\n",
        "def parse_structure(change):\n",
        "  upload_path = Path(STRUCTURE_HOME)\n",
        "  upload_path.mkdir(parents=True, exist_ok=True)\n",
        "  basepath = Path().resolve()\n",
        "\n",
        "  # Write to specific path\n",
        "  name = list(upload_btn.value.keys())[0]\n",
        "  content = upload_btn.value[name][\"content\"]\n",
        "  save_path = upload_path / name\n",
        "  with open(save_path, \"wb\") as wb:\n",
        "    wb.write(content)\n",
        "\n",
        "  chain = input_chain.value\n",
        "  protein_list = [(name, chain)]\n",
        "  mprs = MultipleProcessRunnerSimplifier(protein_list, pdb2sequence, n_process=2, return_results=True, verbose=False)\n",
        "  seqs = mprs.run()\n",
        "\n",
        "  sa_seq = seqs[0].split(\"\\t\")[-1]\n",
        "  aa_seq = sa_seq[0::2]\n",
        "  struc_seq = sa_seq[1::2].replace(\"#\", \"\\#\")\n",
        "\n",
        "  seq_info = HTML(markdown.markdown(f\"**3Di sequence:**\\n\\n{struc_seq}\\n\\n**Backbone visualization of {name} ({len(struc_seq)} amino acids):**\"))\n",
        "\n",
        "  # Sampling config\n",
        "  sampling_method_box = ipywidgets.Dropdown(\n",
        "            options=['argmax', 'muiltinomial'],\n",
        "            value='argmax',\n",
        "            description='Sampling methodology:',\n",
        "            disabled=False,\n",
        "            layout=Layout(width=WIDTH, height=HEIGHT),\n",
        "            style={'description_width': 'initial'},\n",
        "          )\n",
        "  sampling_num = ipywidgets.BoundedIntText(\n",
        "        value=1,\n",
        "        min=1,\n",
        "        max=100,\n",
        "        step=1,\n",
        "        description='Sampling num:',\n",
        "        disabled=False,\n",
        "        style={'description_width': 'initial'},\n",
        "        layout=Layout(width=WIDTH, height=HEIGHT)\n",
        "      )\n",
        "  save_name = ipywidgets.Text(\n",
        "        value=\"predicted_seq\",\n",
        "        description='Save name:',\n",
        "        disabled=False,\n",
        "        style={'description_width': 'initial'},\n",
        "        layout=Layout(width=WIDTH, height=HEIGHT)\n",
        "      )\n",
        "\n",
        "\n",
        "  pred_hint = HTML(markdown.markdown(f\"### {'-'*75}\"))\n",
        "  pred_btn = Button(description='Predict protein sequence', layout=Layout(width=WIDTH, height=HEIGHT), button_style=\"info\")\n",
        "\n",
        "  def change_sampling_method(change):\n",
        "    sampling_method = change[\"new\"]\n",
        "    if sampling_method == \"muiltinomial\":\n",
        "      sampling_num.layout.display = None\n",
        "    else:\n",
        "      sampling_num.layout.display = \"none\"\n",
        "\n",
        "  def predict(button):\n",
        "    print(\"Predicting sequence...\")\n",
        "\n",
        "    struc_seq = sa_seq[1::2]\n",
        "    masked_aa_seq = \"#\" * len(struc_seq)\n",
        "    methods = sampling_method_box.value\n",
        "    num_samples = sampling_num.value\n",
        "\n",
        "    pred_seqs = inverse_folding(masked_aa_seq, struc_seq, methods, num_samples)\n",
        "    print(pred_seqs)\n",
        "\n",
        "    save_path = f\"{root_dir}/SaprotHub/output/{save_name.value}.fasta\"\n",
        "    with open(save_path, \"w\") as w:\n",
        "      for i, aa_seq in enumerate(pred_seqs):\n",
        "        print(aa_seq)\n",
        "        w.write(f\">predicted_seq_{i}\\n{aa_seq}\\n\\n\")\n",
        "\n",
        "    file_download(save_path)\n",
        "\n",
        "  # Set click events\n",
        "  sampling_method_box.observe(change_sampling_method, names=\"value\")\n",
        "  pred_btn.on_click(predict)\n",
        "\n",
        "  # Set default state\n",
        "  sampling_num.layout.display = \"none\"\n",
        "\n",
        "  # Update the display\n",
        "  clear_output()\n",
        "  display(*items)\n",
        "  display(seq_info)\n",
        "  show_pdb(save_path, color=\"chain\").show()\n",
        "  display(sampling_method_box, sampling_num, save_name, pred_hint, pred_btn, HINT)\n",
        "\n",
        "upload_btn.observe(parse_structure, names='value')\n",
        "display(*items)"
      ],
      "metadata": {
        "id": "0OvkkQ5Ai38I",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000,
          "referenced_widgets": [
            "a9723407cf1f40ad9f9ddc6472cd5725",
            "569e239ab6bb40b591d356977d8c4f4b",
            "1093f8bd744746b2a06238e85741860d",
            "175b30b1575440b3b59099e84d2e0f13",
            "9999c853c5384d24a80347132be628a3",
            "ffb3964e5dcd4c5dac74bd6e26f55512",
            "2c1676e71126401fb39d189d9f8ab49e",
            "dc38bc3e6cc64324a147ed689ce71cd7",
            "d99808b256714956bf9d06c177d54ca8",
            "f084fff8fb7b4b5898a2d1634b3a93a9",
            "b6752eaaae2941888af318478d2c631d",
            "d634a2c416694a58b7e80eba8d7bc622",
            "479015be7f2348f383ee6b89e9d47235",
            "47b9e1fd9e91497296fe4ffb9cc4eff2",
            "20465a9b162b475b8bdae2c41b2afd00",
            "df1ef4d191984d08aa998bce741c2b06",
            "ae7ec3398a7a409c811deb7cd651e02e",
            "0ee820d318304ea0985ac9e4c685a934",
            "c57b814c8c5f4bcb853f1d0acd08fb76",
            "6715e54762c04feb848b8a9e856e5f96",
            "be6a0ff96fad472297de0c2b74dd3136",
            "6947b47837da4ccba98baa1385f4d3d7",
            "42cf804e522b4cbfb5d08031d98221f9",
            "70397e15a0154fbdb8b6099adeb7f703",
            "bcee82164a8b4a07a197ac405c3eac1f",
            "60ae6a0b0b384f84992ed3015a88a711",
            "b23e6852167f40d681fe2978573541ec",
            "191833433bae4921b9e70bb8c07ef71c",
            "8a45ad0aefa74b269a1c219a85fdf815",
            "92226c261557406fb560cf70e9f4465f",
            "7010fbdcef0b4981a83185a5cdb777a2",
            "5be7f153467c43a4b3c9f04cdd46abcd",
            "27bca44291ae48498199a4bec2736665",
            "2d2fb545e57048d1a8b26d16167531ef",
            "18a9e1774f3147a4a8628625278cebc1",
            "c31819dac666410c90f97b98deeed15f",
            "77b44d6bb79d42229f01c2078cb8a58b",
            "ac513893c04945a59f3338269b07c62d",
            "5d2a9091f7c24f4b95bdd3c2c2cf0c70",
            "5a8c37150ef54eaabdc6ba0c797907d6",
            "87fb0778cd97466a95bbda0ac26003bb",
            "ad8f2a36a7e14ecba5b7c670f7e2c3e7"
          ]
        },
        "outputId": "d2c8a4a5-3eba-4a33-a639-aa6962188833"
      },
      "id": "0OvkkQ5Ai38I",
      "execution_count": 78,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "HTML(value='<h2>Protein sequence design</h2>')"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "a9723407cf1f40ad9f9ddc6472cd5725"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "HTML(value='<h3>Please upload the structure backbone:</h3>')"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "175b30b1575440b3b59099e84d2e0f13"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "HTML(value='<p>Chain (to be extracted from the structure):</p>')"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "2c1676e71126401fb39d189d9f8ab49e"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "Text(value='A', layout=Layout(height='30px', width='500px'), placeholder='Enter the name of chain here')"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "f084fff8fb7b4b5898a2d1634b3a93a9"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "HTML(value='<p>Upload the protein structure:</p>')"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "479015be7f2348f383ee6b89e9d47235"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "FileUpload(value={'LCC.pdb': {'metadata': {'name': 'LCC.pdb', 'type': '', 'size': 217323, 'lastModified': 1727"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "df1ef4d191984d08aa998bce741c2b06"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "HTML(value='<p><strong>3Di sequence:</strong></p>\\n<p>dfplddadlddlvqlldqadfdwdkdwfcqvvddlfrtfmkiagppdqeaaeeeee"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "c57b814c8c5f4bcb853f1d0acd08fb76"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "application/3dmoljs_load.v0": "<div id=\"3dmolviewer_17290926055486805\"  style=\"position: relative; width: 640px; height: 480px;\">\n        <p id=\"3dmolwarning_17290926055486805\" style=\"background-color:#ffcccc;color:black\">3Dmol.js failed to load for some reason.  Please check your browser console for error messages.<br></p>\n        </div>\n<script>\n\nvar loadScriptAsync = function(uri){\n  return new Promise((resolve, reject) => {\n    //this is to ignore the existence of requirejs amd\n    var savedexports, savedmodule;\n    if (typeof exports !== 'undefined') savedexports = exports;\n    else exports = {}\n    if (typeof module !== 'undefined') savedmodule = module;\n    else module = {}\n\n    var tag = document.createElement('script');\n    tag.src = uri;\n    tag.async = true;\n    tag.onload = () => {\n        exports = savedexports;\n        module = savedmodule;\n        resolve();\n    };\n  var firstScriptTag = document.getElementsByTagName('script')[0];\n  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);\n});\n};\n\nif(typeof $3Dmolpromise === 'undefined') {\n$3Dmolpromise = null;\n  $3Dmolpromise = loadScriptAsync('https://3dmol.org/build/3Dmol.js');\n}\n\nvar viewer_17290926055486805 = null;\nvar warn = document.getElementById(\"3dmolwarning_17290926055486805\");\nif(warn) {\n    warn.parentNode.removeChild(warn);\n}\n$3Dmolpromise.then(function() {\nviewer_17290926055486805 = $3Dmol.createViewer(document.getElementById(\"3dmolviewer_17290926055486805\"),{backgroundColor:\"white\"});\nviewer_17290926055486805.zoomTo();\n\tviewer_17290926055486805.addModel(\"HEADER    HYDROLASE                               23-MAR-12   4EB0              \\nTITLE     CRYSTAL STRUCTURE OF LEAF-BRANCH COMPOST BACTERIAL CUTINASE HOMOLOG   \\nCOMPND    MOL_ID: 1;                                                            \\nCOMPND   2 MOLECULE: LCC;                                                       \\nCOMPND   3 CHAIN: A;                                                            \\nCOMPND   4 FRAGMENT: UNP RESIDUES 36-293;                                       \\nCOMPND   5 EC: 3.1.1.74;                                                        \\nCOMPND   6 ENGINEERED: YES                                                      \\nSOURCE    MOL_ID: 1;                                                            \\nSOURCE   2 ORGANISM_SCIENTIFIC: UNCULTURED BACTERIUM;                           \\nSOURCE   3 ORGANISM_TAXID: 77133;                                               \\nSOURCE   4 EXPRESSION_SYSTEM: ESCHERICHIA COLI;                                 \\nSOURCE   5 EXPRESSION_SYSTEM_TAXID: 562;                                        \\nSOURCE   6 EXPRESSION_SYSTEM_STRAIN: BL21-CODONPLUS(DE3)-RP;                    \\nSOURCE   7 EXPRESSION_SYSTEM_VECTOR_TYPE: PLASMID;                              \\nSOURCE   8 EXPRESSION_SYSTEM_PLASMID: PET25B                                    \\nKEYWDS    HYDROLASE, SERINE ESTERASE, CUTINASE HOMOLOG, PET DEGRADATION,        \\nKEYWDS   2 METAGENOME                                                           \\nEXPDTA    X-RAY DIFFRACTION                                                     \\nAUTHOR    S.SULAIMAN,D.J.YOU,K.EIKO,Y.KOGA,S.KANAYA                             \\nREVDAT   3   08-NOV-23 4EB0    1       REMARK                                   \\nREVDAT   2   15-NOV-17 4EB0    1       REMARK                                   \\nREVDAT   1   27-MAR-13 4EB0    0                                                \\nJRNL        AUTH   S.SULAIMAN,D.J.YOU,K.EIKO,Y.KOGA,S.KANAYA                    \\nJRNL        TITL   CRYSTAL STRUCTURE OF LEAF-BRANCH COMPOST BACTERIAL CUTINASE  \\nJRNL        TITL 2 HOMOLOG                                                      \\nJRNL        REF    TO BE PUBLISHED                                              \\nJRNL        REFN                                                                \\nREMARK   2                                                                      \\nREMARK   2 RESOLUTION.    1.50 ANGSTROMS.                                       \\nREMARK   3                                                                      \\nREMARK   3 REFINEMENT.                                                          \\nREMARK   3   PROGRAM     : REFMAC REFMAC_5.5.0109                               \\nREMARK   3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,              \\nREMARK   3               : NICHOLLS,WINN,LONG,VAGIN                             \\nREMARK   3                                                                      \\nREMARK   3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD                            \\nREMARK   3                                                                      \\nREMARK   3  DATA USED IN REFINEMENT.                                            \\nREMARK   3   RESOLUTION RANGE HIGH (ANGSTROMS) : 1.50                           \\nREMARK   3   RESOLUTION RANGE LOW  (ANGSTROMS) : 50.00                          \\nREMARK   3   DATA CUTOFF            (SIGMA(F)) : NULL                           \\nREMARK   3   COMPLETENESS FOR RANGE        (%) : 99.6                           \\nREMARK   3   NUMBER OF REFLECTIONS             : 32818                          \\nREMARK   3                                                                      \\nREMARK   3  FIT TO DATA USED IN REFINEMENT.                                     \\nREMARK   3   CROSS-VALIDATION METHOD          : THROUGHOUT                      \\nREMARK   3   FREE R VALUE TEST SET SELECTION  : RANDOM                          \\nREMARK   3   R VALUE     (WORKING + TEST SET) : 0.158                           \\nREMARK   3   R VALUE            (WORKING SET) : 0.157                           \\nREMARK   3   FREE R VALUE                     : 0.186                           \\nREMARK   3   FREE R VALUE TEST SET SIZE   (%) : 5.000                           \\nREMARK   3   FREE R VALUE TEST SET COUNT      : 1741                            \\nREMARK   3                                                                      \\nREMARK   3  FIT IN THE HIGHEST RESOLUTION BIN.                                  \\nREMARK   3   TOTAL NUMBER OF BINS USED           : 20                           \\nREMARK   3   BIN RESOLUTION RANGE HIGH       (A) : 1.50                         \\nREMARK   3   BIN RESOLUTION RANGE LOW        (A) : 1.54                         \\nREMARK   3   REFLECTION IN BIN     (WORKING SET) : 2374                         \\nREMARK   3   BIN COMPLETENESS (WORKING+TEST) (%) : 99.64                        \\nREMARK   3   BIN R VALUE           (WORKING SET) : 0.1880                       \\nREMARK   3   BIN FREE R VALUE SET COUNT          : 127                          \\nREMARK   3   BIN FREE R VALUE                    : 0.2500                       \\nREMARK   3                                                                      \\nREMARK   3  NUMBER OF NON-HYDROGEN ATOMS USED IN REFINEMENT.                    \\nREMARK   3   PROTEIN ATOMS            : 1962                                    \\nREMARK   3   NUCLEIC ACID ATOMS       : 0                                       \\nREMARK   3   HETEROGEN ATOMS          : 12                                      \\nREMARK   3   SOLVENT ATOMS            : 315                                     \\nREMARK   3                                                                      \\nREMARK   3  B VALUES.                                                           \\nREMARK   3   FROM WILSON PLOT           (A**2) : NULL                           \\nREMARK   3   MEAN B VALUE      (OVERALL, A**2) : 17.17                          \\nREMARK   3   OVERALL ANISOTROPIC B VALUE.                                       \\nREMARK   3    B11 (A**2) : -0.67000                                             \\nREMARK   3    B22 (A**2) : -0.04000                                             \\nREMARK   3    B33 (A**2) : 0.70000                                              \\nREMARK   3    B12 (A**2) : 0.00000                                              \\nREMARK   3    B13 (A**2) : 0.00000                                              \\nREMARK   3    B23 (A**2) : 0.00000                                              \\nREMARK   3                                                                      \\nREMARK   3  ESTIMATED OVERALL COORDINATE ERROR.                                 \\nREMARK   3   ESU BASED ON R VALUE                            (A): 0.074         \\nREMARK   3   ESU BASED ON FREE R VALUE                       (A): 0.075         \\nREMARK   3   ESU BASED ON MAXIMUM LIKELIHOOD                 (A): 0.043         \\nREMARK   3   ESU FOR B VALUES BASED ON MAXIMUM LIKELIHOOD (A**2): 1.126         \\nREMARK   3                                                                      \\nREMARK   3 CORRELATION COEFFICIENTS.                                            \\nREMARK   3   CORRELATION COEFFICIENT FO-FC      : 0.971                         \\nREMARK   3   CORRELATION COEFFICIENT FO-FC FREE : 0.959                         \\nREMARK   3                                                                      \\nREMARK   3  RMS DEVIATIONS FROM IDEAL VALUES        COUNT    RMS    WEIGHT      \\nREMARK   3   BOND LENGTHS REFINED ATOMS        (A):  2023 ; 0.007 ; 0.022       \\nREMARK   3   BOND LENGTHS OTHERS               (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   BOND ANGLES REFINED ATOMS   (DEGREES):  2764 ; 1.000 ; 1.941       \\nREMARK   3   BOND ANGLES OTHERS          (DEGREES):  NULL ;  NULL ;  NULL       \\nREMARK   3   TORSION ANGLES, PERIOD 1    (DEGREES):   257 ; 7.429 ; 5.000       \\nREMARK   3   TORSION ANGLES, PERIOD 2    (DEGREES):    87 ;32.304 ;22.759       \\nREMARK   3   TORSION ANGLES, PERIOD 3    (DEGREES):   283 ;10.840 ;15.000       \\nREMARK   3   TORSION ANGLES, PERIOD 4    (DEGREES):    16 ;12.062 ;15.000       \\nREMARK   3   CHIRAL-CENTER RESTRAINTS       (A**3):   310 ; 0.084 ; 0.200       \\nREMARK   3   GENERAL PLANES REFINED ATOMS      (A):  1578 ; 0.020 ; 0.021       \\nREMARK   3   GENERAL PLANES OTHERS             (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   NON-BONDED CONTACTS REFINED ATOMS (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   NON-BONDED CONTACTS OTHERS        (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   NON-BONDED TORSION REFINED ATOMS  (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   NON-BONDED TORSION OTHERS         (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   H-BOND (X...Y) REFINED ATOMS      (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   H-BOND (X...Y) OTHERS             (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   POTENTIAL METAL-ION REFINED ATOMS (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   POTENTIAL METAL-ION OTHERS        (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   SYMMETRY VDW REFINED ATOMS        (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   SYMMETRY VDW OTHERS               (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   SYMMETRY H-BOND REFINED ATOMS     (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   SYMMETRY H-BOND OTHERS            (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   SYMMETRY METAL-ION REFINED ATOMS  (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   SYMMETRY METAL-ION OTHERS         (A):  NULL ;  NULL ;  NULL       \\nREMARK   3                                                                      \\nREMARK   3  ISOTROPIC THERMAL FACTOR RESTRAINTS.     COUNT   RMS    WEIGHT      \\nREMARK   3   MAIN-CHAIN BOND REFINED ATOMS  (A**2):  1298 ; 1.466 ; 1.500       \\nREMARK   3   MAIN-CHAIN BOND OTHER ATOMS    (A**2):  NULL ;  NULL ;  NULL       \\nREMARK   3   MAIN-CHAIN ANGLE REFINED ATOMS (A**2):  2088 ; 2.175 ; 2.000       \\nREMARK   3   MAIN-CHAIN ANGLE OTHER ATOMS   (A**2):  NULL ;  NULL ;  NULL       \\nREMARK   3   SIDE-CHAIN BOND REFINED ATOMS  (A**2):   725 ; 3.398 ; 3.000       \\nREMARK   3   SIDE-CHAIN BOND OTHER ATOMS    (A**2):  NULL ;  NULL ;  NULL       \\nREMARK   3   SIDE-CHAIN ANGLE REFINED ATOMS (A**2):   676 ; 4.458 ; 4.500       \\nREMARK   3   SIDE-CHAIN ANGLE OTHER ATOMS   (A**2):  NULL ;  NULL ;  NULL       \\nREMARK   3   LONG RANGE B REFINED ATOMS     (A**2):  NULL ;  NULL ;  NULL       \\nREMARK   3   LONG RANGE B OTHER ATOMS       (A**2):  NULL ;  NULL ;  NULL       \\nREMARK   3                                                                      \\nREMARK   3 ANISOTROPIC THERMAL FACTOR RESTRAINTS.    COUNT   RMS   WEIGHT       \\nREMARK   3   RIGID-BOND RESTRAINTS          (A**2):  NULL ;  NULL ;  NULL       \\nREMARK   3   SPHERICITY; FREE ATOMS         (A**2):  NULL ;  NULL ;  NULL       \\nREMARK   3   SPHERICITY; BONDED ATOMS       (A**2):  NULL ;  NULL ;  NULL       \\nREMARK   3                                                                      \\nREMARK   3  NCS RESTRAINTS STATISTICS                                           \\nREMARK   3   NUMBER OF DIFFERENT NCS GROUPS : NULL                              \\nREMARK   3                                                                      \\nREMARK   3  TLS DETAILS                                                         \\nREMARK   3   NUMBER OF TLS GROUPS  : NULL                                       \\nREMARK   3                                                                      \\nREMARK   3  BULK SOLVENT MODELLING.                                             \\nREMARK   3   METHOD USED : MASK                                                 \\nREMARK   3   PARAMETERS FOR MASK CALCULATION                                    \\nREMARK   3   VDW PROBE RADIUS   : 1.40                                          \\nREMARK   3   ION PROBE RADIUS   : 0.80                                          \\nREMARK   3   SHRINKAGE RADIUS   : 0.80                                          \\nREMARK   3                                                                      \\nREMARK   3  OTHER REFINEMENT REMARKS: HYDROGENS HAVE BEEN ADDED IN THE RIDING   \\nREMARK   3  POSITIONS                                                           \\nREMARK   4                                                                      \\nREMARK   4 4EB0 COMPLIES WITH FORMAT V. 3.30, 13-JUL-11                         \\nREMARK 100                                                                      \\nREMARK 100 THIS ENTRY HAS BEEN PROCESSED BY PDBJ ON 30-MAR-12.                  \\nREMARK 100 THE DEPOSITION ID IS D_1000071392.                                   \\nREMARK 200                                                                      \\nREMARK 200 EXPERIMENTAL DETAILS                                                 \\nREMARK 200  EXPERIMENT TYPE                : X-RAY DIFFRACTION                  \\nREMARK 200  DATE OF DATA COLLECTION        : 10-OCT-11                          \\nREMARK 200  TEMPERATURE           (KELVIN) : 100                                \\nREMARK 200  PH                             : 7.0                                \\nREMARK 200  NUMBER OF CRYSTALS USED        : 1                                  \\nREMARK 200                                                                      \\nREMARK 200  SYNCHROTRON              (Y/N) : Y                                  \\nREMARK 200  RADIATION SOURCE               : SPRING-8                           \\nREMARK 200  BEAMLINE                       : BL44XU                             \\nREMARK 200  X-RAY GENERATOR MODEL          : NULL                               \\nREMARK 200  MONOCHROMATIC OR LAUE    (M/L) : M                                  \\nREMARK 200  WAVELENGTH OR RANGE        (A) : 0.9                                \\nREMARK 200  MONOCHROMATOR                  : HORIZONTAL FOCUSING MIRROR         \\nREMARK 200  OPTICS                         : NULL                               \\nREMARK 200                                                                      \\nREMARK 200  DETECTOR TYPE                  : CCD                                \\nREMARK 200  DETECTOR MANUFACTURER          : BRUKER SMART 6500                  \\nREMARK 200  INTENSITY-INTEGRATION SOFTWARE : DENZO                              \\nREMARK 200  DATA SCALING SOFTWARE          : SCALEPACK                          \\nREMARK 200                                                                      \\nREMARK 200  NUMBER OF UNIQUE REFLECTIONS   : 35110                              \\nREMARK 200  RESOLUTION RANGE HIGH      (A) : 1.500                              \\nREMARK 200  RESOLUTION RANGE LOW       (A) : 50.000                             \\nREMARK 200  REJECTION CRITERIA  (SIGMA(I)) : -3.000                             \\nREMARK 200                                                                      \\nREMARK 200 OVERALL.                                                             \\nREMARK 200  COMPLETENESS FOR RANGE     (%) : 99.9                               \\nREMARK 200  DATA REDUNDANCY                : 13.70                              \\nREMARK 200  R MERGE                    (I) : 0.10600                            \\nREMARK 200  R SYM                      (I) : 0.10600                            \\nREMARK 200  <I/SIGMA(I)> FOR THE DATA SET  : 31.6570                            \\nREMARK 200                                                                      \\nREMARK 200 IN THE HIGHEST RESOLUTION SHELL.                                     \\nREMARK 200  HIGHEST RESOLUTION SHELL, RANGE HIGH (A) : 1.50                     \\nREMARK 200  HIGHEST RESOLUTION SHELL, RANGE LOW  (A) : 1.53                     \\nREMARK 200  COMPLETENESS FOR SHELL     (%) : 100.0                              \\nREMARK 200  DATA REDUNDANCY IN SHELL       : 12.70                              \\nREMARK 200  R MERGE FOR SHELL          (I) : 0.41900                            \\nREMARK 200  R SYM FOR SHELL            (I) : 0.41900                            \\nREMARK 200  <I/SIGMA(I)> FOR SHELL         : 0.419                              \\nREMARK 200                                                                      \\nREMARK 200 DIFFRACTION PROTOCOL: SINGLE WAVELENGTH                              \\nREMARK 200 METHOD USED TO DETERMINE THE STRUCTURE: MOLECULAR REPLACEMENT        \\nREMARK 200 SOFTWARE USED: MOLREP, DM 6.1                                        \\nREMARK 200 STARTING MODEL: 1JFR                                                 \\nREMARK 200                                                                      \\nREMARK 200 REMARK: NULL                                                         \\nREMARK 280                                                                      \\nREMARK 280 CRYSTAL                                                              \\nREMARK 280 SOLVENT CONTENT, VS   (%): 35.33                                     \\nREMARK 280 MATTHEWS COEFFICIENT, VM (ANGSTROMS**3/DA): 1.90                     \\nREMARK 280                                                                      \\nREMARK 280 CRYSTALLIZATION CONDITIONS: 20% (W/V) POLYETHYLENE GLYCOL (PEG)      \\nREMARK 280  3350, 200MM SODIUM THIOCYANATE, PH 7.0, VAPOR DIFFUSION, SITTING    \\nREMARK 280  DROP, TEMPERATURE 293K                                              \\nREMARK 290                                                                      \\nREMARK 290 CRYSTALLOGRAPHIC SYMMETRY                                            \\nREMARK 290 SYMMETRY OPERATORS FOR SPACE GROUP: P 21 21 21                       \\nREMARK 290                                                                      \\nREMARK 290      SYMOP   SYMMETRY                                                \\nREMARK 290     NNNMMM   OPERATOR                                                \\nREMARK 290       1555   X,Y,Z                                                   \\nREMARK 290       2555   -X+1/2,-Y,Z+1/2                                         \\nREMARK 290       3555   -X,Y+1/2,-Z+1/2                                         \\nREMARK 290       4555   X+1/2,-Y+1/2,-Z                                         \\nREMARK 290                                                                      \\nREMARK 290     WHERE NNN -> OPERATOR NUMBER                                     \\nREMARK 290           MMM -> TRANSLATION VECTOR                                  \\nREMARK 290                                                                      \\nREMARK 290 CRYSTALLOGRAPHIC SYMMETRY TRANSFORMATIONS                            \\nREMARK 290 THE FOLLOWING TRANSFORMATIONS OPERATE ON THE ATOM/HETATM             \\nREMARK 290 RECORDS IN THIS ENTRY TO PRODUCE CRYSTALLOGRAPHICALLY                \\nREMARK 290 RELATED MOLECULES.                                                   \\nREMARK 290   SMTRY1   1  1.000000  0.000000  0.000000        0.00000            \\nREMARK 290   SMTRY2   1  0.000000  1.000000  0.000000        0.00000            \\nREMARK 290   SMTRY3   1  0.000000  0.000000  1.000000        0.00000            \\nREMARK 290   SMTRY1   2 -1.000000  0.000000  0.000000       20.45550            \\nREMARK 290   SMTRY2   2  0.000000 -1.000000  0.000000        0.00000            \\nREMARK 290   SMTRY3   2  0.000000  0.000000  1.000000       36.36300            \\nREMARK 290   SMTRY1   3 -1.000000  0.000000  0.000000        0.00000            \\nREMARK 290   SMTRY2   3  0.000000  1.000000  0.000000       35.54200            \\nREMARK 290   SMTRY3   3  0.000000  0.000000 -1.000000       36.36300            \\nREMARK 290   SMTRY1   4  1.000000  0.000000  0.000000       20.45550            \\nREMARK 290   SMTRY2   4  0.000000 -1.000000  0.000000       35.54200            \\nREMARK 290   SMTRY3   4  0.000000  0.000000 -1.000000        0.00000            \\nREMARK 290                                                                      \\nREMARK 290 REMARK: NULL                                                         \\nREMARK 300                                                                      \\nREMARK 300 BIOMOLECULE: 1                                                       \\nREMARK 300 SEE REMARK 350 FOR THE AUTHOR PROVIDED AND/OR PROGRAM                \\nREMARK 300 GENERATED ASSEMBLY INFORMATION FOR THE STRUCTURE IN                  \\nREMARK 300 THIS ENTRY. THE REMARK MAY ALSO PROVIDE INFORMATION ON               \\nREMARK 300 BURIED SURFACE AREA.                                                 \\nREMARK 350                                                                      \\nREMARK 350 COORDINATES FOR A COMPLETE MULTIMER REPRESENTING THE KNOWN           \\nREMARK 350 BIOLOGICALLY SIGNIFICANT OLIGOMERIZATION STATE OF THE                \\nREMARK 350 MOLECULE CAN BE GENERATED BY APPLYING BIOMT TRANSFORMATIONS          \\nREMARK 350 GIVEN BELOW.  BOTH NON-CRYSTALLOGRAPHIC AND                          \\nREMARK 350 CRYSTALLOGRAPHIC OPERATIONS ARE GIVEN.                               \\nREMARK 350                                                                      \\nREMARK 350 BIOMOLECULE: 1                                                       \\nREMARK 350 AUTHOR DETERMINED BIOLOGICAL UNIT: MONOMERIC                         \\nREMARK 350 SOFTWARE DETERMINED QUATERNARY STRUCTURE: MONOMERIC                  \\nREMARK 350 SOFTWARE USED: PISA                                                  \\nREMARK 350 APPLY THE FOLLOWING TO CHAINS: A                                     \\nREMARK 350   BIOMT1   1  1.000000  0.000000  0.000000        0.00000            \\nREMARK 350   BIOMT2   1  0.000000  1.000000  0.000000        0.00000            \\nREMARK 350   BIOMT3   1  0.000000  0.000000  1.000000        0.00000            \\nREMARK 500                                                                      \\nREMARK 500 GEOMETRY AND STEREOCHEMISTRY                                         \\nREMARK 500 SUBTOPIC: TORSION ANGLES                                             \\nREMARK 500                                                                      \\nREMARK 500 TORSION ANGLES OUTSIDE THE EXPECTED RAMACHANDRAN REGIONS:            \\nREMARK 500 (M=MODEL NUMBER; RES=RESIDUE NAME; C=CHAIN IDENTIFIER;               \\nREMARK 500 SSEQ=SEQUENCE NUMBER; I=INSERTION CODE).                             \\nREMARK 500                                                                      \\nREMARK 500 STANDARD TABLE:                                                      \\nREMARK 500 FORMAT:(10X,I3,1X,A3,1X,A1,I4,A1,4X,F7.2,3X,F7.2)                    \\nREMARK 500                                                                      \\nREMARK 500 EXPECTED VALUES: GJ KLEYWEGT AND TA JONES (1996). PHI/PSI-           \\nREMARK 500 CHOLOGY: RAMACHANDRAN REVISITED. STRUCTURE 4, 1395 - 1400            \\nREMARK 500                                                                      \\nREMARK 500  M RES CSSEQI        PSI       PHI                                   \\nREMARK 500    SER A 165     -127.89     60.19                                   \\nREMARK 500    HIS A 218      -83.73   -126.70                                   \\nREMARK 500                                                                      \\nREMARK 500 REMARK: NULL                                                         \\nREMARK 500                                                                      \\nREMARK 500 GEOMETRY AND STEREOCHEMISTRY                                         \\nREMARK 500 SUBTOPIC: NON-CIS, NON-TRANS                                         \\nREMARK 500                                                                      \\nREMARK 500 THE FOLLOWING PEPTIDE BONDS DEVIATE SIGNIFICANTLY FROM BOTH          \\nREMARK 500 CIS AND TRANS CONFORMATION.  CIS BONDS, IF ANY, ARE LISTED           \\nREMARK 500 ON CISPEP RECORDS.  TRANS IS DEFINED AS 180 +/- 30 AND               \\nREMARK 500 CIS IS DEFINED AS 0 +/- 30 DEGREES.                                  \\nREMARK 500                                 MODEL     OMEGA                      \\nREMARK 500 ILE A  119     ASN A  120                  149.22                    \\nREMARK 500                                                                      \\nREMARK 500 REMARK: NULL                                                         \\nREMARK 800                                                                      \\nREMARK 800 SITE                                                                 \\nREMARK 800 SITE_IDENTIFIER: AC1                                                 \\nREMARK 800 EVIDENCE_CODE: SOFTWARE                                              \\nREMARK 800 SITE_DESCRIPTION: BINDING SITE FOR RESIDUE SCN A 301                 \\nREMARK 800                                                                      \\nREMARK 800 SITE_IDENTIFIER: AC2                                                 \\nREMARK 800 EVIDENCE_CODE: SOFTWARE                                              \\nREMARK 800 SITE_DESCRIPTION: BINDING SITE FOR RESIDUE SCN A 302                 \\nREMARK 800                                                                      \\nREMARK 800 SITE_IDENTIFIER: AC3                                                 \\nREMARK 800 EVIDENCE_CODE: SOFTWARE                                              \\nREMARK 800 SITE_DESCRIPTION: BINDING SITE FOR RESIDUE SCN A 303                 \\nREMARK 800                                                                      \\nREMARK 800 SITE_IDENTIFIER: AC4                                                 \\nREMARK 800 EVIDENCE_CODE: SOFTWARE                                              \\nREMARK 800 SITE_DESCRIPTION: BINDING SITE FOR RESIDUE SCN A 304                 \\nDBREF  4EB0 A   36   293  UNP    G9BY57   G9BY57_9BACT    36    293             \\nSEQRES   1 A  258  SER ASN PRO TYR GLN ARG GLY PRO ASN PRO THR ARG SER          \\nSEQRES   2 A  258  ALA LEU THR ALA ASP GLY PRO PHE SER VAL ALA THR TYR          \\nSEQRES   3 A  258  THR VAL SER ARG LEU SER VAL SER GLY PHE GLY GLY GLY          \\nSEQRES   4 A  258  VAL ILE TYR TYR PRO THR GLY THR SER LEU THR PHE GLY          \\nSEQRES   5 A  258  GLY ILE ALA MET SER PRO GLY TYR THR ALA ASP ALA SER          \\nSEQRES   6 A  258  SER LEU ALA TRP LEU GLY ARG ARG LEU ALA SER HIS GLY          \\nSEQRES   7 A  258  PHE VAL VAL LEU VAL ILE ASN THR ASN SER ARG PHE ASP          \\nSEQRES   8 A  258  TYR PRO ASP SER ARG ALA SER GLN LEU SER ALA ALA LEU          \\nSEQRES   9 A  258  ASN TYR LEU ARG THR SER SER PRO SER ALA VAL ARG ALA          \\nSEQRES  10 A  258  ARG LEU ASP ALA ASN ARG LEU ALA VAL ALA GLY HIS SER          \\nSEQRES  11 A  258  MET GLY GLY GLY GLY THR LEU ARG ILE ALA GLU GLN ASN          \\nSEQRES  12 A  258  PRO SER LEU LYS ALA ALA VAL PRO LEU THR PRO TRP HIS          \\nSEQRES  13 A  258  THR ASP LYS THR PHE ASN THR SER VAL PRO VAL LEU ILE          \\nSEQRES  14 A  258  VAL GLY ALA GLU ALA ASP THR VAL ALA PRO VAL SER GLN          \\nSEQRES  15 A  258  HIS ALA ILE PRO PHE TYR GLN ASN LEU PRO SER THR THR          \\nSEQRES  16 A  258  PRO LYS VAL TYR VAL GLU LEU ASP ASN ALA SER HIS PHE          \\nSEQRES  17 A  258  ALA PRO ASN SER ASN ASN ALA ALA ILE SER VAL TYR THR          \\nSEQRES  18 A  258  ILE SER TRP MET LYS LEU TRP VAL ASP ASN ASP THR ARG          \\nSEQRES  19 A  258  TYR ARG GLN PHE LEU CYS ASN VAL ASN ASP PRO ALA LEU          \\nSEQRES  20 A  258  SER ASP PHE ARG THR ASN ASN ARG HIS CYS GLN                  \\nHET    SCN  A 301       3                                                       \\nHET    SCN  A 302       3                                                       \\nHET    SCN  A 303       3                                                       \\nHET    SCN  A 304       3                                                       \\nHETNAM     SCN THIOCYANATE ION                                                  \\nFORMUL   2  SCN    4(C N S 1-)                                                  \\nFORMUL   6  HOH   *315(H2 O)                                                    \\nHELIX    1   1 ARG A   47  ALA A   52  5                                   6    \\nHELIX    2   2 SER A   64  VAL A   68  5                                   5    \\nHELIX    3   3 ASP A   98  SER A  101  5                                   4    \\nHELIX    4   4 LEU A  102  HIS A  112  1                                  11    \\nHELIX    5   5 TYR A  127  SER A  145  1                                  19    \\nHELIX    6   6 PRO A  147  ALA A  152  1                                   6    \\nHELIX    7   7 SER A  165  ASN A  178  1                                  14    \\nHELIX    8   8 HIS A  218  LEU A  226  1                                   9    \\nHELIX    9   9 PHE A  243  SER A  247  5                                   5    \\nHELIX   10  10 ASN A  249  ASP A  265  1                                  17    \\nHELIX   11  11 ASP A  267  LEU A  274  5                                   8    \\nHELIX   12  12 ASN A  289  GLN A  293  5                                   5    \\nSHEET    1   A 6 VAL A  58  VAL A  63  0                                        \\nSHEET    2   A 6 GLY A  74  PRO A  79 -1  O  ILE A  76   N  TYR A  61           \\nSHEET    3   A 6 VAL A 115  ILE A 119 -1  O  VAL A 116   N  TYR A  77           \\nSHEET    4   A 6 PHE A  86  SER A  92  1  N  ILE A  89   O  VAL A 115           \\nSHEET    5   A 6 LEU A 154  HIS A 164  1  O  ASP A 155   N  PHE A  86           \\nSHEET    6   A 6 ALA A 184  LEU A 187  1  O  LEU A 187   N  GLY A 163           \\nSHEET    1   B 3 VAL A 202  ALA A 207  0                                        \\nSHEET    2   B 3 LYS A 232  LEU A 237  1  O  VAL A 235   N  GLY A 206           \\nSHEET    3   B 3 LEU A 282  THR A 287 -1  O  ARG A 286   N  TYR A 234           \\nSSBOND   1 CYS A  275    CYS A  292                          1555   1555  2.05  \\nSITE     1 AC1  5 GLY A  42  TYR A 223  ARG A 286  HOH A 452                    \\nSITE     2 AC1  5 HOH A 527                                                     \\nSITE     1 AC2  5 THR A  96  ARG A 124  PRO A 231  ARG A 290                    \\nSITE     2 AC2  5 HOH A 469                                                     \\nSITE     1 AC3  4 SER A  48  SER A  69  THR A 144  SER A 145                    \\nSITE     1 AC4  8 ARG A  47  ASN A 276  VAL A 277  ASN A 278                    \\nSITE     2 AC4  8 ASP A 279  HOH A 458  HOH A 493  HOH A 544                    \\nCRYST1   40.911   71.084   72.726  90.00  90.00  90.00 P 21 21 21    4          \\nORIGX1      1.000000  0.000000  0.000000        0.00000                         \\nORIGX2      0.000000  1.000000  0.000000        0.00000                         \\nORIGX3      0.000000  0.000000  1.000000        0.00000                         \\nSCALE1      0.024443  0.000000  0.000000        0.00000                         \\nSCALE2      0.000000  0.014068  0.000000        0.00000                         \\nSCALE3      0.000000  0.000000  0.013750        0.00000                         \\nATOM      1  N   SER A  36      10.488 -16.119   4.405  1.00 32.20           N  \\nATOM      2  CA  SER A  36       9.773 -16.823   5.499  1.00 24.29           C  \\nATOM      3  C   SER A  36       8.613 -15.977   6.014  1.00 19.49           C  \\nATOM      4  O   SER A  36       7.724 -16.476   6.701  1.00 19.40           O  \\nATOM      5  CB  SER A  36      10.740 -17.189   6.637  1.00 24.44           C  \\nATOM      6  OG  SER A  36      11.354 -16.036   7.201  1.00 36.14           O  \\nATOM      7  N   ASN A  37       8.614 -14.697   5.655  1.00 18.01           N  \\nATOM      8  CA  ASN A  37       7.627 -13.754   6.184  1.00 15.55           C  \\nATOM      9  C   ASN A  37       6.744 -13.209   5.064  1.00 13.92           C  \\nATOM     10  O   ASN A  37       7.197 -12.391   4.265  1.00 16.39           O  \\nATOM     11  CB  ASN A  37       8.333 -12.581   6.879  1.00 14.38           C  \\nATOM     12  CG  ASN A  37       7.358 -11.487   7.295  1.00 13.17           C  \\nATOM     13  OD1 ASN A  37       6.154 -11.687   7.248  1.00 13.91           O  \\nATOM     14  ND2 ASN A  37       7.877 -10.393   7.830  1.00 13.94           N  \\nATOM     15  N   PRO A  38       5.481 -13.648   5.004  1.00 14.35           N  \\nATOM     16  CA  PRO A  38       4.616 -13.313   3.884  1.00 14.53           C  \\nATOM     17  C   PRO A  38       4.218 -11.835   3.856  1.00 13.55           C  \\nATOM     18  O   PRO A  38       3.690 -11.361   2.843  1.00 17.67           O  \\nATOM     19  CB  PRO A  38       3.382 -14.196   4.118  1.00 17.64           C  \\nATOM     20  CG  PRO A  38       3.363 -14.425   5.594  1.00 16.09           C  \\nATOM     21  CD  PRO A  38       4.815 -14.544   5.972  1.00 15.23           C  \\nATOM     22  N   TYR A  39       4.476 -11.118   4.942  1.00 14.55           N  \\nATOM     23  CA  TYR A  39       4.073  -9.707   5.002  1.00 15.93           C  \\nATOM     24  C   TYR A  39       5.250  -8.796   4.696  1.00 15.42           C  \\nATOM     25  O   TYR A  39       5.103  -7.576   4.692  1.00 13.44           O  \\nATOM     26  CB  TYR A  39       3.477  -9.346   6.366  1.00 14.24           C  \\nATOM     27  CG  TYR A  39       2.477 -10.339   6.913  1.00 15.58           C  \\nATOM     28  CD1 TYR A  39       1.335 -10.677   6.201  1.00 16.49           C  \\nATOM     29  CD2 TYR A  39       2.742 -11.017   8.099  1.00 16.84           C  \\nATOM     30  CE1 TYR A  39       0.456 -11.664   6.680  1.00 12.92           C  \\nATOM     31  CE2 TYR A  39       1.884 -11.975   8.590  1.00 16.72           C  \\nATOM     32  CZ  TYR A  39       0.771 -12.324   7.863  1.00 14.38           C  \\nATOM     33  OH  TYR A  39      -0.063 -13.296   8.385  1.00 19.31           O  \\nATOM     34  N   GLN A  40       6.429  -9.376   4.513  1.00 12.55           N  \\nATOM     35  CA  GLN A  40       7.629  -8.593   4.219  1.00 14.21           C  \\nATOM     36  C   GLN A  40       7.543  -7.885   2.857  1.00 12.42           C  \\nATOM     37  O   GLN A  40       7.192  -8.506   1.840  1.00 14.35           O  \\nATOM     38  CB  GLN A  40       8.856  -9.493   4.281  1.00 16.80           C  \\nATOM     39  CG  GLN A  40      10.160  -8.751   4.422  1.00 18.32           C  \\nATOM     40  CD  GLN A  40      11.342  -9.693   4.418  1.00 21.13           C  \\nATOM     41  OE1 GLN A  40      11.542 -10.439   5.373  1.00 22.91           O  \\nATOM     42  NE2 GLN A  40      11.980  -9.821   3.266  1.00 18.33           N  \\nATOM     43  N   ARG A  41       7.895  -6.599   2.835  1.00 12.41           N  \\nATOM     44  CA  ARG A  41       7.883  -5.811   1.599  1.00 13.22           C  \\nATOM     45  C   ARG A  41       9.157  -4.998   1.461  1.00 13.07           C  \\nATOM     46  O   ARG A  41       9.717  -4.542   2.460  1.00 12.24           O  \\nATOM     47  CB  ARG A  41       6.697  -4.840   1.571  1.00 14.84           C  \\nATOM     48  CG  ARG A  41       5.334  -5.472   1.854  1.00 13.01           C  \\nATOM     49  CD  ARG A  41       4.777  -6.221   0.640  1.00 12.90           C  \\nATOM     50  NE  ARG A  41       3.476  -6.835   0.931  1.00 14.33           N  \\nATOM     51  CZ  ARG A  41       3.314  -8.082   1.374  1.00 12.18           C  \\nATOM     52  NH1 ARG A  41       4.355  -8.917   1.468  1.00 13.50           N  \\nATOM     53  NH2 ARG A  41       2.111  -8.499   1.774  1.00 13.55           N  \\nATOM     54  N   GLY A  42       9.586  -4.791   0.217  1.00 13.78           N  \\nATOM     55  CA  GLY A  42      10.607  -3.797  -0.088  1.00 13.51           C  \\nATOM     56  C   GLY A  42      12.026  -4.290   0.170  1.00 15.04           C  \\nATOM     57  O   GLY A  42      12.266  -5.464   0.505  1.00 16.09           O  \\nATOM     58  N   PRO A  43      12.997  -3.376   0.049  1.00 13.88           N  \\nATOM     59  CA  PRO A  43      14.410  -3.745   0.078  1.00 14.51           C  \\nATOM     60  C   PRO A  43      14.880  -4.100   1.485  1.00 14.74           C  \\nATOM     61  O   PRO A  43      14.275  -3.693   2.480  1.00 14.03           O  \\nATOM     62  CB  PRO A  43      15.107  -2.458  -0.387  1.00 13.15           C  \\nATOM     63  CG  PRO A  43      14.173  -1.377   0.078  1.00 12.86           C  \\nATOM     64  CD  PRO A  43      12.806  -1.939  -0.201  1.00 13.81           C  \\nATOM     65  N   ASN A  44      16.031  -4.743   1.583  1.00 13.09           N  \\nATOM     66  CA  ASN A  44      16.710  -4.757   2.876  1.00 13.27           C  \\nATOM     67  C   ASN A  44      17.007  -3.308   3.264  1.00 12.73           C  \\nATOM     68  O   ASN A  44      17.529  -2.548   2.447  1.00 12.11           O  \\nATOM     69  CB  ASN A  44      18.012  -5.561   2.790  1.00 13.25           C  \\nATOM     70  CG  ASN A  44      17.779  -7.006   2.362  1.00 15.42           C  \\nATOM     71  OD1 ASN A  44      18.439  -7.510   1.439  1.00 21.55           O  \\nATOM     72  ND2 ASN A  44      16.788  -7.651   2.967  1.00 14.09           N  \\nATOM     73  N   PRO A  45      16.554  -2.877   4.454  1.00 12.79           N  \\nATOM     74  CA  PRO A  45      16.597  -1.446   4.769  1.00 12.39           C  \\nATOM     75  C   PRO A  45      17.997  -0.968   5.136  1.00 12.50           C  \\nATOM     76  O   PRO A  45      18.773  -1.695   5.756  1.00 14.14           O  \\nATOM     77  CB  PRO A  45      15.657  -1.322   5.973  1.00 12.93           C  \\nATOM     78  CG  PRO A  45      15.690  -2.663   6.609  1.00 12.05           C  \\nATOM     79  CD  PRO A  45      15.827  -3.646   5.482  1.00 10.79           C  \\nATOM     80  N   THR A  46      18.308   0.260   4.745  1.00 12.81           N  \\nATOM     81  CA  THR A  46      19.444   0.965   5.310  1.00 12.44           C  \\nATOM     82  C   THR A  46      18.929   2.209   6.004  1.00 12.50           C  \\nATOM     83  O   THR A  46      17.871   2.741   5.638  1.00 13.67           O  \\nATOM     84  CB  THR A  46      20.439   1.402   4.227  1.00 13.35           C  \\nATOM     85  OG1 THR A  46      19.835   2.417   3.414  1.00 16.38           O  \\nATOM     86  CG2 THR A  46      20.872   0.214   3.358  1.00 14.54           C  \\nATOM     87  N   ARG A  47      19.667   2.687   7.000  1.00 13.43           N  \\nATOM     88  CA  ARG A  47      19.192   3.857   7.722  1.00 13.46           C  \\nATOM     89  C   ARG A  47      18.993   5.045   6.776  1.00 15.48           C  \\nATOM     90  O   ARG A  47      18.004   5.780   6.872  1.00 14.79           O  \\nATOM     91  CB  ARG A  47      20.164   4.229   8.834  1.00 14.48           C  \\nATOM     92  CG  ARG A  47      19.611   5.359   9.684  1.00 21.77           C  \\nATOM     93  CD  ARG A  47      20.584   5.729  10.769  1.00 18.47           C  \\nATOM     94  NE  ARG A  47      20.582   4.689  11.786  1.00 30.70           N  \\nATOM     95  CZ  ARG A  47      19.710   4.641  12.788  1.00 27.61           C  \\nATOM     96  NH1 ARG A  47      18.978   5.708  13.074  1.00 26.21           N  \\nATOM     97  NH2 ARG A  47      19.644   3.560  13.550  1.00 31.22           N  \\nATOM     98  N   SER A  48      19.946   5.243   5.870  1.00 13.72           N  \\nATOM     99  CA  SER A  48      19.879   6.321   4.897  1.00 15.69           C  \\nATOM    100  C   SER A  48      18.576   6.262   4.112  1.00 15.30           C  \\nATOM    101  O   SER A  48      17.902   7.273   3.934  1.00 15.80           O  \\nATOM    102  CB  SER A  48      21.086   6.248   3.952  1.00 19.02           C  \\nATOM    103  OG  SER A  48      21.135   7.378   3.095  1.00 27.12           O  \\nATOM    104  N   ALA A  49      18.220   5.075   3.635  1.00 13.74           N  \\nATOM    105  CA  ALA A  49      17.040   4.959   2.798  1.00 15.05           C  \\nATOM    106  C   ALA A  49      15.777   5.312   3.569  1.00 14.66           C  \\nATOM    107  O   ALA A  49      14.785   5.741   2.974  1.00 13.36           O  \\nATOM    108  CB  ALA A  49      16.944   3.554   2.189  1.00 12.91           C  \\nATOM    109  N   LEU A  50      15.816   5.146   4.890  1.00 13.32           N  \\nATOM    110  CA  LEU A  50      14.622   5.394   5.698  1.00 13.31           C  \\nATOM    111  C   LEU A  50      14.457   6.877   5.978  1.00 15.50           C  \\nATOM    112  O   LEU A  50      13.517   7.270   6.648  1.00 14.81           O  \\nATOM    113  CB  LEU A  50      14.666   4.605   7.013  1.00 13.33           C  \\nATOM    114  CG  LEU A  50      14.768   3.091   6.842  1.00 11.84           C  \\nATOM    115  CD1 LEU A  50      14.633   2.407   8.197  1.00 14.70           C  \\nATOM    116  CD2 LEU A  50      13.768   2.556   5.824  1.00 12.79           C  \\nATOM    117  N   THR A  51      15.334   7.704   5.410  1.00 14.18           N  \\nATOM    118  CA  THR A  51      15.197   9.147   5.577  1.00 15.61           C  \\nATOM    119  C   THR A  51      14.633   9.800   4.329  1.00 15.98           C  \\nATOM    120  O   THR A  51      14.528  11.023   4.263  1.00 17.82           O  \\nATOM    121  CB  THR A  51      16.535   9.837   5.921  1.00 16.11           C  \\nATOM    122  OG1 THR A  51      17.418   9.776   4.795  1.00 15.77           O  \\nATOM    123  CG2 THR A  51      17.190   9.223   7.157  1.00 16.38           C  \\nATOM    124  N   ALA A  52      14.221   8.984   3.367  1.00 15.16           N  \\nATOM    125  CA  ALA A  52      13.675   9.478   2.115  1.00 14.03           C  \\nATOM    126  C   ALA A  52      12.480   8.614   1.731  1.00 13.87           C  \\nATOM    127  O   ALA A  52      12.274   7.550   2.310  1.00 13.54           O  \\nATOM    128  CB  ALA A  52      14.739   9.423   1.019  1.00 17.28           C  \\nATOM    129  N   ASP A  53      11.687   9.066   0.767  1.00 13.86           N  \\nATOM    130  CA  ASP A  53      10.587   8.233   0.267  1.00 12.95           C  \\nATOM    131  C   ASP A  53      11.098   6.900  -0.290  1.00 12.99           C  \\nATOM    132  O   ASP A  53      12.249   6.795  -0.767  1.00 13.92           O  \\nATOM    133  CB  ASP A  53       9.766   8.960  -0.804  1.00 14.43           C  \\nATOM    134  CG  ASP A  53       8.895  10.067  -0.249  1.00 15.74           C  \\nATOM    135  OD1 ASP A  53       8.834  10.283   0.973  1.00 15.56           O  \\nATOM    136  OD2 ASP A  53       8.248  10.735  -1.082  1.00 21.72           O  \\nATOM    137  N   GLY A  54      10.293   5.859  -0.098  1.00 13.26           N  \\nATOM    138  CA  GLY A  54      10.693   4.478  -0.402  1.00 13.17           C  \\nATOM    139  C   GLY A  54      10.397   4.047  -1.829  1.00 14.35           C  \\nATOM    140  O   GLY A  54      10.260   4.883  -2.741  1.00 13.88           O  \\nATOM    141  N   PRO A  55      10.297   2.733  -2.045  1.00 15.11           N  \\nATOM    142  CA  PRO A  55      10.301   2.189  -3.398  1.00 16.43           C  \\nATOM    143  C   PRO A  55       8.932   2.163  -4.061  1.00 13.17           C  \\nATOM    144  O   PRO A  55       8.795   1.656  -5.173  1.00 14.53           O  \\nATOM    145  CB  PRO A  55      10.810   0.765  -3.192  1.00 19.68           C  \\nATOM    146  CG  PRO A  55      10.350   0.404  -1.831  1.00 16.18           C  \\nATOM    147  CD  PRO A  55      10.448   1.684  -1.026  1.00 15.87           C  \\nATOM    148  N   PHE A  56       7.893   2.595  -3.351  1.00 13.43           N  \\nATOM    149  CA  PHE A  56       6.561   2.588  -3.943  1.00 13.16           C  \\nATOM    150  C   PHE A  56       6.160   4.002  -4.303  1.00 13.86           C  \\nATOM    151  O   PHE A  56       6.206   4.896  -3.454  1.00 14.95           O  \\nATOM    152  CB  PHE A  56       5.525   1.969  -2.995  1.00 16.49           C  \\nATOM    153  CG  PHE A  56       5.841   0.555  -2.601  1.00 13.24           C  \\nATOM    154  CD1 PHE A  56       5.570  -0.494  -3.470  1.00 15.32           C  \\nATOM    155  CD2 PHE A  56       6.467   0.272  -1.400  1.00 16.54           C  \\nATOM    156  CE1 PHE A  56       5.896  -1.804  -3.123  1.00 12.79           C  \\nATOM    157  CE2 PHE A  56       6.844  -1.026  -1.072  1.00 15.97           C  \\nATOM    158  CZ  PHE A  56       6.537  -2.072  -1.934  1.00 15.49           C  \\nATOM    159  N   SER A  57       5.707   4.203  -5.537  1.00 16.55           N  \\nATOM    160  CA  SER A  57       5.063   5.449  -5.896  1.00 16.76           C  \\nATOM    161  C   SER A  57       3.752   5.513  -5.123  1.00 14.11           C  \\nATOM    162  O   SER A  57       3.192   4.484  -4.750  1.00 17.18           O  \\nATOM    163  CB  SER A  57       4.797   5.499  -7.402  1.00 20.25           C  \\nATOM    164  OG  SER A  57       4.194   6.726  -7.773  1.00 26.47           O  \\nATOM    165  N   VAL A  58       3.303   6.732  -4.855  1.00 14.08           N  \\nATOM    166  CA  VAL A  58       2.169   6.959  -3.973  1.00 13.90           C  \\nATOM    167  C   VAL A  58       1.073   7.688  -4.731  1.00 14.89           C  \\nATOM    168  O   VAL A  58       1.310   8.747  -5.342  1.00 16.54           O  \\nATOM    169  CB  VAL A  58       2.597   7.795  -2.759  1.00 13.74           C  \\nATOM    170  CG1 VAL A  58       1.388   8.172  -1.910  1.00 14.20           C  \\nATOM    171  CG2 VAL A  58       3.621   7.029  -1.931  1.00 15.33           C  \\nATOM    172  N   ALA A  59      -0.137   7.141  -4.668  1.00 14.02           N  \\nATOM    173  CA  ALA A  59      -1.336   7.880  -5.078  1.00 13.71           C  \\nATOM    174  C   ALA A  59      -2.036   8.474  -3.866  1.00 14.79           C  \\nATOM    175  O   ALA A  59      -1.869   7.996  -2.737  1.00 13.10           O  \\nATOM    176  CB  ALA A  59      -2.282   6.963  -5.830  1.00 14.72           C  \\nATOM    177  N   THR A  60      -2.874   9.483  -4.108  1.00 15.39           N  \\nATOM    178  CA  THR A  60      -3.667  10.075  -3.034  1.00 14.85           C  \\nATOM    179  C   THR A  60      -5.138  10.095  -3.390  1.00 16.62           C  \\nATOM    180  O   THR A  60      -5.509  10.297  -4.553  1.00 19.57           O  \\nATOM    181  CB  THR A  60      -3.228  11.523  -2.717  1.00 19.13           C  \\nATOM    182  OG1 THR A  60      -3.256  12.305  -3.921  1.00 22.06           O  \\nATOM    183  CG2 THR A  60      -1.833  11.542  -2.141  1.00 17.93           C  \\nATOM    184  N   TYR A  61      -5.969   9.873  -2.387  1.00 17.13           N  \\nATOM    185  CA  TYR A  61      -7.398  10.055  -2.537  1.00 15.97           C  \\nATOM    186  C   TYR A  61      -7.905  10.997  -1.449  1.00 15.96           C  \\nATOM    187  O   TYR A  61      -7.649  10.794  -0.266  1.00 15.65           O  \\nATOM    188  CB  TYR A  61      -8.109   8.707  -2.453  1.00 17.82           C  \\nATOM    189  CG  TYR A  61      -9.609   8.827  -2.468  1.00 19.72           C  \\nATOM    190  CD1 TYR A  61     -10.293   9.059  -3.655  1.00 26.93           C  \\nATOM    191  CD2 TYR A  61     -10.344   8.703  -1.297  1.00 20.69           C  \\nATOM    192  CE1 TYR A  61     -11.674   9.172  -3.671  1.00 22.45           C  \\nATOM    193  CE2 TYR A  61     -11.725   8.819  -1.306  1.00 23.69           C  \\nATOM    194  CZ  TYR A  61     -12.374   9.081  -2.491  1.00 25.09           C  \\nATOM    195  OH  TYR A  61     -13.752   9.133  -2.519  1.00 30.36           O  \\nATOM    196  N   THR A  62      -8.634  12.026  -1.860  1.00 17.65           N  \\nATOM    197  CA  THR A  62      -9.234  12.956  -0.925  1.00 20.24           C  \\nATOM    198  C   THR A  62     -10.574  12.423  -0.417  1.00 20.45           C  \\nATOM    199  O   THR A  62     -11.517  12.209  -1.195  1.00 22.31           O  \\nATOM    200  CB  THR A  62      -9.433  14.341  -1.569  1.00 19.54           C  \\nATOM    201  OG1 THR A  62      -8.158  14.927  -1.849  1.00 24.49           O  \\nATOM    202  CG2 THR A  62     -10.209  15.262  -0.630  1.00 24.78           C  \\nATOM    203  N   VAL A  63     -10.659  12.239   0.897  1.00 18.61           N  \\nATOM    204  CA  VAL A  63     -11.896  11.835   1.546  1.00 20.02           C  \\nATOM    205  C   VAL A  63     -12.611  13.097   2.012  1.00 21.29           C  \\nATOM    206  O   VAL A  63     -12.138  13.792   2.896  1.00 18.73           O  \\nATOM    207  CB  VAL A  63     -11.617  10.943   2.772  1.00 19.52           C  \\nATOM    208  CG1 VAL A  63     -12.921  10.536   3.432  1.00 19.42           C  \\nATOM    209  CG2 VAL A  63     -10.814   9.721   2.350  1.00 20.01           C  \\nATOM    210  N   SER A  64     -13.696  13.455   1.334  1.00 23.39           N  \\nATOM    211  CA  SER A  64     -14.392  14.680   1.668  1.00 24.89           C  \\nATOM    212  C   SER A  64     -15.031  14.605   3.056  1.00 24.64           C  \\nATOM    213  O   SER A  64     -15.461  13.537   3.501  1.00 23.20           O  \\nATOM    214  CB  SER A  64     -15.453  14.990   0.611  1.00 27.36           C  \\nATOM    215  OG  SER A  64     -16.645  14.282   0.876  1.00 31.25           O  \\nATOM    216  N   ARG A  65     -15.035  15.740   3.747  1.00 20.99           N  \\nATOM    217  CA  ARG A  65     -15.555  15.837   5.106  1.00 24.26           C  \\nATOM    218  C   ARG A  65     -17.015  15.398   5.169  1.00 26.21           C  \\nATOM    219  O   ARG A  65     -17.411  14.631   6.049  1.00 25.64           O  \\nATOM    220  CB  ARG A  65     -15.403  17.269   5.624  1.00 23.53           C  \\nATOM    221  CG  ARG A  65     -16.206  17.576   6.875  1.00 29.79           C  \\nATOM    222  CD  ARG A  65     -15.969  19.004   7.326  1.00 29.85           C  \\nATOM    223  NE  ARG A  65     -14.549  19.296   7.464  1.00 36.57           N  \\nATOM    224  CZ  ARG A  65     -13.877  19.221   8.610  1.00 37.01           C  \\nATOM    225  NH1 ARG A  65     -14.523  19.000   9.748  1.00 33.30           N  \\nATOM    226  NH2 ARG A  65     -12.576  19.474   8.630  1.00 31.36           N  \\nATOM    227  N   LEU A  66     -17.766  15.740   4.128  1.00 26.63           N  \\nATOM    228  CA  LEU A  66     -19.194  15.470   4.100  1.00 24.37           C  \\nATOM    229  C   LEU A  66     -19.500  13.983   3.951  1.00 25.17           C  \\nATOM    230  O   LEU A  66     -20.559  13.523   4.369  1.00 27.77           O  \\nATOM    231  CB  LEU A  66     -19.848  16.268   2.967  1.00 25.63           C  \\nATOM    232  CG  LEU A  66     -21.366  16.429   3.035  1.00 30.96           C  \\nATOM    233  CD1 LEU A  66     -21.803  16.999   4.382  1.00 30.40           C  \\nATOM    234  CD2 LEU A  66     -21.846  17.317   1.893  1.00 30.30           C  \\nATOM    235  N   SER A  67     -18.617  13.251   3.273  1.00 22.88           N  \\nATOM    236  CA  SER A  67     -18.881  11.861   2.924  1.00 26.81           C  \\nATOM    237  C   SER A  67     -18.749  10.926   4.126  1.00 26.37           C  \\nATOM    238  O   SER A  67     -19.212   9.788   4.080  1.00 29.83           O  \\nATOM    239  CB  SER A  67     -17.964  11.398   1.788  1.00 28.66           C  \\nATOM    240  OG  SER A  67     -16.619  11.267   2.228  1.00 34.23           O  \\nATOM    241  N   VAL A  68     -18.194  11.437   5.223  1.00 27.64           N  \\nATOM    242  CA  VAL A  68     -18.019  10.627   6.438  1.00 26.80           C  \\nATOM    243  C   VAL A  68     -18.625  11.231   7.706  1.00 28.29           C  \\nATOM    244  O   VAL A  68     -18.361  12.385   8.054  1.00 30.84           O  \\nATOM    245  CB  VAL A  68     -16.529  10.237   6.669  1.00 24.49           C  \\nATOM    246  CG1 VAL A  68     -16.022   9.405   5.512  1.00 22.97           C  \\nATOM    247  CG2 VAL A  68     -15.659  11.477   6.843  1.00 21.49           C  \\nATOM    248  N   SER A  69     -19.291  10.379   8.482  1.00 31.63           N  \\nATOM    249  CA  SER A  69     -19.671  10.702   9.854  1.00 31.30           C  \\nATOM    250  C   SER A  69     -18.779   9.881  10.789  1.00 31.98           C  \\nATOM    251  O   SER A  69     -18.637   8.669  10.608  1.00 38.76           O  \\nATOM    252  CB  SER A  69     -21.157  10.360  10.085  1.00 32.26           C  \\nATOM    253  OG  SER A  69     -21.654  10.885  11.311  1.00 33.82           O  \\nATOM    254  N   GLY A  70     -17.960  10.582  11.563  1.00 29.43           N  \\nATOM    255  CA  GLY A  70     -17.023   9.920  12.470  1.00 24.84           C  \\nATOM    256  C   GLY A  70     -15.631  10.521  12.405  1.00 23.98           C  \\nATOM    257  O   GLY A  70     -14.785  10.267  13.272  1.00 21.91           O  \\nATOM    258  N   PHE A  71     -15.389  11.317  11.371  1.00 19.05           N  \\nATOM    259  CA  PHE A  71     -14.192  12.142  11.306  1.00 19.36           C  \\nATOM    260  C   PHE A  71     -14.386  13.177  10.201  1.00 17.81           C  \\nATOM    261  O   PHE A  71     -15.466  13.264   9.607  1.00 19.30           O  \\nATOM    262  CB  PHE A  71     -12.931  11.280  11.084  1.00 16.13           C  \\nATOM    263  CG  PHE A  71     -12.943  10.476   9.808  1.00 16.20           C  \\nATOM    264  CD1 PHE A  71     -12.077  10.794   8.765  1.00 15.45           C  \\nATOM    265  CD2 PHE A  71     -13.716   9.318   9.695  1.00 16.30           C  \\nATOM    266  CE1 PHE A  71     -12.022  10.007   7.598  1.00 16.54           C  \\nATOM    267  CE2 PHE A  71     -13.706   8.561   8.522  1.00 16.08           C  \\nATOM    268  CZ  PHE A  71     -12.858   8.906   7.467  1.00 15.36           C  \\nATOM    269  N   GLY A  72     -13.417  14.070  10.063  1.00 16.71           N  \\nATOM    270  CA  GLY A  72     -13.555  15.221   9.171  1.00 16.15           C  \\nATOM    271  C   GLY A  72     -13.038  14.990   7.761  1.00 17.43           C  \\nATOM    272  O   GLY A  72     -12.601  15.932   7.099  1.00 19.71           O  \\nATOM    273  N   GLY A  73     -13.103  13.743   7.289  1.00 15.98           N  \\nATOM    274  CA  GLY A  73     -12.442  13.377   6.036  1.00 16.32           C  \\nATOM    275  C   GLY A  73     -10.935  13.358   6.189  1.00 13.41           C  \\nATOM    276  O   GLY A  73     -10.427  13.120   7.275  1.00 16.11           O  \\nATOM    277  N   GLY A  74     -10.220  13.536   5.084  1.00 13.62           N  \\nATOM    278  CA  GLY A  74      -8.767  13.454   5.109  1.00 15.62           C  \\nATOM    279  C   GLY A  74      -8.189  13.061   3.768  1.00 16.29           C  \\nATOM    280  O   GLY A  74      -8.799  13.290   2.722  1.00 17.37           O  \\nATOM    281  N   VAL A  75      -6.981  12.504   3.792  1.00 13.86           N  \\nATOM    282  CA  VAL A  75      -6.301  12.090   2.567  1.00 13.31           C  \\nATOM    283  C   VAL A  75      -5.748  10.687   2.769  1.00 11.95           C  \\nATOM    284  O   VAL A  75      -5.125  10.407   3.806  1.00 14.58           O  \\nATOM    285  CB  VAL A  75      -5.144  13.034   2.186  1.00 14.13           C  \\nATOM    286  CG1 VAL A  75      -4.490  12.536   0.903  1.00 15.76           C  \\nATOM    287  CG2 VAL A  75      -5.652  14.469   2.012  1.00 15.82           C  \\nATOM    288  N   ILE A  76      -6.021   9.798   1.818  1.00 13.85           N  \\nATOM    289  CA  ILE A  76      -5.410   8.456   1.817  1.00 14.03           C  \\nATOM    290  C   ILE A  76      -4.205   8.445   0.892  1.00 13.87           C  \\nATOM    291  O   ILE A  76      -4.310   8.881  -0.251  1.00 15.32           O  \\nATOM    292  CB  ILE A  76      -6.381   7.374   1.323  1.00 13.07           C  \\nATOM    293  CG1 ILE A  76      -7.695   7.435   2.098  1.00 15.14           C  \\nATOM    294  CG2 ILE A  76      -5.778   5.977   1.524  1.00 14.10           C  \\nATOM    295  CD1 ILE A  76      -8.781   6.579   1.461  1.00 19.64           C  \\nATOM    296  N   TYR A  77      -3.059   8.010   1.417  1.00 12.99           N  \\nATOM    297  CA  TYR A  77      -1.834   7.820   0.639  1.00 11.27           C  \\nATOM    298  C   TYR A  77      -1.646   6.326   0.504  1.00 12.43           C  \\nATOM    299  O   TYR A  77      -1.744   5.609   1.489  1.00 13.22           O  \\nATOM    300  CB  TYR A  77      -0.627   8.415   1.387  1.00 12.86           C  \\nATOM    301  CG  TYR A  77      -0.762   9.887   1.702  1.00 12.30           C  \\nATOM    302  CD1 TYR A  77      -0.201  10.855   0.881  1.00 13.05           C  \\nATOM    303  CD2 TYR A  77      -1.456  10.307   2.823  1.00 11.14           C  \\nATOM    304  CE1 TYR A  77      -0.381  12.221   1.145  1.00 13.63           C  \\nATOM    305  CE2 TYR A  77      -1.570  11.666   3.129  1.00 12.28           C  \\nATOM    306  CZ  TYR A  77      -1.126  12.606   2.229  1.00 12.39           C  \\nATOM    307  OH  TYR A  77      -1.243  13.954   2.568  1.00 15.37           O  \\nATOM    308  N   TYR A  78      -1.465   5.851  -0.716  1.00 13.44           N  \\nATOM    309  CA  TYR A  78      -1.384   4.408  -0.911  1.00 12.09           C  \\nATOM    310  C   TYR A  78      -0.478   4.013  -2.065  1.00 14.27           C  \\nATOM    311  O   TYR A  78      -0.207   4.814  -2.952  1.00 12.66           O  \\nATOM    312  CB  TYR A  78      -2.799   3.838  -1.078  1.00 14.14           C  \\nATOM    313  CG  TYR A  78      -3.520   4.295  -2.334  1.00 12.70           C  \\nATOM    314  CD1 TYR A  78      -3.643   3.444  -3.427  1.00 15.11           C  \\nATOM    315  CD2 TYR A  78      -4.208   5.501  -2.364  1.00 13.86           C  \\nATOM    316  CE1 TYR A  78      -4.296   3.848  -4.573  1.00 16.07           C  \\nATOM    317  CE2 TYR A  78      -4.936   5.879  -3.481  1.00 16.42           C  \\nATOM    318  CZ  TYR A  78      -4.927   5.072  -4.605  1.00 17.96           C  \\nATOM    319  OH  TYR A  78      -5.625   5.457  -5.741  1.00 20.07           O  \\nATOM    320  N   PRO A  79       0.087   2.796  -2.013  1.00 13.54           N  \\nATOM    321  CA  PRO A  79       1.053   2.412  -3.037  1.00 13.33           C  \\nATOM    322  C   PRO A  79       0.393   2.172  -4.394  1.00 13.39           C  \\nATOM    323  O   PRO A  79      -0.723   1.633  -4.461  1.00 13.83           O  \\nATOM    324  CB  PRO A  79       1.591   1.074  -2.517  1.00 12.70           C  \\nATOM    325  CG  PRO A  79       1.374   1.133  -1.032  1.00 14.86           C  \\nATOM    326  CD  PRO A  79       0.056   1.836  -0.895  1.00 14.30           C  \\nATOM    327  N   THR A  80       1.119   2.520  -5.453  1.00 13.72           N  \\nATOM    328  CA  THR A  80       0.797   2.072  -6.809  1.00 14.39           C  \\nATOM    329  C   THR A  80       1.910   1.181  -7.354  1.00 13.64           C  \\nATOM    330  O   THR A  80       2.922   0.966  -6.691  1.00 13.77           O  \\nATOM    331  CB  THR A  80       0.603   3.257  -7.770  1.00 14.19           C  \\nATOM    332  OG1 THR A  80       1.834   3.977  -7.907  1.00 16.50           O  \\nATOM    333  CG2 THR A  80      -0.480   4.195  -7.257  1.00 20.12           C  \\nATOM    334  N   GLY A  81       1.724   0.687  -8.577  1.00 14.17           N  \\nATOM    335  CA  GLY A  81       2.748  -0.121  -9.233  1.00 14.38           C  \\nATOM    336  C   GLY A  81       3.009  -1.434  -8.517  1.00 13.29           C  \\nATOM    337  O   GLY A  81       4.116  -1.981  -8.595  1.00 13.26           O  \\nATOM    338  N   THR A  82       1.964  -2.020  -7.950  1.00 13.84           N  \\nATOM    339  CA  THR A  82       2.119  -3.259  -7.216  1.00 13.50           C  \\nATOM    340  C   THR A  82       0.810  -4.031  -7.173  1.00 14.37           C  \\nATOM    341  O   THR A  82      -0.258  -3.440  -7.135  1.00 16.02           O  \\nATOM    342  CB  THR A  82       2.625  -2.983  -5.771  1.00 14.63           C  \\nATOM    343  OG1 THR A  82       2.802  -4.222  -5.067  1.00 15.75           O  \\nATOM    344  CG2 THR A  82       1.659  -2.074  -4.993  1.00 14.93           C  \\nATOM    345  N   SER A  83       0.918  -5.348  -7.038  1.00 14.41           N  \\nATOM    346  CA  SER A  83      -0.257  -6.174  -6.793  1.00 15.51           C  \\nATOM    347  C   SER A  83      -0.275  -6.742  -5.385  1.00 16.39           C  \\nATOM    348  O   SER A  83      -1.169  -7.522  -5.046  1.00 19.30           O  \\nATOM    349  CB  SER A  83      -0.334  -7.312  -7.816  1.00 17.64           C  \\nATOM    350  OG  SER A  83      -0.592  -6.802  -9.115  1.00 21.38           O  \\nATOM    351  N   LEU A  84       0.649  -6.285  -4.544  1.00 14.28           N  \\nATOM    352  CA  LEU A  84       0.723  -6.750  -3.170  1.00 15.06           C  \\nATOM    353  C   LEU A  84      -0.281  -6.000  -2.297  1.00 14.67           C  \\nATOM    354  O   LEU A  84      -0.870  -4.991  -2.707  1.00 14.86           O  \\nATOM    355  CB  LEU A  84       2.147  -6.570  -2.622  1.00 17.14           C  \\nATOM    356  CG  LEU A  84       3.237  -7.284  -3.434  1.00 17.67           C  \\nATOM    357  CD1 LEU A  84       4.622  -6.952  -2.917  1.00 21.53           C  \\nATOM    358  CD2 LEU A  84       3.011  -8.792  -3.442  1.00 27.83           C  \\nATOM    359  N   THR A  85      -0.499  -6.524  -1.100  1.00 13.90           N  \\nATOM    360  CA  THR A  85      -1.297  -5.830  -0.107  1.00 12.43           C  \\nATOM    361  C   THR A  85      -0.442  -5.435   1.092  1.00 12.43           C  \\nATOM    362  O   THR A  85       0.644  -5.973   1.313  1.00 14.08           O  \\nATOM    363  CB  THR A  85      -2.468  -6.703   0.381  1.00 12.03           C  \\nATOM    364  OG1 THR A  85      -1.972  -7.907   0.976  1.00 15.59           O  \\nATOM    365  CG2 THR A  85      -3.419  -7.041  -0.761  1.00 16.22           C  \\nATOM    366  N   PHE A  86      -0.919  -4.446   1.836  1.00 12.33           N  \\nATOM    367  CA  PHE A  86      -0.104  -3.751   2.815  1.00 12.16           C  \\nATOM    368  C   PHE A  86      -0.969  -3.428   4.020  1.00 11.78           C  \\nATOM    369  O   PHE A  86      -2.184  -3.281   3.901  1.00 12.53           O  \\nATOM    370  CB  PHE A  86       0.422  -2.437   2.231  1.00 12.03           C  \\nATOM    371  CG  PHE A  86       1.243  -2.607   0.989  1.00 12.49           C  \\nATOM    372  CD1 PHE A  86       2.618  -2.762   1.073  1.00 13.10           C  \\nATOM    373  CD2 PHE A  86       0.646  -2.560  -0.265  1.00 13.86           C  \\nATOM    374  CE1 PHE A  86       3.393  -2.894  -0.083  1.00 12.96           C  \\nATOM    375  CE2 PHE A  86       1.405  -2.717  -1.420  1.00 14.32           C  \\nATOM    376  CZ  PHE A  86       2.789  -2.825  -1.335  1.00 15.44           C  \\nATOM    377  N   GLY A  87      -0.330  -3.230   5.160  1.00 11.24           N  \\nATOM    378  CA  GLY A  87      -1.048  -2.798   6.350  1.00 11.13           C  \\nATOM    379  C   GLY A  87      -1.566  -1.377   6.216  1.00 10.82           C  \\nATOM    380  O   GLY A  87      -1.080  -0.608   5.376  1.00 12.07           O  \\nATOM    381  N   GLY A  88      -2.570  -1.041   7.019  1.00 11.53           N  \\nATOM    382  CA  GLY A  88      -3.170   0.291   7.000  1.00 12.08           C  \\nATOM    383  C   GLY A  88      -2.782   1.057   8.246  1.00 13.60           C  \\nATOM    384  O   GLY A  88      -2.698   0.484   9.335  1.00 12.07           O  \\nATOM    385  N   ILE A  89      -2.644   2.364   8.104  1.00 12.60           N  \\nATOM    386  CA  ILE A  89      -2.360   3.223   9.237  1.00 12.14           C  \\nATOM    387  C   ILE A  89      -3.362   4.371   9.205  1.00 12.45           C  \\nATOM    388  O   ILE A  89      -3.495   5.035   8.187  1.00 12.44           O  \\nATOM    389  CB  ILE A  89      -0.917   3.773   9.146  1.00 14.19           C  \\nATOM    390  CG1 ILE A  89       0.091   2.606   9.134  1.00 12.77           C  \\nATOM    391  CG2 ILE A  89      -0.640   4.726  10.296  1.00 13.58           C  \\nATOM    392  CD1 ILE A  89       1.522   3.037   8.781  1.00 13.64           C  \\nATOM    393  N   ALA A  90      -4.005   4.648  10.334  1.00 11.32           N  \\nATOM    394  CA  ALA A  90      -4.740   5.913  10.495  1.00 11.64           C  \\nATOM    395  C   ALA A  90      -3.909   6.869  11.338  1.00 10.83           C  \\nATOM    396  O   ALA A  90      -3.361   6.469  12.366  1.00 12.55           O  \\nATOM    397  CB  ALA A  90      -6.108   5.659  11.159  1.00 11.97           C  \\nATOM    398  N   MET A  91      -3.829   8.134  10.908  1.00 10.50           N  \\nATOM    399  CA  MET A  91      -3.008   9.131  11.606  1.00 11.59           C  \\nATOM    400  C   MET A  91      -3.828  10.382  11.916  1.00 10.95           C  \\nATOM    401  O   MET A  91      -4.562  10.888  11.047  1.00 12.21           O  \\nATOM    402  CB  MET A  91      -1.807   9.529  10.742  1.00 16.52           C  \\nATOM    403  CG  MET A  91      -0.957   8.382  10.285  1.00 20.85           C  \\nATOM    404  SD  MET A  91       0.625   8.980   9.673  1.00 34.83           S  \\nATOM    405  CE  MET A  91       0.945  10.232  10.902  1.00  9.36           C  \\nATOM    406  N   SER A  92      -3.687  10.889  13.138  1.00 11.25           N  \\nATOM    407  CA  SER A  92      -4.408  12.088  13.559  1.00 12.64           C  \\nATOM    408  C   SER A  92      -3.460  13.263  13.708  1.00 10.68           C  \\nATOM    409  O   SER A  92      -2.439  13.149  14.384  1.00 10.94           O  \\nATOM    410  CB  SER A  92      -5.101  11.839  14.893  1.00 12.62           C  \\nATOM    411  OG  SER A  92      -5.718  13.033  15.353  1.00 13.23           O  \\nATOM    412  N   PRO A  93      -3.868  14.449  13.220  1.00 11.07           N  \\nATOM    413  CA  PRO A  93      -3.141  15.654  13.574  1.00 12.70           C  \\nATOM    414  C   PRO A  93      -3.468  16.114  14.991  1.00 13.59           C  \\nATOM    415  O   PRO A  93      -4.203  15.424  15.719  1.00 12.31           O  \\nATOM    416  CB  PRO A  93      -3.614  16.663  12.527  1.00 13.11           C  \\nATOM    417  CG  PRO A  93      -5.028  16.238  12.219  1.00 15.13           C  \\nATOM    418  CD  PRO A  93      -5.033  14.713  12.359  1.00 12.18           C  \\nATOM    419  N   GLY A  94      -2.829  17.200  15.422  1.00 14.51           N  \\nATOM    420  CA  GLY A  94      -3.004  17.699  16.783  1.00 15.70           C  \\nATOM    421  C   GLY A  94      -3.931  18.899  16.895  1.00 13.86           C  \\nATOM    422  O   GLY A  94      -4.639  19.240  15.941  1.00 13.59           O  \\nATOM    423  N   TYR A  95      -4.056  19.398  18.122  1.00 14.03           N  \\nATOM    424  CA  TYR A  95      -4.866  20.580  18.419  1.00 15.71           C  \\nATOM    425  C   TYR A  95      -4.508  21.710  17.454  1.00 15.16           C  \\nATOM    426  O   TYR A  95      -3.335  22.041  17.287  1.00 15.70           O  \\nATOM    427  CB  TYR A  95      -4.648  21.003  19.879  1.00 16.11           C  \\nATOM    428  CG  TYR A  95      -5.637  22.002  20.429  1.00 15.97           C  \\nATOM    429  CD1 TYR A  95      -6.943  21.628  20.718  1.00 14.59           C  \\nATOM    430  CD2 TYR A  95      -5.230  23.291  20.771  1.00 14.59           C  \\nATOM    431  CE1 TYR A  95      -7.783  22.475  21.418  1.00 16.86           C  \\nATOM    432  CE2 TYR A  95      -6.092  24.174  21.374  1.00 16.15           C  \\nATOM    433  CZ  TYR A  95      -7.373  23.767  21.689  1.00 16.48           C  \\nATOM    434  OH  TYR A  95      -8.249  24.644  22.305  1.00 18.99           O  \\nATOM    435  N   THR A  96      -5.545  22.342  16.901  1.00 16.19           N  \\nATOM    436  CA  THR A  96      -5.423  23.440  15.916  1.00 14.85           C  \\nATOM    437  C   THR A  96      -4.995  23.038  14.513  1.00 16.68           C  \\nATOM    438  O   THR A  96      -4.791  23.900  13.660  1.00 15.88           O  \\nATOM    439  CB  THR A  96      -4.540  24.638  16.377  1.00 17.91           C  \\nATOM    440  OG1 THR A  96      -3.149  24.313  16.243  1.00 18.28           O  \\nATOM    441  CG2 THR A  96      -4.853  25.061  17.812  1.00 20.79           C  \\nATOM    442  N   ALA A  97      -4.793  21.745  14.275  1.00 15.11           N  \\nATOM    443  CA  ALA A  97      -4.102  21.322  13.068  1.00 14.31           C  \\nATOM    444  C   ALA A  97      -4.992  20.569  12.095  1.00 13.93           C  \\nATOM    445  O   ALA A  97      -5.949  19.890  12.490  1.00 13.99           O  \\nATOM    446  CB  ALA A  97      -2.853  20.487  13.413  1.00 13.13           C  \\nATOM    447  N   ASP A  98      -4.618  20.639  10.817  1.00 15.34           N  \\nATOM    448  CA  ASP A  98      -5.279  19.898   9.748  1.00 14.32           C  \\nATOM    449  C   ASP A  98      -4.435  18.686   9.348  1.00 13.19           C  \\nATOM    450  O   ASP A  98      -3.244  18.613   9.686  1.00 14.04           O  \\nATOM    451  CB  ASP A  98      -5.483  20.800   8.510  1.00 15.35           C  \\nATOM    452  CG  ASP A  98      -6.431  21.960   8.771  1.00 19.23           C  \\nATOM    453  OD1 ASP A  98      -7.360  21.777   9.582  1.00 17.86           O  \\nATOM    454  OD2 ASP A  98      -6.339  22.965   8.029  1.00 19.49           O  \\nATOM    455  N   ALA A  99      -5.024  17.842   8.506  1.00 13.71           N  \\nATOM    456  CA  ALA A  99      -4.340  16.676   7.930  1.00 12.28           C  \\nATOM    457  C   ALA A  99      -3.073  17.107   7.191  1.00 14.64           C  \\nATOM    458  O   ALA A  99      -2.067  16.387   7.176  1.00 14.36           O  \\nATOM    459  CB  ALA A  99      -5.284  15.905   7.002  1.00 15.45           C  \\nATOM    460  N   SER A 100      -3.110  18.294   6.597  1.00 13.37           N  \\nATOM    461  CA  SER A 100      -1.978  18.775   5.810  1.00 13.75           C  \\nATOM    462  C   SER A 100      -0.699  18.898   6.643  1.00 12.87           C  \\nATOM    463  O   SER A 100       0.413  18.866   6.094  1.00 15.17           O  \\nATOM    464  CB  SER A 100      -2.309  20.109   5.153  1.00 12.46           C  \\nATOM    465  OG  SER A 100      -2.604  21.090   6.124  1.00 12.73           O  \\nATOM    466  N   SER A 101      -0.851  18.987   7.961  1.00 12.63           N  \\nATOM    467  CA  SER A 101       0.290  19.138   8.854  1.00 12.78           C  \\nATOM    468  C   SER A 101       1.114  17.857   8.979  1.00 13.44           C  \\nATOM    469  O   SER A 101       2.230  17.882   9.510  1.00 13.97           O  \\nATOM    470  CB  SER A 101      -0.154  19.611  10.246  1.00 13.91           C  \\nATOM    471  OG  SER A 101      -0.893  18.613  10.939  1.00 14.94           O  \\nATOM    472  N   LEU A 102       0.583  16.775   8.421  1.00 13.37           N  \\nATOM    473  CA  LEU A 102       1.285  15.480   8.432  1.00 13.27           C  \\nATOM    474  C   LEU A 102       1.350  14.862   7.042  1.00 15.29           C  \\nATOM    475  O   LEU A 102       1.673  13.685   6.893  1.00 13.86           O  \\nATOM    476  CB  LEU A 102       0.619  14.510   9.409  1.00 12.64           C  \\nATOM    477  CG  LEU A 102       0.700  14.896  10.891  1.00 11.99           C  \\nATOM    478  CD1 LEU A 102      -0.185  13.985  11.735  1.00 14.47           C  \\nATOM    479  CD2 LEU A 102       2.136  14.872  11.412  1.00 15.54           C  \\nATOM    480  N   ALA A 103       1.174  15.685   6.019  1.00 13.65           N  \\nATOM    481  CA  ALA A 103       1.146  15.181   4.656  1.00 13.50           C  \\nATOM    482  C   ALA A 103       2.479  14.523   4.277  1.00 13.22           C  \\nATOM    483  O   ALA A 103       2.498  13.451   3.662  1.00 13.27           O  \\nATOM    484  CB  ALA A 103       0.812  16.306   3.686  1.00 12.81           C  \\nATOM    485  N   TRP A 104       3.590  15.154   4.651  1.00 12.28           N  \\nATOM    486  CA  TRP A 104       4.893  14.583   4.328  1.00 12.73           C  \\nATOM    487  C   TRP A 104       5.044  13.176   4.897  1.00 13.16           C  \\nATOM    488  O   TRP A 104       5.618  12.290   4.245  1.00 13.66           O  \\nATOM    489  CB  TRP A 104       6.024  15.493   4.808  1.00 13.40           C  \\nATOM    490  CG  TRP A 104       6.310  15.360   6.266  1.00 14.80           C  \\nATOM    491  CD1 TRP A 104       5.759  16.089   7.283  1.00 15.28           C  \\nATOM    492  CD2 TRP A 104       7.220  14.438   6.882  1.00 13.55           C  \\nATOM    493  NE1 TRP A 104       6.219  15.632   8.495  1.00 17.02           N  \\nATOM    494  CE2 TRP A 104       7.141  14.640   8.273  1.00 15.95           C  \\nATOM    495  CE3 TRP A 104       8.059  13.427   6.393  1.00 15.58           C  \\nATOM    496  CZ2 TRP A 104       7.869  13.875   9.179  1.00 17.77           C  \\nATOM    497  CZ3 TRP A 104       8.849  12.728   7.299  1.00 14.19           C  \\nATOM    498  CH2 TRP A 104       8.720  12.928   8.672  1.00 16.08           C  \\nATOM    499  N   LEU A 105       4.488  12.953   6.083  1.00 13.67           N  \\nATOM    500  CA  LEU A 105       4.694  11.695   6.788  1.00 12.52           C  \\nATOM    501  C   LEU A 105       3.713  10.625   6.312  1.00 12.90           C  \\nATOM    502  O   LEU A 105       4.079   9.452   6.182  1.00 13.55           O  \\nATOM    503  CB  LEU A 105       4.622  11.880   8.305  1.00 12.68           C  \\nATOM    504  CG  LEU A 105       4.831  10.626   9.149  1.00 12.97           C  \\nATOM    505  CD1 LEU A 105       6.232  10.069   8.896  1.00 14.73           C  \\nATOM    506  CD2 LEU A 105       4.657  10.977  10.629  1.00 17.23           C  \\nATOM    507  N   GLY A 106       2.500  11.023   5.936  1.00 12.79           N  \\nATOM    508  CA  GLY A 106       1.594  10.088   5.260  1.00 13.83           C  \\nATOM    509  C   GLY A 106       2.176   9.567   3.958  1.00 11.79           C  \\nATOM    510  O   GLY A 106       2.166   8.363   3.697  1.00 13.35           O  \\nATOM    511  N   ARG A 107       2.741  10.474   3.162  1.00 13.65           N  \\nATOM    512  CA  ARG A 107       3.398  10.097   1.923  1.00 12.60           C  \\nATOM    513  C   ARG A 107       4.641   9.233   2.187  1.00 11.31           C  \\nATOM    514  O   ARG A 107       4.856   8.228   1.517  1.00 13.57           O  \\nATOM    515  CB  ARG A 107       3.808  11.337   1.137  1.00 14.34           C  \\nATOM    516  CG  ARG A 107       4.549  11.004  -0.145  1.00 17.34           C  \\nATOM    517  CD  ARG A 107       4.692  12.238  -1.021  1.00 19.23           C  \\nATOM    518  NE  ARG A 107       3.401  12.903  -1.199  1.00 25.77           N  \\nATOM    519  CZ  ARG A 107       2.543  12.625  -2.178  1.00 34.13           C  \\nATOM    520  NH1 ARG A 107       2.839  11.707  -3.087  1.00 35.90           N  \\nATOM    521  NH2 ARG A 107       1.376  13.251  -2.232  1.00 34.88           N  \\nATOM    522  N   ARG A 108       5.470   9.653   3.134  1.00 12.45           N  \\nATOM    523  CA  ARG A 108       6.675   8.914   3.491  1.00 11.03           C  \\nATOM    524  C   ARG A 108       6.341   7.472   3.859  1.00 12.07           C  \\nATOM    525  O   ARG A 108       6.940   6.534   3.317  1.00 12.47           O  \\nATOM    526  CB  ARG A 108       7.388   9.620   4.655  1.00 12.67           C  \\nATOM    527  CG  ARG A 108       8.638   8.888   5.184  1.00 13.64           C  \\nATOM    528  CD  ARG A 108       9.758   8.735   4.123  1.00 13.22           C  \\nATOM    529  NE  ARG A 108      10.071   9.985   3.423  1.00 13.02           N  \\nATOM    530  CZ  ARG A 108      10.704  11.027   3.962  1.00 15.27           C  \\nATOM    531  NH1 ARG A 108      11.258  10.957   5.169  1.00 16.79           N  \\nATOM    532  NH2 ARG A 108      10.866  12.120   3.235  1.00 17.73           N  \\nATOM    533  N   LEU A 109       5.376   7.285   4.751  1.00 10.53           N  \\nATOM    534  CA  LEU A 109       4.982   5.930   5.143  1.00 10.54           C  \\nATOM    535  C   LEU A 109       4.416   5.144   3.980  1.00 11.70           C  \\nATOM    536  O   LEU A 109       4.823   4.000   3.757  1.00 11.29           O  \\nATOM    537  CB  LEU A 109       3.991   5.956   6.310  1.00 11.61           C  \\nATOM    538  CG  LEU A 109       4.614   6.312   7.653  1.00 13.14           C  \\nATOM    539  CD1 LEU A 109       3.520   6.676   8.661  1.00 15.02           C  \\nATOM    540  CD2 LEU A 109       5.417   5.104   8.137  1.00 17.54           C  \\nATOM    541  N   ALA A 110       3.478   5.720   3.234  1.00 11.73           N  \\nATOM    542  CA  ALA A 110       2.876   4.989   2.126  1.00 11.25           C  \\nATOM    543  C   ALA A 110       3.951   4.531   1.134  1.00 11.82           C  \\nATOM    544  O   ALA A 110       3.886   3.413   0.613  1.00 12.06           O  \\nATOM    545  CB  ALA A 110       1.826   5.835   1.427  1.00 12.32           C  \\nATOM    546  N   SER A 111       4.955   5.381   0.903  1.00 11.62           N  \\nATOM    547  CA  SER A 111       6.010   5.094  -0.071  1.00 11.66           C  \\nATOM    548  C   SER A 111       6.902   3.932   0.348  1.00 12.00           C  \\nATOM    549  O   SER A 111       7.651   3.395  -0.472  1.00 13.61           O  \\nATOM    550  CB  SER A 111       6.866   6.342  -0.344  1.00 14.37           C  \\nATOM    551  OG  SER A 111       7.724   6.646   0.740  1.00 11.88           O  \\nATOM    552  N   HIS A 112       6.891   3.610   1.639  1.00 10.86           N  \\nATOM    553  CA  HIS A 112       7.602   2.435   2.143  1.00 11.43           C  \\nATOM    554  C   HIS A 112       6.704   1.219   2.316  1.00 10.98           C  \\nATOM    555  O   HIS A 112       7.131   0.222   2.902  1.00 12.14           O  \\nATOM    556  CB  HIS A 112       8.281   2.751   3.476  1.00 10.98           C  \\nATOM    557  CG  HIS A 112       9.587   3.462   3.320  1.00 12.05           C  \\nATOM    558  ND1 HIS A 112       9.702   4.833   3.173  1.00 15.15           N  \\nATOM    559  CD2 HIS A 112      10.832   2.960   3.158  1.00  9.13           C  \\nATOM    560  CE1 HIS A 112      10.975   5.142   2.955  1.00  9.10           C  \\nATOM    561  NE2 HIS A 112      11.679   4.023   2.968  1.00 15.24           N  \\nATOM    562  N   GLY A 113       5.493   1.297   1.771  1.00 11.64           N  \\nATOM    563  CA  GLY A 113       4.570   0.155   1.718  1.00 12.72           C  \\nATOM    564  C   GLY A 113       3.487   0.095   2.798  1.00 11.52           C  \\nATOM    565  O   GLY A 113       3.443  -0.842   3.598  1.00 12.21           O  \\nATOM    566  N   PHE A 114       2.656   1.134   2.871  1.00 12.19           N  \\nATOM    567  CA  PHE A 114       1.505   1.154   3.769  1.00 11.20           C  \\nATOM    568  C   PHE A 114       0.372   1.904   3.086  1.00 12.30           C  \\nATOM    569  O   PHE A 114       0.615   2.742   2.217  1.00 12.02           O  \\nATOM    570  CB  PHE A 114       1.848   1.852   5.089  1.00 12.24           C  \\nATOM    571  CG  PHE A 114       2.882   1.135   5.904  1.00 10.41           C  \\nATOM    572  CD1 PHE A 114       2.539   0.004   6.625  1.00 11.28           C  \\nATOM    573  CD2 PHE A 114       4.230   1.492   5.819  1.00 11.80           C  \\nATOM    574  CE1 PHE A 114       3.497  -0.732   7.305  1.00 14.86           C  \\nATOM    575  CE2 PHE A 114       5.190   0.776   6.527  1.00 12.75           C  \\nATOM    576  CZ  PHE A 114       4.818  -0.347   7.265  1.00 13.05           C  \\nATOM    577  N   VAL A 115      -0.859   1.612   3.504  1.00 10.94           N  \\nATOM    578  CA  VAL A 115      -2.013   2.438   3.153  1.00 11.23           C  \\nATOM    579  C   VAL A 115      -2.292   3.366   4.321  1.00 11.59           C  \\nATOM    580  O   VAL A 115      -2.513   2.900   5.431  1.00 11.39           O  \\nATOM    581  CB  VAL A 115      -3.260   1.560   2.889  1.00 11.46           C  \\nATOM    582  CG1 VAL A 115      -4.483   2.427   2.619  1.00 15.58           C  \\nATOM    583  CG2 VAL A 115      -2.997   0.609   1.744  1.00 12.49           C  \\nATOM    584  N   VAL A 116      -2.238   4.673   4.097  1.00 12.18           N  \\nATOM    585  CA  VAL A 116      -2.222   5.619   5.220  1.00 12.70           C  \\nATOM    586  C   VAL A 116      -3.336   6.637   5.042  1.00 12.03           C  \\nATOM    587  O   VAL A 116      -3.349   7.393   4.069  1.00 14.15           O  \\nATOM    588  CB  VAL A 116      -0.871   6.364   5.345  1.00 13.57           C  \\nATOM    589  CG1 VAL A 116      -0.838   7.215   6.611  1.00 11.21           C  \\nATOM    590  CG2 VAL A 116       0.298   5.386   5.342  1.00 13.51           C  \\nATOM    591  N   LEU A 117      -4.253   6.677   6.001  1.00 12.48           N  \\nATOM    592  CA  LEU A 117      -5.286   7.705   6.016  1.00 13.74           C  \\nATOM    593  C   LEU A 117      -4.973   8.754   7.085  1.00 11.84           C  \\nATOM    594  O   LEU A 117      -4.985   8.469   8.291  1.00 12.45           O  \\nATOM    595  CB  LEU A 117      -6.656   7.069   6.265  1.00 13.49           C  \\nATOM    596  CG  LEU A 117      -7.848   7.988   6.584  1.00 12.36           C  \\nATOM    597  CD1 LEU A 117      -7.988   9.139   5.581  1.00 14.31           C  \\nATOM    598  CD2 LEU A 117      -9.105   7.138   6.614  1.00 15.13           C  \\nATOM    599  N   VAL A 118      -4.589   9.943   6.628  1.00 11.90           N  \\nATOM    600  CA  VAL A 118      -4.390  11.067   7.544  1.00 12.45           C  \\nATOM    601  C   VAL A 118      -5.682  11.857   7.649  1.00 11.74           C  \\nATOM    602  O   VAL A 118      -6.207  12.319   6.628  1.00 12.72           O  \\nATOM    603  CB  VAL A 118      -3.269  12.004   7.057  1.00 11.83           C  \\nATOM    604  CG1 VAL A 118      -2.985  13.047   8.136  1.00 11.05           C  \\nATOM    605  CG2 VAL A 118      -1.988  11.200   6.748  1.00 13.36           C  \\nATOM    606  N   ILE A 119      -6.281  11.906   8.900  1.00 11.69           N  \\nATOM    607  CA  ILE A 119      -7.596  12.542   8.958  1.00 13.03           C  \\nATOM    608  C   ILE A 119      -7.516  14.033   9.240  1.00 13.56           C  \\nATOM    609  O   ILE A 119      -6.592  14.518   9.878  1.00 12.51           O  \\nATOM    610  CB  ILE A 119      -8.457  11.878  10.048  1.00 20.00           C  \\nATOM    611  CG1 ILE A 119      -7.798  12.036  11.419  1.00 20.00           C  \\nATOM    612  CG2 ILE A 119      -8.684  10.409   9.727  1.00 20.00           C  \\nATOM    613  CD1 ILE A 119      -8.676  11.604  12.572  1.00 20.00           C  \\nATOM    614  N   ASN A 120      -8.547  14.715   8.870  1.00 13.27           N  \\nATOM    615  CA  ASN A 120      -8.954  15.910   9.584  1.00 13.98           C  \\nATOM    616  C   ASN A 120      -9.956  15.532  10.654  1.00 13.51           C  \\nATOM    617  O   ASN A 120     -10.705  14.558  10.504  1.00 14.30           O  \\nATOM    618  CB  ASN A 120      -9.574  16.910   8.606  1.00 13.30           C  \\nATOM    619  CG  ASN A 120      -8.529  17.647   7.811  1.00 17.23           C  \\nATOM    620  OD1 ASN A 120      -8.599  17.715   6.589  1.00 24.44           O  \\nATOM    621  ND2 ASN A 120      -7.624  18.305   8.515  1.00 18.36           N  \\nATOM    622  N   THR A 121      -9.922  16.240  11.773  1.00 13.06           N  \\nATOM    623  CA  THR A 121     -10.885  15.987  12.836  1.00 14.61           C  \\nATOM    624  C   THR A 121     -12.155  16.796  12.618  1.00 16.03           C  \\nATOM    625  O   THR A 121     -12.199  17.699  11.774  1.00 16.42           O  \\nATOM    626  CB  THR A 121     -10.288  16.271  14.220  1.00 14.00           C  \\nATOM    627  OG1 THR A 121      -9.789  17.609  14.261  1.00 14.06           O  \\nATOM    628  CG2 THR A 121      -9.113  15.329  14.494  1.00 15.45           C  \\nATOM    629  N   ASN A 122     -13.227  16.395  13.287  1.00 15.99           N  \\nATOM    630  CA  ASN A 122     -14.491  17.120  13.155  1.00 14.68           C  \\nATOM    631  C   ASN A 122     -14.331  18.583  13.545  1.00 18.22           C  \\nATOM    632  O   ASN A 122     -14.839  19.493  12.865  1.00 19.00           O  \\nATOM    633  CB  ASN A 122     -15.549  16.467  14.046  1.00 15.08           C  \\nATOM    634  CG  ASN A 122     -15.965  15.111  13.549  1.00 16.28           C  \\nATOM    635  OD1 ASN A 122     -15.920  14.828  12.355  1.00 18.63           O  \\nATOM    636  ND2 ASN A 122     -16.220  14.208  14.493  1.00 22.03           N  \\nATOM    637  N   SER A 123     -13.550  18.811  14.592  1.00 13.70           N  \\nATOM    638  CA  SER A 123     -13.148  20.158  14.965  1.00 14.87           C  \\nATOM    639  C   SER A 123     -11.670  20.165  15.311  1.00 15.04           C  \\nATOM    640  O   SER A 123     -11.168  19.233  15.960  1.00 14.47           O  \\nATOM    641  CB  SER A 123     -13.957  20.644  16.167  1.00 15.91           C  \\nATOM    642  OG  SER A 123     -13.371  21.817  16.710  1.00 18.92           O  \\nATOM    643  N   ARG A 124     -10.973  21.226  14.905  1.00 13.69           N  \\nATOM    644  CA  ARG A 124      -9.581  21.394  15.283  1.00 14.01           C  \\nATOM    645  C   ARG A 124      -9.376  21.522  16.790  1.00 11.91           C  \\nATOM    646  O   ARG A 124      -8.248  21.446  17.279  1.00 14.50           O  \\nATOM    647  CB  ARG A 124      -8.955  22.590  14.553  1.00 15.50           C  \\nATOM    648  CG  ARG A 124      -8.579  22.253  13.111  1.00 15.31           C  \\nATOM    649  CD  ARG A 124      -7.952  23.457  12.401  1.00 14.64           C  \\nATOM    650  NE  ARG A 124      -8.973  24.467  12.169  1.00 15.06           N  \\nATOM    651  CZ  ARG A 124      -9.810  24.460  11.140  1.00 14.77           C  \\nATOM    652  NH1 ARG A 124      -9.608  23.616  10.132  1.00 16.20           N  \\nATOM    653  NH2 ARG A 124     -10.823  25.328  11.096  1.00 14.26           N  \\nATOM    654  N   PHE A 125     -10.467  21.752  17.524  1.00 14.33           N  \\nATOM    655  CA  PHE A 125     -10.376  21.845  18.974  1.00 15.86           C  \\nATOM    656  C   PHE A 125     -10.749  20.551  19.700  1.00 15.21           C  \\nATOM    657  O   PHE A 125     -10.805  20.531  20.935  1.00 15.13           O  \\nATOM    658  CB  PHE A 125     -11.221  23.011  19.509  1.00 16.30           C  \\nATOM    659  CG  PHE A 125     -10.699  24.378  19.124  1.00 16.75           C  \\nATOM    660  CD1 PHE A 125      -9.347  24.594  18.916  1.00 26.21           C  \\nATOM    661  CD2 PHE A 125     -11.558  25.460  19.059  1.00 19.27           C  \\nATOM    662  CE1 PHE A 125      -8.872  25.859  18.597  1.00 25.54           C  \\nATOM    663  CE2 PHE A 125     -11.082  26.724  18.755  1.00 19.47           C  \\nATOM    664  CZ  PHE A 125      -9.743  26.921  18.519  1.00 23.96           C  \\nATOM    665  N   ASP A 126     -10.928  19.458  18.955  1.00 15.39           N  \\nATOM    666  CA  ASP A 126     -11.254  18.182  19.599  1.00 14.69           C  \\nATOM    667  C   ASP A 126     -10.090  17.735  20.462  1.00 14.50           C  \\nATOM    668  O   ASP A 126      -8.915  17.940  20.118  1.00 14.62           O  \\nATOM    669  CB  ASP A 126     -11.543  17.094  18.566  1.00 14.69           C  \\nATOM    670  CG  ASP A 126     -12.832  17.327  17.796  1.00 16.50           C  \\nATOM    671  OD1 ASP A 126     -13.773  17.944  18.350  1.00 18.15           O  \\nATOM    672  OD2 ASP A 126     -12.959  16.765  16.686  1.00 14.57           O  \\nATOM    673  N   TYR A 127     -10.418  17.098  21.579  1.00 14.23           N  \\nATOM    674  CA  TYR A 127      -9.404  16.652  22.522  1.00 13.56           C  \\nATOM    675  C   TYR A 127      -8.974  15.215  22.236  1.00 13.57           C  \\nATOM    676  O   TYR A 127      -9.486  14.581  21.312  1.00 13.61           O  \\nATOM    677  CB  TYR A 127      -9.912  16.816  23.963  1.00 15.48           C  \\nATOM    678  CG  TYR A 127      -9.466  18.122  24.592  1.00 12.65           C  \\nATOM    679  CD1 TYR A 127      -8.756  18.126  25.792  1.00 14.60           C  \\nATOM    680  CD2 TYR A 127      -9.549  19.318  23.884  1.00 14.75           C  \\nATOM    681  CE1 TYR A 127      -8.289  19.301  26.341  1.00 15.34           C  \\nATOM    682  CE2 TYR A 127      -9.072  20.501  24.424  1.00 16.31           C  \\nATOM    683  CZ  TYR A 127      -8.492  20.490  25.681  1.00 17.88           C  \\nATOM    684  OH  TYR A 127      -7.974  21.655  26.191  1.00 19.70           O  \\nATOM    685  N   PRO A 128      -7.960  14.728  22.965  1.00 13.95           N  \\nATOM    686  CA  PRO A 128      -7.362  13.444  22.597  1.00 13.68           C  \\nATOM    687  C   PRO A 128      -8.367  12.291  22.507  1.00 14.18           C  \\nATOM    688  O   PRO A 128      -8.249  11.446  21.618  1.00 13.72           O  \\nATOM    689  CB  PRO A 128      -6.361  13.203  23.727  1.00 12.83           C  \\nATOM    690  CG  PRO A 128      -5.918  14.605  24.096  1.00 14.10           C  \\nATOM    691  CD  PRO A 128      -7.203  15.402  24.035  1.00 13.55           C  \\nATOM    692  N   ASP A 129      -9.248  12.160  23.494  1.00 14.73           N  \\nATOM    693  CA  ASP A 129     -10.125  11.001  23.494  1.00 14.14           C  \\nATOM    694  C   ASP A 129     -11.127  11.036  22.342  1.00 13.98           C  \\nATOM    695  O   ASP A 129     -11.456   9.995  21.765  1.00 14.25           O  \\nATOM    696  CB  ASP A 129     -10.846  10.849  24.838  1.00 14.49           C  \\nATOM    697  CG  ASP A 129      -9.909  10.438  25.959  1.00 16.16           C  \\nATOM    698  OD1 ASP A 129      -8.743  10.108  25.677  1.00 16.88           O  \\nATOM    699  OD2 ASP A 129     -10.316  10.554  27.131  1.00 21.95           O  \\nATOM    700  N   SER A 130     -11.502  12.246  21.911  1.00 14.52           N  \\nATOM    701  CA  SER A 130     -12.308  12.408  20.699  1.00 12.47           C  \\nATOM    702  C   SER A 130     -11.503  11.974  19.479  1.00 12.90           C  \\nATOM    703  O   SER A 130     -12.009  11.290  18.574  1.00 13.86           O  \\nATOM    704  CB  SER A 130     -12.745  13.870  20.569  1.00 13.99           C  \\nATOM    705  OG  SER A 130     -13.368  14.098  19.325  1.00 14.74           O  \\nATOM    706  N   ARG A 131     -10.237  12.351  19.462  1.00 12.80           N  \\nATOM    707  CA  ARG A 131      -9.400  11.952  18.330  1.00 12.68           C  \\nATOM    708  C   ARG A 131      -9.234  10.438  18.278  1.00 11.88           C  \\nATOM    709  O   ARG A 131      -9.205   9.890  17.259  1.00 13.93           O  \\nATOM    710  CB  ARG A 131      -8.027  12.675  18.355  1.00 20.00           C  \\nATOM    711  CG  ARG A 131      -8.128  14.207  18.219  1.00 20.00           C  \\nATOM    712  CD  ARG A 131      -6.878  14.969  18.570  1.00 20.00           C  \\nATOM    713  NE  ARG A 131      -7.254  16.373  18.586  1.00 20.00           N  \\nATOM    714  CZ  ARG A 131      -7.194  17.198  17.557  1.00 20.00           C  \\nATOM    715  NH1 ARG A 131      -6.636  16.828  16.424  1.00 20.00           N  \\nATOM    716  NH2 ARG A 131      -7.706  18.380  17.661  1.00 20.00           N  \\nATOM    717  N   ALA A 132      -9.139   9.816  19.435  1.00 13.10           N  \\nATOM    718  CA  ALA A 132      -9.096   8.354  19.464  1.00 12.48           C  \\nATOM    719  C   ALA A 132     -10.301   7.756  18.744  1.00 12.12           C  \\nATOM    720  O   ALA A 132     -10.160   6.834  17.935  1.00 13.38           O  \\nATOM    721  CB  ALA A 132      -9.020   7.864  20.900  1.00 13.65           C  \\nATOM    722  N   SER A 133     -11.492   8.275  19.045  1.00 13.55           N  \\nATOM    723  CA  SER A 133     -12.707   7.736  18.461  1.00 13.42           C  \\nATOM    724  C   SER A 133     -12.792   8.033  16.965  1.00 12.03           C  \\nATOM    725  O   SER A 133     -13.311   7.239  16.175  1.00 13.47           O  \\nATOM    726  CB  SER A 133     -13.942   8.242  19.212  1.00 15.56           C  \\nATOM    727  OG  SER A 133     -14.268   9.560  18.818  1.00 26.41           O  \\nATOM    728  N   GLN A 134     -12.164   9.128  16.550  1.00 14.13           N  \\nATOM    729  CA  GLN A 134     -12.127   9.466  15.135  1.00 12.69           C  \\nATOM    730  C   GLN A 134     -11.124   8.603  14.359  1.00 13.84           C  \\nATOM    731  O   GLN A 134     -11.373   8.230  13.207  1.00 13.15           O  \\nATOM    732  CB  GLN A 134     -11.819  10.953  14.966  1.00 12.23           C  \\nATOM    733  CG  GLN A 134     -12.940  11.819  15.530  1.00 11.42           C  \\nATOM    734  CD  GLN A 134     -12.699  13.288  15.327  1.00 14.07           C  \\nATOM    735  OE1 GLN A 134     -12.679  13.776  14.188  1.00 15.21           O  \\nATOM    736  NE2 GLN A 134     -12.650  14.025  16.430  1.00 14.40           N  \\nATOM    737  N   LEU A 135     -10.021   8.246  15.017  1.00 12.65           N  \\nATOM    738  CA  LEU A 135      -9.122   7.233  14.466  1.00 12.79           C  \\nATOM    739  C   LEU A 135      -9.827   5.907  14.259  1.00 12.73           C  \\nATOM    740  O   LEU A 135      -9.640   5.259  13.235  1.00 13.59           O  \\nATOM    741  CB  LEU A 135      -7.883   7.034  15.345  1.00 12.15           C  \\nATOM    742  CG  LEU A 135      -6.876   8.183  15.272  1.00 13.55           C  \\nATOM    743  CD1 LEU A 135      -5.853   8.035  16.383  1.00 14.10           C  \\nATOM    744  CD2 LEU A 135      -6.191   8.262  13.900  1.00 15.07           C  \\nATOM    745  N   SER A 136     -10.689   5.541  15.204  1.00 14.33           N  \\nATOM    746  CA  SER A 136     -11.452   4.293  15.064  1.00 14.43           C  \\nATOM    747  C   SER A 136     -12.409   4.351  13.887  1.00 14.52           C  \\nATOM    748  O   SER A 136     -12.523   3.391  13.119  1.00 15.58           O  \\nATOM    749  CB  SER A 136     -12.240   3.994  16.335  1.00 17.16           C  \\nATOM    750  OG  SER A 136     -11.375   3.581  17.366  1.00 22.03           O  \\nATOM    751  N   ALA A 137     -13.090   5.489  13.740  1.00 14.22           N  \\nATOM    752  CA  ALA A 137     -13.984   5.687  12.609  1.00 14.61           C  \\nATOM    753  C   ALA A 137     -13.213   5.656  11.304  1.00 15.45           C  \\nATOM    754  O   ALA A 137     -13.694   5.136  10.300  1.00 15.83           O  \\nATOM    755  CB  ALA A 137     -14.727   6.985  12.739  1.00 14.20           C  \\nATOM    756  N   ALA A 138     -12.024   6.257  11.308  1.00 14.30           N  \\nATOM    757  CA  ALA A 138     -11.189   6.262  10.114  1.00 15.09           C  \\nATOM    758  C   ALA A 138     -10.782   4.857   9.683  1.00 13.51           C  \\nATOM    759  O   ALA A 138     -10.753   4.549   8.482  1.00 16.44           O  \\nATOM    760  CB  ALA A 138      -9.957   7.137  10.330  1.00 15.02           C  \\nATOM    761  N   LEU A 139     -10.416   4.022  10.655  1.00 14.13           N  \\nATOM    762  CA  LEU A 139     -10.069   2.640  10.345  1.00 14.83           C  \\nATOM    763  C   LEU A 139     -11.275   1.859   9.814  1.00 13.85           C  \\nATOM    764  O   LEU A 139     -11.152   1.121   8.845  1.00 15.47           O  \\nATOM    765  CB  LEU A 139      -9.438   1.946  11.554  1.00 14.69           C  \\nATOM    766  CG  LEU A 139      -8.052   2.479  11.918  1.00 15.63           C  \\nATOM    767  CD1 LEU A 139      -7.632   1.973  13.283  1.00 18.11           C  \\nATOM    768  CD2 LEU A 139      -7.018   2.099  10.851  1.00 17.07           C  \\nATOM    769  N   ASN A 140     -12.462   2.129  10.349  1.00 16.01           N  \\nATOM    770  CA  ASN A 140     -13.652   1.470   9.827  1.00 14.58           C  \\nATOM    771  C   ASN A 140     -13.957   1.902   8.400  1.00 15.71           C  \\nATOM    772  O   ASN A 140     -14.270   1.078   7.541  1.00 16.46           O  \\nATOM    773  CB  ASN A 140     -14.855   1.720  10.738  1.00 14.23           C  \\nATOM    774  CG  ASN A 140     -16.161   1.167  10.153  1.00 22.64           C  \\nATOM    775  OD1 ASN A 140     -16.422  -0.038  10.228  1.00 26.02           O  \\nATOM    776  ND2 ASN A 140     -16.973   2.041   9.557  1.00 20.37           N  \\nATOM    777  N   TYR A 141     -13.737   3.182   8.117  1.00 15.96           N  \\nATOM    778  CA  TYR A 141     -13.940   3.692   6.771  1.00 14.82           C  \\nATOM    779  C   TYR A 141     -12.957   3.034   5.802  1.00 17.29           C  \\nATOM    780  O   TYR A 141     -13.316   2.664   4.681  1.00 17.03           O  \\nATOM    781  CB  TYR A 141     -13.757   5.214   6.747  1.00 16.00           C  \\nATOM    782  CG  TYR A 141     -13.582   5.762   5.351  1.00 17.47           C  \\nATOM    783  CD1 TYR A 141     -14.677   6.118   4.580  1.00 18.37           C  \\nATOM    784  CD2 TYR A 141     -12.316   5.950   4.812  1.00 17.82           C  \\nATOM    785  CE1 TYR A 141     -14.523   6.487   3.255  1.00 19.36           C  \\nATOM    786  CE2 TYR A 141     -12.150   6.382   3.506  1.00 24.18           C  \\nATOM    787  CZ  TYR A 141     -13.259   6.634   2.730  1.00 24.53           C  \\nATOM    788  OH  TYR A 141     -13.087   7.073   1.431  1.00 26.69           O  \\nATOM    789  N   LEU A 142     -11.686   3.006   6.192  1.00 15.66           N  \\nATOM    790  CA  LEU A 142     -10.656   2.445   5.324  1.00 18.65           C  \\nATOM    791  C   LEU A 142     -11.017   1.017   4.921  1.00 16.84           C  \\nATOM    792  O   LEU A 142     -10.841   0.629   3.762  1.00 19.94           O  \\nATOM    793  CB  LEU A 142      -9.310   2.469   6.046  1.00 23.08           C  \\nATOM    794  CG  LEU A 142      -8.016   2.552   5.247  1.00 24.31           C  \\nATOM    795  CD1 LEU A 142      -8.010   3.700   4.238  1.00 16.34           C  \\nATOM    796  CD2 LEU A 142      -6.839   2.662   6.216  1.00 17.79           C  \\nATOM    797  N   ARG A 143     -11.495   0.237   5.893  1.00 16.11           N  \\nATOM    798  CA  ARG A 143     -11.838  -1.173   5.688  1.00 18.76           C  \\nATOM    799  C   ARG A 143     -13.082  -1.340   4.816  1.00 19.13           C  \\nATOM    800  O   ARG A 143     -13.232  -2.340   4.111  1.00 20.71           O  \\nATOM    801  CB  ARG A 143     -12.052  -1.850   7.046  1.00 19.07           C  \\nATOM    802  CG  ARG A 143     -12.406  -3.327   6.993  1.00 29.71           C  \\nATOM    803  CD  ARG A 143     -12.399  -3.933   8.397  1.00 34.22           C  \\nATOM    804  NE  ARG A 143     -13.122  -3.099   9.359  1.00 39.71           N  \\nATOM    805  CZ  ARG A 143     -14.441  -3.122   9.548  1.00 37.84           C  \\nATOM    806  NH1 ARG A 143     -15.196  -4.052   8.964  1.00 32.84           N  \\nATOM    807  NH2 ARG A 143     -15.001  -2.247  10.374  1.00 25.97           N  \\nATOM    808  N   THR A 144     -14.033  -0.423   4.954  1.00 16.98           N  \\nATOM    809  CA  THR A 144     -15.378  -0.691   4.439  1.00 17.04           C  \\nATOM    810  C   THR A 144     -15.789   0.182   3.277  1.00 17.15           C  \\nATOM    811  O   THR A 144     -16.729  -0.159   2.563  1.00 17.98           O  \\nATOM    812  CB  THR A 144     -16.469  -0.599   5.525  1.00 15.59           C  \\nATOM    813  OG1 THR A 144     -16.533   0.737   6.052  1.00 16.30           O  \\nATOM    814  CG2 THR A 144     -16.206  -1.583   6.636  1.00 15.23           C  \\nATOM    815  N   SER A 145     -15.209   1.372   3.166  1.00 17.57           N  \\nATOM    816  CA  SER A 145     -15.802   2.401   2.322  1.00 17.97           C  \\nATOM    817  C   SER A 145     -14.822   3.086   1.378  1.00 16.25           C  \\nATOM    818  O   SER A 145     -15.181   4.029   0.684  1.00 19.12           O  \\nATOM    819  CB  SER A 145     -16.514   3.449   3.176  1.00 19.70           C  \\nATOM    820  OG  SER A 145     -17.577   2.845   3.888  1.00 29.31           O  \\nATOM    821  N   SER A 146     -13.557   2.709   1.459  1.00 19.20           N  \\nATOM    822  CA  SER A 146     -12.554   3.370   0.633  1.00 19.28           C  \\nATOM    823  C   SER A 146     -12.731   2.955  -0.826  1.00 16.78           C  \\nATOM    824  O   SER A 146     -13.337   1.921  -1.109  1.00 18.38           O  \\nATOM    825  CB  SER A 146     -11.152   3.028   1.135  1.00 20.22           C  \\nATOM    826  OG  SER A 146     -10.996   1.622   1.248  1.00 21.93           O  \\nATOM    827  N   PRO A 147     -12.174   3.747  -1.755  1.00 15.77           N  \\nATOM    828  CA  PRO A 147     -12.257   3.449  -3.180  1.00 15.72           C  \\nATOM    829  C   PRO A 147     -11.637   2.106  -3.537  1.00 16.28           C  \\nATOM    830  O   PRO A 147     -10.729   1.634  -2.839  1.00 17.53           O  \\nATOM    831  CB  PRO A 147     -11.458   4.583  -3.834  1.00 18.07           C  \\nATOM    832  CG  PRO A 147     -11.342   5.650  -2.802  1.00 22.55           C  \\nATOM    833  CD  PRO A 147     -11.417   4.978  -1.474  1.00 17.41           C  \\nATOM    834  N   SER A 148     -11.997   1.586  -4.708  1.00 17.26           N  \\nATOM    835  CA  SER A 148     -11.563   0.250  -5.097  1.00 17.13           C  \\nATOM    836  C   SER A 148     -10.043   0.121  -5.159  1.00 16.48           C  \\nATOM    837  O   SER A 148      -9.491  -0.910  -4.797  1.00 18.72           O  \\nATOM    838  CB  SER A 148     -12.183  -0.178  -6.432  1.00 19.26           C  \\nATOM    839  OG  SER A 148     -11.850   0.728  -7.480  1.00 22.11           O  \\nATOM    840  N   ALA A 149      -9.367   1.169  -5.620  1.00 18.59           N  \\nATOM    841  CA  ALA A 149      -7.917   1.099  -5.773  1.00 17.38           C  \\nATOM    842  C   ALA A 149      -7.233   1.002  -4.418  1.00 18.27           C  \\nATOM    843  O   ALA A 149      -6.180   0.379  -4.289  1.00 18.73           O  \\nATOM    844  CB  ALA A 149      -7.401   2.307  -6.536  1.00 18.96           C  \\nATOM    845  N   VAL A 150      -7.785   1.700  -3.433  1.00 15.91           N  \\nATOM    846  CA  VAL A 150      -7.278   1.627  -2.070  1.00 15.03           C  \\nATOM    847  C   VAL A 150      -7.556   0.261  -1.445  1.00 16.25           C  \\nATOM    848  O   VAL A 150      -6.651  -0.385  -0.904  1.00 16.25           O  \\nATOM    849  CB  VAL A 150      -7.899   2.737  -1.194  1.00 15.49           C  \\nATOM    850  CG1 VAL A 150      -7.484   2.567   0.259  1.00 16.92           C  \\nATOM    851  CG2 VAL A 150      -7.518   4.122  -1.728  1.00 16.04           C  \\nATOM    852  N   ARG A 151      -8.801  -0.199  -1.546  1.00 15.36           N  \\nATOM    853  CA  ARG A 151      -9.172  -1.480  -0.970  1.00 18.31           C  \\nATOM    854  C   ARG A 151      -8.303  -2.625  -1.475  1.00 15.62           C  \\nATOM    855  O   ARG A 151      -7.926  -3.517  -0.710  1.00 16.97           O  \\nATOM    856  CB  ARG A 151     -10.640  -1.779  -1.239  1.00 21.22           C  \\nATOM    857  CG  ARG A 151     -11.276  -2.666  -0.171  1.00 29.31           C  \\nATOM    858  CD  ARG A 151     -11.732  -1.849   1.037  1.00 30.34           C  \\nATOM    859  NE  ARG A 151     -12.718  -0.836   0.658  1.00 32.36           N  \\nATOM    860  CZ  ARG A 151     -14.021  -1.072   0.513  1.00 34.73           C  \\nATOM    861  NH1 ARG A 151     -14.534  -2.244   0.879  1.00 26.48           N  \\nATOM    862  NH2 ARG A 151     -14.819  -0.118   0.045  1.00 27.93           N  \\nATOM    863  N   ALA A 152      -7.963  -2.586  -2.759  1.00 16.45           N  \\nATOM    864  CA  ALA A 152      -7.136  -3.628  -3.367  1.00 15.18           C  \\nATOM    865  C   ALA A 152      -5.725  -3.680  -2.788  1.00 14.72           C  \\nATOM    866  O   ALA A 152      -5.082  -4.721  -2.850  1.00 16.19           O  \\nATOM    867  CB  ALA A 152      -7.077  -3.452  -4.876  1.00 18.14           C  \\nATOM    868  N   ARG A 153      -5.258  -2.563  -2.224  1.00 14.62           N  \\nATOM    869  CA  ARG A 153      -3.900  -2.483  -1.664  1.00 13.94           C  \\nATOM    870  C   ARG A 153      -3.871  -2.757  -0.165  1.00 13.24           C  \\nATOM    871  O   ARG A 153      -2.800  -2.903   0.408  1.00 14.34           O  \\nATOM    872  CB  ARG A 153      -3.242  -1.125  -1.958  1.00 15.25           C  \\nATOM    873  CG  ARG A 153      -3.074  -0.815  -3.440  1.00 13.78           C  \\nATOM    874  CD  ARG A 153      -1.887  -1.541  -4.089  1.00 16.55           C  \\nATOM    875  NE  ARG A 153      -2.092  -2.983  -4.191  1.00 16.57           N  \\nATOM    876  CZ  ARG A 153      -2.836  -3.580  -5.114  1.00 17.03           C  \\nATOM    877  NH1 ARG A 153      -2.899  -4.908  -5.147  1.00 18.90           N  \\nATOM    878  NH2 ARG A 153      -3.512  -2.863  -6.003  1.00 18.88           N  \\nATOM    879  N   LEU A 154      -5.046  -2.758   0.465  1.00 13.08           N  \\nATOM    880  CA  LEU A 154      -5.157  -2.832   1.915  1.00 14.27           C  \\nATOM    881  C   LEU A 154      -5.346  -4.277   2.382  1.00 13.24           C  \\nATOM    882  O   LEU A 154      -6.290  -4.960   1.969  1.00 14.44           O  \\nATOM    883  CB  LEU A 154      -6.348  -2.004   2.411  1.00 13.92           C  \\nATOM    884  CG  LEU A 154      -6.576  -2.028   3.930  1.00 13.47           C  \\nATOM    885  CD1 LEU A 154      -5.385  -1.442   4.687  1.00 16.90           C  \\nATOM    886  CD2 LEU A 154      -7.866  -1.269   4.282  1.00 15.44           C  \\nATOM    887  N   ASP A 155      -4.478  -4.708   3.292  1.00 13.98           N  \\nATOM    888  CA  ASP A 155      -4.752  -5.884   4.121  1.00 11.93           C  \\nATOM    889  C   ASP A 155      -5.319  -5.424   5.446  1.00 12.32           C  \\nATOM    890  O   ASP A 155      -4.591  -4.972   6.319  1.00 13.05           O  \\nATOM    891  CB  ASP A 155      -3.470  -6.676   4.369  1.00 13.32           C  \\nATOM    892  CG  ASP A 155      -3.708  -7.914   5.207  1.00 13.35           C  \\nATOM    893  OD1 ASP A 155      -4.864  -8.124   5.649  1.00 16.69           O  \\nATOM    894  OD2 ASP A 155      -2.755  -8.695   5.393  1.00 13.23           O  \\nATOM    895  N   ALA A 156      -6.641  -5.436   5.540  1.00 12.99           N  \\nATOM    896  CA  ALA A 156      -7.323  -4.837   6.676  1.00 13.40           C  \\nATOM    897  C   ALA A 156      -7.136  -5.636   7.961  1.00 13.73           C  \\nATOM    898  O   ALA A 156      -7.510  -5.180   9.032  1.00 14.92           O  \\nATOM    899  CB  ALA A 156      -8.809  -4.643   6.369  1.00 18.44           C  \\nATOM    900  N   ASN A 157      -6.508  -6.804   7.859  1.00 13.64           N  \\nATOM    901  CA  ASN A 157      -6.127  -7.564   9.044  1.00 14.44           C  \\nATOM    902  C   ASN A 157      -5.045  -6.871   9.853  1.00 12.63           C  \\nATOM    903  O   ASN A 157      -4.849  -7.188  11.024  1.00 14.01           O  \\nATOM    904  CB  ASN A 157      -5.637  -8.975   8.674  1.00 11.92           C  \\nATOM    905  CG  ASN A 157      -6.736  -9.854   8.131  1.00 16.18           C  \\nATOM    906  OD1 ASN A 157      -7.910  -9.651   8.433  1.00 17.50           O  \\nATOM    907  ND2 ASN A 157      -6.371 -10.768   7.247  1.00 18.22           N  \\nATOM    908  N   ARG A 158      -4.279  -5.991   9.204  1.00 12.66           N  \\nATOM    909  CA  ARG A 158      -3.058  -5.481   9.812  1.00 14.21           C  \\nATOM    910  C   ARG A 158      -3.067  -3.960   9.874  1.00 12.59           C  \\nATOM    911  O   ARG A 158      -2.816  -3.296   8.874  1.00 12.19           O  \\nATOM    912  CB  ARG A 158      -1.821  -6.005   9.064  1.00 11.96           C  \\nATOM    913  CG  ARG A 158      -1.798  -7.551   8.951  1.00 13.80           C  \\nATOM    914  CD  ARG A 158      -0.512  -8.046   8.289  1.00 14.38           C  \\nATOM    915  NE  ARG A 158      -0.477  -7.717   6.867  1.00 13.11           N  \\nATOM    916  CZ  ARG A 158       0.489  -7.012   6.269  1.00 11.25           C  \\nATOM    917  NH1 ARG A 158       1.386  -6.361   7.000  1.00 11.99           N  \\nATOM    918  NH2 ARG A 158       0.480  -6.856   4.949  1.00 14.26           N  \\nATOM    919  N   LEU A 159      -3.464  -3.418  11.026  1.00 12.41           N  \\nATOM    920  CA  LEU A 159      -3.770  -1.995  11.147  1.00 11.80           C  \\nATOM    921  C   LEU A 159      -2.992  -1.389  12.298  1.00 11.73           C  \\nATOM    922  O   LEU A 159      -2.703  -2.056  13.282  1.00 11.47           O  \\nATOM    923  CB  LEU A 159      -5.248  -1.797  11.437  1.00 11.26           C  \\nATOM    924  CG  LEU A 159      -6.206  -2.330  10.376  1.00 12.73           C  \\nATOM    925  CD1 LEU A 159      -7.656  -2.139  10.844  1.00 16.17           C  \\nATOM    926  CD2 LEU A 159      -5.996  -1.607   9.064  1.00 14.52           C  \\nATOM    927  N   ALA A 160      -2.677  -0.106  12.174  1.00 11.56           N  \\nATOM    928  CA  ALA A 160      -1.911   0.587  13.198  1.00 11.53           C  \\nATOM    929  C   ALA A 160      -2.293   2.066  13.203  1.00 12.83           C  \\nATOM    930  O   ALA A 160      -3.076   2.515  12.362  1.00 12.23           O  \\nATOM    931  CB  ALA A 160      -0.412   0.426  12.951  1.00 10.53           C  \\nATOM    932  N   VAL A 161      -1.857   2.775  14.235  1.00 11.91           N  \\nATOM    933  CA  VAL A 161      -2.224   4.184  14.415  1.00 11.14           C  \\nATOM    934  C   VAL A 161      -1.022   5.056  14.758  1.00 12.94           C  \\nATOM    935  O   VAL A 161      -0.027   4.577  15.289  1.00 11.68           O  \\nATOM    936  CB  VAL A 161      -3.296   4.352  15.518  1.00 12.71           C  \\nATOM    937  CG1 VAL A 161      -4.618   3.784  15.061  1.00 12.35           C  \\nATOM    938  CG2 VAL A 161      -2.851   3.680  16.806  1.00 14.76           C  \\nATOM    939  N   ALA A 162      -1.108   6.331  14.384  1.00 12.47           N  \\nATOM    940  CA  ALA A 162      -0.110   7.325  14.764  1.00 12.07           C  \\nATOM    941  C   ALA A 162      -0.790   8.672  14.880  1.00 11.38           C  \\nATOM    942  O   ALA A 162      -1.915   8.847  14.417  1.00 12.12           O  \\nATOM    943  CB  ALA A 162       1.002   7.397  13.709  1.00 14.51           C  \\nATOM    944  N   GLY A 163      -0.127   9.624  15.515  1.00 11.62           N  \\nATOM    945  CA  GLY A 163      -0.733  10.935  15.636  1.00 12.21           C  \\nATOM    946  C   GLY A 163       0.190  11.929  16.278  1.00 11.41           C  \\nATOM    947  O   GLY A 163       1.077  11.575  17.051  1.00 13.03           O  \\nATOM    948  N   HIS A 164      -0.097  13.201  16.025  1.00 11.36           N  \\nATOM    949  CA  HIS A 164       0.718  14.300  16.523  1.00 12.72           C  \\nATOM    950  C   HIS A 164       0.009  14.968  17.703  1.00 13.27           C  \\nATOM    951  O   HIS A 164      -1.169  15.323  17.603  1.00 12.97           O  \\nATOM    952  CB  HIS A 164       0.971  15.295  15.391  1.00 10.52           C  \\nATOM    953  CG  HIS A 164       1.872  16.420  15.782  1.00 11.24           C  \\nATOM    954  ND1 HIS A 164       1.730  17.699  15.288  1.00 13.00           N  \\nATOM    955  CD2 HIS A 164       2.934  16.454  16.620  1.00 11.81           C  \\nATOM    956  CE1 HIS A 164       2.669  18.473  15.806  1.00 14.28           C  \\nATOM    957  NE2 HIS A 164       3.383  17.753  16.655  1.00 12.06           N  \\nATOM    958  N   SER A 165       0.737  15.142  18.799  1.00 10.55           N  \\nATOM    959  CA  SER A 165       0.351  16.013  19.900  1.00 13.84           C  \\nATOM    960  C   SER A 165      -0.981  15.476  20.427  1.00 14.25           C  \\nATOM    961  O   SER A 165      -1.052  14.302  20.747  1.00 13.30           O  \\nATOM    962  CB  SER A 165       0.287  17.472  19.470  1.00 13.92           C  \\nATOM    963  OG  SER A 165       0.528  18.311  20.600  1.00 16.90           O  \\nATOM    964  N   MET A 166      -2.020  16.280  20.563  1.00 12.97           N  \\nATOM    965  CA  MET A 166      -3.285  15.696  21.024  1.00 13.40           C  \\nATOM    966  C   MET A 166      -3.698  14.467  20.198  1.00 13.52           C  \\nATOM    967  O   MET A 166      -4.370  13.568  20.704  1.00 12.91           O  \\nATOM    968  CB  MET A 166      -4.416  16.726  21.048  1.00 15.71           C  \\nATOM    969  CG  MET A 166      -4.331  17.697  22.204  1.00 12.94           C  \\nATOM    970  SD  MET A 166      -5.884  18.589  22.360  1.00 14.49           S  \\nATOM    971  CE  MET A 166      -5.391  19.863  23.515  1.00 15.06           C  \\nATOM    972  N   GLY A 167      -3.328  14.443  18.919  1.00 12.87           N  \\nATOM    973  CA  GLY A 167      -3.521  13.238  18.110  1.00 14.82           C  \\nATOM    974  C   GLY A 167      -2.684  12.074  18.617  1.00 13.25           C  \\nATOM    975  O   GLY A 167      -3.054  10.903  18.442  1.00 13.52           O  \\nATOM    976  N   GLY A 168      -1.532  12.379  19.204  1.00 13.42           N  \\nATOM    977  CA  GLY A 168      -0.711  11.333  19.810  1.00 13.62           C  \\nATOM    978  C   GLY A 168      -1.322  10.856  21.111  1.00 12.88           C  \\nATOM    979  O   GLY A 168      -1.247   9.665  21.446  1.00 12.75           O  \\nATOM    980  N   GLY A 169      -1.965  11.765  21.845  1.00 13.28           N  \\nATOM    981  CA  GLY A 169      -2.816  11.357  22.974  1.00 13.89           C  \\nATOM    982  C   GLY A 169      -3.947  10.441  22.544  1.00 14.71           C  \\nATOM    983  O   GLY A 169      -4.191   9.408  23.175  1.00 13.74           O  \\nATOM    984  N   GLY A 170      -4.576  10.759  21.420  1.00 13.32           N  \\nATOM    985  CA  GLY A 170      -5.589   9.883  20.829  1.00 12.66           C  \\nATOM    986  C   GLY A 170      -5.075   8.508  20.436  1.00 12.72           C  \\nATOM    987  O   GLY A 170      -5.758   7.504  20.634  1.00 13.85           O  \\nATOM    988  N   THR A 171      -3.846   8.471  19.926  1.00 11.41           N  \\nATOM    989  CA  THR A 171      -3.167   7.225  19.553  1.00 11.98           C  \\nATOM    990  C   THR A 171      -2.972   6.327  20.782  1.00 12.75           C  \\nATOM    991  O   THR A 171      -3.259   5.124  20.735  1.00 12.15           O  \\nATOM    992  CB  THR A 171      -1.803   7.551  18.906  1.00 14.92           C  \\nATOM    993  OG1 THR A 171      -2.026   8.247  17.670  1.00 14.51           O  \\nATOM    994  CG2 THR A 171      -0.999   6.286  18.648  1.00 14.28           C  \\nATOM    995  N   LEU A 172      -2.575   6.919  21.906  1.00 12.16           N  \\nATOM    996  CA  LEU A 172      -2.423   6.148  23.132  1.00 11.87           C  \\nATOM    997  C   LEU A 172      -3.760   5.604  23.638  1.00 15.63           C  \\nATOM    998  O   LEU A 172      -3.849   4.441  24.041  1.00 14.29           O  \\nATOM    999  CB  LEU A 172      -1.737   6.995  24.203  1.00 11.86           C  \\nATOM   1000  CG  LEU A 172      -0.238   7.201  23.973  1.00 13.83           C  \\nATOM   1001  CD1 LEU A 172       0.229   8.468  24.704  1.00 14.02           C  \\nATOM   1002  CD2 LEU A 172       0.509   5.982  24.477  1.00 15.52           C  \\nATOM   1003  N   ARG A 173      -4.812   6.414  23.553  1.00 14.01           N  \\nATOM   1004  CA  ARG A 173      -6.133   5.974  23.996  1.00 13.09           C  \\nATOM   1005  C   ARG A 173      -6.717   4.857  23.121  1.00 14.16           C  \\nATOM   1006  O   ARG A 173      -7.232   3.875  23.645  1.00 14.52           O  \\nATOM   1007  CB  ARG A 173      -7.097   7.164  24.073  1.00 14.07           C  \\nATOM   1008  CG  ARG A 173      -8.504   6.791  24.536  1.00 13.91           C  \\nATOM   1009  CD  ARG A 173      -8.502   6.401  26.010  1.00 16.38           C  \\nATOM   1010  NE  ARG A 173      -8.342   7.593  26.841  1.00 15.22           N  \\nATOM   1011  CZ  ARG A 173      -8.033   7.595  28.131  1.00 19.64           C  \\nATOM   1012  NH1 ARG A 173      -7.816   6.454  28.776  1.00 24.62           N  \\nATOM   1013  NH2 ARG A 173      -8.068   8.736  28.808  1.00 22.51           N  \\nATOM   1014  N   ILE A 174      -6.650   5.001  21.804  1.00 14.15           N  \\nATOM   1015  CA  ILE A 174      -7.203   3.976  20.927  1.00 13.00           C  \\nATOM   1016  C   ILE A 174      -6.419   2.677  21.077  1.00 15.24           C  \\nATOM   1017  O   ILE A 174      -7.001   1.595  21.023  1.00 16.08           O  \\nATOM   1018  CB  ILE A 174      -7.289   4.427  19.457  1.00 12.26           C  \\nATOM   1019  CG1 ILE A 174      -8.173   3.447  18.670  1.00 15.65           C  \\nATOM   1020  CG2 ILE A 174      -5.892   4.507  18.835  1.00 14.88           C  \\nATOM   1021  CD1 ILE A 174      -8.286   3.782  17.199  1.00 16.20           C  \\nATOM   1022  N   ALA A 175      -5.144   2.790  21.439  1.00 13.98           N  \\nATOM   1023  CA  ALA A 175      -4.342   1.593  21.694  1.00 15.21           C  \\nATOM   1024  C   ALA A 175      -4.877   0.822  22.898  1.00 14.90           C  \\nATOM   1025  O   ALA A 175      -4.938  -0.404  22.882  1.00 16.52           O  \\nATOM   1026  CB  ALA A 175      -2.903   1.960  21.890  1.00 15.11           C  \\nATOM   1027  N   GLU A 176      -5.322   1.540  23.922  1.00 16.94           N  \\nATOM   1028  CA  GLU A 176      -6.019   0.911  25.048  1.00 18.07           C  \\nATOM   1029  C   GLU A 176      -7.273   0.206  24.574  1.00 17.50           C  \\nATOM   1030  O   GLU A 176      -7.619  -0.871  25.056  1.00 21.18           O  \\nATOM   1031  CB  GLU A 176      -6.459   1.981  26.028  1.00 24.57           C  \\nATOM   1032  CG  GLU A 176      -5.495   2.309  27.099  1.00 28.65           C  \\nATOM   1033  CD  GLU A 176      -6.181   3.062  28.209  1.00 21.95           C  \\nATOM   1034  OE1 GLU A 176      -7.166   3.786  27.919  1.00 19.39           O  \\nATOM   1035  OE2 GLU A 176      -5.708   2.978  29.349  1.00 19.79           O  \\nATOM   1036  N   GLN A 177      -8.056   0.927  23.783  1.00 18.41           N  \\nATOM   1037  CA  GLN A 177      -9.399   0.501  23.424  1.00 16.95           C  \\nATOM   1038  C   GLN A 177      -9.385  -0.717  22.482  1.00 19.62           C  \\nATOM   1039  O   GLN A 177     -10.361  -1.467  22.412  1.00 21.33           O  \\nATOM   1040  CB  GLN A 177     -10.155   1.672  22.781  1.00 17.76           C  \\nATOM   1041  CG  GLN A 177     -10.467   2.805  23.757  1.00 21.19           C  \\nATOM   1042  CD  GLN A 177     -10.901   4.103  23.078  1.00 23.31           C  \\nATOM   1043  OE1 GLN A 177     -10.624   4.348  21.900  1.00 23.46           O  \\nATOM   1044  NE2 GLN A 177     -11.522   4.975  23.852  1.00 23.78           N  \\nATOM   1045  N   ASN A 178      -8.324  -0.852  21.691  1.00 17.29           N  \\nATOM   1046  CA  ASN A 178      -8.311  -1.795  20.573  1.00 17.91           C  \\nATOM   1047  C   ASN A 178      -7.011  -2.587  20.567  1.00 17.22           C  \\nATOM   1048  O   ASN A 178      -6.061  -2.238  19.868  1.00 16.35           O  \\nATOM   1049  CB  ASN A 178      -8.511  -1.071  19.240  1.00 15.82           C  \\nATOM   1050  CG  ASN A 178      -8.651  -2.031  18.065  1.00 17.24           C  \\nATOM   1051  OD1 ASN A 178      -8.386  -3.234  18.197  1.00 19.78           O  \\nATOM   1052  ND2 ASN A 178      -9.041  -1.499  16.900  1.00 20.95           N  \\nATOM   1053  N   PRO A 179      -6.991  -3.711  21.296  1.00 17.57           N  \\nATOM   1054  CA  PRO A 179      -5.778  -4.512  21.431  1.00 17.25           C  \\nATOM   1055  C   PRO A 179      -5.418  -5.273  20.157  1.00 17.14           C  \\nATOM   1056  O   PRO A 179      -4.368  -5.915  20.101  1.00 18.24           O  \\nATOM   1057  CB  PRO A 179      -6.114  -5.492  22.565  1.00 21.56           C  \\nATOM   1058  CG  PRO A 179      -7.587  -5.488  22.661  1.00 20.33           C  \\nATOM   1059  CD  PRO A 179      -8.033  -4.120  22.253  1.00 18.33           C  \\nATOM   1060  N   SER A 180      -6.282  -5.217  19.146  1.00 16.20           N  \\nATOM   1061  CA  SER A 180      -5.993  -5.875  17.877  1.00 16.56           C  \\nATOM   1062  C   SER A 180      -5.032  -5.060  17.001  1.00 14.08           C  \\nATOM   1063  O   SER A 180      -4.480  -5.578  16.027  1.00 14.61           O  \\nATOM   1064  CB  SER A 180      -7.284  -6.145  17.109  1.00 16.32           C  \\nATOM   1065  OG  SER A 180      -7.769  -4.962  16.495  1.00 23.67           O  \\nATOM   1066  N   LEU A 181      -4.832  -3.789  17.339  1.00 13.92           N  \\nATOM   1067  CA  LEU A 181      -3.915  -2.954  16.564  1.00 12.79           C  \\nATOM   1068  C   LEU A 181      -2.480  -3.483  16.657  1.00 12.45           C  \\nATOM   1069  O   LEU A 181      -2.058  -4.014  17.698  1.00 12.44           O  \\nATOM   1070  CB  LEU A 181      -3.978  -1.507  17.071  1.00 11.77           C  \\nATOM   1071  CG  LEU A 181      -5.292  -0.775  16.785  1.00 13.07           C  \\nATOM   1072  CD1 LEU A 181      -5.323   0.557  17.504  1.00 16.74           C  \\nATOM   1073  CD2 LEU A 181      -5.453  -0.572  15.296  1.00 17.98           C  \\nATOM   1074  N   LYS A 182      -1.740  -3.376  15.556  1.00 11.64           N  \\nATOM   1075  CA  LYS A 182      -0.384  -3.928  15.529  1.00 12.63           C  \\nATOM   1076  C   LYS A 182       0.638  -3.012  16.198  1.00 11.88           C  \\nATOM   1077  O   LYS A 182       1.663  -3.484  16.687  1.00 13.28           O  \\nATOM   1078  CB  LYS A 182       0.078  -4.243  14.099  1.00 12.15           C  \\nATOM   1079  CG  LYS A 182      -0.842  -5.161  13.280  1.00 13.97           C  \\nATOM   1080  CD  LYS A 182      -1.142  -6.447  14.025  1.00 19.87           C  \\nATOM   1081  CE  LYS A 182       0.121  -7.244  14.270  1.00 21.99           C  \\nATOM   1082  NZ  LYS A 182      -0.198  -8.681  14.572  1.00 24.10           N  \\nATOM   1083  N   ALA A 183       0.408  -1.698  16.136  1.00 11.72           N  \\nATOM   1084  CA  ALA A 183       1.396  -0.739  16.607  1.00 12.23           C  \\nATOM   1085  C   ALA A 183       0.725   0.612  16.794  1.00 10.64           C  \\nATOM   1086  O   ALA A 183      -0.293   0.906  16.151  1.00 12.16           O  \\nATOM   1087  CB  ALA A 183       2.554  -0.619  15.610  1.00 12.50           C  \\nATOM   1088  N   ALA A 184       1.364   1.452  17.610  1.00 11.39           N  \\nATOM   1089  CA  ALA A 184       0.948   2.836  17.833  1.00 11.29           C  \\nATOM   1090  C   ALA A 184       2.193   3.704  17.915  1.00 12.54           C  \\nATOM   1091  O   ALA A 184       3.161   3.358  18.596  1.00 12.77           O  \\nATOM   1092  CB  ALA A 184       0.164   2.930  19.136  1.00 13.38           C  \\nATOM   1093  N   VAL A 185       2.172   4.833  17.208  1.00 10.19           N  \\nATOM   1094  CA  VAL A 185       3.272   5.798  17.265  1.00 10.41           C  \\nATOM   1095  C   VAL A 185       2.742   7.190  17.622  1.00 10.58           C  \\nATOM   1096  O   VAL A 185       2.349   7.968  16.743  1.00 11.35           O  \\nATOM   1097  CB  VAL A 185       4.069   5.858  15.944  1.00 10.81           C  \\nATOM   1098  CG1 VAL A 185       5.185   6.902  16.048  1.00 13.31           C  \\nATOM   1099  CG2 VAL A 185       4.651   4.466  15.614  1.00 11.15           C  \\nATOM   1100  N   PRO A 186       2.777   7.520  18.916  1.00 11.18           N  \\nATOM   1101  CA  PRO A 186       2.457   8.886  19.342  1.00 11.18           C  \\nATOM   1102  C   PRO A 186       3.630   9.829  19.112  1.00 11.41           C  \\nATOM   1103  O   PRO A 186       4.726   9.595  19.613  1.00 12.12           O  \\nATOM   1104  CB  PRO A 186       2.164   8.744  20.850  1.00 10.92           C  \\nATOM   1105  CG  PRO A 186       1.940   7.208  21.059  1.00 11.19           C  \\nATOM   1106  CD  PRO A 186       2.875   6.590  20.058  1.00 11.05           C  \\nATOM   1107  N   LEU A 187       3.400  10.857  18.298  1.00 10.93           N  \\nATOM   1108  CA  LEU A 187       4.425  11.848  17.973  1.00 11.20           C  \\nATOM   1109  C   LEU A 187       4.300  13.166  18.715  1.00 11.70           C  \\nATOM   1110  O   LEU A 187       3.387  13.948  18.434  1.00 12.10           O  \\nATOM   1111  CB  LEU A 187       4.340  12.226  16.495  1.00 11.57           C  \\nATOM   1112  CG  LEU A 187       4.468  11.071  15.502  1.00 10.71           C  \\nATOM   1113  CD1 LEU A 187       4.121  11.565  14.107  1.00 15.77           C  \\nATOM   1114  CD2 LEU A 187       5.878  10.507  15.535  1.00 15.13           C  \\nATOM   1115  N   THR A 188       5.209  13.474  19.531  1.00 11.54           N  \\nATOM   1116  CA  THR A 188       5.194  14.570  20.508  1.00 11.47           C  \\nATOM   1117  C   THR A 188       3.824  14.689  21.176  1.00 13.23           C  \\nATOM   1118  O   THR A 188       3.214  15.751  21.198  1.00 13.47           O  \\nATOM   1119  CB  THR A 188       5.605  15.928  19.911  1.00 20.00           C  \\nATOM   1120  OG1 THR A 188       4.702  16.284  18.859  1.00 20.00           O  \\nATOM   1121  CG2 THR A 188       6.945  15.815  19.200  1.00 20.00           C  \\nATOM   1122  N   PRO A 189       3.425  13.604  21.854  1.00 11.17           N  \\nATOM   1123  CA  PRO A 189       2.031  13.446  22.267  1.00 11.81           C  \\nATOM   1124  C   PRO A 189       1.645  14.371  23.423  1.00 12.06           C  \\nATOM   1125  O   PRO A 189       2.501  14.770  24.217  1.00 11.32           O  \\nATOM   1126  CB  PRO A 189       1.979  11.978  22.722  1.00 12.95           C  \\nATOM   1127  CG  PRO A 189       3.374  11.694  23.169  1.00 11.91           C  \\nATOM   1128  CD  PRO A 189       4.239  12.415  22.165  1.00 11.73           C  \\nATOM   1129  N   TRP A 190       0.353  14.660  23.538  1.00 11.97           N  \\nATOM   1130  CA  TRP A 190      -0.192  15.481  24.631  1.00 12.66           C  \\nATOM   1131  C   TRP A 190      -1.428  14.760  25.140  1.00 11.64           C  \\nATOM   1132  O   TRP A 190      -2.287  14.361  24.343  1.00 12.90           O  \\nATOM   1133  CB  TRP A 190      -0.586  16.859  24.070  1.00 12.75           C  \\nATOM   1134  CG  TRP A 190      -1.007  17.893  25.098  1.00 13.85           C  \\nATOM   1135  CD1 TRP A 190      -0.307  19.016  25.455  1.00 14.45           C  \\nATOM   1136  CD2 TRP A 190      -2.301  18.016  25.721  1.00 13.89           C  \\nATOM   1137  NE1 TRP A 190      -1.062  19.797  26.313  1.00 14.91           N  \\nATOM   1138  CE2 TRP A 190      -2.278  19.193  26.504  1.00 14.93           C  \\nATOM   1139  CE3 TRP A 190      -3.472  17.245  25.691  1.00 14.32           C  \\nATOM   1140  CZ2 TRP A 190      -3.369  19.600  27.272  1.00 16.85           C  \\nATOM   1141  CZ3 TRP A 190      -4.557  17.651  26.450  1.00 13.92           C  \\nATOM   1142  CH2 TRP A 190      -4.509  18.840  27.202  1.00 15.47           C  \\nATOM   1143  N   HIS A 191      -1.536  14.646  26.463  1.00 13.40           N  \\nATOM   1144  CA  HIS A 191      -2.751  14.156  27.109  1.00 12.47           C  \\nATOM   1145  C   HIS A 191      -2.669  14.446  28.604  1.00 13.92           C  \\nATOM   1146  O   HIS A 191      -1.576  14.431  29.188  1.00 14.91           O  \\nATOM   1147  CB  HIS A 191      -2.901  12.643  26.891  1.00 12.22           C  \\nATOM   1148  CG  HIS A 191      -4.317  12.167  26.911  1.00 13.31           C  \\nATOM   1149  ND1 HIS A 191      -5.221  12.558  27.877  1.00 20.06           N  \\nATOM   1150  CD2 HIS A 191      -4.967  11.272  26.129  1.00 13.78           C  \\nATOM   1151  CE1 HIS A 191      -6.400  12.026  27.607  1.00 19.67           C  \\nATOM   1152  NE2 HIS A 191      -6.251  11.170  26.609  1.00 13.66           N  \\nATOM   1153  N   THR A 192      -3.817  14.699  29.228  1.00 15.39           N  \\nATOM   1154  CA  THR A 192      -3.835  14.912  30.671  1.00 15.37           C  \\nATOM   1155  C   THR A 192      -3.849  13.603  31.448  1.00 15.35           C  \\nATOM   1156  O   THR A 192      -3.305  13.529  32.545  1.00 17.10           O  \\nATOM   1157  CB  THR A 192      -5.038  15.757  31.099  1.00 16.13           C  \\nATOM   1158  OG1 THR A 192      -6.224  15.181  30.552  1.00 17.05           O  \\nATOM   1159  CG2 THR A 192      -4.889  17.199  30.574  1.00 17.23           C  \\nATOM   1160  N   ASP A 193      -4.307  12.541  30.794  1.00 13.69           N  \\nATOM   1161  CA  ASP A 193      -4.171  11.191  31.340  1.00 15.39           C  \\nATOM   1162  C   ASP A 193      -2.766  10.681  31.036  1.00 16.25           C  \\nATOM   1163  O   ASP A 193      -2.313  10.755  29.895  1.00 16.36           O  \\nATOM   1164  CB  ASP A 193      -5.227  10.275  30.713  1.00 19.17           C  \\nATOM   1165  CG  ASP A 193      -5.075   8.821  31.134  1.00 30.88           C  \\nATOM   1166  OD1 ASP A 193      -4.449   8.549  32.184  1.00 27.86           O  \\nATOM   1167  OD2 ASP A 193      -5.709   7.963  30.481  1.00 41.54           O  \\nATOM   1168  N   LYS A 194      -2.060  10.215  32.063  1.00 15.12           N  \\nATOM   1169  CA  LYS A 194      -0.662   9.831  31.910  1.00 15.82           C  \\nATOM   1170  C   LYS A 194      -0.452   8.314  31.819  1.00 15.90           C  \\nATOM   1171  O   LYS A 194       0.641   7.854  31.485  1.00 15.53           O  \\nATOM   1172  CB  LYS A 194       0.194  10.412  33.036  1.00 13.96           C  \\nATOM   1173  CG  LYS A 194      -0.011  11.896  33.312  1.00 16.61           C  \\nATOM   1174  CD  LYS A 194       0.269  12.756  32.095  1.00 17.47           C  \\nATOM   1175  CE  LYS A 194      -0.009  14.223  32.408  1.00 16.47           C  \\nATOM   1176  NZ  LYS A 194       0.175  15.081  31.186  1.00 16.69           N  \\nATOM   1177  N   THR A 195      -1.508   7.540  32.058  1.00 14.80           N  \\nATOM   1178  CA  THR A 195      -1.369   6.094  32.176  1.00 14.43           C  \\nATOM   1179  C   THR A 195      -2.271   5.360  31.199  1.00 14.45           C  \\nATOM   1180  O   THR A 195      -3.502   5.444  31.287  1.00 17.24           O  \\nATOM   1181  CB  THR A 195      -1.700   5.637  33.601  1.00 15.27           C  \\nATOM   1182  OG1 THR A 195      -0.846   6.325  34.524  1.00 18.45           O  \\nATOM   1183  CG2 THR A 195      -1.503   4.124  33.735  1.00 18.77           C  \\nATOM   1184  N   PHE A 196      -1.666   4.622  30.276  1.00 14.28           N  \\nATOM   1185  CA  PHE A 196      -2.424   3.926  29.245  1.00 14.26           C  \\nATOM   1186  C   PHE A 196      -2.169   2.440  29.324  1.00 15.83           C  \\nATOM   1187  O   PHE A 196      -1.029   1.977  29.257  1.00 18.61           O  \\nATOM   1188  CB  PHE A 196      -2.070   4.468  27.853  1.00 13.48           C  \\nATOM   1189  CG  PHE A 196      -2.450   5.906  27.663  1.00 14.72           C  \\nATOM   1190  CD1 PHE A 196      -3.747   6.241  27.325  1.00 13.32           C  \\nATOM   1191  CD2 PHE A 196      -1.559   6.915  27.979  1.00 15.24           C  \\nATOM   1192  CE1 PHE A 196      -4.148   7.568  27.271  1.00 15.06           C  \\nATOM   1193  CE2 PHE A 196      -1.960   8.235  27.952  1.00 13.18           C  \\nATOM   1194  CZ  PHE A 196      -3.254   8.553  27.610  1.00 14.24           C  \\nATOM   1195  N   ASN A 197      -3.233   1.692  29.534  1.00 14.46           N  \\nATOM   1196  CA  ASN A 197      -3.111   0.279  29.777  1.00 14.55           C  \\nATOM   1197  C   ASN A 197      -3.363  -0.457  28.476  1.00 14.59           C  \\nATOM   1198  O   ASN A 197      -4.501  -0.750  28.123  1.00 18.25           O  \\nATOM   1199  CB  ASN A 197      -4.106  -0.126  30.865  1.00 14.41           C  \\nATOM   1200  CG  ASN A 197      -3.938  -1.548  31.308  1.00 19.85           C  \\nATOM   1201  OD1 ASN A 197      -3.143  -2.297  30.741  1.00 22.89           O  \\nATOM   1202  ND2 ASN A 197      -4.797  -1.976  32.232  1.00 25.51           N  \\nATOM   1203  N   THR A 198      -2.316  -0.539  27.663  1.00 13.16           N  \\nATOM   1204  CA  THR A 198      -2.434  -1.111  26.319  1.00 14.67           C  \\nATOM   1205  C   THR A 198      -1.413  -2.215  26.142  1.00 14.66           C  \\nATOM   1206  O   THR A 198      -0.326  -2.151  26.729  1.00 15.40           O  \\nATOM   1207  CB  THR A 198      -2.241  -0.051  25.192  1.00 15.56           C  \\nATOM   1208  OG1 THR A 198      -2.326  -0.693  23.912  1.00 15.15           O  \\nATOM   1209  CG2 THR A 198      -0.895   0.658  25.313  1.00 17.94           C  \\nATOM   1210  N   SER A 199      -1.777  -3.229  25.360  1.00 13.15           N  \\nATOM   1211  CA  SER A 199      -0.847  -4.281  24.959  1.00 13.30           C  \\nATOM   1212  C   SER A 199      -0.313  -4.079  23.535  1.00 13.92           C  \\nATOM   1213  O   SER A 199       0.398  -4.930  22.994  1.00 15.13           O  \\nATOM   1214  CB  SER A 199      -1.536  -5.646  25.062  1.00 12.71           C  \\nATOM   1215  OG  SER A 199      -2.773  -5.660  24.364  1.00 16.84           O  \\nATOM   1216  N   VAL A 200      -0.679  -2.958  22.914  1.00 12.85           N  \\nATOM   1217  CA  VAL A 200      -0.200  -2.645  21.572  1.00 13.67           C  \\nATOM   1218  C   VAL A 200       1.208  -2.050  21.666  1.00 12.82           C  \\nATOM   1219  O   VAL A 200       1.435  -1.137  22.448  1.00 13.66           O  \\nATOM   1220  CB  VAL A 200      -1.136  -1.608  20.916  1.00 10.23           C  \\nATOM   1221  CG1 VAL A 200      -0.624  -1.194  19.555  1.00 13.33           C  \\nATOM   1222  CG2 VAL A 200      -2.551  -2.164  20.800  1.00 14.72           C  \\nATOM   1223  N   PRO A 201       2.152  -2.544  20.849  1.00 11.02           N  \\nATOM   1224  CA  PRO A 201       3.506  -1.969  20.855  1.00 11.42           C  \\nATOM   1225  C   PRO A 201       3.522  -0.470  20.548  1.00 10.64           C  \\nATOM   1226  O   PRO A 201       2.941  -0.024  19.543  1.00 12.55           O  \\nATOM   1227  CB  PRO A 201       4.208  -2.724  19.732  1.00 11.54           C  \\nATOM   1228  CG  PRO A 201       3.535  -4.096  19.750  1.00 11.49           C  \\nATOM   1229  CD  PRO A 201       2.082  -3.781  20.035  1.00 11.46           C  \\nATOM   1230  N   VAL A 202       4.170   0.296  21.424  1.00 10.67           N  \\nATOM   1231  CA  VAL A 202       4.130   1.755  21.354  1.00 12.12           C  \\nATOM   1232  C   VAL A 202       5.553   2.287  21.193  1.00 12.04           C  \\nATOM   1233  O   VAL A 202       6.418   1.974  21.999  1.00 11.96           O  \\nATOM   1234  CB  VAL A 202       3.558   2.331  22.655  1.00 10.58           C  \\nATOM   1235  CG1 VAL A 202       3.724   3.851  22.675  1.00 13.83           C  \\nATOM   1236  CG2 VAL A 202       2.081   1.979  22.807  1.00 12.92           C  \\nATOM   1237  N   LEU A 203       5.759   3.153  20.196  1.00 11.16           N  \\nATOM   1238  CA  LEU A 203       6.968   3.964  20.122  1.00 10.70           C  \\nATOM   1239  C   LEU A 203       6.588   5.436  20.146  1.00 10.31           C  \\nATOM   1240  O   LEU A 203       5.800   5.893  19.313  1.00 11.65           O  \\nATOM   1241  CB  LEU A 203       7.773   3.650  18.854  1.00 10.89           C  \\nATOM   1242  CG  LEU A 203       9.094   4.417  18.721  1.00 10.45           C  \\nATOM   1243  CD1 LEU A 203      10.105   3.627  17.903  1.00 11.91           C  \\nATOM   1244  CD2 LEU A 203       8.895   5.806  18.101  1.00 13.96           C  \\nATOM   1245  N   ILE A 204       7.019   6.115  21.202  1.00 11.42           N  \\nATOM   1246  CA  ILE A 204       6.737   7.543  21.383  1.00 11.26           C  \\nATOM   1247  C   ILE A 204       7.928   8.358  20.945  1.00 11.89           C  \\nATOM   1248  O   ILE A 204       9.072   8.056  21.314  1.00 12.16           O  \\nATOM   1249  CB  ILE A 204       6.444   7.856  22.863  1.00 10.91           C  \\nATOM   1250  CG1 ILE A 204       5.141   7.189  23.322  1.00 11.23           C  \\nATOM   1251  CG2 ILE A 204       6.386   9.377  23.098  1.00 12.88           C  \\nATOM   1252  CD1 ILE A 204       5.076   6.933  24.833  1.00 15.36           C  \\nATOM   1253  N   VAL A 205       7.672   9.375  20.120  1.00 11.62           N  \\nATOM   1254  CA  VAL A 205       8.694  10.386  19.820  1.00 11.80           C  \\nATOM   1255  C   VAL A 205       8.446  11.602  20.694  1.00 12.82           C  \\nATOM   1256  O   VAL A 205       7.388  12.222  20.614  1.00 14.31           O  \\nATOM   1257  CB  VAL A 205       8.642  10.815  18.345  1.00 12.08           C  \\nATOM   1258  CG1 VAL A 205       9.667  11.911  18.073  1.00 13.30           C  \\nATOM   1259  CG2 VAL A 205       8.890   9.609  17.428  1.00 13.77           C  \\nATOM   1260  N   GLY A 206       9.436  11.959  21.504  1.00 12.29           N  \\nATOM   1261  CA  GLY A 206       9.412  13.245  22.211  1.00 13.24           C  \\nATOM   1262  C   GLY A 206      10.312  14.259  21.519  1.00 13.59           C  \\nATOM   1263  O   GLY A 206      11.221  13.886  20.763  1.00 12.87           O  \\nATOM   1264  N   ALA A 207      10.011  15.543  21.714  1.00 12.77           N  \\nATOM   1265  CA  ALA A 207      10.881  16.621  21.237  1.00 14.48           C  \\nATOM   1266  C   ALA A 207      11.503  17.336  22.433  1.00 13.52           C  \\nATOM   1267  O   ALA A 207      10.792  17.837  23.302  1.00 14.74           O  \\nATOM   1268  CB  ALA A 207      10.081  17.602  20.379  1.00 14.12           C  \\nATOM   1269  N   GLU A 208      12.825  17.266  22.547  1.00 14.92           N  \\nATOM   1270  CA  GLU A 208      13.475  17.691  23.783  1.00 15.58           C  \\nATOM   1271  C   GLU A 208      12.967  19.058  24.231  1.00 14.92           C  \\nATOM   1272  O   GLU A 208      12.730  19.272  25.413  1.00 16.18           O  \\nATOM   1273  CB  GLU A 208      14.997  17.723  23.638  1.00 15.18           C  \\nATOM   1274  CG  GLU A 208      15.701  17.942  24.969  1.00 19.06           C  \\nATOM   1275  CD  GLU A 208      17.217  17.980  24.850  1.00 21.51           C  \\nATOM   1276  OE1 GLU A 208      17.732  18.269  23.756  1.00 20.34           O  \\nATOM   1277  OE2 GLU A 208      17.889  17.797  25.884  1.00 26.51           O  \\nATOM   1278  N   ALA A 209      12.909  20.006  23.295  1.00 16.76           N  \\nATOM   1279  CA  ALA A 209      12.605  21.400  23.630  1.00 17.31           C  \\nATOM   1280  C   ALA A 209      11.128  21.749  23.482  1.00 19.50           C  \\nATOM   1281  O   ALA A 209      10.762  22.920  23.435  1.00 19.62           O  \\nATOM   1282  CB  ALA A 209      13.448  22.338  22.783  1.00 16.74           C  \\nATOM   1283  N   ASP A 210      10.270  20.736  23.476  1.00 16.08           N  \\nATOM   1284  CA  ASP A 210       8.844  20.939  23.241  1.00 14.87           C  \\nATOM   1285  C   ASP A 210       8.283  21.831  24.340  1.00 16.14           C  \\nATOM   1286  O   ASP A 210       8.425  21.526  25.517  1.00 17.30           O  \\nATOM   1287  CB  ASP A 210       8.138  19.573  23.301  1.00 14.52           C  \\nATOM   1288  CG  ASP A 210       6.768  19.580  22.641  1.00 15.16           C  \\nATOM   1289  OD1 ASP A 210       6.052  20.602  22.719  1.00 14.74           O  \\nATOM   1290  OD2 ASP A 210       6.366  18.532  22.092  1.00 14.08           O  \\nATOM   1291  N   THR A 211       7.630  22.914  23.935  1.00 14.27           N  \\nATOM   1292  CA  THR A 211       6.963  23.810  24.889  1.00 15.04           C  \\nATOM   1293  C   THR A 211       5.453  23.625  24.914  1.00 15.44           C  \\nATOM   1294  O   THR A 211       4.777  24.159  25.802  1.00 17.34           O  \\nATOM   1295  CB  THR A 211       7.257  25.296  24.577  1.00 16.01           C  \\nATOM   1296  OG1 THR A 211       6.830  25.592  23.242  1.00 17.13           O  \\nATOM   1297  CG2 THR A 211       8.736  25.612  24.724  1.00 17.95           C  \\nATOM   1298  N   VAL A 212       4.914  22.902  23.936  1.00 13.06           N  \\nATOM   1299  CA  VAL A 212       3.472  22.672  23.840  1.00 13.74           C  \\nATOM   1300  C   VAL A 212       3.086  21.443  24.672  1.00 15.19           C  \\nATOM   1301  O   VAL A 212       2.135  21.474  25.470  1.00 15.25           O  \\nATOM   1302  CB  VAL A 212       3.052  22.446  22.377  1.00 15.64           C  \\nATOM   1303  CG1 VAL A 212       1.594  22.047  22.292  1.00 17.94           C  \\nATOM   1304  CG2 VAL A 212       3.342  23.697  21.544  1.00 16.27           C  \\nATOM   1305  N   ALA A 213       3.867  20.374  24.512  1.00 13.87           N  \\nATOM   1306  CA  ALA A 213       3.661  19.148  25.283  1.00 13.06           C  \\nATOM   1307  C   ALA A 213       4.988  18.744  25.921  1.00 13.86           C  \\nATOM   1308  O   ALA A 213       5.690  17.873  25.413  1.00 13.70           O  \\nATOM   1309  CB  ALA A 213       3.139  18.024  24.361  1.00 12.73           C  \\nATOM   1310  N   PRO A 214       5.430  19.489  26.934  1.00 13.40           N  \\nATOM   1311  CA  PRO A 214       6.774  19.251  27.452  1.00 13.09           C  \\nATOM   1312  C   PRO A 214       6.991  17.785  27.803  1.00 14.09           C  \\nATOM   1313  O   PRO A 214       6.117  17.165  28.402  1.00 14.10           O  \\nATOM   1314  CB  PRO A 214       6.805  20.103  28.724  1.00 16.34           C  \\nATOM   1315  CG  PRO A 214       5.862  21.206  28.443  1.00 15.88           C  \\nATOM   1316  CD  PRO A 214       4.757  20.613  27.612  1.00 16.50           C  \\nATOM   1317  N   VAL A 215       8.156  17.250  27.459  1.00 12.48           N  \\nATOM   1318  CA  VAL A 215       8.360  15.798  27.578  1.00 13.09           C  \\nATOM   1319  C   VAL A 215       8.345  15.321  29.032  1.00 13.90           C  \\nATOM   1320  O   VAL A 215       7.891  14.220  29.308  1.00 15.25           O  \\nATOM   1321  CB  VAL A 215       9.636  15.299  26.853  1.00 12.90           C  \\nATOM   1322  CG1 VAL A 215       9.534  15.555  25.355  1.00 14.15           C  \\nATOM   1323  CG2 VAL A 215      10.890  15.940  27.432  1.00 15.47           C  \\nATOM   1324  N   SER A 216       8.706  16.194  29.974  1.00 15.46           N  \\nATOM   1325  CA  SER A 216       8.652  15.825  31.398  1.00 14.83           C  \\nATOM   1326  C   SER A 216       7.226  15.623  31.903  1.00 16.24           C  \\nATOM   1327  O   SER A 216       7.005  14.893  32.875  1.00 16.57           O  \\nATOM   1328  CB  SER A 216       9.363  16.873  32.263  1.00 19.67           C  \\nATOM   1329  OG  SER A 216       8.633  18.085  32.277  1.00 23.97           O  \\nATOM   1330  N   GLN A 217       6.258  16.219  31.211  1.00 15.78           N  \\nATOM   1331  CA  GLN A 217       4.870  16.188  31.637  1.00 16.56           C  \\nATOM   1332  C   GLN A 217       3.998  15.265  30.800  1.00 15.27           C  \\nATOM   1333  O   GLN A 217       2.906  14.879  31.216  1.00 17.01           O  \\nATOM   1334  CB  GLN A 217       4.287  17.601  31.656  1.00 15.97           C  \\nATOM   1335  CG  GLN A 217       4.871  18.446  32.775  1.00 24.82           C  \\nATOM   1336  CD  GLN A 217       4.599  19.923  32.598  1.00 39.63           C  \\nATOM   1337  OE1 GLN A 217       5.298  20.607  31.850  1.00 37.51           O  \\nATOM   1338  NE2 GLN A 217       3.673  20.450  33.394  1.00 48.52           N  \\nATOM   1339  N   HIS A 218       4.385  14.942  29.608  1.00 13.21           N  \\nATOM   1340  CA  HIS A 218       3.621  14.173  28.628  1.00 13.20           C  \\nATOM   1341  C   HIS A 218       4.418  12.974  28.103  1.00 12.89           C  \\nATOM   1342  O   HIS A 218       4.424  11.916  28.738  1.00 13.69           O  \\nATOM   1343  CB  HIS A 218       3.196  15.068  27.463  1.00 20.00           C  \\nATOM   1344  CG  HIS A 218       2.222  16.137  27.847  1.00 20.00           C  \\nATOM   1345  ND1 HIS A 218       2.613  17.338  28.401  1.00 20.00           N  \\nATOM   1346  CD2 HIS A 218       0.872  16.189  27.755  1.00 20.00           C  \\nATOM   1347  CE1 HIS A 218       1.547  18.081  28.635  1.00 20.00           C  \\nATOM   1348  NE2 HIS A 218       0.477  17.409  28.253  1.00 20.00           N  \\nATOM   1349  N   ALA A 219       5.252  13.059  27.166  1.00 12.26           N  \\nATOM   1350  CA  ALA A 219       5.910  11.895  26.558  1.00 12.73           C  \\nATOM   1351  C   ALA A 219       6.533  10.942  27.564  1.00 14.21           C  \\nATOM   1352  O   ALA A 219       6.317   9.726  27.480  1.00 13.41           O  \\nATOM   1353  CB  ALA A 219       6.942  12.346  25.543  1.00 12.39           C  \\nATOM   1354  N   ILE A 220       7.324  11.478  28.492  1.00 13.27           N  \\nATOM   1355  CA  ILE A 220       8.060  10.633  29.434  1.00 14.36           C  \\nATOM   1356  C   ILE A 220       7.166   9.879  30.425  1.00 15.12           C  \\nATOM   1357  O   ILE A 220       7.208   8.643  30.484  1.00 14.11           O  \\nATOM   1358  CB  ILE A 220       9.192  11.399  30.120  1.00 14.19           C  \\nATOM   1359  CG1 ILE A 220      10.259  11.709  29.071  1.00 15.86           C  \\nATOM   1360  CG2 ILE A 220       9.810  10.548  31.228  1.00 14.80           C  \\nATOM   1361  CD1 ILE A 220      11.324  12.683  29.540  1.00 16.14           C  \\nATOM   1362  N   PRO A 221       6.191  10.578  31.027  1.00 12.13           N  \\nATOM   1363  CA  PRO A 221       5.230   9.836  31.836  1.00 12.42           C  \\nATOM   1364  C   PRO A 221       4.433   8.816  31.052  1.00 11.13           C  \\nATOM   1365  O   PRO A 221       4.239   7.702  31.548  1.00 13.20           O  \\nATOM   1366  CB  PRO A 221       4.324  10.935  32.408  1.00 13.37           C  \\nATOM   1367  CG  PRO A 221       5.230  12.119  32.492  1.00 15.10           C  \\nATOM   1368  CD  PRO A 221       6.106  12.032  31.268  1.00 13.36           C  \\nATOM   1369  N   PHE A 222       4.039   9.123  29.810  1.00 13.07           N  \\nATOM   1370  CA  PHE A 222       3.339   8.107  29.000  1.00 11.14           C  \\nATOM   1371  C   PHE A 222       4.202   6.866  28.842  1.00 12.77           C  \\nATOM   1372  O   PHE A 222       3.738   5.746  29.049  1.00 13.00           O  \\nATOM   1373  CB  PHE A 222       2.955   8.603  27.601  1.00 13.45           C  \\nATOM   1374  CG  PHE A 222       2.170   9.884  27.581  1.00 13.38           C  \\nATOM   1375  CD1 PHE A 222       1.355  10.253  28.651  1.00 14.69           C  \\nATOM   1376  CD2 PHE A 222       2.306  10.760  26.513  1.00 13.37           C  \\nATOM   1377  CE1 PHE A 222       0.609  11.417  28.597  1.00 16.94           C  \\nATOM   1378  CE2 PHE A 222       1.604  11.961  26.471  1.00 13.69           C  \\nATOM   1379  CZ  PHE A 222       0.772  12.306  27.527  1.00 17.25           C  \\nATOM   1380  N   TYR A 223       5.506   7.163  28.413  1.00 11.59           N  \\nATOM   1381  CA  TYR A 223       6.334   5.964  28.321  1.00 12.85           C  \\nATOM   1382  C   TYR A 223       6.337   5.189  29.619  1.00 12.79           C  \\nATOM   1383  O   TYR A 223       6.077   3.981  29.625  1.00 14.46           O  \\nATOM   1384  CB  TYR A 223       7.766   6.331  27.928  1.00 20.00           C  \\nATOM   1385  CG  TYR A 223       8.717   5.155  27.903  1.00 20.00           C  \\nATOM   1386  CD1 TYR A 223       8.706   4.252  26.846  1.00 20.00           C  \\nATOM   1387  CD2 TYR A 223       9.623   4.950  28.933  1.00 20.00           C  \\nATOM   1388  CE1 TYR A 223       9.572   3.177  26.819  1.00 20.00           C  \\nATOM   1389  CE2 TYR A 223      10.495   3.876  28.914  1.00 20.00           C  \\nATOM   1390  CZ  TYR A 223      10.465   2.994  27.855  1.00 20.00           C  \\nATOM   1391  OH  TYR A 223      11.328   1.922  27.831  1.00 20.00           O  \\nATOM   1392  N   GLN A 224       6.727   5.847  30.679  1.00 12.16           N  \\nATOM   1393  CA  GLN A 224       7.019   5.174  31.938  1.00 13.90           C  \\nATOM   1394  C   GLN A 224       5.845   4.333  32.407  1.00 14.27           C  \\nATOM   1395  O   GLN A 224       6.041   3.303  33.040  1.00 17.63           O  \\nATOM   1396  CB  GLN A 224       7.375   6.190  33.015  1.00 14.30           C  \\nATOM   1397  CG  GLN A 224       8.733   6.827  32.816  1.00 14.69           C  \\nATOM   1398  CD  GLN A 224       8.995   7.931  33.797  1.00 16.62           C  \\nATOM   1399  OE1 GLN A 224       8.057   8.544  34.314  1.00 22.84           O  \\nATOM   1400  NE2 GLN A 224      10.264   8.239  34.017  1.00 21.42           N  \\nATOM   1401  N   ASN A 225       4.634   4.810  32.158  1.00 14.71           N  \\nATOM   1402  CA  ASN A 225       3.451   4.203  32.764  1.00 14.26           C  \\nATOM   1403  C   ASN A 225       2.760   3.161  31.902  1.00 15.33           C  \\nATOM   1404  O   ASN A 225       1.788   2.543  32.339  1.00 16.76           O  \\nATOM   1405  CB  ASN A 225       2.435   5.269  33.184  1.00 14.55           C  \\nATOM   1406  CG  ASN A 225       2.978   6.226  34.226  1.00 19.09           C  \\nATOM   1407  OD1 ASN A 225       3.681   5.826  35.147  1.00 22.80           O  \\nATOM   1408  ND2 ASN A 225       2.507   7.467  34.176  1.00 18.49           N  \\nATOM   1409  N   LEU A 226       3.229   2.968  30.674  1.00 14.79           N  \\nATOM   1410  CA  LEU A 226       2.760   1.843  29.865  1.00 15.57           C  \\nATOM   1411  C   LEU A 226       3.143   0.539  30.554  1.00 14.36           C  \\nATOM   1412  O   LEU A 226       4.163   0.474  31.229  1.00 15.75           O  \\nATOM   1413  CB  LEU A 226       3.396   1.877  28.478  1.00 15.47           C  \\nATOM   1414  CG  LEU A 226       2.910   2.955  27.502  1.00 14.92           C  \\nATOM   1415  CD1 LEU A 226       3.921   3.071  26.367  1.00 17.80           C  \\nATOM   1416  CD2 LEU A 226       1.542   2.624  26.939  1.00 18.00           C  \\nATOM   1417  N   PRO A 227       2.339  -0.519  30.368  1.00 14.23           N  \\nATOM   1418  CA  PRO A 227       2.670  -1.777  31.040  1.00 14.15           C  \\nATOM   1419  C   PRO A 227       4.093  -2.259  30.749  1.00 12.90           C  \\nATOM   1420  O   PRO A 227       4.567  -2.171  29.613  1.00 13.39           O  \\nATOM   1421  CB  PRO A 227       1.653  -2.760  30.465  1.00 14.31           C  \\nATOM   1422  CG  PRO A 227       0.492  -1.896  30.003  1.00 16.99           C  \\nATOM   1423  CD  PRO A 227       1.153  -0.639  29.508  1.00 14.29           C  \\nATOM   1424  N   SER A 228       4.727  -2.869  31.743  1.00 13.58           N  \\nATOM   1425  CA  SER A 228       6.090  -3.357  31.583  1.00 13.34           C  \\nATOM   1426  C   SER A 228       6.197  -4.533  30.630  1.00 12.75           C  \\nATOM   1427  O   SER A 228       7.243  -4.733  29.996  1.00 13.76           O  \\nATOM   1428  CB  SER A 228       6.688  -3.718  32.932  1.00 14.32           C  \\nATOM   1429  OG  SER A 228       6.789  -2.542  33.718  1.00 19.93           O  \\nATOM   1430  N   THR A 229       5.058  -5.177  30.382  1.00 12.69           N  \\nATOM   1431  CA  THR A 229       4.993  -6.317  29.488  1.00 12.66           C  \\nATOM   1432  C   THR A 229       4.679  -5.931  28.034  1.00 11.40           C  \\nATOM   1433  O   THR A 229       4.667  -6.791  27.155  1.00 13.06           O  \\nATOM   1434  CB  THR A 229       3.934  -7.327  29.975  1.00 13.83           C  \\nATOM   1435  OG1 THR A 229       2.729  -6.632  30.313  1.00 15.60           O  \\nATOM   1436  CG2 THR A 229       4.437  -8.096  31.211  1.00 16.94           C  \\nATOM   1437  N   THR A 230       4.417  -4.647  27.770  1.00 12.80           N  \\nATOM   1438  CA  THR A 230       4.119  -4.196  26.398  1.00 11.90           C  \\nATOM   1439  C   THR A 230       5.384  -3.632  25.747  1.00 12.65           C  \\nATOM   1440  O   THR A 230       6.066  -2.801  26.346  1.00 12.26           O  \\nATOM   1441  CB  THR A 230       3.082  -3.087  26.430  1.00 10.26           C  \\nATOM   1442  OG1 THR A 230       1.872  -3.591  27.027  1.00 12.31           O  \\nATOM   1443  CG2 THR A 230       2.802  -2.552  25.033  1.00 13.59           C  \\nATOM   1444  N   PRO A 231       5.733  -4.116  24.546  1.00 11.15           N  \\nATOM   1445  CA  PRO A 231       6.909  -3.544  23.878  1.00 10.81           C  \\nATOM   1446  C   PRO A 231       6.773  -2.039  23.751  1.00 12.26           C  \\nATOM   1447  O   PRO A 231       5.749  -1.549  23.269  1.00 12.84           O  \\nATOM   1448  CB  PRO A 231       6.897  -4.211  22.505  1.00 12.75           C  \\nATOM   1449  CG  PRO A 231       6.200  -5.526  22.761  1.00 13.69           C  \\nATOM   1450  CD  PRO A 231       5.105  -5.176  23.731  1.00 10.22           C  \\nATOM   1451  N   LYS A 232       7.772  -1.319  24.254  1.00 11.13           N  \\nATOM   1452  CA  LYS A 232       7.708   0.141  24.255  1.00 12.12           C  \\nATOM   1453  C   LYS A 232       9.075   0.752  24.021  1.00 13.25           C  \\nATOM   1454  O   LYS A 232      10.092   0.221  24.459  1.00 13.28           O  \\nATOM   1455  CB  LYS A 232       7.068   0.674  25.543  1.00 13.29           C  \\nATOM   1456  CG  LYS A 232       7.776   0.288  26.851  1.00 13.46           C  \\nATOM   1457  CD  LYS A 232       7.053   0.951  28.015  1.00 15.98           C  \\nATOM   1458  CE  LYS A 232       7.625   0.573  29.360  1.00 16.25           C  \\nATOM   1459  NZ  LYS A 232       6.825   1.253  30.446  1.00 20.08           N  \\nATOM   1460  N   VAL A 233       9.080   1.852  23.280  1.00 12.53           N  \\nATOM   1461  CA  VAL A 233      10.294   2.632  23.009  1.00 12.23           C  \\nATOM   1462  C   VAL A 233       9.955   4.108  23.145  1.00 12.19           C  \\nATOM   1463  O   VAL A 233       8.849   4.537  22.816  1.00 13.40           O  \\nATOM   1464  CB  VAL A 233      10.807   2.385  21.572  1.00 12.09           C  \\nATOM   1465  CG1 VAL A 233      11.880   3.392  21.204  1.00 11.26           C  \\nATOM   1466  CG2 VAL A 233      11.354   0.958  21.441  1.00 13.52           C  \\nATOM   1467  N   TYR A 234      10.896   4.859  23.703  1.00 12.07           N  \\nATOM   1468  CA  TYR A 234      10.835   6.322  23.674  1.00 10.41           C  \\nATOM   1469  C   TYR A 234      12.080   6.843  22.955  1.00 11.74           C  \\nATOM   1470  O   TYR A 234      13.215   6.494  23.310  1.00 13.17           O  \\nATOM   1471  CB  TYR A 234      10.795   6.874  25.105  1.00 13.52           C  \\nATOM   1472  CG  TYR A 234      10.926   8.382  25.194  1.00 13.81           C  \\nATOM   1473  CD1 TYR A 234       9.966   9.206  24.637  1.00 13.95           C  \\nATOM   1474  CD2 TYR A 234      11.941   8.964  25.935  1.00 13.95           C  \\nATOM   1475  CE1 TYR A 234      10.064  10.591  24.723  1.00 12.49           C  \\nATOM   1476  CE2 TYR A 234      12.039  10.349  26.063  1.00 14.30           C  \\nATOM   1477  CZ  TYR A 234      11.108  11.149  25.430  1.00 11.80           C  \\nATOM   1478  OH  TYR A 234      11.172  12.535  25.555  1.00 15.44           O  \\nATOM   1479  N   VAL A 235      11.868   7.661  21.926  1.00 11.98           N  \\nATOM   1480  CA  VAL A 235      12.967   8.383  21.289  1.00 12.74           C  \\nATOM   1481  C   VAL A 235      12.759   9.883  21.429  1.00 11.48           C  \\nATOM   1482  O   VAL A 235      11.647  10.373  21.227  1.00 13.02           O  \\nATOM   1483  CB  VAL A 235      13.146   8.001  19.810  1.00 13.23           C  \\nATOM   1484  CG1 VAL A 235      13.240   6.482  19.666  1.00 15.06           C  \\nATOM   1485  CG2 VAL A 235      12.024   8.536  18.967  1.00 16.40           C  \\nATOM   1486  N   GLU A 236      13.755  10.556  21.993  1.00 12.87           N  \\nATOM   1487  CA  GLU A 236      13.651  11.988  22.208  1.00 14.22           C  \\nATOM   1488  C   GLU A 236      14.553  12.725  21.243  1.00 14.04           C  \\nATOM   1489  O   GLU A 236      15.777  12.615  21.338  1.00 15.44           O  \\nATOM   1490  CB  GLU A 236      14.039  12.326  23.645  1.00 12.71           C  \\nATOM   1491  CG  GLU A 236      13.776  13.769  24.027  1.00 14.27           C  \\nATOM   1492  CD  GLU A 236      14.184  14.044  25.457  1.00 17.93           C  \\nATOM   1493  OE1 GLU A 236      13.440  13.646  26.372  1.00 15.59           O  \\nATOM   1494  OE2 GLU A 236      15.316  14.533  25.674  1.00 20.69           O  \\nATOM   1495  N   LEU A 237      13.949  13.554  20.396  1.00 13.73           N  \\nATOM   1496  CA  LEU A 237      14.690  14.316  19.386  1.00 13.08           C  \\nATOM   1497  C   LEU A 237      15.473  15.458  20.012  1.00 15.49           C  \\nATOM   1498  O   LEU A 237      14.903  16.343  20.640  1.00 15.37           O  \\nATOM   1499  CB  LEU A 237      13.739  14.885  18.331  1.00 12.20           C  \\nATOM   1500  CG  LEU A 237      12.798  13.894  17.639  1.00 15.27           C  \\nATOM   1501  CD1 LEU A 237      11.886  14.635  16.687  1.00 16.55           C  \\nATOM   1502  CD2 LEU A 237      13.577  12.816  16.885  1.00 18.35           C  \\nATOM   1503  N   ASP A 238      16.790  15.428  19.817  1.00 17.04           N  \\nATOM   1504  CA  ASP A 238      17.702  16.408  20.390  1.00 19.74           C  \\nATOM   1505  C   ASP A 238      17.303  17.826  19.967  1.00 17.63           C  \\nATOM   1506  O   ASP A 238      17.189  18.109  18.777  1.00 18.16           O  \\nATOM   1507  CB  ASP A 238      19.116  16.095  19.880  1.00 19.24           C  \\nATOM   1508  CG  ASP A 238      20.159  17.017  20.442  1.00 24.11           C  \\nATOM   1509  OD1 ASP A 238      20.037  17.427  21.608  1.00 24.89           O  \\nATOM   1510  OD2 ASP A 238      21.174  17.216  19.749  1.00 33.38           O  \\nATOM   1511  N   ASN A 239      17.047  18.675  20.962  1.00 19.41           N  \\nATOM   1512  CA  ASN A 239      16.772  20.103  20.770  1.00 20.66           C  \\nATOM   1513  C   ASN A 239      15.589  20.435  19.872  1.00 21.24           C  \\nATOM   1514  O   ASN A 239      15.459  21.563  19.397  1.00 22.64           O  \\nATOM   1515  CB  ASN A 239      18.011  20.864  20.305  1.00 22.58           C  \\nATOM   1516  CG  ASN A 239      17.990  22.315  20.748  1.00 29.52           C  \\nATOM   1517  OD1 ASN A 239      17.396  22.651  21.776  1.00 36.97           O  \\nATOM   1518  ND2 ASN A 239      18.536  23.193  19.919  1.00 37.81           N  \\nATOM   1519  N   ALA A 240      14.656  19.499  19.745  1.00 16.63           N  \\nATOM   1520  CA  ALA A 240      13.575  19.659  18.785  1.00 16.07           C  \\nATOM   1521  C   ALA A 240      12.357  20.365  19.354  1.00 15.03           C  \\nATOM   1522  O   ALA A 240      12.074  20.310  20.556  1.00 16.12           O  \\nATOM   1523  CB  ALA A 240      13.175  18.312  18.212  1.00 14.17           C  \\nATOM   1524  N   SER A 241      11.587  20.958  18.453  1.00 16.05           N  \\nATOM   1525  CA  SER A 241      10.371  21.637  18.843  1.00 16.86           C  \\nATOM   1526  C   SER A 241       9.132  20.779  18.642  1.00 16.61           C  \\nATOM   1527  O   SER A 241       9.163  19.729  17.968  1.00 15.47           O  \\nATOM   1528  CB  SER A 241      10.230  22.952  18.072  1.00 17.99           C  \\nATOM   1529  OG  SER A 241       9.590  22.743  16.827  1.00 23.43           O  \\nATOM   1530  N   HIS A 242       8.032  21.237  19.219  1.00 14.73           N  \\nATOM   1531  CA  HIS A 242       6.744  20.625  18.985  1.00 15.64           C  \\nATOM   1532  C   HIS A 242       6.402  20.459  17.506  1.00 16.01           C  \\nATOM   1533  O   HIS A 242       5.612  19.582  17.145  1.00 16.98           O  \\nATOM   1534  CB  HIS A 242       5.652  21.420  19.692  1.00 16.33           C  \\nATOM   1535  CG  HIS A 242       4.432  20.616  19.975  1.00 24.08           C  \\nATOM   1536  ND1 HIS A 242       4.476  19.481  20.758  1.00 25.00           N  \\nATOM   1537  CD2 HIS A 242       3.216  20.615  19.382  1.00 12.36           C  \\nATOM   1538  CE1 HIS A 242       3.319  18.856  20.692  1.00 15.95           C  \\nATOM   1539  NE2 HIS A 242       2.518  19.550  19.901  1.00 17.12           N  \\nATOM   1540  N   PHE A 243       6.915  21.355  16.667  1.00 13.57           N  \\nATOM   1541  CA  PHE A 243       6.547  21.361  15.258  1.00 16.72           C  \\nATOM   1542  C   PHE A 243       7.366  20.374  14.440  1.00 15.55           C  \\nATOM   1543  O   PHE A 243       7.219  20.314  13.216  1.00 15.91           O  \\nATOM   1544  CB  PHE A 243       6.720  22.760  14.658  1.00 18.63           C  \\nATOM   1545  CG  PHE A 243       5.804  23.785  15.243  1.00 26.36           C  \\nATOM   1546  CD1 PHE A 243       4.436  23.687  15.070  1.00 32.76           C  \\nATOM   1547  CD2 PHE A 243       6.318  24.893  15.890  1.00 27.79           C  \\nATOM   1548  CE1 PHE A 243       3.584  24.646  15.595  1.00 40.92           C  \\nATOM   1549  CE2 PHE A 243       5.473  25.883  16.379  1.00 41.38           C  \\nATOM   1550  CZ  PHE A 243       4.103  25.731  16.273  1.00 38.04           C  \\nATOM   1551  N   ALA A 244       8.326  19.713  15.081  1.00 15.54           N  \\nATOM   1552  CA  ALA A 244       9.200  18.798  14.342  1.00 14.42           C  \\nATOM   1553  C   ALA A 244       8.450  17.782  13.487  1.00 13.99           C  \\nATOM   1554  O   ALA A 244       8.851  17.535  12.358  1.00 13.78           O  \\nATOM   1555  CB  ALA A 244      10.181  18.109  15.264  1.00 14.62           C  \\nATOM   1556  N   PRO A 245       7.295  17.272  13.965  1.00 14.34           N  \\nATOM   1557  CA  PRO A 245       6.632  16.260  13.146  1.00 14.45           C  \\nATOM   1558  C   PRO A 245       5.901  16.831  11.932  1.00 14.65           C  \\nATOM   1559  O   PRO A 245       5.404  16.066  11.107  1.00 13.90           O  \\nATOM   1560  CB  PRO A 245       5.610  15.631  14.110  1.00 15.13           C  \\nATOM   1561  CG  PRO A 245       6.190  15.859  15.460  1.00 14.45           C  \\nATOM   1562  CD  PRO A 245       6.851  17.220  15.368  1.00 13.80           C  \\nATOM   1563  N   ASN A 246       5.859  18.158  11.797  1.00 13.15           N  \\nATOM   1564  CA  ASN A 246       5.094  18.784  10.721  1.00 13.15           C  \\nATOM   1565  C   ASN A 246       5.918  19.144   9.480  1.00 13.74           C  \\nATOM   1566  O   ASN A 246       5.390  19.725   8.519  1.00 17.00           O  \\nATOM   1567  CB  ASN A 246       4.397  20.053  11.226  1.00 14.77           C  \\nATOM   1568  CG  ASN A 246       3.460  19.791  12.397  1.00 18.01           C  \\nATOM   1569  OD1 ASN A 246       3.455  20.540  13.367  1.00 19.04           O  \\nATOM   1570  ND2 ASN A 246       2.718  18.696  12.340  1.00 14.36           N  \\nATOM   1571  N   SER A 247       7.192  18.767   9.489  1.00 14.73           N  \\nATOM   1572  CA  SER A 247       7.997  18.803   8.279  1.00 15.76           C  \\nATOM   1573  C   SER A 247       8.979  17.649   8.249  1.00 14.89           C  \\nATOM   1574  O   SER A 247       9.230  17.010   9.274  1.00 15.67           O  \\nATOM   1575  CB  SER A 247       8.717  20.150   8.120  1.00 19.00           C  \\nATOM   1576  OG  SER A 247       9.660  20.360   9.136  1.00 22.89           O  \\nATOM   1577  N   ASN A 248       9.507  17.363   7.067  1.00 14.98           N  \\nATOM   1578  CA  ASN A 248      10.438  16.250   6.884  1.00 16.66           C  \\nATOM   1579  C   ASN A 248      11.461  16.201   8.008  1.00 16.80           C  \\nATOM   1580  O   ASN A 248      12.116  17.202   8.329  1.00 18.51           O  \\nATOM   1581  CB  ASN A 248      11.135  16.387   5.526  1.00 16.82           C  \\nATOM   1582  CG  ASN A 248      11.868  15.126   5.095  1.00 22.21           C  \\nATOM   1583  OD1 ASN A 248      11.660  14.039   5.644  1.00 18.87           O  \\nATOM   1584  ND2 ASN A 248      12.609  15.242   3.996  1.00 28.19           N  \\nATOM   1585  N   ASN A 249      11.556  15.039   8.648  1.00 13.85           N  \\nATOM   1586  CA  ASN A 249      12.422  14.870   9.791  1.00 14.06           C  \\nATOM   1587  C   ASN A 249      13.013  13.469   9.754  1.00 14.71           C  \\nATOM   1588  O   ASN A 249      12.280  12.471   9.773  1.00 15.38           O  \\nATOM   1589  CB  ASN A 249      11.619  15.050  11.084  1.00 16.46           C  \\nATOM   1590  CG  ASN A 249      12.499  15.079  12.308  1.00 15.90           C  \\nATOM   1591  OD1 ASN A 249      12.750  16.150  12.885  1.00 23.40           O  \\nATOM   1592  ND2 ASN A 249      12.958  13.910  12.742  1.00 12.98           N  \\nATOM   1593  N   ALA A 250      14.329  13.393   9.608  1.00 13.09           N  \\nATOM   1594  CA  ALA A 250      14.979  12.108   9.378  1.00 13.93           C  \\nATOM   1595  C   ALA A 250      14.713  11.115  10.504  1.00 14.90           C  \\nATOM   1596  O   ALA A 250      14.417   9.954  10.247  1.00 13.15           O  \\nATOM   1597  CB  ALA A 250      16.473  12.307   9.178  1.00 13.69           C  \\nATOM   1598  N   ALA A 251      14.894  11.536  11.749  1.00 12.35           N  \\nATOM   1599  CA  ALA A 251      14.758  10.607  12.854  1.00 14.31           C  \\nATOM   1600  C   ALA A 251      13.313  10.146  13.016  1.00 15.10           C  \\nATOM   1601  O   ALA A 251      13.059   8.953  13.191  1.00 12.54           O  \\nATOM   1602  CB  ALA A 251      15.298  11.197  14.141  1.00 15.15           C  \\nATOM   1603  N   ILE A 252      12.357  11.055  12.822  1.00 11.99           N  \\nATOM   1604  CA  ILE A 252      10.970  10.645  12.864  1.00 13.31           C  \\nATOM   1605  C   ILE A 252      10.669   9.619  11.766  1.00 14.73           C  \\nATOM   1606  O   ILE A 252      10.000   8.606  12.001  1.00 12.93           O  \\nATOM   1607  CB  ILE A 252      10.020  11.846  12.752  1.00 13.20           C  \\nATOM   1608  CG1 ILE A 252      10.138  12.709  14.006  1.00 14.29           C  \\nATOM   1609  CG2 ILE A 252       8.583  11.362  12.565  1.00 13.66           C  \\nATOM   1610  CD1 ILE A 252       9.250  13.966  13.954  1.00 15.11           C  \\nATOM   1611  N   SER A 253      11.269   9.818  10.603  1.00 12.61           N  \\nATOM   1612  CA  SER A 253      11.075   8.909   9.487  1.00 12.16           C  \\nATOM   1613  C   SER A 253      11.663   7.539   9.822  1.00 12.26           C  \\nATOM   1614  O   SER A 253      10.979   6.517   9.700  1.00 12.44           O  \\nATOM   1615  CB  SER A 253      11.761   9.492   8.246  1.00 12.68           C  \\nATOM   1616  OG  SER A 253      11.414   8.777   7.081  1.00 15.57           O  \\nATOM   1617  N   VAL A 254      12.901   7.519  10.306  1.00 11.27           N  \\nATOM   1618  CA  VAL A 254      13.559   6.252  10.644  1.00 12.07           C  \\nATOM   1619  C   VAL A 254      12.741   5.461  11.658  1.00 11.96           C  \\nATOM   1620  O   VAL A 254      12.476   4.265  11.462  1.00 12.40           O  \\nATOM   1621  CB  VAL A 254      14.992   6.464  11.157  1.00 10.86           C  \\nATOM   1622  CG1 VAL A 254      15.581   5.158  11.730  1.00 12.84           C  \\nATOM   1623  CG2 VAL A 254      15.880   7.002  10.040  1.00 13.26           C  \\nATOM   1624  N   TYR A 255      12.381   6.093  12.764  1.00 11.53           N  \\nATOM   1625  CA  TYR A 255      11.747   5.357  13.842  1.00 11.10           C  \\nATOM   1626  C   TYR A 255      10.305   4.974  13.542  1.00 12.18           C  \\nATOM   1627  O   TYR A 255       9.843   3.904  13.965  1.00 12.26           O  \\nATOM   1628  CB  TYR A 255      11.881   6.096  15.172  1.00 12.18           C  \\nATOM   1629  CG  TYR A 255      13.309   6.058  15.671  1.00 12.31           C  \\nATOM   1630  CD1 TYR A 255      13.885   4.863  16.097  1.00 11.99           C  \\nATOM   1631  CD2 TYR A 255      14.093   7.208  15.683  1.00 10.85           C  \\nATOM   1632  CE1 TYR A 255      15.205   4.815  16.519  1.00 15.66           C  \\nATOM   1633  CE2 TYR A 255      15.447   7.145  15.994  1.00 14.01           C  \\nATOM   1634  CZ  TYR A 255      15.977   5.961  16.464  1.00 18.53           C  \\nATOM   1635  OH  TYR A 255      17.307   5.916  16.832  1.00 17.67           O  \\nATOM   1636  N   THR A 256       9.581   5.852  12.871  1.00 11.41           N  \\nATOM   1637  CA  THR A 256       8.188   5.561  12.544  1.00  9.80           C  \\nATOM   1638  C   THR A 256       8.096   4.448  11.500  1.00 10.41           C  \\nATOM   1639  O   THR A 256       7.359   3.465  11.692  1.00 11.30           O  \\nATOM   1640  CB  THR A 256       7.438   6.809  12.057  1.00 10.89           C  \\nATOM   1641  OG1 THR A 256       7.558   7.833  13.052  1.00 14.03           O  \\nATOM   1642  CG2 THR A 256       5.967   6.511  11.887  1.00 11.36           C  \\nATOM   1643  N   ILE A 257       8.893   4.536  10.442  1.00 11.82           N  \\nATOM   1644  CA  ILE A 257       8.934   3.431   9.482  1.00 10.72           C  \\nATOM   1645  C   ILE A 257       9.344   2.148  10.176  1.00 11.64           C  \\nATOM   1646  O   ILE A 257       8.749   1.097   9.933  1.00 11.99           O  \\nATOM   1647  CB  ILE A 257       9.870   3.713   8.293  1.00 10.94           C  \\nATOM   1648  CG1 ILE A 257       9.352   4.894   7.464  1.00 11.67           C  \\nATOM   1649  CG2 ILE A 257      10.008   2.463   7.412  1.00 11.41           C  \\nATOM   1650  CD1 ILE A 257      10.409   5.450   6.493  1.00 13.52           C  \\nATOM   1651  N   SER A 258      10.393   2.206  10.997  1.00 11.08           N  \\nATOM   1652  CA  SER A 258      10.916   0.979  11.582  1.00 12.38           C  \\nATOM   1653  C   SER A 258       9.905   0.302  12.482  1.00 12.71           C  \\nATOM   1654  O   SER A 258       9.760  -0.936  12.457  1.00 11.61           O  \\nATOM   1655  CB  SER A 258      12.205   1.223  12.349  1.00 12.31           C  \\nATOM   1656  OG  SER A 258      13.232   1.589  11.445  1.00 12.48           O  \\nATOM   1657  N   TRP A 259       9.231   1.094  13.318  1.00 10.42           N  \\nATOM   1658  CA  TRP A 259       8.256   0.528  14.244  1.00 10.55           C  \\nATOM   1659  C   TRP A 259       7.067  -0.056  13.479  1.00 11.05           C  \\nATOM   1660  O   TRP A 259       6.619  -1.178  13.752  1.00 11.96           O  \\nATOM   1661  CB  TRP A 259       7.781   1.591  15.235  1.00 10.06           C  \\nATOM   1662  CG  TRP A 259       7.144   1.016  16.478  1.00  9.86           C  \\nATOM   1663  CD1 TRP A 259       5.842   1.161  16.874  1.00  9.92           C  \\nATOM   1664  CD2 TRP A 259       7.816   0.341  17.550  1.00  9.64           C  \\nATOM   1665  NE1 TRP A 259       5.651   0.591  18.115  1.00 11.58           N  \\nATOM   1666  CE2 TRP A 259       6.855   0.111  18.567  1.00 10.94           C  \\nATOM   1667  CE3 TRP A 259       9.138  -0.071  17.761  1.00 11.21           C  \\nATOM   1668  CZ2 TRP A 259       7.168  -0.556  19.748  1.00 12.86           C  \\nATOM   1669  CZ3 TRP A 259       9.442  -0.759  18.931  1.00 12.14           C  \\nATOM   1670  CH2 TRP A 259       8.468  -0.953  19.927  1.00 10.93           C  \\nATOM   1671  N   MET A 260       6.581   0.670  12.480  1.00 11.09           N  \\nATOM   1672  CA  MET A 260       5.481   0.161  11.665  1.00 11.36           C  \\nATOM   1673  C   MET A 260       5.887  -1.108  10.940  1.00 12.08           C  \\nATOM   1674  O   MET A 260       5.135  -2.085  10.936  1.00 11.80           O  \\nATOM   1675  CB  MET A 260       4.977   1.214  10.667  1.00 12.28           C  \\nATOM   1676  CG  MET A 260       4.328   2.411  11.340  1.00 10.30           C  \\nATOM   1677  SD  MET A 260       2.881   1.956  12.315  1.00 12.18           S  \\nATOM   1678  CE  MET A 260       2.325   3.602  12.805  1.00 12.32           C  \\nATOM   1679  N   LYS A 261       7.090  -1.122  10.359  1.00 10.85           N  \\nATOM   1680  CA  LYS A 261       7.564  -2.332   9.685  1.00 11.87           C  \\nATOM   1681  C   LYS A 261       7.678  -3.519  10.634  1.00 12.02           C  \\nATOM   1682  O   LYS A 261       7.237  -4.612  10.315  1.00 11.37           O  \\nATOM   1683  CB  LYS A 261       8.910  -2.080   9.019  1.00 12.14           C  \\nATOM   1684  CG  LYS A 261       8.851  -1.239   7.736  1.00 11.69           C  \\nATOM   1685  CD  LYS A 261       8.422  -2.121   6.561  1.00 13.34           C  \\nATOM   1686  CE  LYS A 261       8.625  -1.414   5.222  1.00 12.11           C  \\nATOM   1687  NZ  LYS A 261       8.301  -2.351   4.099  1.00 12.41           N  \\nATOM   1688  N   LEU A 262       8.299  -3.299  11.788  1.00 12.69           N  \\nATOM   1689  CA  LEU A 262       8.559  -4.407  12.712  1.00 11.45           C  \\nATOM   1690  C   LEU A 262       7.254  -5.087  13.122  1.00 12.76           C  \\nATOM   1691  O   LEU A 262       7.196  -6.317  13.234  1.00 12.42           O  \\nATOM   1692  CB  LEU A 262       9.293  -3.916  13.956  1.00 11.74           C  \\nATOM   1693  CG  LEU A 262       9.666  -4.981  15.001  1.00 10.45           C  \\nATOM   1694  CD1 LEU A 262      10.516  -6.087  14.346  1.00 13.77           C  \\nATOM   1695  CD2 LEU A 262      10.405  -4.359  16.170  1.00 14.47           C  \\nATOM   1696  N   TRP A 263       6.242  -4.289  13.459  1.00 12.47           N  \\nATOM   1697  CA  TRP A 263       5.024  -4.837  14.041  1.00 11.16           C  \\nATOM   1698  C   TRP A 263       3.888  -5.055  13.047  1.00 12.63           C  \\nATOM   1699  O   TRP A 263       3.199  -6.088  13.089  1.00 15.08           O  \\nATOM   1700  CB  TRP A 263       4.558  -3.964  15.200  1.00 11.92           C  \\nATOM   1701  CG  TRP A 263       5.603  -3.822  16.246  1.00 11.48           C  \\nATOM   1702  CD1 TRP A 263       6.341  -2.708  16.505  1.00 11.90           C  \\nATOM   1703  CD2 TRP A 263       6.047  -4.822  17.185  1.00 11.81           C  \\nATOM   1704  NE1 TRP A 263       7.214  -2.948  17.532  1.00 13.10           N  \\nATOM   1705  CE2 TRP A 263       6.997  -4.211  18.021  1.00 12.19           C  \\nATOM   1706  CE3 TRP A 263       5.668  -6.144  17.451  1.00 13.62           C  \\nATOM   1707  CZ2 TRP A 263       7.680  -4.912  19.013  1.00 14.21           C  \\nATOM   1708  CZ3 TRP A 263       6.337  -6.838  18.464  1.00 14.11           C  \\nATOM   1709  CH2 TRP A 263       7.299  -6.206  19.254  1.00 13.99           C  \\nATOM   1710  N   VAL A 264       3.664  -4.104  12.144  1.00 11.14           N  \\nATOM   1711  CA  VAL A 264       2.608  -4.290  11.166  1.00 11.10           C  \\nATOM   1712  C   VAL A 264       2.957  -5.429  10.215  1.00 12.77           C  \\nATOM   1713  O   VAL A 264       2.091  -6.209   9.844  1.00 12.18           O  \\nATOM   1714  CB  VAL A 264       2.288  -2.993  10.383  1.00 11.44           C  \\nATOM   1715  CG1 VAL A 264       1.176  -3.253   9.351  1.00 13.95           C  \\nATOM   1716  CG2 VAL A 264       1.870  -1.883  11.358  1.00 11.45           C  \\nATOM   1717  N   ASP A 265       4.228  -5.506   9.814  1.00 11.40           N  \\nATOM   1718  CA  ASP A 265       4.654  -6.466   8.793  1.00 12.67           C  \\nATOM   1719  C   ASP A 265       5.423  -7.641   9.384  1.00 12.59           C  \\nATOM   1720  O   ASP A 265       6.003  -8.427   8.623  1.00 14.26           O  \\nATOM   1721  CB  ASP A 265       5.562  -5.808   7.747  1.00 12.01           C  \\nATOM   1722  CG  ASP A 265       4.873  -4.731   6.918  1.00 13.70           C  \\nATOM   1723  OD1 ASP A 265       3.640  -4.515   7.043  1.00 12.66           O  \\nATOM   1724  OD2 ASP A 265       5.592  -4.146   6.080  1.00 13.42           O  \\nATOM   1725  N   ASN A 266       5.632  -7.633  10.697  1.00 12.52           N  \\nATOM   1726  CA  ASN A 266       6.521  -8.611  11.341  1.00 11.18           C  \\nATOM   1727  C   ASN A 266       7.944  -8.545  10.784  1.00 13.24           C  \\nATOM   1728  O   ASN A 266       8.655  -9.567  10.712  1.00 15.02           O  \\nATOM   1729  CB  ASN A 266       5.983 -10.054  11.202  1.00 12.09           C  \\nATOM   1730  CG  ASN A 266       4.612 -10.252  11.838  1.00 15.02           C  \\nATOM   1731  OD1 ASN A 266       4.126  -9.405  12.573  1.00 14.55           O  \\nATOM   1732  ND2 ASN A 266       4.036 -11.432  11.630  1.00 18.58           N  \\nATOM   1733  N   ASP A 267       8.328  -7.374  10.276  1.00 11.89           N  \\nATOM   1734  CA  ASP A 267       9.573  -7.276   9.527  1.00 11.27           C  \\nATOM   1735  C   ASP A 267      10.786  -7.080  10.427  1.00 13.18           C  \\nATOM   1736  O   ASP A 267      11.176  -5.950  10.764  1.00 12.99           O  \\nATOM   1737  CB  ASP A 267       9.494  -6.170   8.489  1.00 15.07           C  \\nATOM   1738  CG  ASP A 267      10.526  -6.328   7.390  1.00 12.54           C  \\nATOM   1739  OD1 ASP A 267      11.588  -6.929   7.643  1.00 14.33           O  \\nATOM   1740  OD2 ASP A 267      10.342  -5.731   6.311  1.00 12.28           O  \\nATOM   1741  N   THR A 268      11.427  -8.189  10.757  1.00 12.31           N  \\nATOM   1742  CA  THR A 268      12.497  -8.171  11.736  1.00 12.82           C  \\nATOM   1743  C   THR A 268      13.795  -7.648  11.156  1.00 11.68           C  \\nATOM   1744  O   THR A 268      14.725  -7.371  11.920  1.00 12.95           O  \\nATOM   1745  CB  THR A 268      12.741  -9.554  12.354  1.00 11.24           C  \\nATOM   1746  OG1 THR A 268      12.969 -10.499  11.301  1.00 14.17           O  \\nATOM   1747  CG2 THR A 268      11.541  -9.972  13.202  1.00 15.91           C  \\nATOM   1748  N   ARG A 269      13.789  -7.312   9.858  1.00 12.31           N  \\nATOM   1749  CA  ARG A 269      14.920  -6.607   9.261  1.00 13.14           C  \\nATOM   1750  C   ARG A 269      15.073  -5.216   9.845  1.00 11.75           C  \\nATOM   1751  O   ARG A 269      16.130  -4.600   9.707  1.00 13.84           O  \\nATOM   1752  CB  ARG A 269      14.818  -6.522   7.739  1.00 13.40           C  \\nATOM   1753  CG  ARG A 269      14.826  -7.866   7.052  1.00 12.81           C  \\nATOM   1754  CD  ARG A 269      14.604  -7.722   5.558  1.00 12.45           C  \\nATOM   1755  NE  ARG A 269      13.348  -7.034   5.262  1.00 12.27           N  \\nATOM   1756  CZ  ARG A 269      12.980  -6.668   4.042  1.00 12.37           C  \\nATOM   1757  NH1 ARG A 269      13.759  -6.960   3.006  1.00 15.72           N  \\nATOM   1758  NH2 ARG A 269      11.777  -6.132   3.832  1.00 14.97           N  \\nATOM   1759  N   TYR A 270      14.028  -4.730  10.508  1.00 13.33           N  \\nATOM   1760  CA  TYR A 270      14.066  -3.408  11.101  1.00 12.35           C  \\nATOM   1761  C   TYR A 270      14.439  -3.377  12.579  1.00 11.91           C  \\nATOM   1762  O   TYR A 270      14.615  -2.298  13.157  1.00 13.58           O  \\nATOM   1763  CB  TYR A 270      12.749  -2.684  10.845  1.00 12.29           C  \\nATOM   1764  CG  TYR A 270      12.558  -2.333   9.391  1.00 11.77           C  \\nATOM   1765  CD1 TYR A 270      12.785  -1.049   8.944  1.00 12.58           C  \\nATOM   1766  CD2 TYR A 270      12.173  -3.297   8.459  1.00 13.45           C  \\nATOM   1767  CE1 TYR A 270      12.648  -0.728   7.599  1.00 12.13           C  \\nATOM   1768  CE2 TYR A 270      12.037  -2.979   7.106  1.00 12.44           C  \\nATOM   1769  CZ  TYR A 270      12.323  -1.698   6.687  1.00 11.58           C  \\nATOM   1770  OH  TYR A 270      12.168  -1.323   5.373  1.00 12.39           O  \\nATOM   1771  N   ARG A 271      14.714  -4.554  13.156  1.00 12.19           N  \\nATOM   1772  CA  ARG A 271      15.200  -4.637  14.523  1.00 12.97           C  \\nATOM   1773  C   ARG A 271      16.505  -3.878  14.686  1.00 12.52           C  \\nATOM   1774  O   ARG A 271      16.766  -3.280  15.722  1.00 14.16           O  \\nATOM   1775  CB  ARG A 271      15.405  -6.098  14.951  1.00 16.58           C  \\nATOM   1776  CG  ARG A 271      14.143  -6.825  15.295  1.00 19.34           C  \\nATOM   1777  CD  ARG A 271      14.447  -8.288  15.614  1.00 22.29           C  \\nATOM   1778  NE  ARG A 271      14.729  -8.501  17.026  1.00 31.63           N  \\nATOM   1779  CZ  ARG A 271      15.019  -9.684  17.556  1.00 38.66           C  \\nATOM   1780  NH1 ARG A 271      15.304 -10.708  16.761  1.00 38.16           N  \\nATOM   1781  NH2 ARG A 271      14.952  -9.864  18.872  1.00 34.63           N  \\nATOM   1782  N   GLN A 272      17.325  -3.882  13.646  1.00 13.50           N  \\nATOM   1783  CA  GLN A 272      18.649  -3.283  13.733  1.00 12.42           C  \\nATOM   1784  C   GLN A 272      18.595  -1.798  14.086  1.00 14.05           C  \\nATOM   1785  O   GLN A 272      19.556  -1.269  14.641  1.00 16.37           O  \\nATOM   1786  CB  GLN A 272      19.409  -3.486  12.425  1.00 15.05           C  \\nATOM   1787  CG  GLN A 272      18.705  -2.887  11.218  1.00 13.31           C  \\nATOM   1788  CD  GLN A 272      19.459  -3.162   9.939  1.00 15.55           C  \\nATOM   1789  OE1 GLN A 272      20.625  -2.762   9.795  1.00 15.68           O  \\nATOM   1790  NE2 GLN A 272      18.823  -3.878   9.015  1.00 13.01           N  \\nATOM   1791  N   PHE A 273      17.517  -1.109  13.713  1.00 12.08           N  \\nATOM   1792  CA  PHE A 273      17.402   0.326  13.984  1.00 12.72           C  \\nATOM   1793  C   PHE A 273      16.799   0.588  15.363  1.00 14.19           C  \\nATOM   1794  O   PHE A 273      16.764   1.729  15.832  1.00 14.03           O  \\nATOM   1795  CB  PHE A 273      16.563   1.006  12.898  1.00 11.48           C  \\nATOM   1796  CG  PHE A 273      16.990   0.649  11.505  1.00 12.19           C  \\nATOM   1797  CD1 PHE A 273      18.253   0.981  11.051  1.00 14.52           C  \\nATOM   1798  CD2 PHE A 273      16.158  -0.081  10.682  1.00 12.15           C  \\nATOM   1799  CE1 PHE A 273      18.674   0.595   9.787  1.00 14.45           C  \\nATOM   1800  CE2 PHE A 273      16.581  -0.507   9.440  1.00 12.61           C  \\nATOM   1801  CZ  PHE A 273      17.844  -0.149   8.977  1.00 13.43           C  \\nATOM   1802  N   LEU A 274      16.292  -0.475  15.987  1.00 11.82           N  \\nATOM   1803  CA  LEU A 274      15.478  -0.366  17.202  1.00 12.32           C  \\nATOM   1804  C   LEU A 274      16.104  -1.080  18.389  1.00 12.08           C  \\nATOM   1805  O   LEU A 274      15.568  -1.020  19.494  1.00 12.89           O  \\nATOM   1806  CB  LEU A 274      14.073  -0.937  16.971  1.00 13.57           C  \\nATOM   1807  CG  LEU A 274      13.237  -0.202  15.923  1.00 13.06           C  \\nATOM   1808  CD1 LEU A 274      12.047  -1.051  15.529  1.00 12.74           C  \\nATOM   1809  CD2 LEU A 274      12.778   1.151  16.456  1.00 15.14           C  \\nATOM   1810  N   CYS A 275      17.267  -1.681  18.171  1.00 14.32           N  \\nATOM   1811  CA  CYS A 275      17.969  -2.443  19.196  1.00 12.98           C  \\nATOM   1812  C   CYS A 275      19.408  -1.942  19.246  1.00 14.03           C  \\nATOM   1813  O   CYS A 275      19.965  -1.526  18.223  1.00 14.45           O  \\nATOM   1814  CB  CYS A 275      17.939  -3.932  18.851  1.00 13.90           C  \\nATOM   1815  SG  CYS A 275      16.281  -4.658  18.936  1.00 14.13           S  \\nATOM   1816  N   ASN A 276      20.018  -2.008  20.428  1.00 15.42           N  \\nATOM   1817  CA  ASN A 276      21.405  -1.575  20.607  1.00 14.74           C  \\nATOM   1818  C   ASN A 276      21.614  -0.162  20.096  1.00 16.18           C  \\nATOM   1819  O   ASN A 276      22.522   0.103  19.303  1.00 17.65           O  \\nATOM   1820  CB  ASN A 276      22.395  -2.544  19.955  1.00 17.75           C  \\nATOM   1821  CG  ASN A 276      22.204  -3.982  20.422  1.00 20.60           C  \\nATOM   1822  OD1 ASN A 276      21.820  -4.860  19.642  1.00 22.74           O  \\nATOM   1823  ND2 ASN A 276      22.469  -4.225  21.696  1.00 21.17           N  \\nATOM   1824  N   VAL A 277      20.750   0.740  20.544  1.00 15.01           N  \\nATOM   1825  CA  VAL A 277      20.635   2.054  19.940  1.00 13.99           C  \\nATOM   1826  C   VAL A 277      21.592   3.042  20.614  1.00 16.12           C  \\nATOM   1827  O   VAL A 277      21.612   3.166  21.837  1.00 19.10           O  \\nATOM   1828  CB  VAL A 277      19.186   2.566  19.997  1.00 14.89           C  \\nATOM   1829  CG1 VAL A 277      19.100   3.990  19.466  1.00 15.66           C  \\nATOM   1830  CG2 VAL A 277      18.281   1.647  19.181  1.00 15.63           C  \\nATOM   1831  N   ASN A 278      22.404   3.719  19.801  1.00 16.50           N  \\nATOM   1832  CA  ASN A 278      23.162   4.877  20.255  1.00 16.44           C  \\nATOM   1833  C   ASN A 278      23.144   5.951  19.184  1.00 19.04           C  \\nATOM   1834  O   ASN A 278      23.975   5.948  18.272  1.00 19.03           O  \\nATOM   1835  CB  ASN A 278      24.614   4.493  20.592  1.00 17.33           C  \\nATOM   1836  CG  ASN A 278      25.430   5.684  21.083  1.00 19.48           C  \\nATOM   1837  OD1 ASN A 278      24.867   6.663  21.560  1.00 29.88           O  \\nATOM   1838  ND2 ASN A 278      26.740   5.499  21.183  1.00 30.33           N  \\nATOM   1839  N   ASP A 279      22.099   6.772  19.215  1.00 15.28           N  \\nATOM   1840  CA  ASP A 279      21.730   7.592  18.068  1.00 15.24           C  \\nATOM   1841  C   ASP A 279      21.992   9.058  18.390  1.00 16.97           C  \\nATOM   1842  O   ASP A 279      21.451   9.582  19.364  1.00 17.35           O  \\nATOM   1843  CB  ASP A 279      20.244   7.379  17.763  1.00 17.73           C  \\nATOM   1844  CG  ASP A 279      19.757   8.222  16.606  1.00 16.04           C  \\nATOM   1845  OD1 ASP A 279      20.509   9.116  16.157  1.00 18.33           O  \\nATOM   1846  OD2 ASP A 279      18.648   7.946  16.084  1.00 16.99           O  \\nATOM   1847  N   PRO A 280      22.967   9.671  17.693  1.00 14.38           N  \\nATOM   1848  CA  PRO A 280      23.356  11.029  18.042  1.00 15.32           C  \\nATOM   1849  C   PRO A 280      22.273  12.072  17.778  1.00 17.79           C  \\nATOM   1850  O   PRO A 280      22.365  13.176  18.304  1.00 18.14           O  \\nATOM   1851  CB  PRO A 280      24.567  11.304  17.125  1.00 18.12           C  \\nATOM   1852  CG  PRO A 280      24.451  10.343  16.038  1.00 21.23           C  \\nATOM   1853  CD  PRO A 280      23.786   9.116  16.603  1.00 16.66           C  \\nATOM   1854  N   ALA A 281      21.234  11.705  17.031  1.00 16.98           N  \\nATOM   1855  CA  ALA A 281      20.101  12.599  16.792  1.00 19.03           C  \\nATOM   1856  C   ALA A 281      19.179  12.697  18.004  1.00 18.93           C  \\nATOM   1857  O   ALA A 281      18.263  13.516  18.023  1.00 20.07           O  \\nATOM   1858  CB  ALA A 281      19.327  12.146  15.580  1.00 18.92           C  \\nATOM   1859  N   LEU A 282      19.414  11.850  19.002  1.00 18.83           N  \\nATOM   1860  CA  LEU A 282      18.538  11.749  20.157  1.00 15.60           C  \\nATOM   1861  C   LEU A 282      19.198  12.299  21.419  1.00 19.69           C  \\nATOM   1862  O   LEU A 282      20.342  11.944  21.743  1.00 21.85           O  \\nATOM   1863  CB  LEU A 282      18.142  10.291  20.393  1.00 17.31           C  \\nATOM   1864  CG  LEU A 282      17.433   9.579  19.254  1.00 15.92           C  \\nATOM   1865  CD1 LEU A 282      16.983   8.182  19.697  1.00 16.12           C  \\nATOM   1866  CD2 LEU A 282      16.259  10.399  18.690  1.00 17.02           C  \\nATOM   1867  N   SER A 283      18.421  13.014  22.220  1.00 17.52           N  \\nATOM   1868  CA  SER A 283      18.846  13.373  23.552  1.00 16.40           C  \\nATOM   1869  C   SER A 283      18.529  12.289  24.567  1.00 17.03           C  \\nATOM   1870  O   SER A 283      18.994  12.352  25.695  1.00 20.61           O  \\nATOM   1871  CB  SER A 283      18.247  14.712  23.991  1.00 15.74           C  \\nATOM   1872  OG  SER A 283      16.897  14.824  23.558  1.00 15.31           O  \\nATOM   1873  N   ASP A 284      17.623  11.378  24.209  1.00 16.35           N  \\nATOM   1874  CA  ASP A 284      17.284  10.260  25.093  1.00 15.24           C  \\nATOM   1875  C   ASP A 284      16.714   9.120  24.268  1.00 15.82           C  \\nATOM   1876  O   ASP A 284      16.162   9.335  23.193  1.00 14.91           O  \\nATOM   1877  CB  ASP A 284      16.263  10.692  26.153  1.00 16.30           C  \\nATOM   1878  CG  ASP A 284      16.104   9.670  27.271  1.00 14.35           C  \\nATOM   1879  OD1 ASP A 284      16.985   8.790  27.423  1.00 20.71           O  \\nATOM   1880  OD2 ASP A 284      15.183   9.857  28.091  1.00 18.74           O  \\nATOM   1881  N   PHE A 285      16.958   7.902  24.735  1.00 14.03           N  \\nATOM   1882  CA  PHE A 285      16.357   6.706  24.156  1.00 13.86           C  \\nATOM   1883  C   PHE A 285      16.037   5.748  25.303  1.00 13.50           C  \\nATOM   1884  O   PHE A 285      16.873   5.511  26.176  1.00 14.38           O  \\nATOM   1885  CB  PHE A 285      17.345   6.037  23.195  1.00 14.86           C  \\nATOM   1886  CG  PHE A 285      16.821   4.776  22.570  1.00 13.89           C  \\nATOM   1887  CD1 PHE A 285      16.067   4.836  21.407  1.00 14.72           C  \\nATOM   1888  CD2 PHE A 285      16.951   3.557  23.218  1.00 16.18           C  \\nATOM   1889  CE1 PHE A 285      15.533   3.689  20.839  1.00 15.70           C  \\nATOM   1890  CE2 PHE A 285      16.380   2.413  22.678  1.00 15.98           C  \\nATOM   1891  CZ  PHE A 285      15.662   2.488  21.497  1.00 16.12           C  \\nATOM   1892  N   ARG A 286      14.798   5.261  25.346  1.00 12.39           N  \\nATOM   1893  CA  ARG A 286      14.418   4.300  26.394  1.00 13.37           C  \\nATOM   1894  C   ARG A 286      13.654   3.159  25.752  1.00 13.99           C  \\nATOM   1895  O   ARG A 286      12.837   3.365  24.855  1.00 14.48           O  \\nATOM   1896  CB  ARG A 286      13.516   4.954  27.437  1.00 14.82           C  \\nATOM   1897  CG  ARG A 286      14.049   6.249  28.005  1.00 14.23           C  \\nATOM   1898  CD  ARG A 286      13.145   6.776  29.082  1.00 16.32           C  \\nATOM   1899  NE  ARG A 286      13.674   8.042  29.569  1.00 15.42           N  \\nATOM   1900  CZ  ARG A 286      13.313   8.628  30.700  1.00 18.27           C  \\nATOM   1901  NH1 ARG A 286      12.467   8.032  31.535  1.00 17.87           N  \\nATOM   1902  NH2 ARG A 286      13.754   9.853  30.954  1.00 18.60           N  \\nATOM   1903  N   THR A 287      13.843   1.957  26.281  1.00 13.04           N  \\nATOM   1904  CA  THR A 287      12.997   0.837  25.879  1.00 13.40           C  \\nATOM   1905  C   THR A 287      12.986  -0.217  26.978  1.00 14.27           C  \\nATOM   1906  O   THR A 287      13.945  -0.297  27.769  1.00 14.86           O  \\nATOM   1907  CB  THR A 287      13.504   0.230  24.560  1.00 13.60           C  \\nATOM   1908  OG1 THR A 287      12.617  -0.805  24.119  1.00 12.63           O  \\nATOM   1909  CG2 THR A 287      14.916  -0.320  24.717  1.00 13.40           C  \\nATOM   1910  N   ASN A 288      11.962  -1.077  26.973  1.00 12.99           N  \\nATOM   1911  CA  ASN A 288      12.045  -2.332  27.728  1.00 12.05           C  \\nATOM   1912  C   ASN A 288      12.652  -3.503  26.958  1.00 12.48           C  \\nATOM   1913  O   ASN A 288      12.816  -4.596  27.521  1.00 13.79           O  \\nATOM   1914  CB  ASN A 288      10.699  -2.702  28.364  1.00 13.25           C  \\nATOM   1915  CG  ASN A 288       9.577  -2.813  27.358  1.00 11.91           C  \\nATOM   1916  OD1 ASN A 288       9.761  -2.574  26.174  1.00 12.51           O  \\nATOM   1917  ND2 ASN A 288       8.399  -3.204  27.838  1.00 11.19           N  \\nATOM   1918  N   ASN A 289      13.095  -3.247  25.727  1.00 14.22           N  \\nATOM   1919  CA  ASN A 289      13.905  -4.199  24.956  1.00 12.82           C  \\nATOM   1920  C   ASN A 289      13.230  -5.554  24.749  1.00 13.86           C  \\nATOM   1921  O   ASN A 289      13.911  -6.554  24.515  1.00 14.14           O  \\nATOM   1922  CB  ASN A 289      15.282  -4.429  25.609  1.00 13.03           C  \\nATOM   1923  CG  ASN A 289      16.372  -3.536  25.038  1.00 14.05           C  \\nATOM   1924  OD1 ASN A 289      17.292  -3.109  25.753  1.00 17.24           O  \\nATOM   1925  ND2 ASN A 289      16.316  -3.306  23.750  1.00 11.32           N  \\nATOM   1926  N   ARG A 290      11.906  -5.590  24.784  1.00 12.67           N  \\nATOM   1927  CA  ARG A 290      11.210  -6.853  24.543  1.00 12.75           C  \\nATOM   1928  C   ARG A 290      11.427  -7.369  23.129  1.00 14.87           C  \\nATOM   1929  O   ARG A 290      11.429  -8.583  22.904  1.00 15.36           O  \\nATOM   1930  CB  ARG A 290       9.726  -6.724  24.876  1.00 13.69           C  \\nATOM   1931  CG  ARG A 290       9.476  -6.616  26.370  1.00 15.87           C  \\nATOM   1932  CD  ARG A 290       7.987  -6.671  26.710  1.00 15.41           C  \\nATOM   1933  NE  ARG A 290       7.396  -7.980  26.420  1.00 13.97           N  \\nATOM   1934  CZ  ARG A 290       7.395  -9.010  27.258  1.00 14.09           C  \\nATOM   1935  NH1 ARG A 290       7.984  -8.920  28.437  1.00 17.11           N  \\nATOM   1936  NH2 ARG A 290       6.803 -10.149  26.900  1.00 15.91           N  \\nATOM   1937  N   HIS A 291      11.706  -6.452  22.201  1.00 14.72           N  \\nATOM   1938  CA  HIS A 291      11.853  -6.763  20.779  1.00 13.19           C  \\nATOM   1939  C   HIS A 291      13.326  -6.994  20.403  1.00 14.13           C  \\nATOM   1940  O   HIS A 291      13.643  -7.190  19.229  1.00 16.03           O  \\nATOM   1941  CB  HIS A 291      11.278  -5.610  19.929  1.00 15.31           C  \\nATOM   1942  CG  HIS A 291      11.954  -4.289  20.163  1.00 15.08           C  \\nATOM   1943  ND1 HIS A 291      11.887  -3.618  21.366  1.00 14.95           N  \\nATOM   1944  CD2 HIS A 291      12.788  -3.566  19.377  1.00 14.15           C  \\nATOM   1945  CE1 HIS A 291      12.660  -2.547  21.317  1.00 14.94           C  \\nATOM   1946  NE2 HIS A 291      13.176  -2.463  20.103  1.00 14.50           N  \\nATOM   1947  N   CYS A 292      14.188  -7.143  21.412  1.00 14.33           N  \\nATOM   1948  CA  CYS A 292      15.632  -7.256  21.202  1.00 15.22           C  \\nATOM   1949  C   CYS A 292      16.275  -8.459  21.885  1.00 18.60           C  \\nATOM   1950  O   CYS A 292      17.474  -8.437  22.189  1.00 25.75           O  \\nATOM   1951  CB  CYS A 292      16.340  -5.989  21.695  1.00 13.03           C  \\nATOM   1952  SG  CYS A 292      15.764  -4.445  20.903  1.00 14.24           S  \\nATOM   1953  N   GLN A 293      15.518  -9.531  22.076  1.00 18.53           N  \\nATOM   1954  CA  GLN A 293      16.109 -10.729  22.661  1.00 21.64           C  \\nATOM   1955  C   GLN A 293      17.208 -11.303  21.785  1.00 22.85           C  \\nATOM   1956  O   GLN A 293      17.123 -11.213  20.559  1.00 26.04           O  \\nATOM   1957  CB  GLN A 293      15.054 -11.785  22.882  1.00 21.59           C  \\nATOM   1958  CG  GLN A 293      14.054 -11.408  23.929  1.00 19.13           C  \\nATOM   1959  CD  GLN A 293      12.868 -12.337  23.915  1.00 19.03           C  \\nATOM   1960  OE1 GLN A 293      12.135 -12.377  22.932  1.00 18.39           O  \\nATOM   1961  NE2 GLN A 293      12.803 -13.232  24.901  1.00 16.69           N  \\nATOM   1962  OXT GLN A 293      18.098 -11.998  22.293  1.00 27.75           O  \\nTER    1963      GLN A 293                                                      \\nHETATM 1964  S   SCN A 301      10.134  -3.022  -3.643  1.00 18.08           S  \\nHETATM 1965  C   SCN A 301       8.906  -4.240  -4.067  1.00 17.44           C  \\nHETATM 1966  N   SCN A 301       8.096  -5.052  -4.325  1.00 19.97           N  \\nHETATM 1967  S   SCN A 302      -7.830  26.388  14.724  1.00 16.65           S  \\nHETATM 1968  C   SCN A 302      -6.768  26.643  13.309  1.00 13.89           C  \\nHETATM 1969  N   SCN A 302      -6.089  26.800  12.362  1.00 15.25           N  \\nHETATM 1970  S   SCN A 303     -17.849   4.124   6.314  1.00 21.15           S  \\nHETATM 1971  C   SCN A 303     -17.707   5.737   7.103  1.00 23.48           C  \\nHETATM 1972  N   SCN A 303     -17.633   6.804   7.616  1.00 22.75           N  \\nHETATM 1973  S   SCN A 304      21.308   4.602  16.307  1.00 42.52           S  \\nHETATM 1974  C   SCN A 304      22.292   3.223  16.913  1.00 35.17           C  \\nHETATM 1975  N   SCN A 304      22.922   2.313  17.312  1.00 36.86           N  \\nHETATM 1976  O   HOH A 401       2.501  -3.110   5.115  1.00 13.13           O  \\nHETATM 1977  O   HOH A 402       5.309  15.174  24.789  1.00 12.47           O  \\nHETATM 1978  O   HOH A 403       7.343  15.970  22.731  1.00 13.26           O  \\nHETATM 1979  O   HOH A 404      11.816   0.459   3.095  1.00 12.12           O  \\nHETATM 1980  O   HOH A 405      21.956   1.038   7.613  1.00 15.26           O  \\nHETATM 1981  O   HOH A 406       9.680  -0.971   1.896  1.00 12.61           O  \\nHETATM 1982  O   HOH A 407      11.783  -2.935   3.423  1.00 13.43           O  \\nHETATM 1983  O   HOH A 408       2.886  -5.771   4.136  1.00 13.44           O  \\nHETATM 1984  O   HOH A 409      -2.002  19.349  20.171  1.00 19.40           O  \\nHETATM 1985  O   HOH A 410     -13.166  16.803  22.491  1.00 17.49           O  \\nHETATM 1986  O   HOH A 411      -7.369  18.872  14.627  1.00 14.31           O  \\nHETATM 1987  O   HOH A 412      19.378  -4.342   6.064  1.00 15.50           O  \\nHETATM 1988  O   HOH A 413       3.565  18.229   5.438  1.00 17.71           O  \\nHETATM 1989  O   HOH A 414      -0.909  -8.851   3.314  1.00 15.51           O  \\nHETATM 1990  O   HOH A 415       1.172  -5.979  28.159  1.00 14.95           O  \\nHETATM 1991  O   HOH A 416       2.039  -6.514  24.479  1.00 17.93           O  \\nHETATM 1992  O   HOH A 417      -0.675  17.826  13.670  1.00 15.12           O  \\nHETATM 1993  O   HOH A 418      21.136  -1.500   7.280  1.00 16.25           O  \\nHETATM 1994  O   HOH A 419       5.531  -2.562   3.946  1.00 14.42           O  \\nHETATM 1995  O   HOH A 420      -3.483 -10.997   6.702  1.00 15.99           O  \\nHETATM 1996  O   HOH A 421       1.084   5.569  29.981  1.00 15.75           O  \\nHETATM 1997  O   HOH A 422      16.822   0.259   1.843  1.00 15.92           O  \\nHETATM 1998  O   HOH A 423      10.234  -3.589  23.624  1.00 15.39           O  \\nHETATM 1999  O   HOH A 424       8.074  23.520  21.081  1.00 15.43           O  \\nHETATM 2000  O   HOH A 425     -12.299  14.322  23.874  1.00 18.24           O  \\nHETATM 2001  O   HOH A 426       8.013  -5.029   5.113  1.00 13.34           O  \\nHETATM 2002  O   HOH A 427      17.470  -6.365  11.992  1.00 15.99           O  \\nHETATM 2003  O   HOH A 428      19.105   0.051  22.903  1.00 18.22           O  \\nHETATM 2004  O   HOH A 429       7.366  12.556   2.093  1.00 18.57           O  \\nHETATM 2005  O   HOH A 430      -2.625  14.875   4.636  1.00 16.37           O  \\nHETATM 2006  O   HOH A 431      -4.443  -5.207  13.196  1.00 15.84           O  \\nHETATM 2007  O   HOH A 432       3.382  -9.224  26.962  1.00 16.42           O  \\nHETATM 2008  O   HOH A 433       8.960 -11.815  11.660  1.00 20.02           O  \\nHETATM 2009  O   HOH A 434      18.913  -3.604  22.582  1.00 16.96           O  \\nHETATM 2010  O   HOH A 435      -4.256  -8.297  15.603  1.00 17.89           O  \\nHETATM 2011  O   HOH A 436      -8.202  18.633  11.344  1.00 16.29           O  \\nHETATM 2012  O   HOH A 437      11.202  18.460  11.347  1.00 19.44           O  \\nHETATM 2013  O   HOH A 438      -5.956  19.295   5.381  1.00 18.31           O  \\nHETATM 2014  O   HOH A 439      14.494   5.627   0.349  1.00 16.06           O  \\nHETATM 2015  O   HOH A 440      -2.521  22.481  10.210  1.00 14.86           O  \\nHETATM 2016  O   HOH A 441      10.839 -10.161   8.088  1.00 19.82           O  \\nHETATM 2017  O   HOH A 442       5.316 -13.136   9.538  1.00 15.41           O  \\nHETATM 2018  O   HOH A 443      10.122  18.785  29.368  1.00 20.71           O  \\nHETATM 2019  O   HOH A 444      10.056  19.361  26.534  1.00 18.16           O  \\nHETATM 2020  O   HOH A 445     -16.566   5.093  10.144  1.00 21.26           O  \\nHETATM 2021  O   HOH A 446      19.914  -1.687  25.034  1.00 19.55           O  \\nHETATM 2022  O   HOH A 447      14.204   1.555   2.100  1.00 13.60           O  \\nHETATM 2023  O   HOH A 448      -1.916  20.588  22.734  1.00 19.85           O  \\nHETATM 2024  O   HOH A 449      18.502  -7.294   9.449  1.00 18.65           O  \\nHETATM 2025  O   HOH A 450       1.664  -8.556  11.411  1.00 17.62           O  \\nHETATM 2026  O   HOH A 451      -4.632  -2.766  24.338  1.00 17.64           O  \\nHETATM 2027  O   HOH A 452       5.441  -4.671  -5.467  1.00 19.71           O  \\nHETATM 2028  O   HOH A 453      -9.565  13.770  25.926  1.00 18.96           O  \\nHETATM 2029  O   HOH A 454      18.002  -6.599   6.815  1.00 19.51           O  \\nHETATM 2030  O   HOH A 455       6.183  -2.307  -6.617  1.00 18.31           O  \\nHETATM 2031  O   HOH A 456      -4.402  24.415  10.940  1.00 15.72           O  \\nHETATM 2032  O   HOH A 457       6.440  -2.954  36.171  1.00 16.99           O  \\nHETATM 2033  O   HOH A 458      18.569   3.761  15.800  1.00 20.39           O  \\nHETATM 2034  O   HOH A 459       2.337  24.112  27.120  1.00 18.10           O  \\nHETATM 2035  O   HOH A 460       4.313  26.542  23.430  1.00 18.83           O  \\nHETATM 2036  O   HOH A 461      19.430   4.692  25.898  1.00 18.02           O  \\nHETATM 2037  O   HOH A 462     -17.230  17.780   2.190  1.00 28.37           O  \\nHETATM 2038  O   HOH A 463      -1.344 -14.842   6.471  1.00 22.90           O  \\nHETATM 2039  O   HOH A 464      -4.693  -0.392  -6.433  1.00 23.06           O  \\nHETATM 2040  O   HOH A 465       6.360  25.419  19.828  1.00 21.63           O  \\nHETATM 2041  O   HOH A 466      20.959   5.032  23.577  1.00 23.10           O  \\nHETATM 2042  O   HOH A 467      20.477   6.937  21.530  1.00 18.93           O  \\nHETATM 2043  O   HOH A 468       0.066   5.421  36.865  1.00 24.14           O  \\nHETATM 2044  O   HOH A 469      -3.422  27.137  13.251  1.00 18.59           O  \\nHETATM 2045  O   HOH A 470      16.754  -1.036  22.144  1.00 17.16           O  \\nHETATM 2046  O   HOH A 471      -9.176  -4.337   1.870  1.00 22.05           O  \\nHETATM 2047  O   HOH A 472     -10.524   3.502  -7.053  1.00 23.85           O  \\nHETATM 2048  O   HOH A 473       8.591  -1.062  -5.784  1.00 15.37           O  \\nHETATM 2049  O   HOH A 474      -1.264  -5.904  29.332  1.00 24.09           O  \\nHETATM 2050  O   HOH A 475      -1.610  25.017  13.866  1.00 18.61           O  \\nHETATM 2051  O   HOH A 476      -2.491   1.082  -6.497  1.00 23.02           O  \\nHETATM 2052  O   HOH A 477     -16.014  13.924  19.970  1.00 23.23           O  \\nHETATM 2053  O   HOH A 478       0.306  -9.520  -0.886  1.00 22.20           O  \\nHETATM 2054  O   HOH A 479     -10.847  20.005  11.321  1.00 18.22           O  \\nHETATM 2055  O   HOH A 480      -1.701  15.720   0.413  1.00 21.37           O  \\nHETATM 2056  O   HOH A 481      15.931  15.855   9.805  1.00 21.09           O  \\nHETATM 2057  O   HOH A 482      17.178   6.986  29.390  1.00 24.03           O  \\nHETATM 2058  O   HOH A 483       8.743  18.851   4.573  1.00 23.66           O  \\nHETATM 2059  O   HOH A 484     -17.453  14.527   8.793  1.00 20.97           O  \\nHETATM 2060  O   HOH A 485       0.600  21.082  14.045  1.00 21.89           O  \\nHETATM 2061  O   HOH A 486      10.900  -7.820   1.205  1.00 19.86           O  \\nHETATM 2062  O   HOH A 487      12.268  11.510  -0.858  1.00 26.19           O  \\nHETATM 2063  O   HOH A 488       1.083   8.445  -8.133  1.00 27.32           O  \\nHETATM 2064  O   HOH A 489      16.135  14.277  12.370  1.00 24.27           O  \\nHETATM 2065  O   HOH A 490      -0.663  22.911  12.311  1.00 18.54           O  \\nHETATM 2066  O   HOH A 491       0.990  -7.962  26.406  1.00 21.30           O  \\nHETATM 2067  O   HOH A 492       8.324   6.586  -3.826  1.00 23.36           O  \\nHETATM 2068  O   HOH A 493      20.712   0.557  16.420  1.00 23.91           O  \\nHETATM 2069  O   HOH A 494      23.643  -1.037  17.024  1.00 23.74           O  \\nHETATM 2070  O   HOH A 495      21.220  -4.729  16.862  1.00 27.42           O  \\nHETATM 2071  O   HOH A 496      -1.273  22.591   7.822  1.00 17.52           O  \\nHETATM 2072  O   HOH A 497      -0.367   0.826  32.231  1.00 25.21           O  \\nHETATM 2073  O   HOH A 498       2.493 -10.643  29.119  1.00 21.11           O  \\nHETATM 2074  O   HOH A 499     -10.754  -3.266  -5.058  1.00 24.92           O  \\nHETATM 2075  O   HOH A 500     -12.636  21.290  22.859  1.00 23.87           O  \\nHETATM 2076  O   HOH A 501      -1.294  -9.137  25.736  1.00 26.15           O  \\nHETATM 2077  O   HOH A 502       1.396   1.395  34.892  1.00 26.64           O  \\nHETATM 2078  O   HOH A 503     -12.552   7.632  22.969  1.00 21.60           O  \\nHETATM 2079  O   HOH A 504      12.606  20.986  15.629  1.00 24.03           O  \\nHETATM 2080  O   HOH A 505      -2.853  10.469  -6.927  1.00 25.47           O  \\nHETATM 2081  O   HOH A 506      -8.357  -6.848   3.614  1.00 26.25           O  \\nHETATM 2082  O   HOH A 507       9.143  -6.590  29.925  1.00 21.43           O  \\nHETATM 2083  O   HOH A 508       0.451 -10.861   1.990  1.00 22.95           O  \\nHETATM 2084  O   HOH A 509       5.831   7.701  36.050  1.00 22.50           O  \\nHETATM 2085  O   HOH A 510      -7.752  27.141  21.641  1.00 29.31           O  \\nHETATM 2086  O   HOH A 511      18.987  -6.261  16.350  1.00 27.44           O  \\nHETATM 2087  O   HOH A 512      -4.420  23.292   6.128  1.00 23.72           O  \\nHETATM 2088  O   HOH A 513      25.288   5.604  16.179  1.00 22.84           O  \\nHETATM 2089  O   HOH A 514     -11.086  16.943   5.223  1.00 23.65           O  \\nHETATM 2090  O   HOH A 515       3.451  -7.997  15.203  1.00 24.84           O  \\nHETATM 2091  O   HOH A 516      11.696 -12.740  11.405  1.00 23.03           O  \\nHETATM 2092  O   HOH A 517      -6.488  -2.080  27.440  1.00 22.75           O  \\nHETATM 2093  O   HOH A 518      -0.121  -6.719  21.067  1.00 21.24           O  \\nHETATM 2094  O   HOH A 519      14.136  13.190   2.560  1.00 25.34           O  \\nHETATM 2095  O   HOH A 520       1.867  -6.237  17.068  1.00 21.13           O  \\nHETATM 2096  O   HOH A 521      -3.230  10.384  34.807  1.00 24.69           O  \\nHETATM 2097  O   HOH A 522      11.792  -6.194  29.485  1.00 25.21           O  \\nHETATM 2098  O   HOH A 523     -15.116  16.691  20.592  1.00 24.44           O  \\nHETATM 2099  O   HOH A 524     -16.634  16.454  10.080  1.00 23.10           O  \\nHETATM 2100  O   HOH A 525       3.151  22.266  31.121  1.00 28.48           O  \\nHETATM 2101  O   HOH A 526     -16.330  17.889  17.292  1.00 23.95           O  \\nHETATM 2102  O   HOH A 527       7.954  -6.110  -1.655  1.00 23.49           O  \\nHETATM 2103  O   HOH A 528      14.086  12.796   6.286  1.00 26.52           O  \\nHETATM 2104  O   HOH A 529      -7.493  13.711  32.372  1.00 25.17           O  \\nHETATM 2105  O   HOH A 530     -12.035   0.794  14.003  1.00 23.16           O  \\nHETATM 2106  O   HOH A 531      -3.309  -4.005  28.548  1.00 27.07           O  \\nHETATM 2107  O   HOH A 532      23.636   3.850  12.431  1.00 24.41           O  \\nHETATM 2108  O   HOH A 533      -1.323  17.756  31.679  1.00 25.85           O  \\nHETATM 2109  O   HOH A 534       2.919  10.080  35.948  1.00 30.32           O  \\nHETATM 2110  O   HOH A 535     -13.499   3.190  -6.466  1.00 25.42           O  \\nHETATM 2111  O   HOH A 536      -9.742   3.373  28.208  1.00 27.52           O  \\nHETATM 2112  O   HOH A 537      -8.443 -12.581   5.987  1.00 26.58           O  \\nHETATM 2113  O   HOH A 538      19.675   1.968  24.856  1.00 27.33           O  \\nHETATM 2114  O   HOH A 539     -10.748   1.020  16.643  1.00 28.53           O  \\nHETATM 2115  O   HOH A 540      10.710  24.482  21.216  1.00 24.80           O  \\nHETATM 2116  O   HOH A 541       1.526   6.287  -9.175  1.00 29.12           O  \\nHETATM 2117  O   HOH A 542       1.476  22.563   8.363  1.00 28.80           O  \\nHETATM 2118  O   HOH A 543     -18.869   7.647  13.414  1.00 27.84           O  \\nHETATM 2119  O   HOH A 544      23.030   5.469  14.548  1.00 27.58           O  \\nHETATM 2120  O   HOH A 545     -18.758  -0.963  11.170  1.00 25.32           O  \\nHETATM 2121  O   HOH A 546      14.813 -12.928  17.607  1.00 32.33           O  \\nHETATM 2122  O   HOH A 547      12.802 -10.178  21.211  1.00 22.66           O  \\nHETATM 2123  O   HOH A 548      -0.057  -8.931 -10.914  1.00 27.52           O  \\nHETATM 2124  O   HOH A 549      -1.963   8.552  35.625  1.00 29.30           O  \\nHETATM 2125  O   HOH A 550      -9.030  12.516  -4.703  1.00 27.73           O  \\nHETATM 2126  O   HOH A 551     -18.681  12.548  13.638  1.00 25.24           O  \\nHETATM 2127  O   HOH A 552      -0.243 -10.581  11.152  1.00 25.28           O  \\nHETATM 2128  O   HOH A 553       4.922   8.889  -5.839  1.00 28.28           O  \\nHETATM 2129  O   HOH A 554       7.971  21.802  11.085  1.00 24.35           O  \\nHETATM 2130  O   HOH A 555      -5.424  -5.322  30.533  1.00 27.23           O  \\nHETATM 2131  O   HOH A 556     -10.024  -0.583  -9.006  1.00 28.43           O  \\nHETATM 2132  O   HOH A 557       2.100  -7.448  19.435  1.00 25.60           O  \\nHETATM 2133  O   HOH A 558      -0.784  20.077  16.468  1.00 28.65           O  \\nHETATM 2134  O   HOH A 559      -1.928  -8.578  16.955  1.00 23.70           O  \\nHETATM 2135  O   HOH A 560      14.192 -12.637  27.413  1.00 25.48           O  \\nHETATM 2136  O   HOH A 561      -6.498   0.431  33.347  1.00 27.98           O  \\nHETATM 2137  O   HOH A 562      19.351  -3.046   0.308  1.00 28.79           O  \\nHETATM 2138  O   HOH A 563      -6.013  -7.081  -3.461  1.00 28.32           O  \\nHETATM 2139  O   HOH A 564      -3.567  17.011   3.408  1.00 28.57           O  \\nHETATM 2140  O   HOH A 565      -6.394  -9.681   3.971  1.00 34.38           O  \\nHETATM 2141  O   HOH A 566       8.811  12.931  33.719  1.00 29.23           O  \\nHETATM 2142  O   HOH A 567      15.008 -10.760   9.614  1.00 31.45           O  \\nHETATM 2143  O   HOH A 568      21.883  -2.507  15.470  1.00 24.02           O  \\nHETATM 2144  O   HOH A 569       6.852   0.466  32.955  1.00 29.52           O  \\nHETATM 2145  O   HOH A 570      -4.707   4.910  -8.120  1.00 27.23           O  \\nHETATM 2146  O   HOH A 571      -3.504   2.404  -8.596  1.00 27.75           O  \\nHETATM 2147  O   HOH A 572     -13.398  -4.656   1.365  1.00 28.37           O  \\nHETATM 2148  O   HOH A 573      -0.106 -15.635   4.230  1.00 34.98           O  \\nHETATM 2149  O   HOH A 574      15.285  -2.401  28.819  1.00 24.74           O  \\nHETATM 2150  O   HOH A 575      -5.720  13.739  -2.409  1.00 29.03           O  \\nHETATM 2151  O   HOH A 576      -7.569  16.495  28.679  1.00 25.62           O  \\nHETATM 2152  O   HOH A 577       1.479  23.703  10.810  1.00 26.64           O  \\nHETATM 2153  O   HOH A 578      17.609  15.705  16.723  1.00 31.63           O  \\nHETATM 2154  O   HOH A 579      13.891  18.697  27.718  1.00 28.57           O  \\nHETATM 2155  O   HOH A 580     -14.612   6.083  -1.217  1.00 30.22           O  \\nHETATM 2156  O   HOH A 581      19.825  -6.150  22.871  1.00 34.20           O  \\nHETATM 2157  O   HOH A 582      -9.699  11.555  29.409  1.00 39.22           O  \\nHETATM 2158  O   HOH A 583      -2.479  -6.180  21.862  1.00 28.86           O  \\nHETATM 2159  O   HOH A 584     -11.173  -4.142   3.771  1.00 30.52           O  \\nHETATM 2160  O   HOH A 585      17.649  -9.768   9.689  1.00 28.69           O  \\nHETATM 2161  O   HOH A 586     -10.961  -1.286  12.487  1.00 32.82           O  \\nHETATM 2162  O   HOH A 587     -17.892  -2.726   3.009  1.00 32.10           O  \\nHETATM 2163  O   HOH A 588       9.193  -9.550   0.038  1.00 28.88           O  \\nHETATM 2164  O   HOH A 589      17.898  -6.609  24.861  1.00 26.39           O  \\nHETATM 2165  O   HOH A 590     -17.676  -2.989  13.226  1.00 26.07           O  \\nHETATM 2166  O   HOH A 591       4.103  -7.544  20.887  1.00 29.02           O  \\nHETATM 2167  O   HOH A 592     -14.267  12.342  23.813  1.00 32.73           O  \\nHETATM 2168  O   HOH A 593      11.100 -13.283   4.523  1.00 27.20           O  \\nHETATM 2169  O   HOH A 594       7.962  13.439  -1.003  1.00 30.25           O  \\nHETATM 2170  O   HOH A 595       0.711  11.272  -5.311  1.00 29.23           O  \\nHETATM 2171  O   HOH A 596       1.214  20.430   4.012  1.00 24.79           O  \\nHETATM 2172  O   HOH A 597       2.134 -10.141  15.005  1.00 34.68           O  \\nHETATM 2173  O   HOH A 598      15.100  -8.417  26.239  1.00 29.00           O  \\nHETATM 2174  O   HOH A 599      14.835  12.463  28.616  1.00 28.58           O  \\nHETATM 2175  O   HOH A 600      21.483  -6.407   2.637  1.00 30.79           O  \\nHETATM 2176  O   HOH A 601     -10.082  -7.680   9.068  1.00 33.33           O  \\nHETATM 2177  O   HOH A 602       4.791  10.719  -3.907  1.00 33.33           O  \\nHETATM 2178  O   HOH A 603       3.568  23.167  12.131  1.00 31.18           O  \\nHETATM 2179  O   HOH A 604     -14.844  11.197  -0.637  1.00 30.26           O  \\nHETATM 2180  O   HOH A 605      18.404  -8.909   5.520  1.00 36.71           O  \\nHETATM 2181  O   HOH A 606      18.237  -8.079  18.750  1.00 33.86           O  \\nHETATM 2182  O   HOH A 607      17.892  11.278   2.623  1.00 33.03           O  \\nHETATM 2183  O   HOH A 608      -4.746  -5.126  26.442  1.00 31.95           O  \\nHETATM 2184  O   HOH A 609      11.423  13.535  33.345  1.00 34.37           O  \\nHETATM 2185  O   HOH A 610       8.019  11.105  35.485  1.00 30.19           O  \\nHETATM 2186  O   HOH A 611     -13.051  25.593   9.302  1.00 25.71           O  \\nHETATM 2187  O   HOH A 612       0.455  26.295  14.856  1.00 29.01           O  \\nHETATM 2188  O   HOH A 613      22.425  12.944  22.904  1.00 26.31           O  \\nHETATM 2189  O   HOH A 614       0.668 -10.203  31.075  1.00 27.19           O  \\nHETATM 2190  O   HOH A 615      20.896   3.083   0.992  1.00 30.51           O  \\nHETATM 2191  O   HOH A 616      11.821  20.916  13.026  1.00 30.52           O  \\nHETATM 2192  O   HOH A 617      25.406  -3.335  17.780  1.00 30.97           O  \\nHETATM 2193  O   HOH A 618      20.235  18.958  23.244  1.00 30.81           O  \\nHETATM 2194  O   HOH A 619       0.080  20.806  18.810  1.00 31.80           O  \\nHETATM 2195  O   HOH A 620     -16.417  15.260  17.238  1.00 32.26           O  \\nHETATM 2196  O   HOH A 621       2.669 -11.179  -1.000  1.00 33.81           O  \\nHETATM 2197  O   HOH A 622      17.095 -13.172  18.671  1.00 41.08           O  \\nHETATM 2198  O   HOH A 623     -18.283  13.408  11.025  1.00 39.75           O  \\nHETATM 2199  O   HOH A 624     -10.963  23.384   7.462  1.00 31.27           O  \\nHETATM 2200  O   HOH A 625       3.066 -13.018   0.735  1.00 29.41           O  \\nHETATM 2201  O   HOH A 626      -2.431  14.736  -3.787  1.00 31.91           O  \\nHETATM 2202  O   HOH A 627       8.359 -11.212  30.088  1.00 30.37           O  \\nHETATM 2203  O   HOH A 628      21.199  -6.911   9.685  1.00 32.86           O  \\nHETATM 2204  O   HOH A 629      21.847  15.540  17.463  1.00 36.92           O  \\nHETATM 2205  O   HOH A 630     -11.951  15.001  26.505  1.00 30.61           O  \\nHETATM 2206  O   HOH A 631     -11.851   4.553  26.798  1.00 27.48           O  \\nHETATM 2207  O   HOH A 632      11.620 -12.860   8.761  1.00 31.54           O  \\nHETATM 2208  O   HOH A 633      22.639  -4.493   8.859  1.00 32.32           O  \\nHETATM 2209  O   HOH A 634       2.930  23.060  33.894  1.00 38.31           O  \\nHETATM 2210  O   HOH A 635     -21.436  12.482   7.060  1.00 38.57           O  \\nHETATM 2211  O   HOH A 636      19.009  -7.230  13.877  1.00 29.90           O  \\nHETATM 2212  O   HOH A 637      -3.256  -6.443  -9.515  1.00 37.20           O  \\nHETATM 2213  O   HOH A 638      -8.924  11.048  -6.823  1.00 37.80           O  \\nHETATM 2214  O   HOH A 639      -9.397  -2.731  14.301  1.00 31.73           O  \\nHETATM 2215  O   HOH A 640      -4.774  -5.226  -7.517  1.00 32.79           O  \\nHETATM 2216  O   HOH A 641      -0.495  22.963   3.617  1.00 27.62           O  \\nHETATM 2217  O   HOH A 642      -4.757  -4.432  33.569  1.00 35.12           O  \\nHETATM 2218  O   HOH A 643      -7.531   7.004  -5.586  1.00 31.28           O  \\nHETATM 2219  O   HOH A 644      12.681  15.494  31.199  1.00 35.93           O  \\nHETATM 2220  O   HOH A 645      11.980  24.366  25.647  1.00 35.08           O  \\nHETATM 2221  O   HOH A 646     -12.915  -0.888  23.525  1.00 32.83           O  \\nHETATM 2222  O   HOH A 647      12.605  17.794  30.316  1.00 33.13           O  \\nHETATM 2223  O   HOH A 648     -12.069  -2.223  19.974  1.00 36.20           O  \\nHETATM 2224  O   HOH A 649      13.668   7.953  -2.739  1.00 31.23           O  \\nHETATM 2225  O   HOH A 650      -1.869  -6.215  18.998  1.00 29.49           O  \\nHETATM 2226  O   HOH A 651     -12.349   4.060  19.763  1.00 34.56           O  \\nHETATM 2227  O   HOH A 652       6.876 -11.384   1.592  1.00 30.36           O  \\nHETATM 2228  O   HOH A 653      -3.157  15.529  34.431  1.00 34.43           O  \\nHETATM 2229  O   HOH A 654      -5.776  13.454  -4.908  1.00 31.05           O  \\nHETATM 2230  O   HOH A 655     -22.154  13.467  11.829  1.00 32.13           O  \\nHETATM 2231  O   HOH A 656      16.818   6.255  -0.939  1.00 35.70           O  \\nHETATM 2232  O   HOH A 657       1.575 -12.236  12.905  1.00 28.98           O  \\nHETATM 2233  O   HOH A 658      -1.734  -3.906  32.294  1.00 33.82           O  \\nHETATM 2234  O   HOH A 659      -1.981  -9.511  12.505  1.00 30.75           O  \\nHETATM 2235  O   HOH A 660      -9.982  20.774   7.643  1.00 40.66           O  \\nHETATM 2236  O   HOH A 661      -1.633   3.147 -10.679  1.00 32.56           O  \\nHETATM 2237  O   HOH A 662     -11.218  -4.492  23.129  1.00 33.42           O  \\nHETATM 2238  O   HOH A 663      -6.246  -0.642  -8.555  1.00 33.34           O  \\nHETATM 2239  O   HOH A 664       8.636  26.185  18.066  1.00 39.09           O  \\nHETATM 2240  O   HOH A 665       0.289  19.426   1.506  1.00 33.45           O  \\nHETATM 2241  O   HOH A 666      24.958   0.561  20.504  1.00 38.90           O  \\nHETATM 2242  O   HOH A 667      14.401 -11.408   2.783  1.00 28.78           O  \\nHETATM 2243  O   HOH A 668     -12.040  19.788   5.161  1.00 35.49           O  \\nHETATM 2244  O   HOH A 669       1.190  -6.798  32.445  1.00 33.64           O  \\nHETATM 2245  O   HOH A 670      16.759  15.132  27.840  1.00 35.09           O  \\nHETATM 2246  O   HOH A 671      12.048  19.898   7.782  1.00 34.90           O  \\nHETATM 2247  O   HOH A 672      -8.324  -7.103  -4.872  1.00 37.96           O  \\nHETATM 2248  O   HOH A 673      17.083  21.360  23.947  1.00 40.58           O  \\nHETATM 2249  O   HOH A 674      21.075  10.005  13.881  1.00 31.41           O  \\nHETATM 2250  O   HOH A 675       5.853  19.591   5.392  1.00 42.35           O  \\nHETATM 2251  O   HOH A 676      10.144  24.384  14.904  1.00 37.30           O  \\nHETATM 2252  O   HOH A 677     -14.221  20.062  20.108  1.00 32.96           O  \\nHETATM 2253  O   HOH A 678       2.680  14.556   0.695  1.00 34.21           O  \\nHETATM 2254  O   HOH A 679       9.223  14.918   2.885  1.00 33.96           O  \\nHETATM 2255  O   HOH A 680      13.358  11.418  33.226  1.00 35.89           O  \\nHETATM 2256  O   HOH A 681      23.633  18.090  21.121  1.00 36.63           O  \\nHETATM 2257  O   HOH A 682      15.387 -13.695  10.254  1.00 33.76           O  \\nHETATM 2258  O   HOH A 683      14.328 -11.970  14.778  1.00 33.87           O  \\nHETATM 2259  O   HOH A 684      17.334   8.866  -1.651  1.00 35.14           O  \\nHETATM 2260  O   HOH A 685       4.615  16.131   1.198  1.00 36.92           O  \\nHETATM 2261  O   HOH A 686      -8.578  -5.606  11.346  1.00 30.81           O  \\nHETATM 2262  O   HOH A 687      19.659   5.627   0.375  1.00 31.39           O  \\nHETATM 2263  O   HOH A 688      -9.265  16.184   3.163  1.00 38.30           O  \\nHETATM 2264  O   HOH A 689      13.957 -11.331   6.586  1.00 33.90           O  \\nHETATM 2265  O   HOH A 690       6.051   9.677  -2.343  1.00 31.22           O  \\nHETATM 2266  O   HOH A 691      -9.651   5.785  -7.090  1.00 34.12           O  \\nHETATM 2267  O   HOH A 692      -7.148   9.545  -6.453  1.00 38.88           O  \\nHETATM 2268  O   HOH A 693       5.474  14.596  35.467  1.00 40.30           O  \\nHETATM 2269  O   HOH A 694      20.475  17.556  25.232  1.00 38.54           O  \\nHETATM 2270  O   HOH A 695      16.741  -9.152  13.095  1.00 36.51           O  \\nHETATM 2271  O   HOH A 696       0.082 -13.611   2.271  1.00 34.99           O  \\nHETATM 2272  O   HOH A 697      -1.617  -1.224 -10.462  1.00 38.27           O  \\nHETATM 2273  O   HOH A 698      17.042 -10.212   3.681  1.00 36.40           O  \\nHETATM 2274  O   HOH A 699      12.879  24.932  19.778  1.00 33.80           O  \\nHETATM 2275  O   HOH A 700      21.175   9.391  22.769  1.00 36.00           O  \\nHETATM 2276  O   HOH A 701      -6.919   3.260  31.656  1.00 37.31           O  \\nHETATM 2277  O   HOH A 702      -0.006  -2.257  33.486  1.00 35.12           O  \\nHETATM 2278  O   HOH A 703      21.762  -4.630   4.548  1.00 40.71           O  \\nHETATM 2279  O   HOH A 704      -1.257  -9.744  -3.237  1.00 33.87           O  \\nHETATM 2280  O   HOH A 705       9.496  22.971  27.447  1.00 39.67           O  \\nHETATM 2281  O   HOH A 706     -17.533   0.852  -0.520  1.00 40.92           O  \\nHETATM 2282  O   HOH A 707       7.197 -10.170  32.790  1.00 37.69           O  \\nHETATM 2283  O   HOH A 708      16.513  25.094  22.300  1.00 40.10           O  \\nHETATM 2284  O   HOH A 709       3.069  13.021  -6.476  1.00 35.88           O  \\nHETATM 2285  O   HOH A 710      15.347  15.896  14.729  1.00 32.31           O  \\nHETATM 2286  O   HOH A 711       0.202  22.824  16.213  1.00 31.62           O  \\nHETATM 2287  O   HOH A 712      19.570  21.797  17.267  1.00 40.23           O  \\nHETATM 2288  O   HOH A 713     -13.232  18.517  24.643  1.00 32.87           O  \\nHETATM 2289  O   HOH A 714     -12.326  18.791  26.729  1.00 37.04           O  \\nHETATM 2290  O   HOH A 715      -0.997  -0.871  -7.903  1.00 23.40           O  \\nCONECT 1815 1952                                                                \\nCONECT 1952 1815                                                                \\nCONECT 1964 1965                                                                \\nCONECT 1965 1964 1966                                                           \\nCONECT 1966 1965                                                                \\nCONECT 1967 1968                                                                \\nCONECT 1968 1967 1969                                                           \\nCONECT 1969 1968                                                                \\nCONECT 1970 1971                                                                \\nCONECT 1971 1970 1972                                                           \\nCONECT 1972 1971                                                                \\nCONECT 1973 1974                                                                \\nCONECT 1974 1973 1975                                                           \\nCONECT 1975 1974                                                                \\nMASTER      285    0    4   12    9    0    7    6 2289    1   14   20          \\nEND                                                                             \\n\",\"pdb\");\n\tviewer_17290926055486805.setStyle({\"chain\": \"A\"},{\"cartoon\": {\"color\": \"#33ff33\"}});\n\tviewer_17290926055486805.zoomTo();\nviewer_17290926055486805.render();\n});\n</script>",
            "text/html": [
              "<div id=\"3dmolviewer_17290926055486805\"  style=\"position: relative; width: 640px; height: 480px;\">\n",
              "        <p id=\"3dmolwarning_17290926055486805\" style=\"background-color:#ffcccc;color:black\">3Dmol.js failed to load for some reason.  Please check your browser console for error messages.<br></p>\n",
              "        </div>\n",
              "<script>\n",
              "\n",
              "var loadScriptAsync = function(uri){\n",
              "  return new Promise((resolve, reject) => {\n",
              "    //this is to ignore the existence of requirejs amd\n",
              "    var savedexports, savedmodule;\n",
              "    if (typeof exports !== 'undefined') savedexports = exports;\n",
              "    else exports = {}\n",
              "    if (typeof module !== 'undefined') savedmodule = module;\n",
              "    else module = {}\n",
              "\n",
              "    var tag = document.createElement('script');\n",
              "    tag.src = uri;\n",
              "    tag.async = true;\n",
              "    tag.onload = () => {\n",
              "        exports = savedexports;\n",
              "        module = savedmodule;\n",
              "        resolve();\n",
              "    };\n",
              "  var firstScriptTag = document.getElementsByTagName('script')[0];\n",
              "  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);\n",
              "});\n",
              "};\n",
              "\n",
              "if(typeof $3Dmolpromise === 'undefined') {\n",
              "$3Dmolpromise = null;\n",
              "  $3Dmolpromise = loadScriptAsync('https://3dmol.org/build/3Dmol.js');\n",
              "}\n",
              "\n",
              "var viewer_17290926055486805 = null;\n",
              "var warn = document.getElementById(\"3dmolwarning_17290926055486805\");\n",
              "if(warn) {\n",
              "    warn.parentNode.removeChild(warn);\n",
              "}\n",
              "$3Dmolpromise.then(function() {\n",
              "viewer_17290926055486805 = $3Dmol.createViewer(document.getElementById(\"3dmolviewer_17290926055486805\"),{backgroundColor:\"white\"});\n",
              "viewer_17290926055486805.zoomTo();\n",
              "\tviewer_17290926055486805.addModel(\"HEADER    HYDROLASE                               23-MAR-12   4EB0              \\nTITLE     CRYSTAL STRUCTURE OF LEAF-BRANCH COMPOST BACTERIAL CUTINASE HOMOLOG   \\nCOMPND    MOL_ID: 1;                                                            \\nCOMPND   2 MOLECULE: LCC;                                                       \\nCOMPND   3 CHAIN: A;                                                            \\nCOMPND   4 FRAGMENT: UNP RESIDUES 36-293;                                       \\nCOMPND   5 EC: 3.1.1.74;                                                        \\nCOMPND   6 ENGINEERED: YES                                                      \\nSOURCE    MOL_ID: 1;                                                            \\nSOURCE   2 ORGANISM_SCIENTIFIC: UNCULTURED BACTERIUM;                           \\nSOURCE   3 ORGANISM_TAXID: 77133;                                               \\nSOURCE   4 EXPRESSION_SYSTEM: ESCHERICHIA COLI;                                 \\nSOURCE   5 EXPRESSION_SYSTEM_TAXID: 562;                                        \\nSOURCE   6 EXPRESSION_SYSTEM_STRAIN: BL21-CODONPLUS(DE3)-RP;                    \\nSOURCE   7 EXPRESSION_SYSTEM_VECTOR_TYPE: PLASMID;                              \\nSOURCE   8 EXPRESSION_SYSTEM_PLASMID: PET25B                                    \\nKEYWDS    HYDROLASE, SERINE ESTERASE, CUTINASE HOMOLOG, PET DEGRADATION,        \\nKEYWDS   2 METAGENOME                                                           \\nEXPDTA    X-RAY DIFFRACTION                                                     \\nAUTHOR    S.SULAIMAN,D.J.YOU,K.EIKO,Y.KOGA,S.KANAYA                             \\nREVDAT   3   08-NOV-23 4EB0    1       REMARK                                   \\nREVDAT   2   15-NOV-17 4EB0    1       REMARK                                   \\nREVDAT   1   27-MAR-13 4EB0    0                                                \\nJRNL        AUTH   S.SULAIMAN,D.J.YOU,K.EIKO,Y.KOGA,S.KANAYA                    \\nJRNL        TITL   CRYSTAL STRUCTURE OF LEAF-BRANCH COMPOST BACTERIAL CUTINASE  \\nJRNL        TITL 2 HOMOLOG                                                      \\nJRNL        REF    TO BE PUBLISHED                                              \\nJRNL        REFN                                                                \\nREMARK   2                                                                      \\nREMARK   2 RESOLUTION.    1.50 ANGSTROMS.                                       \\nREMARK   3                                                                      \\nREMARK   3 REFINEMENT.                                                          \\nREMARK   3   PROGRAM     : REFMAC REFMAC_5.5.0109                               \\nREMARK   3   AUTHORS     : MURSHUDOV,SKUBAK,LEBEDEV,PANNU,STEINER,              \\nREMARK   3               : NICHOLLS,WINN,LONG,VAGIN                             \\nREMARK   3                                                                      \\nREMARK   3    REFINEMENT TARGET : MAXIMUM LIKELIHOOD                            \\nREMARK   3                                                                      \\nREMARK   3  DATA USED IN REFINEMENT.                                            \\nREMARK   3   RESOLUTION RANGE HIGH (ANGSTROMS) : 1.50                           \\nREMARK   3   RESOLUTION RANGE LOW  (ANGSTROMS) : 50.00                          \\nREMARK   3   DATA CUTOFF            (SIGMA(F)) : NULL                           \\nREMARK   3   COMPLETENESS FOR RANGE        (%) : 99.6                           \\nREMARK   3   NUMBER OF REFLECTIONS             : 32818                          \\nREMARK   3                                                                      \\nREMARK   3  FIT TO DATA USED IN REFINEMENT.                                     \\nREMARK   3   CROSS-VALIDATION METHOD          : THROUGHOUT                      \\nREMARK   3   FREE R VALUE TEST SET SELECTION  : RANDOM                          \\nREMARK   3   R VALUE     (WORKING + TEST SET) : 0.158                           \\nREMARK   3   R VALUE            (WORKING SET) : 0.157                           \\nREMARK   3   FREE R VALUE                     : 0.186                           \\nREMARK   3   FREE R VALUE TEST SET SIZE   (%) : 5.000                           \\nREMARK   3   FREE R VALUE TEST SET COUNT      : 1741                            \\nREMARK   3                                                                      \\nREMARK   3  FIT IN THE HIGHEST RESOLUTION BIN.                                  \\nREMARK   3   TOTAL NUMBER OF BINS USED           : 20                           \\nREMARK   3   BIN RESOLUTION RANGE HIGH       (A) : 1.50                         \\nREMARK   3   BIN RESOLUTION RANGE LOW        (A) : 1.54                         \\nREMARK   3   REFLECTION IN BIN     (WORKING SET) : 2374                         \\nREMARK   3   BIN COMPLETENESS (WORKING+TEST) (%) : 99.64                        \\nREMARK   3   BIN R VALUE           (WORKING SET) : 0.1880                       \\nREMARK   3   BIN FREE R VALUE SET COUNT          : 127                          \\nREMARK   3   BIN FREE R VALUE                    : 0.2500                       \\nREMARK   3                                                                      \\nREMARK   3  NUMBER OF NON-HYDROGEN ATOMS USED IN REFINEMENT.                    \\nREMARK   3   PROTEIN ATOMS            : 1962                                    \\nREMARK   3   NUCLEIC ACID ATOMS       : 0                                       \\nREMARK   3   HETEROGEN ATOMS          : 12                                      \\nREMARK   3   SOLVENT ATOMS            : 315                                     \\nREMARK   3                                                                      \\nREMARK   3  B VALUES.                                                           \\nREMARK   3   FROM WILSON PLOT           (A**2) : NULL                           \\nREMARK   3   MEAN B VALUE      (OVERALL, A**2) : 17.17                          \\nREMARK   3   OVERALL ANISOTROPIC B VALUE.                                       \\nREMARK   3    B11 (A**2) : -0.67000                                             \\nREMARK   3    B22 (A**2) : -0.04000                                             \\nREMARK   3    B33 (A**2) : 0.70000                                              \\nREMARK   3    B12 (A**2) : 0.00000                                              \\nREMARK   3    B13 (A**2) : 0.00000                                              \\nREMARK   3    B23 (A**2) : 0.00000                                              \\nREMARK   3                                                                      \\nREMARK   3  ESTIMATED OVERALL COORDINATE ERROR.                                 \\nREMARK   3   ESU BASED ON R VALUE                            (A): 0.074         \\nREMARK   3   ESU BASED ON FREE R VALUE                       (A): 0.075         \\nREMARK   3   ESU BASED ON MAXIMUM LIKELIHOOD                 (A): 0.043         \\nREMARK   3   ESU FOR B VALUES BASED ON MAXIMUM LIKELIHOOD (A**2): 1.126         \\nREMARK   3                                                                      \\nREMARK   3 CORRELATION COEFFICIENTS.                                            \\nREMARK   3   CORRELATION COEFFICIENT FO-FC      : 0.971                         \\nREMARK   3   CORRELATION COEFFICIENT FO-FC FREE : 0.959                         \\nREMARK   3                                                                      \\nREMARK   3  RMS DEVIATIONS FROM IDEAL VALUES        COUNT    RMS    WEIGHT      \\nREMARK   3   BOND LENGTHS REFINED ATOMS        (A):  2023 ; 0.007 ; 0.022       \\nREMARK   3   BOND LENGTHS OTHERS               (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   BOND ANGLES REFINED ATOMS   (DEGREES):  2764 ; 1.000 ; 1.941       \\nREMARK   3   BOND ANGLES OTHERS          (DEGREES):  NULL ;  NULL ;  NULL       \\nREMARK   3   TORSION ANGLES, PERIOD 1    (DEGREES):   257 ; 7.429 ; 5.000       \\nREMARK   3   TORSION ANGLES, PERIOD 2    (DEGREES):    87 ;32.304 ;22.759       \\nREMARK   3   TORSION ANGLES, PERIOD 3    (DEGREES):   283 ;10.840 ;15.000       \\nREMARK   3   TORSION ANGLES, PERIOD 4    (DEGREES):    16 ;12.062 ;15.000       \\nREMARK   3   CHIRAL-CENTER RESTRAINTS       (A**3):   310 ; 0.084 ; 0.200       \\nREMARK   3   GENERAL PLANES REFINED ATOMS      (A):  1578 ; 0.020 ; 0.021       \\nREMARK   3   GENERAL PLANES OTHERS             (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   NON-BONDED CONTACTS REFINED ATOMS (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   NON-BONDED CONTACTS OTHERS        (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   NON-BONDED TORSION REFINED ATOMS  (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   NON-BONDED TORSION OTHERS         (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   H-BOND (X...Y) REFINED ATOMS      (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   H-BOND (X...Y) OTHERS             (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   POTENTIAL METAL-ION REFINED ATOMS (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   POTENTIAL METAL-ION OTHERS        (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   SYMMETRY VDW REFINED ATOMS        (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   SYMMETRY VDW OTHERS               (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   SYMMETRY H-BOND REFINED ATOMS     (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   SYMMETRY H-BOND OTHERS            (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   SYMMETRY METAL-ION REFINED ATOMS  (A):  NULL ;  NULL ;  NULL       \\nREMARK   3   SYMMETRY METAL-ION OTHERS         (A):  NULL ;  NULL ;  NULL       \\nREMARK   3                                                                      \\nREMARK   3  ISOTROPIC THERMAL FACTOR RESTRAINTS.     COUNT   RMS    WEIGHT      \\nREMARK   3   MAIN-CHAIN BOND REFINED ATOMS  (A**2):  1298 ; 1.466 ; 1.500       \\nREMARK   3   MAIN-CHAIN BOND OTHER ATOMS    (A**2):  NULL ;  NULL ;  NULL       \\nREMARK   3   MAIN-CHAIN ANGLE REFINED ATOMS (A**2):  2088 ; 2.175 ; 2.000       \\nREMARK   3   MAIN-CHAIN ANGLE OTHER ATOMS   (A**2):  NULL ;  NULL ;  NULL       \\nREMARK   3   SIDE-CHAIN BOND REFINED ATOMS  (A**2):   725 ; 3.398 ; 3.000       \\nREMARK   3   SIDE-CHAIN BOND OTHER ATOMS    (A**2):  NULL ;  NULL ;  NULL       \\nREMARK   3   SIDE-CHAIN ANGLE REFINED ATOMS (A**2):   676 ; 4.458 ; 4.500       \\nREMARK   3   SIDE-CHAIN ANGLE OTHER ATOMS   (A**2):  NULL ;  NULL ;  NULL       \\nREMARK   3   LONG RANGE B REFINED ATOMS     (A**2):  NULL ;  NULL ;  NULL       \\nREMARK   3   LONG RANGE B OTHER ATOMS       (A**2):  NULL ;  NULL ;  NULL       \\nREMARK   3                                                                      \\nREMARK   3 ANISOTROPIC THERMAL FACTOR RESTRAINTS.    COUNT   RMS   WEIGHT       \\nREMARK   3   RIGID-BOND RESTRAINTS          (A**2):  NULL ;  NULL ;  NULL       \\nREMARK   3   SPHERICITY; FREE ATOMS         (A**2):  NULL ;  NULL ;  NULL       \\nREMARK   3   SPHERICITY; BONDED ATOMS       (A**2):  NULL ;  NULL ;  NULL       \\nREMARK   3                                                                      \\nREMARK   3  NCS RESTRAINTS STATISTICS                                           \\nREMARK   3   NUMBER OF DIFFERENT NCS GROUPS : NULL                              \\nREMARK   3                                                                      \\nREMARK   3  TLS DETAILS                                                         \\nREMARK   3   NUMBER OF TLS GROUPS  : NULL                                       \\nREMARK   3                                                                      \\nREMARK   3  BULK SOLVENT MODELLING.                                             \\nREMARK   3   METHOD USED : MASK                                                 \\nREMARK   3   PARAMETERS FOR MASK CALCULATION                                    \\nREMARK   3   VDW PROBE RADIUS   : 1.40                                          \\nREMARK   3   ION PROBE RADIUS   : 0.80                                          \\nREMARK   3   SHRINKAGE RADIUS   : 0.80                                          \\nREMARK   3                                                                      \\nREMARK   3  OTHER REFINEMENT REMARKS: HYDROGENS HAVE BEEN ADDED IN THE RIDING   \\nREMARK   3  POSITIONS                                                           \\nREMARK   4                                                                      \\nREMARK   4 4EB0 COMPLIES WITH FORMAT V. 3.30, 13-JUL-11                         \\nREMARK 100                                                                      \\nREMARK 100 THIS ENTRY HAS BEEN PROCESSED BY PDBJ ON 30-MAR-12.                  \\nREMARK 100 THE DEPOSITION ID IS D_1000071392.                                   \\nREMARK 200                                                                      \\nREMARK 200 EXPERIMENTAL DETAILS                                                 \\nREMARK 200  EXPERIMENT TYPE                : X-RAY DIFFRACTION                  \\nREMARK 200  DATE OF DATA COLLECTION        : 10-OCT-11                          \\nREMARK 200  TEMPERATURE           (KELVIN) : 100                                \\nREMARK 200  PH                             : 7.0                                \\nREMARK 200  NUMBER OF CRYSTALS USED        : 1                                  \\nREMARK 200                                                                      \\nREMARK 200  SYNCHROTRON              (Y/N) : Y                                  \\nREMARK 200  RADIATION SOURCE               : SPRING-8                           \\nREMARK 200  BEAMLINE                       : BL44XU                             \\nREMARK 200  X-RAY GENERATOR MODEL          : NULL                               \\nREMARK 200  MONOCHROMATIC OR LAUE    (M/L) : M                                  \\nREMARK 200  WAVELENGTH OR RANGE        (A) : 0.9                                \\nREMARK 200  MONOCHROMATOR                  : HORIZONTAL FOCUSING MIRROR         \\nREMARK 200  OPTICS                         : NULL                               \\nREMARK 200                                                                      \\nREMARK 200  DETECTOR TYPE                  : CCD                                \\nREMARK 200  DETECTOR MANUFACTURER          : BRUKER SMART 6500                  \\nREMARK 200  INTENSITY-INTEGRATION SOFTWARE : DENZO                              \\nREMARK 200  DATA SCALING SOFTWARE          : SCALEPACK                          \\nREMARK 200                                                                      \\nREMARK 200  NUMBER OF UNIQUE REFLECTIONS   : 35110                              \\nREMARK 200  RESOLUTION RANGE HIGH      (A) : 1.500                              \\nREMARK 200  RESOLUTION RANGE LOW       (A) : 50.000                             \\nREMARK 200  REJECTION CRITERIA  (SIGMA(I)) : -3.000                             \\nREMARK 200                                                                      \\nREMARK 200 OVERALL.                                                             \\nREMARK 200  COMPLETENESS FOR RANGE     (%) : 99.9                               \\nREMARK 200  DATA REDUNDANCY                : 13.70                              \\nREMARK 200  R MERGE                    (I) : 0.10600                            \\nREMARK 200  R SYM                      (I) : 0.10600                            \\nREMARK 200  <I/SIGMA(I)> FOR THE DATA SET  : 31.6570                            \\nREMARK 200                                                                      \\nREMARK 200 IN THE HIGHEST RESOLUTION SHELL.                                     \\nREMARK 200  HIGHEST RESOLUTION SHELL, RANGE HIGH (A) : 1.50                     \\nREMARK 200  HIGHEST RESOLUTION SHELL, RANGE LOW  (A) : 1.53                     \\nREMARK 200  COMPLETENESS FOR SHELL     (%) : 100.0                              \\nREMARK 200  DATA REDUNDANCY IN SHELL       : 12.70                              \\nREMARK 200  R MERGE FOR SHELL          (I) : 0.41900                            \\nREMARK 200  R SYM FOR SHELL            (I) : 0.41900                            \\nREMARK 200  <I/SIGMA(I)> FOR SHELL         : 0.419                              \\nREMARK 200                                                                      \\nREMARK 200 DIFFRACTION PROTOCOL: SINGLE WAVELENGTH                              \\nREMARK 200 METHOD USED TO DETERMINE THE STRUCTURE: MOLECULAR REPLACEMENT        \\nREMARK 200 SOFTWARE USED: MOLREP, DM 6.1                                        \\nREMARK 200 STARTING MODEL: 1JFR                                                 \\nREMARK 200                                                                      \\nREMARK 200 REMARK: NULL                                                         \\nREMARK 280                                                                      \\nREMARK 280 CRYSTAL                                                              \\nREMARK 280 SOLVENT CONTENT, VS   (%): 35.33                                     \\nREMARK 280 MATTHEWS COEFFICIENT, VM (ANGSTROMS**3/DA): 1.90                     \\nREMARK 280                                                                      \\nREMARK 280 CRYSTALLIZATION CONDITIONS: 20% (W/V) POLYETHYLENE GLYCOL (PEG)      \\nREMARK 280  3350, 200MM SODIUM THIOCYANATE, PH 7.0, VAPOR DIFFUSION, SITTING    \\nREMARK 280  DROP, TEMPERATURE 293K                                              \\nREMARK 290                                                                      \\nREMARK 290 CRYSTALLOGRAPHIC SYMMETRY                                            \\nREMARK 290 SYMMETRY OPERATORS FOR SPACE GROUP: P 21 21 21                       \\nREMARK 290                                                                      \\nREMARK 290      SYMOP   SYMMETRY                                                \\nREMARK 290     NNNMMM   OPERATOR                                                \\nREMARK 290       1555   X,Y,Z                                                   \\nREMARK 290       2555   -X+1/2,-Y,Z+1/2                                         \\nREMARK 290       3555   -X,Y+1/2,-Z+1/2                                         \\nREMARK 290       4555   X+1/2,-Y+1/2,-Z                                         \\nREMARK 290                                                                      \\nREMARK 290     WHERE NNN -> OPERATOR NUMBER                                     \\nREMARK 290           MMM -> TRANSLATION VECTOR                                  \\nREMARK 290                                                                      \\nREMARK 290 CRYSTALLOGRAPHIC SYMMETRY TRANSFORMATIONS                            \\nREMARK 290 THE FOLLOWING TRANSFORMATIONS OPERATE ON THE ATOM/HETATM             \\nREMARK 290 RECORDS IN THIS ENTRY TO PRODUCE CRYSTALLOGRAPHICALLY                \\nREMARK 290 RELATED MOLECULES.                                                   \\nREMARK 290   SMTRY1   1  1.000000  0.000000  0.000000        0.00000            \\nREMARK 290   SMTRY2   1  0.000000  1.000000  0.000000        0.00000            \\nREMARK 290   SMTRY3   1  0.000000  0.000000  1.000000        0.00000            \\nREMARK 290   SMTRY1   2 -1.000000  0.000000  0.000000       20.45550            \\nREMARK 290   SMTRY2   2  0.000000 -1.000000  0.000000        0.00000            \\nREMARK 290   SMTRY3   2  0.000000  0.000000  1.000000       36.36300            \\nREMARK 290   SMTRY1   3 -1.000000  0.000000  0.000000        0.00000            \\nREMARK 290   SMTRY2   3  0.000000  1.000000  0.000000       35.54200            \\nREMARK 290   SMTRY3   3  0.000000  0.000000 -1.000000       36.36300            \\nREMARK 290   SMTRY1   4  1.000000  0.000000  0.000000       20.45550            \\nREMARK 290   SMTRY2   4  0.000000 -1.000000  0.000000       35.54200            \\nREMARK 290   SMTRY3   4  0.000000  0.000000 -1.000000        0.00000            \\nREMARK 290                                                                      \\nREMARK 290 REMARK: NULL                                                         \\nREMARK 300                                                                      \\nREMARK 300 BIOMOLECULE: 1                                                       \\nREMARK 300 SEE REMARK 350 FOR THE AUTHOR PROVIDED AND/OR PROGRAM                \\nREMARK 300 GENERATED ASSEMBLY INFORMATION FOR THE STRUCTURE IN                  \\nREMARK 300 THIS ENTRY. THE REMARK MAY ALSO PROVIDE INFORMATION ON               \\nREMARK 300 BURIED SURFACE AREA.                                                 \\nREMARK 350                                                                      \\nREMARK 350 COORDINATES FOR A COMPLETE MULTIMER REPRESENTING THE KNOWN           \\nREMARK 350 BIOLOGICALLY SIGNIFICANT OLIGOMERIZATION STATE OF THE                \\nREMARK 350 MOLECULE CAN BE GENERATED BY APPLYING BIOMT TRANSFORMATIONS          \\nREMARK 350 GIVEN BELOW.  BOTH NON-CRYSTALLOGRAPHIC AND                          \\nREMARK 350 CRYSTALLOGRAPHIC OPERATIONS ARE GIVEN.                               \\nREMARK 350                                                                      \\nREMARK 350 BIOMOLECULE: 1                                                       \\nREMARK 350 AUTHOR DETERMINED BIOLOGICAL UNIT: MONOMERIC                         \\nREMARK 350 SOFTWARE DETERMINED QUATERNARY STRUCTURE: MONOMERIC                  \\nREMARK 350 SOFTWARE USED: PISA                                                  \\nREMARK 350 APPLY THE FOLLOWING TO CHAINS: A                                     \\nREMARK 350   BIOMT1   1  1.000000  0.000000  0.000000        0.00000            \\nREMARK 350   BIOMT2   1  0.000000  1.000000  0.000000        0.00000            \\nREMARK 350   BIOMT3   1  0.000000  0.000000  1.000000        0.00000            \\nREMARK 500                                                                      \\nREMARK 500 GEOMETRY AND STEREOCHEMISTRY                                         \\nREMARK 500 SUBTOPIC: TORSION ANGLES                                             \\nREMARK 500                                                                      \\nREMARK 500 TORSION ANGLES OUTSIDE THE EXPECTED RAMACHANDRAN REGIONS:            \\nREMARK 500 (M=MODEL NUMBER; RES=RESIDUE NAME; C=CHAIN IDENTIFIER;               \\nREMARK 500 SSEQ=SEQUENCE NUMBER; I=INSERTION CODE).                             \\nREMARK 500                                                                      \\nREMARK 500 STANDARD TABLE:                                                      \\nREMARK 500 FORMAT:(10X,I3,1X,A3,1X,A1,I4,A1,4X,F7.2,3X,F7.2)                    \\nREMARK 500                                                                      \\nREMARK 500 EXPECTED VALUES: GJ KLEYWEGT AND TA JONES (1996). PHI/PSI-           \\nREMARK 500 CHOLOGY: RAMACHANDRAN REVISITED. STRUCTURE 4, 1395 - 1400            \\nREMARK 500                                                                      \\nREMARK 500  M RES CSSEQI        PSI       PHI                                   \\nREMARK 500    SER A 165     -127.89     60.19                                   \\nREMARK 500    HIS A 218      -83.73   -126.70                                   \\nREMARK 500                                                                      \\nREMARK 500 REMARK: NULL                                                         \\nREMARK 500                                                                      \\nREMARK 500 GEOMETRY AND STEREOCHEMISTRY                                         \\nREMARK 500 SUBTOPIC: NON-CIS, NON-TRANS                                         \\nREMARK 500                                                                      \\nREMARK 500 THE FOLLOWING PEPTIDE BONDS DEVIATE SIGNIFICANTLY FROM BOTH          \\nREMARK 500 CIS AND TRANS CONFORMATION.  CIS BONDS, IF ANY, ARE LISTED           \\nREMARK 500 ON CISPEP RECORDS.  TRANS IS DEFINED AS 180 +/- 30 AND               \\nREMARK 500 CIS IS DEFINED AS 0 +/- 30 DEGREES.                                  \\nREMARK 500                                 MODEL     OMEGA                      \\nREMARK 500 ILE A  119     ASN A  120                  149.22                    \\nREMARK 500                                                                      \\nREMARK 500 REMARK: NULL                                                         \\nREMARK 800                                                                      \\nREMARK 800 SITE                                                                 \\nREMARK 800 SITE_IDENTIFIER: AC1                                                 \\nREMARK 800 EVIDENCE_CODE: SOFTWARE                                              \\nREMARK 800 SITE_DESCRIPTION: BINDING SITE FOR RESIDUE SCN A 301                 \\nREMARK 800                                                                      \\nREMARK 800 SITE_IDENTIFIER: AC2                                                 \\nREMARK 800 EVIDENCE_CODE: SOFTWARE                                              \\nREMARK 800 SITE_DESCRIPTION: BINDING SITE FOR RESIDUE SCN A 302                 \\nREMARK 800                                                                      \\nREMARK 800 SITE_IDENTIFIER: AC3                                                 \\nREMARK 800 EVIDENCE_CODE: SOFTWARE                                              \\nREMARK 800 SITE_DESCRIPTION: BINDING SITE FOR RESIDUE SCN A 303                 \\nREMARK 800                                                                      \\nREMARK 800 SITE_IDENTIFIER: AC4                                                 \\nREMARK 800 EVIDENCE_CODE: SOFTWARE                                              \\nREMARK 800 SITE_DESCRIPTION: BINDING SITE FOR RESIDUE SCN A 304                 \\nDBREF  4EB0 A   36   293  UNP    G9BY57   G9BY57_9BACT    36    293             \\nSEQRES   1 A  258  SER ASN PRO TYR GLN ARG GLY PRO ASN PRO THR ARG SER          \\nSEQRES   2 A  258  ALA LEU THR ALA ASP GLY PRO PHE SER VAL ALA THR TYR          \\nSEQRES   3 A  258  THR VAL SER ARG LEU SER VAL SER GLY PHE GLY GLY GLY          \\nSEQRES   4 A  258  VAL ILE TYR TYR PRO THR GLY THR SER LEU THR PHE GLY          \\nSEQRES   5 A  258  GLY ILE ALA MET SER PRO GLY TYR THR ALA ASP ALA SER          \\nSEQRES   6 A  258  SER LEU ALA TRP LEU GLY ARG ARG LEU ALA SER HIS GLY          \\nSEQRES   7 A  258  PHE VAL VAL LEU VAL ILE ASN THR ASN SER ARG PHE ASP          \\nSEQRES   8 A  258  TYR PRO ASP SER ARG ALA SER GLN LEU SER ALA ALA LEU          \\nSEQRES   9 A  258  ASN TYR LEU ARG THR SER SER PRO SER ALA VAL ARG ALA          \\nSEQRES  10 A  258  ARG LEU ASP ALA ASN ARG LEU ALA VAL ALA GLY HIS SER          \\nSEQRES  11 A  258  MET GLY GLY GLY GLY THR LEU ARG ILE ALA GLU GLN ASN          \\nSEQRES  12 A  258  PRO SER LEU LYS ALA ALA VAL PRO LEU THR PRO TRP HIS          \\nSEQRES  13 A  258  THR ASP LYS THR PHE ASN THR SER VAL PRO VAL LEU ILE          \\nSEQRES  14 A  258  VAL GLY ALA GLU ALA ASP THR VAL ALA PRO VAL SER GLN          \\nSEQRES  15 A  258  HIS ALA ILE PRO PHE TYR GLN ASN LEU PRO SER THR THR          \\nSEQRES  16 A  258  PRO LYS VAL TYR VAL GLU LEU ASP ASN ALA SER HIS PHE          \\nSEQRES  17 A  258  ALA PRO ASN SER ASN ASN ALA ALA ILE SER VAL TYR THR          \\nSEQRES  18 A  258  ILE SER TRP MET LYS LEU TRP VAL ASP ASN ASP THR ARG          \\nSEQRES  19 A  258  TYR ARG GLN PHE LEU CYS ASN VAL ASN ASP PRO ALA LEU          \\nSEQRES  20 A  258  SER ASP PHE ARG THR ASN ASN ARG HIS CYS GLN                  \\nHET    SCN  A 301       3                                                       \\nHET    SCN  A 302       3                                                       \\nHET    SCN  A 303       3                                                       \\nHET    SCN  A 304       3                                                       \\nHETNAM     SCN THIOCYANATE ION                                                  \\nFORMUL   2  SCN    4(C N S 1-)                                                  \\nFORMUL   6  HOH   *315(H2 O)                                                    \\nHELIX    1   1 ARG A   47  ALA A   52  5                                   6    \\nHELIX    2   2 SER A   64  VAL A   68  5                                   5    \\nHELIX    3   3 ASP A   98  SER A  101  5                                   4    \\nHELIX    4   4 LEU A  102  HIS A  112  1                                  11    \\nHELIX    5   5 TYR A  127  SER A  145  1                                  19    \\nHELIX    6   6 PRO A  147  ALA A  152  1                                   6    \\nHELIX    7   7 SER A  165  ASN A  178  1                                  14    \\nHELIX    8   8 HIS A  218  LEU A  226  1                                   9    \\nHELIX    9   9 PHE A  243  SER A  247  5                                   5    \\nHELIX   10  10 ASN A  249  ASP A  265  1                                  17    \\nHELIX   11  11 ASP A  267  LEU A  274  5                                   8    \\nHELIX   12  12 ASN A  289  GLN A  293  5                                   5    \\nSHEET    1   A 6 VAL A  58  VAL A  63  0                                        \\nSHEET    2   A 6 GLY A  74  PRO A  79 -1  O  ILE A  76   N  TYR A  61           \\nSHEET    3   A 6 VAL A 115  ILE A 119 -1  O  VAL A 116   N  TYR A  77           \\nSHEET    4   A 6 PHE A  86  SER A  92  1  N  ILE A  89   O  VAL A 115           \\nSHEET    5   A 6 LEU A 154  HIS A 164  1  O  ASP A 155   N  PHE A  86           \\nSHEET    6   A 6 ALA A 184  LEU A 187  1  O  LEU A 187   N  GLY A 163           \\nSHEET    1   B 3 VAL A 202  ALA A 207  0                                        \\nSHEET    2   B 3 LYS A 232  LEU A 237  1  O  VAL A 235   N  GLY A 206           \\nSHEET    3   B 3 LEU A 282  THR A 287 -1  O  ARG A 286   N  TYR A 234           \\nSSBOND   1 CYS A  275    CYS A  292                          1555   1555  2.05  \\nSITE     1 AC1  5 GLY A  42  TYR A 223  ARG A 286  HOH A 452                    \\nSITE     2 AC1  5 HOH A 527                                                     \\nSITE     1 AC2  5 THR A  96  ARG A 124  PRO A 231  ARG A 290                    \\nSITE     2 AC2  5 HOH A 469                                                     \\nSITE     1 AC3  4 SER A  48  SER A  69  THR A 144  SER A 145                    \\nSITE     1 AC4  8 ARG A  47  ASN A 276  VAL A 277  ASN A 278                    \\nSITE     2 AC4  8 ASP A 279  HOH A 458  HOH A 493  HOH A 544                    \\nCRYST1   40.911   71.084   72.726  90.00  90.00  90.00 P 21 21 21    4          \\nORIGX1      1.000000  0.000000  0.000000        0.00000                         \\nORIGX2      0.000000  1.000000  0.000000        0.00000                         \\nORIGX3      0.000000  0.000000  1.000000        0.00000                         \\nSCALE1      0.024443  0.000000  0.000000        0.00000                         \\nSCALE2      0.000000  0.014068  0.000000        0.00000                         \\nSCALE3      0.000000  0.000000  0.013750        0.00000                         \\nATOM      1  N   SER A  36      10.488 -16.119   4.405  1.00 32.20           N  \\nATOM      2  CA  SER A  36       9.773 -16.823   5.499  1.00 24.29           C  \\nATOM      3  C   SER A  36       8.613 -15.977   6.014  1.00 19.49           C  \\nATOM      4  O   SER A  36       7.724 -16.476   6.701  1.00 19.40           O  \\nATOM      5  CB  SER A  36      10.740 -17.189   6.637  1.00 24.44           C  \\nATOM      6  OG  SER A  36      11.354 -16.036   7.201  1.00 36.14           O  \\nATOM      7  N   ASN A  37       8.614 -14.697   5.655  1.00 18.01           N  \\nATOM      8  CA  ASN A  37       7.627 -13.754   6.184  1.00 15.55           C  \\nATOM      9  C   ASN A  37       6.744 -13.209   5.064  1.00 13.92           C  \\nATOM     10  O   ASN A  37       7.197 -12.391   4.265  1.00 16.39           O  \\nATOM     11  CB  ASN A  37       8.333 -12.581   6.879  1.00 14.38           C  \\nATOM     12  CG  ASN A  37       7.358 -11.487   7.295  1.00 13.17           C  \\nATOM     13  OD1 ASN A  37       6.154 -11.687   7.248  1.00 13.91           O  \\nATOM     14  ND2 ASN A  37       7.877 -10.393   7.830  1.00 13.94           N  \\nATOM     15  N   PRO A  38       5.481 -13.648   5.004  1.00 14.35           N  \\nATOM     16  CA  PRO A  38       4.616 -13.313   3.884  1.00 14.53           C  \\nATOM     17  C   PRO A  38       4.218 -11.835   3.856  1.00 13.55           C  \\nATOM     18  O   PRO A  38       3.690 -11.361   2.843  1.00 17.67           O  \\nATOM     19  CB  PRO A  38       3.382 -14.196   4.118  1.00 17.64           C  \\nATOM     20  CG  PRO A  38       3.363 -14.425   5.594  1.00 16.09           C  \\nATOM     21  CD  PRO A  38       4.815 -14.544   5.972  1.00 15.23           C  \\nATOM     22  N   TYR A  39       4.476 -11.118   4.942  1.00 14.55           N  \\nATOM     23  CA  TYR A  39       4.073  -9.707   5.002  1.00 15.93           C  \\nATOM     24  C   TYR A  39       5.250  -8.796   4.696  1.00 15.42           C  \\nATOM     25  O   TYR A  39       5.103  -7.576   4.692  1.00 13.44           O  \\nATOM     26  CB  TYR A  39       3.477  -9.346   6.366  1.00 14.24           C  \\nATOM     27  CG  TYR A  39       2.477 -10.339   6.913  1.00 15.58           C  \\nATOM     28  CD1 TYR A  39       1.335 -10.677   6.201  1.00 16.49           C  \\nATOM     29  CD2 TYR A  39       2.742 -11.017   8.099  1.00 16.84           C  \\nATOM     30  CE1 TYR A  39       0.456 -11.664   6.680  1.00 12.92           C  \\nATOM     31  CE2 TYR A  39       1.884 -11.975   8.590  1.00 16.72           C  \\nATOM     32  CZ  TYR A  39       0.771 -12.324   7.863  1.00 14.38           C  \\nATOM     33  OH  TYR A  39      -0.063 -13.296   8.385  1.00 19.31           O  \\nATOM     34  N   GLN A  40       6.429  -9.376   4.513  1.00 12.55           N  \\nATOM     35  CA  GLN A  40       7.629  -8.593   4.219  1.00 14.21           C  \\nATOM     36  C   GLN A  40       7.543  -7.885   2.857  1.00 12.42           C  \\nATOM     37  O   GLN A  40       7.192  -8.506   1.840  1.00 14.35           O  \\nATOM     38  CB  GLN A  40       8.856  -9.493   4.281  1.00 16.80           C  \\nATOM     39  CG  GLN A  40      10.160  -8.751   4.422  1.00 18.32           C  \\nATOM     40  CD  GLN A  40      11.342  -9.693   4.418  1.00 21.13           C  \\nATOM     41  OE1 GLN A  40      11.542 -10.439   5.373  1.00 22.91           O  \\nATOM     42  NE2 GLN A  40      11.980  -9.821   3.266  1.00 18.33           N  \\nATOM     43  N   ARG A  41       7.895  -6.599   2.835  1.00 12.41           N  \\nATOM     44  CA  ARG A  41       7.883  -5.811   1.599  1.00 13.22           C  \\nATOM     45  C   ARG A  41       9.157  -4.998   1.461  1.00 13.07           C  \\nATOM     46  O   ARG A  41       9.717  -4.542   2.460  1.00 12.24           O  \\nATOM     47  CB  ARG A  41       6.697  -4.840   1.571  1.00 14.84           C  \\nATOM     48  CG  ARG A  41       5.334  -5.472   1.854  1.00 13.01           C  \\nATOM     49  CD  ARG A  41       4.777  -6.221   0.640  1.00 12.90           C  \\nATOM     50  NE  ARG A  41       3.476  -6.835   0.931  1.00 14.33           N  \\nATOM     51  CZ  ARG A  41       3.314  -8.082   1.374  1.00 12.18           C  \\nATOM     52  NH1 ARG A  41       4.355  -8.917   1.468  1.00 13.50           N  \\nATOM     53  NH2 ARG A  41       2.111  -8.499   1.774  1.00 13.55           N  \\nATOM     54  N   GLY A  42       9.586  -4.791   0.217  1.00 13.78           N  \\nATOM     55  CA  GLY A  42      10.607  -3.797  -0.088  1.00 13.51           C  \\nATOM     56  C   GLY A  42      12.026  -4.290   0.170  1.00 15.04           C  \\nATOM     57  O   GLY A  42      12.266  -5.464   0.505  1.00 16.09           O  \\nATOM     58  N   PRO A  43      12.997  -3.376   0.049  1.00 13.88           N  \\nATOM     59  CA  PRO A  43      14.410  -3.745   0.078  1.00 14.51           C  \\nATOM     60  C   PRO A  43      14.880  -4.100   1.485  1.00 14.74           C  \\nATOM     61  O   PRO A  43      14.275  -3.693   2.480  1.00 14.03           O  \\nATOM     62  CB  PRO A  43      15.107  -2.458  -0.387  1.00 13.15           C  \\nATOM     63  CG  PRO A  43      14.173  -1.377   0.078  1.00 12.86           C  \\nATOM     64  CD  PRO A  43      12.806  -1.939  -0.201  1.00 13.81           C  \\nATOM     65  N   ASN A  44      16.031  -4.743   1.583  1.00 13.09           N  \\nATOM     66  CA  ASN A  44      16.710  -4.757   2.876  1.00 13.27           C  \\nATOM     67  C   ASN A  44      17.007  -3.308   3.264  1.00 12.73           C  \\nATOM     68  O   ASN A  44      17.529  -2.548   2.447  1.00 12.11           O  \\nATOM     69  CB  ASN A  44      18.012  -5.561   2.790  1.00 13.25           C  \\nATOM     70  CG  ASN A  44      17.779  -7.006   2.362  1.00 15.42           C  \\nATOM     71  OD1 ASN A  44      18.439  -7.510   1.439  1.00 21.55           O  \\nATOM     72  ND2 ASN A  44      16.788  -7.651   2.967  1.00 14.09           N  \\nATOM     73  N   PRO A  45      16.554  -2.877   4.454  1.00 12.79           N  \\nATOM     74  CA  PRO A  45      16.597  -1.446   4.769  1.00 12.39           C  \\nATOM     75  C   PRO A  45      17.997  -0.968   5.136  1.00 12.50           C  \\nATOM     76  O   PRO A  45      18.773  -1.695   5.756  1.00 14.14           O  \\nATOM     77  CB  PRO A  45      15.657  -1.322   5.973  1.00 12.93           C  \\nATOM     78  CG  PRO A  45      15.690  -2.663   6.609  1.00 12.05           C  \\nATOM     79  CD  PRO A  45      15.827  -3.646   5.482  1.00 10.79           C  \\nATOM     80  N   THR A  46      18.308   0.260   4.745  1.00 12.81           N  \\nATOM     81  CA  THR A  46      19.444   0.965   5.310  1.00 12.44           C  \\nATOM     82  C   THR A  46      18.929   2.209   6.004  1.00 12.50           C  \\nATOM     83  O   THR A  46      17.871   2.741   5.638  1.00 13.67           O  \\nATOM     84  CB  THR A  46      20.439   1.402   4.227  1.00 13.35           C  \\nATOM     85  OG1 THR A  46      19.835   2.417   3.414  1.00 16.38           O  \\nATOM     86  CG2 THR A  46      20.872   0.214   3.358  1.00 14.54           C  \\nATOM     87  N   ARG A  47      19.667   2.687   7.000  1.00 13.43           N  \\nATOM     88  CA  ARG A  47      19.192   3.857   7.722  1.00 13.46           C  \\nATOM     89  C   ARG A  47      18.993   5.045   6.776  1.00 15.48           C  \\nATOM     90  O   ARG A  47      18.004   5.780   6.872  1.00 14.79           O  \\nATOM     91  CB  ARG A  47      20.164   4.229   8.834  1.00 14.48           C  \\nATOM     92  CG  ARG A  47      19.611   5.359   9.684  1.00 21.77           C  \\nATOM     93  CD  ARG A  47      20.584   5.729  10.769  1.00 18.47           C  \\nATOM     94  NE  ARG A  47      20.582   4.689  11.786  1.00 30.70           N  \\nATOM     95  CZ  ARG A  47      19.710   4.641  12.788  1.00 27.61           C  \\nATOM     96  NH1 ARG A  47      18.978   5.708  13.074  1.00 26.21           N  \\nATOM     97  NH2 ARG A  47      19.644   3.560  13.550  1.00 31.22           N  \\nATOM     98  N   SER A  48      19.946   5.243   5.870  1.00 13.72           N  \\nATOM     99  CA  SER A  48      19.879   6.321   4.897  1.00 15.69           C  \\nATOM    100  C   SER A  48      18.576   6.262   4.112  1.00 15.30           C  \\nATOM    101  O   SER A  48      17.902   7.273   3.934  1.00 15.80           O  \\nATOM    102  CB  SER A  48      21.086   6.248   3.952  1.00 19.02           C  \\nATOM    103  OG  SER A  48      21.135   7.378   3.095  1.00 27.12           O  \\nATOM    104  N   ALA A  49      18.220   5.075   3.635  1.00 13.74           N  \\nATOM    105  CA  ALA A  49      17.040   4.959   2.798  1.00 15.05           C  \\nATOM    106  C   ALA A  49      15.777   5.312   3.569  1.00 14.66           C  \\nATOM    107  O   ALA A  49      14.785   5.741   2.974  1.00 13.36           O  \\nATOM    108  CB  ALA A  49      16.944   3.554   2.189  1.00 12.91           C  \\nATOM    109  N   LEU A  50      15.816   5.146   4.890  1.00 13.32           N  \\nATOM    110  CA  LEU A  50      14.622   5.394   5.698  1.00 13.31           C  \\nATOM    111  C   LEU A  50      14.457   6.877   5.978  1.00 15.50           C  \\nATOM    112  O   LEU A  50      13.517   7.270   6.648  1.00 14.81           O  \\nATOM    113  CB  LEU A  50      14.666   4.605   7.013  1.00 13.33           C  \\nATOM    114  CG  LEU A  50      14.768   3.091   6.842  1.00 11.84           C  \\nATOM    115  CD1 LEU A  50      14.633   2.407   8.197  1.00 14.70           C  \\nATOM    116  CD2 LEU A  50      13.768   2.556   5.824  1.00 12.79           C  \\nATOM    117  N   THR A  51      15.334   7.704   5.410  1.00 14.18           N  \\nATOM    118  CA  THR A  51      15.197   9.147   5.577  1.00 15.61           C  \\nATOM    119  C   THR A  51      14.633   9.800   4.329  1.00 15.98           C  \\nATOM    120  O   THR A  51      14.528  11.023   4.263  1.00 17.82           O  \\nATOM    121  CB  THR A  51      16.535   9.837   5.921  1.00 16.11           C  \\nATOM    122  OG1 THR A  51      17.418   9.776   4.795  1.00 15.77           O  \\nATOM    123  CG2 THR A  51      17.190   9.223   7.157  1.00 16.38           C  \\nATOM    124  N   ALA A  52      14.221   8.984   3.367  1.00 15.16           N  \\nATOM    125  CA  ALA A  52      13.675   9.478   2.115  1.00 14.03           C  \\nATOM    126  C   ALA A  52      12.480   8.614   1.731  1.00 13.87           C  \\nATOM    127  O   ALA A  52      12.274   7.550   2.310  1.00 13.54           O  \\nATOM    128  CB  ALA A  52      14.739   9.423   1.019  1.00 17.28           C  \\nATOM    129  N   ASP A  53      11.687   9.066   0.767  1.00 13.86           N  \\nATOM    130  CA  ASP A  53      10.587   8.233   0.267  1.00 12.95           C  \\nATOM    131  C   ASP A  53      11.098   6.900  -0.290  1.00 12.99           C  \\nATOM    132  O   ASP A  53      12.249   6.795  -0.767  1.00 13.92           O  \\nATOM    133  CB  ASP A  53       9.766   8.960  -0.804  1.00 14.43           C  \\nATOM    134  CG  ASP A  53       8.895  10.067  -0.249  1.00 15.74           C  \\nATOM    135  OD1 ASP A  53       8.834  10.283   0.973  1.00 15.56           O  \\nATOM    136  OD2 ASP A  53       8.248  10.735  -1.082  1.00 21.72           O  \\nATOM    137  N   GLY A  54      10.293   5.859  -0.098  1.00 13.26           N  \\nATOM    138  CA  GLY A  54      10.693   4.478  -0.402  1.00 13.17           C  \\nATOM    139  C   GLY A  54      10.397   4.047  -1.829  1.00 14.35           C  \\nATOM    140  O   GLY A  54      10.260   4.883  -2.741  1.00 13.88           O  \\nATOM    141  N   PRO A  55      10.297   2.733  -2.045  1.00 15.11           N  \\nATOM    142  CA  PRO A  55      10.301   2.189  -3.398  1.00 16.43           C  \\nATOM    143  C   PRO A  55       8.932   2.163  -4.061  1.00 13.17           C  \\nATOM    144  O   PRO A  55       8.795   1.656  -5.173  1.00 14.53           O  \\nATOM    145  CB  PRO A  55      10.810   0.765  -3.192  1.00 19.68           C  \\nATOM    146  CG  PRO A  55      10.350   0.404  -1.831  1.00 16.18           C  \\nATOM    147  CD  PRO A  55      10.448   1.684  -1.026  1.00 15.87           C  \\nATOM    148  N   PHE A  56       7.893   2.595  -3.351  1.00 13.43           N  \\nATOM    149  CA  PHE A  56       6.561   2.588  -3.943  1.00 13.16           C  \\nATOM    150  C   PHE A  56       6.160   4.002  -4.303  1.00 13.86           C  \\nATOM    151  O   PHE A  56       6.206   4.896  -3.454  1.00 14.95           O  \\nATOM    152  CB  PHE A  56       5.525   1.969  -2.995  1.00 16.49           C  \\nATOM    153  CG  PHE A  56       5.841   0.555  -2.601  1.00 13.24           C  \\nATOM    154  CD1 PHE A  56       5.570  -0.494  -3.470  1.00 15.32           C  \\nATOM    155  CD2 PHE A  56       6.467   0.272  -1.400  1.00 16.54           C  \\nATOM    156  CE1 PHE A  56       5.896  -1.804  -3.123  1.00 12.79           C  \\nATOM    157  CE2 PHE A  56       6.844  -1.026  -1.072  1.00 15.97           C  \\nATOM    158  CZ  PHE A  56       6.537  -2.072  -1.934  1.00 15.49           C  \\nATOM    159  N   SER A  57       5.707   4.203  -5.537  1.00 16.55           N  \\nATOM    160  CA  SER A  57       5.063   5.449  -5.896  1.00 16.76           C  \\nATOM    161  C   SER A  57       3.752   5.513  -5.123  1.00 14.11           C  \\nATOM    162  O   SER A  57       3.192   4.484  -4.750  1.00 17.18           O  \\nATOM    163  CB  SER A  57       4.797   5.499  -7.402  1.00 20.25           C  \\nATOM    164  OG  SER A  57       4.194   6.726  -7.773  1.00 26.47           O  \\nATOM    165  N   VAL A  58       3.303   6.732  -4.855  1.00 14.08           N  \\nATOM    166  CA  VAL A  58       2.169   6.959  -3.973  1.00 13.90           C  \\nATOM    167  C   VAL A  58       1.073   7.688  -4.731  1.00 14.89           C  \\nATOM    168  O   VAL A  58       1.310   8.747  -5.342  1.00 16.54           O  \\nATOM    169  CB  VAL A  58       2.597   7.795  -2.759  1.00 13.74           C  \\nATOM    170  CG1 VAL A  58       1.388   8.172  -1.910  1.00 14.20           C  \\nATOM    171  CG2 VAL A  58       3.621   7.029  -1.931  1.00 15.33           C  \\nATOM    172  N   ALA A  59      -0.137   7.141  -4.668  1.00 14.02           N  \\nATOM    173  CA  ALA A  59      -1.336   7.880  -5.078  1.00 13.71           C  \\nATOM    174  C   ALA A  59      -2.036   8.474  -3.866  1.00 14.79           C  \\nATOM    175  O   ALA A  59      -1.869   7.996  -2.737  1.00 13.10           O  \\nATOM    176  CB  ALA A  59      -2.282   6.963  -5.830  1.00 14.72           C  \\nATOM    177  N   THR A  60      -2.874   9.483  -4.108  1.00 15.39           N  \\nATOM    178  CA  THR A  60      -3.667  10.075  -3.034  1.00 14.85           C  \\nATOM    179  C   THR A  60      -5.138  10.095  -3.390  1.00 16.62           C  \\nATOM    180  O   THR A  60      -5.509  10.297  -4.553  1.00 19.57           O  \\nATOM    181  CB  THR A  60      -3.228  11.523  -2.717  1.00 19.13           C  \\nATOM    182  OG1 THR A  60      -3.256  12.305  -3.921  1.00 22.06           O  \\nATOM    183  CG2 THR A  60      -1.833  11.542  -2.141  1.00 17.93           C  \\nATOM    184  N   TYR A  61      -5.969   9.873  -2.387  1.00 17.13           N  \\nATOM    185  CA  TYR A  61      -7.398  10.055  -2.537  1.00 15.97           C  \\nATOM    186  C   TYR A  61      -7.905  10.997  -1.449  1.00 15.96           C  \\nATOM    187  O   TYR A  61      -7.649  10.794  -0.266  1.00 15.65           O  \\nATOM    188  CB  TYR A  61      -8.109   8.707  -2.453  1.00 17.82           C  \\nATOM    189  CG  TYR A  61      -9.609   8.827  -2.468  1.00 19.72           C  \\nATOM    190  CD1 TYR A  61     -10.293   9.059  -3.655  1.00 26.93           C  \\nATOM    191  CD2 TYR A  61     -10.344   8.703  -1.297  1.00 20.69           C  \\nATOM    192  CE1 TYR A  61     -11.674   9.172  -3.671  1.00 22.45           C  \\nATOM    193  CE2 TYR A  61     -11.725   8.819  -1.306  1.00 23.69           C  \\nATOM    194  CZ  TYR A  61     -12.374   9.081  -2.491  1.00 25.09           C  \\nATOM    195  OH  TYR A  61     -13.752   9.133  -2.519  1.00 30.36           O  \\nATOM    196  N   THR A  62      -8.634  12.026  -1.860  1.00 17.65           N  \\nATOM    197  CA  THR A  62      -9.234  12.956  -0.925  1.00 20.24           C  \\nATOM    198  C   THR A  62     -10.574  12.423  -0.417  1.00 20.45           C  \\nATOM    199  O   THR A  62     -11.517  12.209  -1.195  1.00 22.31           O  \\nATOM    200  CB  THR A  62      -9.433  14.341  -1.569  1.00 19.54           C  \\nATOM    201  OG1 THR A  62      -8.158  14.927  -1.849  1.00 24.49           O  \\nATOM    202  CG2 THR A  62     -10.209  15.262  -0.630  1.00 24.78           C  \\nATOM    203  N   VAL A  63     -10.659  12.239   0.897  1.00 18.61           N  \\nATOM    204  CA  VAL A  63     -11.896  11.835   1.546  1.00 20.02           C  \\nATOM    205  C   VAL A  63     -12.611  13.097   2.012  1.00 21.29           C  \\nATOM    206  O   VAL A  63     -12.138  13.792   2.896  1.00 18.73           O  \\nATOM    207  CB  VAL A  63     -11.617  10.943   2.772  1.00 19.52           C  \\nATOM    208  CG1 VAL A  63     -12.921  10.536   3.432  1.00 19.42           C  \\nATOM    209  CG2 VAL A  63     -10.814   9.721   2.350  1.00 20.01           C  \\nATOM    210  N   SER A  64     -13.696  13.455   1.334  1.00 23.39           N  \\nATOM    211  CA  SER A  64     -14.392  14.680   1.668  1.00 24.89           C  \\nATOM    212  C   SER A  64     -15.031  14.605   3.056  1.00 24.64           C  \\nATOM    213  O   SER A  64     -15.461  13.537   3.501  1.00 23.20           O  \\nATOM    214  CB  SER A  64     -15.453  14.990   0.611  1.00 27.36           C  \\nATOM    215  OG  SER A  64     -16.645  14.282   0.876  1.00 31.25           O  \\nATOM    216  N   ARG A  65     -15.035  15.740   3.747  1.00 20.99           N  \\nATOM    217  CA  ARG A  65     -15.555  15.837   5.106  1.00 24.26           C  \\nATOM    218  C   ARG A  65     -17.015  15.398   5.169  1.00 26.21           C  \\nATOM    219  O   ARG A  65     -17.411  14.631   6.049  1.00 25.64           O  \\nATOM    220  CB  ARG A  65     -15.403  17.269   5.624  1.00 23.53           C  \\nATOM    221  CG  ARG A  65     -16.206  17.576   6.875  1.00 29.79           C  \\nATOM    222  CD  ARG A  65     -15.969  19.004   7.326  1.00 29.85           C  \\nATOM    223  NE  ARG A  65     -14.549  19.296   7.464  1.00 36.57           N  \\nATOM    224  CZ  ARG A  65     -13.877  19.221   8.610  1.00 37.01           C  \\nATOM    225  NH1 ARG A  65     -14.523  19.000   9.748  1.00 33.30           N  \\nATOM    226  NH2 ARG A  65     -12.576  19.474   8.630  1.00 31.36           N  \\nATOM    227  N   LEU A  66     -17.766  15.740   4.128  1.00 26.63           N  \\nATOM    228  CA  LEU A  66     -19.194  15.470   4.100  1.00 24.37           C  \\nATOM    229  C   LEU A  66     -19.500  13.983   3.951  1.00 25.17           C  \\nATOM    230  O   LEU A  66     -20.559  13.523   4.369  1.00 27.77           O  \\nATOM    231  CB  LEU A  66     -19.848  16.268   2.967  1.00 25.63           C  \\nATOM    232  CG  LEU A  66     -21.366  16.429   3.035  1.00 30.96           C  \\nATOM    233  CD1 LEU A  66     -21.803  16.999   4.382  1.00 30.40           C  \\nATOM    234  CD2 LEU A  66     -21.846  17.317   1.893  1.00 30.30           C  \\nATOM    235  N   SER A  67     -18.617  13.251   3.273  1.00 22.88           N  \\nATOM    236  CA  SER A  67     -18.881  11.861   2.924  1.00 26.81           C  \\nATOM    237  C   SER A  67     -18.749  10.926   4.126  1.00 26.37           C  \\nATOM    238  O   SER A  67     -19.212   9.788   4.080  1.00 29.83           O  \\nATOM    239  CB  SER A  67     -17.964  11.398   1.788  1.00 28.66           C  \\nATOM    240  OG  SER A  67     -16.619  11.267   2.228  1.00 34.23           O  \\nATOM    241  N   VAL A  68     -18.194  11.437   5.223  1.00 27.64           N  \\nATOM    242  CA  VAL A  68     -18.019  10.627   6.438  1.00 26.80           C  \\nATOM    243  C   VAL A  68     -18.625  11.231   7.706  1.00 28.29           C  \\nATOM    244  O   VAL A  68     -18.361  12.385   8.054  1.00 30.84           O  \\nATOM    245  CB  VAL A  68     -16.529  10.237   6.669  1.00 24.49           C  \\nATOM    246  CG1 VAL A  68     -16.022   9.405   5.512  1.00 22.97           C  \\nATOM    247  CG2 VAL A  68     -15.659  11.477   6.843  1.00 21.49           C  \\nATOM    248  N   SER A  69     -19.291  10.379   8.482  1.00 31.63           N  \\nATOM    249  CA  SER A  69     -19.671  10.702   9.854  1.00 31.30           C  \\nATOM    250  C   SER A  69     -18.779   9.881  10.789  1.00 31.98           C  \\nATOM    251  O   SER A  69     -18.637   8.669  10.608  1.00 38.76           O  \\nATOM    252  CB  SER A  69     -21.157  10.360  10.085  1.00 32.26           C  \\nATOM    253  OG  SER A  69     -21.654  10.885  11.311  1.00 33.82           O  \\nATOM    254  N   GLY A  70     -17.960  10.582  11.563  1.00 29.43           N  \\nATOM    255  CA  GLY A  70     -17.023   9.920  12.470  1.00 24.84           C  \\nATOM    256  C   GLY A  70     -15.631  10.521  12.405  1.00 23.98           C  \\nATOM    257  O   GLY A  70     -14.785  10.267  13.272  1.00 21.91           O  \\nATOM    258  N   PHE A  71     -15.389  11.317  11.371  1.00 19.05           N  \\nATOM    259  CA  PHE A  71     -14.192  12.142  11.306  1.00 19.36           C  \\nATOM    260  C   PHE A  71     -14.386  13.177  10.201  1.00 17.81           C  \\nATOM    261  O   PHE A  71     -15.466  13.264   9.607  1.00 19.30           O  \\nATOM    262  CB  PHE A  71     -12.931  11.280  11.084  1.00 16.13           C  \\nATOM    263  CG  PHE A  71     -12.943  10.476   9.808  1.00 16.20           C  \\nATOM    264  CD1 PHE A  71     -12.077  10.794   8.765  1.00 15.45           C  \\nATOM    265  CD2 PHE A  71     -13.716   9.318   9.695  1.00 16.30           C  \\nATOM    266  CE1 PHE A  71     -12.022  10.007   7.598  1.00 16.54           C  \\nATOM    267  CE2 PHE A  71     -13.706   8.561   8.522  1.00 16.08           C  \\nATOM    268  CZ  PHE A  71     -12.858   8.906   7.467  1.00 15.36           C  \\nATOM    269  N   GLY A  72     -13.417  14.070  10.063  1.00 16.71           N  \\nATOM    270  CA  GLY A  72     -13.555  15.221   9.171  1.00 16.15           C  \\nATOM    271  C   GLY A  72     -13.038  14.990   7.761  1.00 17.43           C  \\nATOM    272  O   GLY A  72     -12.601  15.932   7.099  1.00 19.71           O  \\nATOM    273  N   GLY A  73     -13.103  13.743   7.289  1.00 15.98           N  \\nATOM    274  CA  GLY A  73     -12.442  13.377   6.036  1.00 16.32           C  \\nATOM    275  C   GLY A  73     -10.935  13.358   6.189  1.00 13.41           C  \\nATOM    276  O   GLY A  73     -10.427  13.120   7.275  1.00 16.11           O  \\nATOM    277  N   GLY A  74     -10.220  13.536   5.084  1.00 13.62           N  \\nATOM    278  CA  GLY A  74      -8.767  13.454   5.109  1.00 15.62           C  \\nATOM    279  C   GLY A  74      -8.189  13.061   3.768  1.00 16.29           C  \\nATOM    280  O   GLY A  74      -8.799  13.290   2.722  1.00 17.37           O  \\nATOM    281  N   VAL A  75      -6.981  12.504   3.792  1.00 13.86           N  \\nATOM    282  CA  VAL A  75      -6.301  12.090   2.567  1.00 13.31           C  \\nATOM    283  C   VAL A  75      -5.748  10.687   2.769  1.00 11.95           C  \\nATOM    284  O   VAL A  75      -5.125  10.407   3.806  1.00 14.58           O  \\nATOM    285  CB  VAL A  75      -5.144  13.034   2.186  1.00 14.13           C  \\nATOM    286  CG1 VAL A  75      -4.490  12.536   0.903  1.00 15.76           C  \\nATOM    287  CG2 VAL A  75      -5.652  14.469   2.012  1.00 15.82           C  \\nATOM    288  N   ILE A  76      -6.021   9.798   1.818  1.00 13.85           N  \\nATOM    289  CA  ILE A  76      -5.410   8.456   1.817  1.00 14.03           C  \\nATOM    290  C   ILE A  76      -4.205   8.445   0.892  1.00 13.87           C  \\nATOM    291  O   ILE A  76      -4.310   8.881  -0.251  1.00 15.32           O  \\nATOM    292  CB  ILE A  76      -6.381   7.374   1.323  1.00 13.07           C  \\nATOM    293  CG1 ILE A  76      -7.695   7.435   2.098  1.00 15.14           C  \\nATOM    294  CG2 ILE A  76      -5.778   5.977   1.524  1.00 14.10           C  \\nATOM    295  CD1 ILE A  76      -8.781   6.579   1.461  1.00 19.64           C  \\nATOM    296  N   TYR A  77      -3.059   8.010   1.417  1.00 12.99           N  \\nATOM    297  CA  TYR A  77      -1.834   7.820   0.639  1.00 11.27           C  \\nATOM    298  C   TYR A  77      -1.646   6.326   0.504  1.00 12.43           C  \\nATOM    299  O   TYR A  77      -1.744   5.609   1.489  1.00 13.22           O  \\nATOM    300  CB  TYR A  77      -0.627   8.415   1.387  1.00 12.86           C  \\nATOM    301  CG  TYR A  77      -0.762   9.887   1.702  1.00 12.30           C  \\nATOM    302  CD1 TYR A  77      -0.201  10.855   0.881  1.00 13.05           C  \\nATOM    303  CD2 TYR A  77      -1.456  10.307   2.823  1.00 11.14           C  \\nATOM    304  CE1 TYR A  77      -0.381  12.221   1.145  1.00 13.63           C  \\nATOM    305  CE2 TYR A  77      -1.570  11.666   3.129  1.00 12.28           C  \\nATOM    306  CZ  TYR A  77      -1.126  12.606   2.229  1.00 12.39           C  \\nATOM    307  OH  TYR A  77      -1.243  13.954   2.568  1.00 15.37           O  \\nATOM    308  N   TYR A  78      -1.465   5.851  -0.716  1.00 13.44           N  \\nATOM    309  CA  TYR A  78      -1.384   4.408  -0.911  1.00 12.09           C  \\nATOM    310  C   TYR A  78      -0.478   4.013  -2.065  1.00 14.27           C  \\nATOM    311  O   TYR A  78      -0.207   4.814  -2.952  1.00 12.66           O  \\nATOM    312  CB  TYR A  78      -2.799   3.838  -1.078  1.00 14.14           C  \\nATOM    313  CG  TYR A  78      -3.520   4.295  -2.334  1.00 12.70           C  \\nATOM    314  CD1 TYR A  78      -3.643   3.444  -3.427  1.00 15.11           C  \\nATOM    315  CD2 TYR A  78      -4.208   5.501  -2.364  1.00 13.86           C  \\nATOM    316  CE1 TYR A  78      -4.296   3.848  -4.573  1.00 16.07           C  \\nATOM    317  CE2 TYR A  78      -4.936   5.879  -3.481  1.00 16.42           C  \\nATOM    318  CZ  TYR A  78      -4.927   5.072  -4.605  1.00 17.96           C  \\nATOM    319  OH  TYR A  78      -5.625   5.457  -5.741  1.00 20.07           O  \\nATOM    320  N   PRO A  79       0.087   2.796  -2.013  1.00 13.54           N  \\nATOM    321  CA  PRO A  79       1.053   2.412  -3.037  1.00 13.33           C  \\nATOM    322  C   PRO A  79       0.393   2.172  -4.394  1.00 13.39           C  \\nATOM    323  O   PRO A  79      -0.723   1.633  -4.461  1.00 13.83           O  \\nATOM    324  CB  PRO A  79       1.591   1.074  -2.517  1.00 12.70           C  \\nATOM    325  CG  PRO A  79       1.374   1.133  -1.032  1.00 14.86           C  \\nATOM    326  CD  PRO A  79       0.056   1.836  -0.895  1.00 14.30           C  \\nATOM    327  N   THR A  80       1.119   2.520  -5.453  1.00 13.72           N  \\nATOM    328  CA  THR A  80       0.797   2.072  -6.809  1.00 14.39           C  \\nATOM    329  C   THR A  80       1.910   1.181  -7.354  1.00 13.64           C  \\nATOM    330  O   THR A  80       2.922   0.966  -6.691  1.00 13.77           O  \\nATOM    331  CB  THR A  80       0.603   3.257  -7.770  1.00 14.19           C  \\nATOM    332  OG1 THR A  80       1.834   3.977  -7.907  1.00 16.50           O  \\nATOM    333  CG2 THR A  80      -0.480   4.195  -7.257  1.00 20.12           C  \\nATOM    334  N   GLY A  81       1.724   0.687  -8.577  1.00 14.17           N  \\nATOM    335  CA  GLY A  81       2.748  -0.121  -9.233  1.00 14.38           C  \\nATOM    336  C   GLY A  81       3.009  -1.434  -8.517  1.00 13.29           C  \\nATOM    337  O   GLY A  81       4.116  -1.981  -8.595  1.00 13.26           O  \\nATOM    338  N   THR A  82       1.964  -2.020  -7.950  1.00 13.84           N  \\nATOM    339  CA  THR A  82       2.119  -3.259  -7.216  1.00 13.50           C  \\nATOM    340  C   THR A  82       0.810  -4.031  -7.173  1.00 14.37           C  \\nATOM    341  O   THR A  82      -0.258  -3.440  -7.135  1.00 16.02           O  \\nATOM    342  CB  THR A  82       2.625  -2.983  -5.771  1.00 14.63           C  \\nATOM    343  OG1 THR A  82       2.802  -4.222  -5.067  1.00 15.75           O  \\nATOM    344  CG2 THR A  82       1.659  -2.074  -4.993  1.00 14.93           C  \\nATOM    345  N   SER A  83       0.918  -5.348  -7.038  1.00 14.41           N  \\nATOM    346  CA  SER A  83      -0.257  -6.174  -6.793  1.00 15.51           C  \\nATOM    347  C   SER A  83      -0.275  -6.742  -5.385  1.00 16.39           C  \\nATOM    348  O   SER A  83      -1.169  -7.522  -5.046  1.00 19.30           O  \\nATOM    349  CB  SER A  83      -0.334  -7.312  -7.816  1.00 17.64           C  \\nATOM    350  OG  SER A  83      -0.592  -6.802  -9.115  1.00 21.38           O  \\nATOM    351  N   LEU A  84       0.649  -6.285  -4.544  1.00 14.28           N  \\nATOM    352  CA  LEU A  84       0.723  -6.750  -3.170  1.00 15.06           C  \\nATOM    353  C   LEU A  84      -0.281  -6.000  -2.297  1.00 14.67           C  \\nATOM    354  O   LEU A  84      -0.870  -4.991  -2.707  1.00 14.86           O  \\nATOM    355  CB  LEU A  84       2.147  -6.570  -2.622  1.00 17.14           C  \\nATOM    356  CG  LEU A  84       3.237  -7.284  -3.434  1.00 17.67           C  \\nATOM    357  CD1 LEU A  84       4.622  -6.952  -2.917  1.00 21.53           C  \\nATOM    358  CD2 LEU A  84       3.011  -8.792  -3.442  1.00 27.83           C  \\nATOM    359  N   THR A  85      -0.499  -6.524  -1.100  1.00 13.90           N  \\nATOM    360  CA  THR A  85      -1.297  -5.830  -0.107  1.00 12.43           C  \\nATOM    361  C   THR A  85      -0.442  -5.435   1.092  1.00 12.43           C  \\nATOM    362  O   THR A  85       0.644  -5.973   1.313  1.00 14.08           O  \\nATOM    363  CB  THR A  85      -2.468  -6.703   0.381  1.00 12.03           C  \\nATOM    364  OG1 THR A  85      -1.972  -7.907   0.976  1.00 15.59           O  \\nATOM    365  CG2 THR A  85      -3.419  -7.041  -0.761  1.00 16.22           C  \\nATOM    366  N   PHE A  86      -0.919  -4.446   1.836  1.00 12.33           N  \\nATOM    367  CA  PHE A  86      -0.104  -3.751   2.815  1.00 12.16           C  \\nATOM    368  C   PHE A  86      -0.969  -3.428   4.020  1.00 11.78           C  \\nATOM    369  O   PHE A  86      -2.184  -3.281   3.901  1.00 12.53           O  \\nATOM    370  CB  PHE A  86       0.422  -2.437   2.231  1.00 12.03           C  \\nATOM    371  CG  PHE A  86       1.243  -2.607   0.989  1.00 12.49           C  \\nATOM    372  CD1 PHE A  86       2.618  -2.762   1.073  1.00 13.10           C  \\nATOM    373  CD2 PHE A  86       0.646  -2.560  -0.265  1.00 13.86           C  \\nATOM    374  CE1 PHE A  86       3.393  -2.894  -0.083  1.00 12.96           C  \\nATOM    375  CE2 PHE A  86       1.405  -2.717  -1.420  1.00 14.32           C  \\nATOM    376  CZ  PHE A  86       2.789  -2.825  -1.335  1.00 15.44           C  \\nATOM    377  N   GLY A  87      -0.330  -3.230   5.160  1.00 11.24           N  \\nATOM    378  CA  GLY A  87      -1.048  -2.798   6.350  1.00 11.13           C  \\nATOM    379  C   GLY A  87      -1.566  -1.377   6.216  1.00 10.82           C  \\nATOM    380  O   GLY A  87      -1.080  -0.608   5.376  1.00 12.07           O  \\nATOM    381  N   GLY A  88      -2.570  -1.041   7.019  1.00 11.53           N  \\nATOM    382  CA  GLY A  88      -3.170   0.291   7.000  1.00 12.08           C  \\nATOM    383  C   GLY A  88      -2.782   1.057   8.246  1.00 13.60           C  \\nATOM    384  O   GLY A  88      -2.698   0.484   9.335  1.00 12.07           O  \\nATOM    385  N   ILE A  89      -2.644   2.364   8.104  1.00 12.60           N  \\nATOM    386  CA  ILE A  89      -2.360   3.223   9.237  1.00 12.14           C  \\nATOM    387  C   ILE A  89      -3.362   4.371   9.205  1.00 12.45           C  \\nATOM    388  O   ILE A  89      -3.495   5.035   8.187  1.00 12.44           O  \\nATOM    389  CB  ILE A  89      -0.917   3.773   9.146  1.00 14.19           C  \\nATOM    390  CG1 ILE A  89       0.091   2.606   9.134  1.00 12.77           C  \\nATOM    391  CG2 ILE A  89      -0.640   4.726  10.296  1.00 13.58           C  \\nATOM    392  CD1 ILE A  89       1.522   3.037   8.781  1.00 13.64           C  \\nATOM    393  N   ALA A  90      -4.005   4.648  10.334  1.00 11.32           N  \\nATOM    394  CA  ALA A  90      -4.740   5.913  10.495  1.00 11.64           C  \\nATOM    395  C   ALA A  90      -3.909   6.869  11.338  1.00 10.83           C  \\nATOM    396  O   ALA A  90      -3.361   6.469  12.366  1.00 12.55           O  \\nATOM    397  CB  ALA A  90      -6.108   5.659  11.159  1.00 11.97           C  \\nATOM    398  N   MET A  91      -3.829   8.134  10.908  1.00 10.50           N  \\nATOM    399  CA  MET A  91      -3.008   9.131  11.606  1.00 11.59           C  \\nATOM    400  C   MET A  91      -3.828  10.382  11.916  1.00 10.95           C  \\nATOM    401  O   MET A  91      -4.562  10.888  11.047  1.00 12.21           O  \\nATOM    402  CB  MET A  91      -1.807   9.529  10.742  1.00 16.52           C  \\nATOM    403  CG  MET A  91      -0.957   8.382  10.285  1.00 20.85           C  \\nATOM    404  SD  MET A  91       0.625   8.980   9.673  1.00 34.83           S  \\nATOM    405  CE  MET A  91       0.945  10.232  10.902  1.00  9.36           C  \\nATOM    406  N   SER A  92      -3.687  10.889  13.138  1.00 11.25           N  \\nATOM    407  CA  SER A  92      -4.408  12.088  13.559  1.00 12.64           C  \\nATOM    408  C   SER A  92      -3.460  13.263  13.708  1.00 10.68           C  \\nATOM    409  O   SER A  92      -2.439  13.149  14.384  1.00 10.94           O  \\nATOM    410  CB  SER A  92      -5.101  11.839  14.893  1.00 12.62           C  \\nATOM    411  OG  SER A  92      -5.718  13.033  15.353  1.00 13.23           O  \\nATOM    412  N   PRO A  93      -3.868  14.449  13.220  1.00 11.07           N  \\nATOM    413  CA  PRO A  93      -3.141  15.654  13.574  1.00 12.70           C  \\nATOM    414  C   PRO A  93      -3.468  16.114  14.991  1.00 13.59           C  \\nATOM    415  O   PRO A  93      -4.203  15.424  15.719  1.00 12.31           O  \\nATOM    416  CB  PRO A  93      -3.614  16.663  12.527  1.00 13.11           C  \\nATOM    417  CG  PRO A  93      -5.028  16.238  12.219  1.00 15.13           C  \\nATOM    418  CD  PRO A  93      -5.033  14.713  12.359  1.00 12.18           C  \\nATOM    419  N   GLY A  94      -2.829  17.200  15.422  1.00 14.51           N  \\nATOM    420  CA  GLY A  94      -3.004  17.699  16.783  1.00 15.70           C  \\nATOM    421  C   GLY A  94      -3.931  18.899  16.895  1.00 13.86           C  \\nATOM    422  O   GLY A  94      -4.639  19.240  15.941  1.00 13.59           O  \\nATOM    423  N   TYR A  95      -4.056  19.398  18.122  1.00 14.03           N  \\nATOM    424  CA  TYR A  95      -4.866  20.580  18.419  1.00 15.71           C  \\nATOM    425  C   TYR A  95      -4.508  21.710  17.454  1.00 15.16           C  \\nATOM    426  O   TYR A  95      -3.335  22.041  17.287  1.00 15.70           O  \\nATOM    427  CB  TYR A  95      -4.648  21.003  19.879  1.00 16.11           C  \\nATOM    428  CG  TYR A  95      -5.637  22.002  20.429  1.00 15.97           C  \\nATOM    429  CD1 TYR A  95      -6.943  21.628  20.718  1.00 14.59           C  \\nATOM    430  CD2 TYR A  95      -5.230  23.291  20.771  1.00 14.59           C  \\nATOM    431  CE1 TYR A  95      -7.783  22.475  21.418  1.00 16.86           C  \\nATOM    432  CE2 TYR A  95      -6.092  24.174  21.374  1.00 16.15           C  \\nATOM    433  CZ  TYR A  95      -7.373  23.767  21.689  1.00 16.48           C  \\nATOM    434  OH  TYR A  95      -8.249  24.644  22.305  1.00 18.99           O  \\nATOM    435  N   THR A  96      -5.545  22.342  16.901  1.00 16.19           N  \\nATOM    436  CA  THR A  96      -5.423  23.440  15.916  1.00 14.85           C  \\nATOM    437  C   THR A  96      -4.995  23.038  14.513  1.00 16.68           C  \\nATOM    438  O   THR A  96      -4.791  23.900  13.660  1.00 15.88           O  \\nATOM    439  CB  THR A  96      -4.540  24.638  16.377  1.00 17.91           C  \\nATOM    440  OG1 THR A  96      -3.149  24.313  16.243  1.00 18.28           O  \\nATOM    441  CG2 THR A  96      -4.853  25.061  17.812  1.00 20.79           C  \\nATOM    442  N   ALA A  97      -4.793  21.745  14.275  1.00 15.11           N  \\nATOM    443  CA  ALA A  97      -4.102  21.322  13.068  1.00 14.31           C  \\nATOM    444  C   ALA A  97      -4.992  20.569  12.095  1.00 13.93           C  \\nATOM    445  O   ALA A  97      -5.949  19.890  12.490  1.00 13.99           O  \\nATOM    446  CB  ALA A  97      -2.853  20.487  13.413  1.00 13.13           C  \\nATOM    447  N   ASP A  98      -4.618  20.639  10.817  1.00 15.34           N  \\nATOM    448  CA  ASP A  98      -5.279  19.898   9.748  1.00 14.32           C  \\nATOM    449  C   ASP A  98      -4.435  18.686   9.348  1.00 13.19           C  \\nATOM    450  O   ASP A  98      -3.244  18.613   9.686  1.00 14.04           O  \\nATOM    451  CB  ASP A  98      -5.483  20.800   8.510  1.00 15.35           C  \\nATOM    452  CG  ASP A  98      -6.431  21.960   8.771  1.00 19.23           C  \\nATOM    453  OD1 ASP A  98      -7.360  21.777   9.582  1.00 17.86           O  \\nATOM    454  OD2 ASP A  98      -6.339  22.965   8.029  1.00 19.49           O  \\nATOM    455  N   ALA A  99      -5.024  17.842   8.506  1.00 13.71           N  \\nATOM    456  CA  ALA A  99      -4.340  16.676   7.930  1.00 12.28           C  \\nATOM    457  C   ALA A  99      -3.073  17.107   7.191  1.00 14.64           C  \\nATOM    458  O   ALA A  99      -2.067  16.387   7.176  1.00 14.36           O  \\nATOM    459  CB  ALA A  99      -5.284  15.905   7.002  1.00 15.45           C  \\nATOM    460  N   SER A 100      -3.110  18.294   6.597  1.00 13.37           N  \\nATOM    461  CA  SER A 100      -1.978  18.775   5.810  1.00 13.75           C  \\nATOM    462  C   SER A 100      -0.699  18.898   6.643  1.00 12.87           C  \\nATOM    463  O   SER A 100       0.413  18.866   6.094  1.00 15.17           O  \\nATOM    464  CB  SER A 100      -2.309  20.109   5.153  1.00 12.46           C  \\nATOM    465  OG  SER A 100      -2.604  21.090   6.124  1.00 12.73           O  \\nATOM    466  N   SER A 101      -0.851  18.987   7.961  1.00 12.63           N  \\nATOM    467  CA  SER A 101       0.290  19.138   8.854  1.00 12.78           C  \\nATOM    468  C   SER A 101       1.114  17.857   8.979  1.00 13.44           C  \\nATOM    469  O   SER A 101       2.230  17.882   9.510  1.00 13.97           O  \\nATOM    470  CB  SER A 101      -0.154  19.611  10.246  1.00 13.91           C  \\nATOM    471  OG  SER A 101      -0.893  18.613  10.939  1.00 14.94           O  \\nATOM    472  N   LEU A 102       0.583  16.775   8.421  1.00 13.37           N  \\nATOM    473  CA  LEU A 102       1.285  15.480   8.432  1.00 13.27           C  \\nATOM    474  C   LEU A 102       1.350  14.862   7.042  1.00 15.29           C  \\nATOM    475  O   LEU A 102       1.673  13.685   6.893  1.00 13.86           O  \\nATOM    476  CB  LEU A 102       0.619  14.510   9.409  1.00 12.64           C  \\nATOM    477  CG  LEU A 102       0.700  14.896  10.891  1.00 11.99           C  \\nATOM    478  CD1 LEU A 102      -0.185  13.985  11.735  1.00 14.47           C  \\nATOM    479  CD2 LEU A 102       2.136  14.872  11.412  1.00 15.54           C  \\nATOM    480  N   ALA A 103       1.174  15.685   6.019  1.00 13.65           N  \\nATOM    481  CA  ALA A 103       1.146  15.181   4.656  1.00 13.50           C  \\nATOM    482  C   ALA A 103       2.479  14.523   4.277  1.00 13.22           C  \\nATOM    483  O   ALA A 103       2.498  13.451   3.662  1.00 13.27           O  \\nATOM    484  CB  ALA A 103       0.812  16.306   3.686  1.00 12.81           C  \\nATOM    485  N   TRP A 104       3.590  15.154   4.651  1.00 12.28           N  \\nATOM    486  CA  TRP A 104       4.893  14.583   4.328  1.00 12.73           C  \\nATOM    487  C   TRP A 104       5.044  13.176   4.897  1.00 13.16           C  \\nATOM    488  O   TRP A 104       5.618  12.290   4.245  1.00 13.66           O  \\nATOM    489  CB  TRP A 104       6.024  15.493   4.808  1.00 13.40           C  \\nATOM    490  CG  TRP A 104       6.310  15.360   6.266  1.00 14.80           C  \\nATOM    491  CD1 TRP A 104       5.759  16.089   7.283  1.00 15.28           C  \\nATOM    492  CD2 TRP A 104       7.220  14.438   6.882  1.00 13.55           C  \\nATOM    493  NE1 TRP A 104       6.219  15.632   8.495  1.00 17.02           N  \\nATOM    494  CE2 TRP A 104       7.141  14.640   8.273  1.00 15.95           C  \\nATOM    495  CE3 TRP A 104       8.059  13.427   6.393  1.00 15.58           C  \\nATOM    496  CZ2 TRP A 104       7.869  13.875   9.179  1.00 17.77           C  \\nATOM    497  CZ3 TRP A 104       8.849  12.728   7.299  1.00 14.19           C  \\nATOM    498  CH2 TRP A 104       8.720  12.928   8.672  1.00 16.08           C  \\nATOM    499  N   LEU A 105       4.488  12.953   6.083  1.00 13.67           N  \\nATOM    500  CA  LEU A 105       4.694  11.695   6.788  1.00 12.52           C  \\nATOM    501  C   LEU A 105       3.713  10.625   6.312  1.00 12.90           C  \\nATOM    502  O   LEU A 105       4.079   9.452   6.182  1.00 13.55           O  \\nATOM    503  CB  LEU A 105       4.622  11.880   8.305  1.00 12.68           C  \\nATOM    504  CG  LEU A 105       4.831  10.626   9.149  1.00 12.97           C  \\nATOM    505  CD1 LEU A 105       6.232  10.069   8.896  1.00 14.73           C  \\nATOM    506  CD2 LEU A 105       4.657  10.977  10.629  1.00 17.23           C  \\nATOM    507  N   GLY A 106       2.500  11.023   5.936  1.00 12.79           N  \\nATOM    508  CA  GLY A 106       1.594  10.088   5.260  1.00 13.83           C  \\nATOM    509  C   GLY A 106       2.176   9.567   3.958  1.00 11.79           C  \\nATOM    510  O   GLY A 106       2.166   8.363   3.697  1.00 13.35           O  \\nATOM    511  N   ARG A 107       2.741  10.474   3.162  1.00 13.65           N  \\nATOM    512  CA  ARG A 107       3.398  10.097   1.923  1.00 12.60           C  \\nATOM    513  C   ARG A 107       4.641   9.233   2.187  1.00 11.31           C  \\nATOM    514  O   ARG A 107       4.856   8.228   1.517  1.00 13.57           O  \\nATOM    515  CB  ARG A 107       3.808  11.337   1.137  1.00 14.34           C  \\nATOM    516  CG  ARG A 107       4.549  11.004  -0.145  1.00 17.34           C  \\nATOM    517  CD  ARG A 107       4.692  12.238  -1.021  1.00 19.23           C  \\nATOM    518  NE  ARG A 107       3.401  12.903  -1.199  1.00 25.77           N  \\nATOM    519  CZ  ARG A 107       2.543  12.625  -2.178  1.00 34.13           C  \\nATOM    520  NH1 ARG A 107       2.839  11.707  -3.087  1.00 35.90           N  \\nATOM    521  NH2 ARG A 107       1.376  13.251  -2.232  1.00 34.88           N  \\nATOM    522  N   ARG A 108       5.470   9.653   3.134  1.00 12.45           N  \\nATOM    523  CA  ARG A 108       6.675   8.914   3.491  1.00 11.03           C  \\nATOM    524  C   ARG A 108       6.341   7.472   3.859  1.00 12.07           C  \\nATOM    525  O   ARG A 108       6.940   6.534   3.317  1.00 12.47           O  \\nATOM    526  CB  ARG A 108       7.388   9.620   4.655  1.00 12.67           C  \\nATOM    527  CG  ARG A 108       8.638   8.888   5.184  1.00 13.64           C  \\nATOM    528  CD  ARG A 108       9.758   8.735   4.123  1.00 13.22           C  \\nATOM    529  NE  ARG A 108      10.071   9.985   3.423  1.00 13.02           N  \\nATOM    530  CZ  ARG A 108      10.704  11.027   3.962  1.00 15.27           C  \\nATOM    531  NH1 ARG A 108      11.258  10.957   5.169  1.00 16.79           N  \\nATOM    532  NH2 ARG A 108      10.866  12.120   3.235  1.00 17.73           N  \\nATOM    533  N   LEU A 109       5.376   7.285   4.751  1.00 10.53           N  \\nATOM    534  CA  LEU A 109       4.982   5.930   5.143  1.00 10.54           C  \\nATOM    535  C   LEU A 109       4.416   5.144   3.980  1.00 11.70           C  \\nATOM    536  O   LEU A 109       4.823   4.000   3.757  1.00 11.29           O  \\nATOM    537  CB  LEU A 109       3.991   5.956   6.310  1.00 11.61           C  \\nATOM    538  CG  LEU A 109       4.614   6.312   7.653  1.00 13.14           C  \\nATOM    539  CD1 LEU A 109       3.520   6.676   8.661  1.00 15.02           C  \\nATOM    540  CD2 LEU A 109       5.417   5.104   8.137  1.00 17.54           C  \\nATOM    541  N   ALA A 110       3.478   5.720   3.234  1.00 11.73           N  \\nATOM    542  CA  ALA A 110       2.876   4.989   2.126  1.00 11.25           C  \\nATOM    543  C   ALA A 110       3.951   4.531   1.134  1.00 11.82           C  \\nATOM    544  O   ALA A 110       3.886   3.413   0.613  1.00 12.06           O  \\nATOM    545  CB  ALA A 110       1.826   5.835   1.427  1.00 12.32           C  \\nATOM    546  N   SER A 111       4.955   5.381   0.903  1.00 11.62           N  \\nATOM    547  CA  SER A 111       6.010   5.094  -0.071  1.00 11.66           C  \\nATOM    548  C   SER A 111       6.902   3.932   0.348  1.00 12.00           C  \\nATOM    549  O   SER A 111       7.651   3.395  -0.472  1.00 13.61           O  \\nATOM    550  CB  SER A 111       6.866   6.342  -0.344  1.00 14.37           C  \\nATOM    551  OG  SER A 111       7.724   6.646   0.740  1.00 11.88           O  \\nATOM    552  N   HIS A 112       6.891   3.610   1.639  1.00 10.86           N  \\nATOM    553  CA  HIS A 112       7.602   2.435   2.143  1.00 11.43           C  \\nATOM    554  C   HIS A 112       6.704   1.219   2.316  1.00 10.98           C  \\nATOM    555  O   HIS A 112       7.131   0.222   2.902  1.00 12.14           O  \\nATOM    556  CB  HIS A 112       8.281   2.751   3.476  1.00 10.98           C  \\nATOM    557  CG  HIS A 112       9.587   3.462   3.320  1.00 12.05           C  \\nATOM    558  ND1 HIS A 112       9.702   4.833   3.173  1.00 15.15           N  \\nATOM    559  CD2 HIS A 112      10.832   2.960   3.158  1.00  9.13           C  \\nATOM    560  CE1 HIS A 112      10.975   5.142   2.955  1.00  9.10           C  \\nATOM    561  NE2 HIS A 112      11.679   4.023   2.968  1.00 15.24           N  \\nATOM    562  N   GLY A 113       5.493   1.297   1.771  1.00 11.64           N  \\nATOM    563  CA  GLY A 113       4.570   0.155   1.718  1.00 12.72           C  \\nATOM    564  C   GLY A 113       3.487   0.095   2.798  1.00 11.52           C  \\nATOM    565  O   GLY A 113       3.443  -0.842   3.598  1.00 12.21           O  \\nATOM    566  N   PHE A 114       2.656   1.134   2.871  1.00 12.19           N  \\nATOM    567  CA  PHE A 114       1.505   1.154   3.769  1.00 11.20           C  \\nATOM    568  C   PHE A 114       0.372   1.904   3.086  1.00 12.30           C  \\nATOM    569  O   PHE A 114       0.615   2.742   2.217  1.00 12.02           O  \\nATOM    570  CB  PHE A 114       1.848   1.852   5.089  1.00 12.24           C  \\nATOM    571  CG  PHE A 114       2.882   1.135   5.904  1.00 10.41           C  \\nATOM    572  CD1 PHE A 114       2.539   0.004   6.625  1.00 11.28           C  \\nATOM    573  CD2 PHE A 114       4.230   1.492   5.819  1.00 11.80           C  \\nATOM    574  CE1 PHE A 114       3.497  -0.732   7.305  1.00 14.86           C  \\nATOM    575  CE2 PHE A 114       5.190   0.776   6.527  1.00 12.75           C  \\nATOM    576  CZ  PHE A 114       4.818  -0.347   7.265  1.00 13.05           C  \\nATOM    577  N   VAL A 115      -0.859   1.612   3.504  1.00 10.94           N  \\nATOM    578  CA  VAL A 115      -2.013   2.438   3.153  1.00 11.23           C  \\nATOM    579  C   VAL A 115      -2.292   3.366   4.321  1.00 11.59           C  \\nATOM    580  O   VAL A 115      -2.513   2.900   5.431  1.00 11.39           O  \\nATOM    581  CB  VAL A 115      -3.260   1.560   2.889  1.00 11.46           C  \\nATOM    582  CG1 VAL A 115      -4.483   2.427   2.619  1.00 15.58           C  \\nATOM    583  CG2 VAL A 115      -2.997   0.609   1.744  1.00 12.49           C  \\nATOM    584  N   VAL A 116      -2.238   4.673   4.097  1.00 12.18           N  \\nATOM    585  CA  VAL A 116      -2.222   5.619   5.220  1.00 12.70           C  \\nATOM    586  C   VAL A 116      -3.336   6.637   5.042  1.00 12.03           C  \\nATOM    587  O   VAL A 116      -3.349   7.393   4.069  1.00 14.15           O  \\nATOM    588  CB  VAL A 116      -0.871   6.364   5.345  1.00 13.57           C  \\nATOM    589  CG1 VAL A 116      -0.838   7.215   6.611  1.00 11.21           C  \\nATOM    590  CG2 VAL A 116       0.298   5.386   5.342  1.00 13.51           C  \\nATOM    591  N   LEU A 117      -4.253   6.677   6.001  1.00 12.48           N  \\nATOM    592  CA  LEU A 117      -5.286   7.705   6.016  1.00 13.74           C  \\nATOM    593  C   LEU A 117      -4.973   8.754   7.085  1.00 11.84           C  \\nATOM    594  O   LEU A 117      -4.985   8.469   8.291  1.00 12.45           O  \\nATOM    595  CB  LEU A 117      -6.656   7.069   6.265  1.00 13.49           C  \\nATOM    596  CG  LEU A 117      -7.848   7.988   6.584  1.00 12.36           C  \\nATOM    597  CD1 LEU A 117      -7.988   9.139   5.581  1.00 14.31           C  \\nATOM    598  CD2 LEU A 117      -9.105   7.138   6.614  1.00 15.13           C  \\nATOM    599  N   VAL A 118      -4.589   9.943   6.628  1.00 11.90           N  \\nATOM    600  CA  VAL A 118      -4.390  11.067   7.544  1.00 12.45           C  \\nATOM    601  C   VAL A 118      -5.682  11.857   7.649  1.00 11.74           C  \\nATOM    602  O   VAL A 118      -6.207  12.319   6.628  1.00 12.72           O  \\nATOM    603  CB  VAL A 118      -3.269  12.004   7.057  1.00 11.83           C  \\nATOM    604  CG1 VAL A 118      -2.985  13.047   8.136  1.00 11.05           C  \\nATOM    605  CG2 VAL A 118      -1.988  11.200   6.748  1.00 13.36           C  \\nATOM    606  N   ILE A 119      -6.281  11.906   8.900  1.00 11.69           N  \\nATOM    607  CA  ILE A 119      -7.596  12.542   8.958  1.00 13.03           C  \\nATOM    608  C   ILE A 119      -7.516  14.033   9.240  1.00 13.56           C  \\nATOM    609  O   ILE A 119      -6.592  14.518   9.878  1.00 12.51           O  \\nATOM    610  CB  ILE A 119      -8.457  11.878  10.048  1.00 20.00           C  \\nATOM    611  CG1 ILE A 119      -7.798  12.036  11.419  1.00 20.00           C  \\nATOM    612  CG2 ILE A 119      -8.684  10.409   9.727  1.00 20.00           C  \\nATOM    613  CD1 ILE A 119      -8.676  11.604  12.572  1.00 20.00           C  \\nATOM    614  N   ASN A 120      -8.547  14.715   8.870  1.00 13.27           N  \\nATOM    615  CA  ASN A 120      -8.954  15.910   9.584  1.00 13.98           C  \\nATOM    616  C   ASN A 120      -9.956  15.532  10.654  1.00 13.51           C  \\nATOM    617  O   ASN A 120     -10.705  14.558  10.504  1.00 14.30           O  \\nATOM    618  CB  ASN A 120      -9.574  16.910   8.606  1.00 13.30           C  \\nATOM    619  CG  ASN A 120      -8.529  17.647   7.811  1.00 17.23           C  \\nATOM    620  OD1 ASN A 120      -8.599  17.715   6.589  1.00 24.44           O  \\nATOM    621  ND2 ASN A 120      -7.624  18.305   8.515  1.00 18.36           N  \\nATOM    622  N   THR A 121      -9.922  16.240  11.773  1.00 13.06           N  \\nATOM    623  CA  THR A 121     -10.885  15.987  12.836  1.00 14.61           C  \\nATOM    624  C   THR A 121     -12.155  16.796  12.618  1.00 16.03           C  \\nATOM    625  O   THR A 121     -12.199  17.699  11.774  1.00 16.42           O  \\nATOM    626  CB  THR A 121     -10.288  16.271  14.220  1.00 14.00           C  \\nATOM    627  OG1 THR A 121      -9.789  17.609  14.261  1.00 14.06           O  \\nATOM    628  CG2 THR A 121      -9.113  15.329  14.494  1.00 15.45           C  \\nATOM    629  N   ASN A 122     -13.227  16.395  13.287  1.00 15.99           N  \\nATOM    630  CA  ASN A 122     -14.491  17.120  13.155  1.00 14.68           C  \\nATOM    631  C   ASN A 122     -14.331  18.583  13.545  1.00 18.22           C  \\nATOM    632  O   ASN A 122     -14.839  19.493  12.865  1.00 19.00           O  \\nATOM    633  CB  ASN A 122     -15.549  16.467  14.046  1.00 15.08           C  \\nATOM    634  CG  ASN A 122     -15.965  15.111  13.549  1.00 16.28           C  \\nATOM    635  OD1 ASN A 122     -15.920  14.828  12.355  1.00 18.63           O  \\nATOM    636  ND2 ASN A 122     -16.220  14.208  14.493  1.00 22.03           N  \\nATOM    637  N   SER A 123     -13.550  18.811  14.592  1.00 13.70           N  \\nATOM    638  CA  SER A 123     -13.148  20.158  14.965  1.00 14.87           C  \\nATOM    639  C   SER A 123     -11.670  20.165  15.311  1.00 15.04           C  \\nATOM    640  O   SER A 123     -11.168  19.233  15.960  1.00 14.47           O  \\nATOM    641  CB  SER A 123     -13.957  20.644  16.167  1.00 15.91           C  \\nATOM    642  OG  SER A 123     -13.371  21.817  16.710  1.00 18.92           O  \\nATOM    643  N   ARG A 124     -10.973  21.226  14.905  1.00 13.69           N  \\nATOM    644  CA  ARG A 124      -9.581  21.394  15.283  1.00 14.01           C  \\nATOM    645  C   ARG A 124      -9.376  21.522  16.790  1.00 11.91           C  \\nATOM    646  O   ARG A 124      -8.248  21.446  17.279  1.00 14.50           O  \\nATOM    647  CB  ARG A 124      -8.955  22.590  14.553  1.00 15.50           C  \\nATOM    648  CG  ARG A 124      -8.579  22.253  13.111  1.00 15.31           C  \\nATOM    649  CD  ARG A 124      -7.952  23.457  12.401  1.00 14.64           C  \\nATOM    650  NE  ARG A 124      -8.973  24.467  12.169  1.00 15.06           N  \\nATOM    651  CZ  ARG A 124      -9.810  24.460  11.140  1.00 14.77           C  \\nATOM    652  NH1 ARG A 124      -9.608  23.616  10.132  1.00 16.20           N  \\nATOM    653  NH2 ARG A 124     -10.823  25.328  11.096  1.00 14.26           N  \\nATOM    654  N   PHE A 125     -10.467  21.752  17.524  1.00 14.33           N  \\nATOM    655  CA  PHE A 125     -10.376  21.845  18.974  1.00 15.86           C  \\nATOM    656  C   PHE A 125     -10.749  20.551  19.700  1.00 15.21           C  \\nATOM    657  O   PHE A 125     -10.805  20.531  20.935  1.00 15.13           O  \\nATOM    658  CB  PHE A 125     -11.221  23.011  19.509  1.00 16.30           C  \\nATOM    659  CG  PHE A 125     -10.699  24.378  19.124  1.00 16.75           C  \\nATOM    660  CD1 PHE A 125      -9.347  24.594  18.916  1.00 26.21           C  \\nATOM    661  CD2 PHE A 125     -11.558  25.460  19.059  1.00 19.27           C  \\nATOM    662  CE1 PHE A 125      -8.872  25.859  18.597  1.00 25.54           C  \\nATOM    663  CE2 PHE A 125     -11.082  26.724  18.755  1.00 19.47           C  \\nATOM    664  CZ  PHE A 125      -9.743  26.921  18.519  1.00 23.96           C  \\nATOM    665  N   ASP A 126     -10.928  19.458  18.955  1.00 15.39           N  \\nATOM    666  CA  ASP A 126     -11.254  18.182  19.599  1.00 14.69           C  \\nATOM    667  C   ASP A 126     -10.090  17.735  20.462  1.00 14.50           C  \\nATOM    668  O   ASP A 126      -8.915  17.940  20.118  1.00 14.62           O  \\nATOM    669  CB  ASP A 126     -11.543  17.094  18.566  1.00 14.69           C  \\nATOM    670  CG  ASP A 126     -12.832  17.327  17.796  1.00 16.50           C  \\nATOM    671  OD1 ASP A 126     -13.773  17.944  18.350  1.00 18.15           O  \\nATOM    672  OD2 ASP A 126     -12.959  16.765  16.686  1.00 14.57           O  \\nATOM    673  N   TYR A 127     -10.418  17.098  21.579  1.00 14.23           N  \\nATOM    674  CA  TYR A 127      -9.404  16.652  22.522  1.00 13.56           C  \\nATOM    675  C   TYR A 127      -8.974  15.215  22.236  1.00 13.57           C  \\nATOM    676  O   TYR A 127      -9.486  14.581  21.312  1.00 13.61           O  \\nATOM    677  CB  TYR A 127      -9.912  16.816  23.963  1.00 15.48           C  \\nATOM    678  CG  TYR A 127      -9.466  18.122  24.592  1.00 12.65           C  \\nATOM    679  CD1 TYR A 127      -8.756  18.126  25.792  1.00 14.60           C  \\nATOM    680  CD2 TYR A 127      -9.549  19.318  23.884  1.00 14.75           C  \\nATOM    681  CE1 TYR A 127      -8.289  19.301  26.341  1.00 15.34           C  \\nATOM    682  CE2 TYR A 127      -9.072  20.501  24.424  1.00 16.31           C  \\nATOM    683  CZ  TYR A 127      -8.492  20.490  25.681  1.00 17.88           C  \\nATOM    684  OH  TYR A 127      -7.974  21.655  26.191  1.00 19.70           O  \\nATOM    685  N   PRO A 128      -7.960  14.728  22.965  1.00 13.95           N  \\nATOM    686  CA  PRO A 128      -7.362  13.444  22.597  1.00 13.68           C  \\nATOM    687  C   PRO A 128      -8.367  12.291  22.507  1.00 14.18           C  \\nATOM    688  O   PRO A 128      -8.249  11.446  21.618  1.00 13.72           O  \\nATOM    689  CB  PRO A 128      -6.361  13.203  23.727  1.00 12.83           C  \\nATOM    690  CG  PRO A 128      -5.918  14.605  24.096  1.00 14.10           C  \\nATOM    691  CD  PRO A 128      -7.203  15.402  24.035  1.00 13.55           C  \\nATOM    692  N   ASP A 129      -9.248  12.160  23.494  1.00 14.73           N  \\nATOM    693  CA  ASP A 129     -10.125  11.001  23.494  1.00 14.14           C  \\nATOM    694  C   ASP A 129     -11.127  11.036  22.342  1.00 13.98           C  \\nATOM    695  O   ASP A 129     -11.456   9.995  21.765  1.00 14.25           O  \\nATOM    696  CB  ASP A 129     -10.846  10.849  24.838  1.00 14.49           C  \\nATOM    697  CG  ASP A 129      -9.909  10.438  25.959  1.00 16.16           C  \\nATOM    698  OD1 ASP A 129      -8.743  10.108  25.677  1.00 16.88           O  \\nATOM    699  OD2 ASP A 129     -10.316  10.554  27.131  1.00 21.95           O  \\nATOM    700  N   SER A 130     -11.502  12.246  21.911  1.00 14.52           N  \\nATOM    701  CA  SER A 130     -12.308  12.408  20.699  1.00 12.47           C  \\nATOM    702  C   SER A 130     -11.503  11.974  19.479  1.00 12.90           C  \\nATOM    703  O   SER A 130     -12.009  11.290  18.574  1.00 13.86           O  \\nATOM    704  CB  SER A 130     -12.745  13.870  20.569  1.00 13.99           C  \\nATOM    705  OG  SER A 130     -13.368  14.098  19.325  1.00 14.74           O  \\nATOM    706  N   ARG A 131     -10.237  12.351  19.462  1.00 12.80           N  \\nATOM    707  CA  ARG A 131      -9.400  11.952  18.330  1.00 12.68           C  \\nATOM    708  C   ARG A 131      -9.234  10.438  18.278  1.00 11.88           C  \\nATOM    709  O   ARG A 131      -9.205   9.890  17.259  1.00 13.93           O  \\nATOM    710  CB  ARG A 131      -8.027  12.675  18.355  1.00 20.00           C  \\nATOM    711  CG  ARG A 131      -8.128  14.207  18.219  1.00 20.00           C  \\nATOM    712  CD  ARG A 131      -6.878  14.969  18.570  1.00 20.00           C  \\nATOM    713  NE  ARG A 131      -7.254  16.373  18.586  1.00 20.00           N  \\nATOM    714  CZ  ARG A 131      -7.194  17.198  17.557  1.00 20.00           C  \\nATOM    715  NH1 ARG A 131      -6.636  16.828  16.424  1.00 20.00           N  \\nATOM    716  NH2 ARG A 131      -7.706  18.380  17.661  1.00 20.00           N  \\nATOM    717  N   ALA A 132      -9.139   9.816  19.435  1.00 13.10           N  \\nATOM    718  CA  ALA A 132      -9.096   8.354  19.464  1.00 12.48           C  \\nATOM    719  C   ALA A 132     -10.301   7.756  18.744  1.00 12.12           C  \\nATOM    720  O   ALA A 132     -10.160   6.834  17.935  1.00 13.38           O  \\nATOM    721  CB  ALA A 132      -9.020   7.864  20.900  1.00 13.65           C  \\nATOM    722  N   SER A 133     -11.492   8.275  19.045  1.00 13.55           N  \\nATOM    723  CA  SER A 133     -12.707   7.736  18.461  1.00 13.42           C  \\nATOM    724  C   SER A 133     -12.792   8.033  16.965  1.00 12.03           C  \\nATOM    725  O   SER A 133     -13.311   7.239  16.175  1.00 13.47           O  \\nATOM    726  CB  SER A 133     -13.942   8.242  19.212  1.00 15.56           C  \\nATOM    727  OG  SER A 133     -14.268   9.560  18.818  1.00 26.41           O  \\nATOM    728  N   GLN A 134     -12.164   9.128  16.550  1.00 14.13           N  \\nATOM    729  CA  GLN A 134     -12.127   9.466  15.135  1.00 12.69           C  \\nATOM    730  C   GLN A 134     -11.124   8.603  14.359  1.00 13.84           C  \\nATOM    731  O   GLN A 134     -11.373   8.230  13.207  1.00 13.15           O  \\nATOM    732  CB  GLN A 134     -11.819  10.953  14.966  1.00 12.23           C  \\nATOM    733  CG  GLN A 134     -12.940  11.819  15.530  1.00 11.42           C  \\nATOM    734  CD  GLN A 134     -12.699  13.288  15.327  1.00 14.07           C  \\nATOM    735  OE1 GLN A 134     -12.679  13.776  14.188  1.00 15.21           O  \\nATOM    736  NE2 GLN A 134     -12.650  14.025  16.430  1.00 14.40           N  \\nATOM    737  N   LEU A 135     -10.021   8.246  15.017  1.00 12.65           N  \\nATOM    738  CA  LEU A 135      -9.122   7.233  14.466  1.00 12.79           C  \\nATOM    739  C   LEU A 135      -9.827   5.907  14.259  1.00 12.73           C  \\nATOM    740  O   LEU A 135      -9.640   5.259  13.235  1.00 13.59           O  \\nATOM    741  CB  LEU A 135      -7.883   7.034  15.345  1.00 12.15           C  \\nATOM    742  CG  LEU A 135      -6.876   8.183  15.272  1.00 13.55           C  \\nATOM    743  CD1 LEU A 135      -5.853   8.035  16.383  1.00 14.10           C  \\nATOM    744  CD2 LEU A 135      -6.191   8.262  13.900  1.00 15.07           C  \\nATOM    745  N   SER A 136     -10.689   5.541  15.204  1.00 14.33           N  \\nATOM    746  CA  SER A 136     -11.452   4.293  15.064  1.00 14.43           C  \\nATOM    747  C   SER A 136     -12.409   4.351  13.887  1.00 14.52           C  \\nATOM    748  O   SER A 136     -12.523   3.391  13.119  1.00 15.58           O  \\nATOM    749  CB  SER A 136     -12.240   3.994  16.335  1.00 17.16           C  \\nATOM    750  OG  SER A 136     -11.375   3.581  17.366  1.00 22.03           O  \\nATOM    751  N   ALA A 137     -13.090   5.489  13.740  1.00 14.22           N  \\nATOM    752  CA  ALA A 137     -13.984   5.687  12.609  1.00 14.61           C  \\nATOM    753  C   ALA A 137     -13.213   5.656  11.304  1.00 15.45           C  \\nATOM    754  O   ALA A 137     -13.694   5.136  10.300  1.00 15.83           O  \\nATOM    755  CB  ALA A 137     -14.727   6.985  12.739  1.00 14.20           C  \\nATOM    756  N   ALA A 138     -12.024   6.257  11.308  1.00 14.30           N  \\nATOM    757  CA  ALA A 138     -11.189   6.262  10.114  1.00 15.09           C  \\nATOM    758  C   ALA A 138     -10.782   4.857   9.683  1.00 13.51           C  \\nATOM    759  O   ALA A 138     -10.753   4.549   8.482  1.00 16.44           O  \\nATOM    760  CB  ALA A 138      -9.957   7.137  10.330  1.00 15.02           C  \\nATOM    761  N   LEU A 139     -10.416   4.022  10.655  1.00 14.13           N  \\nATOM    762  CA  LEU A 139     -10.069   2.640  10.345  1.00 14.83           C  \\nATOM    763  C   LEU A 139     -11.275   1.859   9.814  1.00 13.85           C  \\nATOM    764  O   LEU A 139     -11.152   1.121   8.845  1.00 15.47           O  \\nATOM    765  CB  LEU A 139      -9.438   1.946  11.554  1.00 14.69           C  \\nATOM    766  CG  LEU A 139      -8.052   2.479  11.918  1.00 15.63           C  \\nATOM    767  CD1 LEU A 139      -7.632   1.973  13.283  1.00 18.11           C  \\nATOM    768  CD2 LEU A 139      -7.018   2.099  10.851  1.00 17.07           C  \\nATOM    769  N   ASN A 140     -12.462   2.129  10.349  1.00 16.01           N  \\nATOM    770  CA  ASN A 140     -13.652   1.470   9.827  1.00 14.58           C  \\nATOM    771  C   ASN A 140     -13.957   1.902   8.400  1.00 15.71           C  \\nATOM    772  O   ASN A 140     -14.270   1.078   7.541  1.00 16.46           O  \\nATOM    773  CB  ASN A 140     -14.855   1.720  10.738  1.00 14.23           C  \\nATOM    774  CG  ASN A 140     -16.161   1.167  10.153  1.00 22.64           C  \\nATOM    775  OD1 ASN A 140     -16.422  -0.038  10.228  1.00 26.02           O  \\nATOM    776  ND2 ASN A 140     -16.973   2.041   9.557  1.00 20.37           N  \\nATOM    777  N   TYR A 141     -13.737   3.182   8.117  1.00 15.96           N  \\nATOM    778  CA  TYR A 141     -13.940   3.692   6.771  1.00 14.82           C  \\nATOM    779  C   TYR A 141     -12.957   3.034   5.802  1.00 17.29           C  \\nATOM    780  O   TYR A 141     -13.316   2.664   4.681  1.00 17.03           O  \\nATOM    781  CB  TYR A 141     -13.757   5.214   6.747  1.00 16.00           C  \\nATOM    782  CG  TYR A 141     -13.582   5.762   5.351  1.00 17.47           C  \\nATOM    783  CD1 TYR A 141     -14.677   6.118   4.580  1.00 18.37           C  \\nATOM    784  CD2 TYR A 141     -12.316   5.950   4.812  1.00 17.82           C  \\nATOM    785  CE1 TYR A 141     -14.523   6.487   3.255  1.00 19.36           C  \\nATOM    786  CE2 TYR A 141     -12.150   6.382   3.506  1.00 24.18           C  \\nATOM    787  CZ  TYR A 141     -13.259   6.634   2.730  1.00 24.53           C  \\nATOM    788  OH  TYR A 141     -13.087   7.073   1.431  1.00 26.69           O  \\nATOM    789  N   LEU A 142     -11.686   3.006   6.192  1.00 15.66           N  \\nATOM    790  CA  LEU A 142     -10.656   2.445   5.324  1.00 18.65           C  \\nATOM    791  C   LEU A 142     -11.017   1.017   4.921  1.00 16.84           C  \\nATOM    792  O   LEU A 142     -10.841   0.629   3.762  1.00 19.94           O  \\nATOM    793  CB  LEU A 142      -9.310   2.469   6.046  1.00 23.08           C  \\nATOM    794  CG  LEU A 142      -8.016   2.552   5.247  1.00 24.31           C  \\nATOM    795  CD1 LEU A 142      -8.010   3.700   4.238  1.00 16.34           C  \\nATOM    796  CD2 LEU A 142      -6.839   2.662   6.216  1.00 17.79           C  \\nATOM    797  N   ARG A 143     -11.495   0.237   5.893  1.00 16.11           N  \\nATOM    798  CA  ARG A 143     -11.838  -1.173   5.688  1.00 18.76           C  \\nATOM    799  C   ARG A 143     -13.082  -1.340   4.816  1.00 19.13           C  \\nATOM    800  O   ARG A 143     -13.232  -2.340   4.111  1.00 20.71           O  \\nATOM    801  CB  ARG A 143     -12.052  -1.850   7.046  1.00 19.07           C  \\nATOM    802  CG  ARG A 143     -12.406  -3.327   6.993  1.00 29.71           C  \\nATOM    803  CD  ARG A 143     -12.399  -3.933   8.397  1.00 34.22           C  \\nATOM    804  NE  ARG A 143     -13.122  -3.099   9.359  1.00 39.71           N  \\nATOM    805  CZ  ARG A 143     -14.441  -3.122   9.548  1.00 37.84           C  \\nATOM    806  NH1 ARG A 143     -15.196  -4.052   8.964  1.00 32.84           N  \\nATOM    807  NH2 ARG A 143     -15.001  -2.247  10.374  1.00 25.97           N  \\nATOM    808  N   THR A 144     -14.033  -0.423   4.954  1.00 16.98           N  \\nATOM    809  CA  THR A 144     -15.378  -0.691   4.439  1.00 17.04           C  \\nATOM    810  C   THR A 144     -15.789   0.182   3.277  1.00 17.15           C  \\nATOM    811  O   THR A 144     -16.729  -0.159   2.563  1.00 17.98           O  \\nATOM    812  CB  THR A 144     -16.469  -0.599   5.525  1.00 15.59           C  \\nATOM    813  OG1 THR A 144     -16.533   0.737   6.052  1.00 16.30           O  \\nATOM    814  CG2 THR A 144     -16.206  -1.583   6.636  1.00 15.23           C  \\nATOM    815  N   SER A 145     -15.209   1.372   3.166  1.00 17.57           N  \\nATOM    816  CA  SER A 145     -15.802   2.401   2.322  1.00 17.97           C  \\nATOM    817  C   SER A 145     -14.822   3.086   1.378  1.00 16.25           C  \\nATOM    818  O   SER A 145     -15.181   4.029   0.684  1.00 19.12           O  \\nATOM    819  CB  SER A 145     -16.514   3.449   3.176  1.00 19.70           C  \\nATOM    820  OG  SER A 145     -17.577   2.845   3.888  1.00 29.31           O  \\nATOM    821  N   SER A 146     -13.557   2.709   1.459  1.00 19.20           N  \\nATOM    822  CA  SER A 146     -12.554   3.370   0.633  1.00 19.28           C  \\nATOM    823  C   SER A 146     -12.731   2.955  -0.826  1.00 16.78           C  \\nATOM    824  O   SER A 146     -13.337   1.921  -1.109  1.00 18.38           O  \\nATOM    825  CB  SER A 146     -11.152   3.028   1.135  1.00 20.22           C  \\nATOM    826  OG  SER A 146     -10.996   1.622   1.248  1.00 21.93           O  \\nATOM    827  N   PRO A 147     -12.174   3.747  -1.755  1.00 15.77           N  \\nATOM    828  CA  PRO A 147     -12.257   3.449  -3.180  1.00 15.72           C  \\nATOM    829  C   PRO A 147     -11.637   2.106  -3.537  1.00 16.28           C  \\nATOM    830  O   PRO A 147     -10.729   1.634  -2.839  1.00 17.53           O  \\nATOM    831  CB  PRO A 147     -11.458   4.583  -3.834  1.00 18.07           C  \\nATOM    832  CG  PRO A 147     -11.342   5.650  -2.802  1.00 22.55           C  \\nATOM    833  CD  PRO A 147     -11.417   4.978  -1.474  1.00 17.41           C  \\nATOM    834  N   SER A 148     -11.997   1.586  -4.708  1.00 17.26           N  \\nATOM    835  CA  SER A 148     -11.563   0.250  -5.097  1.00 17.13           C  \\nATOM    836  C   SER A 148     -10.043   0.121  -5.159  1.00 16.48           C  \\nATOM    837  O   SER A 148      -9.491  -0.910  -4.797  1.00 18.72           O  \\nATOM    838  CB  SER A 148     -12.183  -0.178  -6.432  1.00 19.26           C  \\nATOM    839  OG  SER A 148     -11.850   0.728  -7.480  1.00 22.11           O  \\nATOM    840  N   ALA A 149      -9.367   1.169  -5.620  1.00 18.59           N  \\nATOM    841  CA  ALA A 149      -7.917   1.099  -5.773  1.00 17.38           C  \\nATOM    842  C   ALA A 149      -7.233   1.002  -4.418  1.00 18.27           C  \\nATOM    843  O   ALA A 149      -6.180   0.379  -4.289  1.00 18.73           O  \\nATOM    844  CB  ALA A 149      -7.401   2.307  -6.536  1.00 18.96           C  \\nATOM    845  N   VAL A 150      -7.785   1.700  -3.433  1.00 15.91           N  \\nATOM    846  CA  VAL A 150      -7.278   1.627  -2.070  1.00 15.03           C  \\nATOM    847  C   VAL A 150      -7.556   0.261  -1.445  1.00 16.25           C  \\nATOM    848  O   VAL A 150      -6.651  -0.385  -0.904  1.00 16.25           O  \\nATOM    849  CB  VAL A 150      -7.899   2.737  -1.194  1.00 15.49           C  \\nATOM    850  CG1 VAL A 150      -7.484   2.567   0.259  1.00 16.92           C  \\nATOM    851  CG2 VAL A 150      -7.518   4.122  -1.728  1.00 16.04           C  \\nATOM    852  N   ARG A 151      -8.801  -0.199  -1.546  1.00 15.36           N  \\nATOM    853  CA  ARG A 151      -9.172  -1.480  -0.970  1.00 18.31           C  \\nATOM    854  C   ARG A 151      -8.303  -2.625  -1.475  1.00 15.62           C  \\nATOM    855  O   ARG A 151      -7.926  -3.517  -0.710  1.00 16.97           O  \\nATOM    856  CB  ARG A 151     -10.640  -1.779  -1.239  1.00 21.22           C  \\nATOM    857  CG  ARG A 151     -11.276  -2.666  -0.171  1.00 29.31           C  \\nATOM    858  CD  ARG A 151     -11.732  -1.849   1.037  1.00 30.34           C  \\nATOM    859  NE  ARG A 151     -12.718  -0.836   0.658  1.00 32.36           N  \\nATOM    860  CZ  ARG A 151     -14.021  -1.072   0.513  1.00 34.73           C  \\nATOM    861  NH1 ARG A 151     -14.534  -2.244   0.879  1.00 26.48           N  \\nATOM    862  NH2 ARG A 151     -14.819  -0.118   0.045  1.00 27.93           N  \\nATOM    863  N   ALA A 152      -7.963  -2.586  -2.759  1.00 16.45           N  \\nATOM    864  CA  ALA A 152      -7.136  -3.628  -3.367  1.00 15.18           C  \\nATOM    865  C   ALA A 152      -5.725  -3.680  -2.788  1.00 14.72           C  \\nATOM    866  O   ALA A 152      -5.082  -4.721  -2.850  1.00 16.19           O  \\nATOM    867  CB  ALA A 152      -7.077  -3.452  -4.876  1.00 18.14           C  \\nATOM    868  N   ARG A 153      -5.258  -2.563  -2.224  1.00 14.62           N  \\nATOM    869  CA  ARG A 153      -3.900  -2.483  -1.664  1.00 13.94           C  \\nATOM    870  C   ARG A 153      -3.871  -2.757  -0.165  1.00 13.24           C  \\nATOM    871  O   ARG A 153      -2.800  -2.903   0.408  1.00 14.34           O  \\nATOM    872  CB  ARG A 153      -3.242  -1.125  -1.958  1.00 15.25           C  \\nATOM    873  CG  ARG A 153      -3.074  -0.815  -3.440  1.00 13.78           C  \\nATOM    874  CD  ARG A 153      -1.887  -1.541  -4.089  1.00 16.55           C  \\nATOM    875  NE  ARG A 153      -2.092  -2.983  -4.191  1.00 16.57           N  \\nATOM    876  CZ  ARG A 153      -2.836  -3.580  -5.114  1.00 17.03           C  \\nATOM    877  NH1 ARG A 153      -2.899  -4.908  -5.147  1.00 18.90           N  \\nATOM    878  NH2 ARG A 153      -3.512  -2.863  -6.003  1.00 18.88           N  \\nATOM    879  N   LEU A 154      -5.046  -2.758   0.465  1.00 13.08           N  \\nATOM    880  CA  LEU A 154      -5.157  -2.832   1.915  1.00 14.27           C  \\nATOM    881  C   LEU A 154      -5.346  -4.277   2.382  1.00 13.24           C  \\nATOM    882  O   LEU A 154      -6.290  -4.960   1.969  1.00 14.44           O  \\nATOM    883  CB  LEU A 154      -6.348  -2.004   2.411  1.00 13.92           C  \\nATOM    884  CG  LEU A 154      -6.576  -2.028   3.930  1.00 13.47           C  \\nATOM    885  CD1 LEU A 154      -5.385  -1.442   4.687  1.00 16.90           C  \\nATOM    886  CD2 LEU A 154      -7.866  -1.269   4.282  1.00 15.44           C  \\nATOM    887  N   ASP A 155      -4.478  -4.708   3.292  1.00 13.98           N  \\nATOM    888  CA  ASP A 155      -4.752  -5.884   4.121  1.00 11.93           C  \\nATOM    889  C   ASP A 155      -5.319  -5.424   5.446  1.00 12.32           C  \\nATOM    890  O   ASP A 155      -4.591  -4.972   6.319  1.00 13.05           O  \\nATOM    891  CB  ASP A 155      -3.470  -6.676   4.369  1.00 13.32           C  \\nATOM    892  CG  ASP A 155      -3.708  -7.914   5.207  1.00 13.35           C  \\nATOM    893  OD1 ASP A 155      -4.864  -8.124   5.649  1.00 16.69           O  \\nATOM    894  OD2 ASP A 155      -2.755  -8.695   5.393  1.00 13.23           O  \\nATOM    895  N   ALA A 156      -6.641  -5.436   5.540  1.00 12.99           N  \\nATOM    896  CA  ALA A 156      -7.323  -4.837   6.676  1.00 13.40           C  \\nATOM    897  C   ALA A 156      -7.136  -5.636   7.961  1.00 13.73           C  \\nATOM    898  O   ALA A 156      -7.510  -5.180   9.032  1.00 14.92           O  \\nATOM    899  CB  ALA A 156      -8.809  -4.643   6.369  1.00 18.44           C  \\nATOM    900  N   ASN A 157      -6.508  -6.804   7.859  1.00 13.64           N  \\nATOM    901  CA  ASN A 157      -6.127  -7.564   9.044  1.00 14.44           C  \\nATOM    902  C   ASN A 157      -5.045  -6.871   9.853  1.00 12.63           C  \\nATOM    903  O   ASN A 157      -4.849  -7.188  11.024  1.00 14.01           O  \\nATOM    904  CB  ASN A 157      -5.637  -8.975   8.674  1.00 11.92           C  \\nATOM    905  CG  ASN A 157      -6.736  -9.854   8.131  1.00 16.18           C  \\nATOM    906  OD1 ASN A 157      -7.910  -9.651   8.433  1.00 17.50           O  \\nATOM    907  ND2 ASN A 157      -6.371 -10.768   7.247  1.00 18.22           N  \\nATOM    908  N   ARG A 158      -4.279  -5.991   9.204  1.00 12.66           N  \\nATOM    909  CA  ARG A 158      -3.058  -5.481   9.812  1.00 14.21           C  \\nATOM    910  C   ARG A 158      -3.067  -3.960   9.874  1.00 12.59           C  \\nATOM    911  O   ARG A 158      -2.816  -3.296   8.874  1.00 12.19           O  \\nATOM    912  CB  ARG A 158      -1.821  -6.005   9.064  1.00 11.96           C  \\nATOM    913  CG  ARG A 158      -1.798  -7.551   8.951  1.00 13.80           C  \\nATOM    914  CD  ARG A 158      -0.512  -8.046   8.289  1.00 14.38           C  \\nATOM    915  NE  ARG A 158      -0.477  -7.717   6.867  1.00 13.11           N  \\nATOM    916  CZ  ARG A 158       0.489  -7.012   6.269  1.00 11.25           C  \\nATOM    917  NH1 ARG A 158       1.386  -6.361   7.000  1.00 11.99           N  \\nATOM    918  NH2 ARG A 158       0.480  -6.856   4.949  1.00 14.26           N  \\nATOM    919  N   LEU A 159      -3.464  -3.418  11.026  1.00 12.41           N  \\nATOM    920  CA  LEU A 159      -3.770  -1.995  11.147  1.00 11.80           C  \\nATOM    921  C   LEU A 159      -2.992  -1.389  12.298  1.00 11.73           C  \\nATOM    922  O   LEU A 159      -2.703  -2.056  13.282  1.00 11.47           O  \\nATOM    923  CB  LEU A 159      -5.248  -1.797  11.437  1.00 11.26           C  \\nATOM    924  CG  LEU A 159      -6.206  -2.330  10.376  1.00 12.73           C  \\nATOM    925  CD1 LEU A 159      -7.656  -2.139  10.844  1.00 16.17           C  \\nATOM    926  CD2 LEU A 159      -5.996  -1.607   9.064  1.00 14.52           C  \\nATOM    927  N   ALA A 160      -2.677  -0.106  12.174  1.00 11.56           N  \\nATOM    928  CA  ALA A 160      -1.911   0.587  13.198  1.00 11.53           C  \\nATOM    929  C   ALA A 160      -2.293   2.066  13.203  1.00 12.83           C  \\nATOM    930  O   ALA A 160      -3.076   2.515  12.362  1.00 12.23           O  \\nATOM    931  CB  ALA A 160      -0.412   0.426  12.951  1.00 10.53           C  \\nATOM    932  N   VAL A 161      -1.857   2.775  14.235  1.00 11.91           N  \\nATOM    933  CA  VAL A 161      -2.224   4.184  14.415  1.00 11.14           C  \\nATOM    934  C   VAL A 161      -1.022   5.056  14.758  1.00 12.94           C  \\nATOM    935  O   VAL A 161      -0.027   4.577  15.289  1.00 11.68           O  \\nATOM    936  CB  VAL A 161      -3.296   4.352  15.518  1.00 12.71           C  \\nATOM    937  CG1 VAL A 161      -4.618   3.784  15.061  1.00 12.35           C  \\nATOM    938  CG2 VAL A 161      -2.851   3.680  16.806  1.00 14.76           C  \\nATOM    939  N   ALA A 162      -1.108   6.331  14.384  1.00 12.47           N  \\nATOM    940  CA  ALA A 162      -0.110   7.325  14.764  1.00 12.07           C  \\nATOM    941  C   ALA A 162      -0.790   8.672  14.880  1.00 11.38           C  \\nATOM    942  O   ALA A 162      -1.915   8.847  14.417  1.00 12.12           O  \\nATOM    943  CB  ALA A 162       1.002   7.397  13.709  1.00 14.51           C  \\nATOM    944  N   GLY A 163      -0.127   9.624  15.515  1.00 11.62           N  \\nATOM    945  CA  GLY A 163      -0.733  10.935  15.636  1.00 12.21           C  \\nATOM    946  C   GLY A 163       0.190  11.929  16.278  1.00 11.41           C  \\nATOM    947  O   GLY A 163       1.077  11.575  17.051  1.00 13.03           O  \\nATOM    948  N   HIS A 164      -0.097  13.201  16.025  1.00 11.36           N  \\nATOM    949  CA  HIS A 164       0.718  14.300  16.523  1.00 12.72           C  \\nATOM    950  C   HIS A 164       0.009  14.968  17.703  1.00 13.27           C  \\nATOM    951  O   HIS A 164      -1.169  15.323  17.603  1.00 12.97           O  \\nATOM    952  CB  HIS A 164       0.971  15.295  15.391  1.00 10.52           C  \\nATOM    953  CG  HIS A 164       1.872  16.420  15.782  1.00 11.24           C  \\nATOM    954  ND1 HIS A 164       1.730  17.699  15.288  1.00 13.00           N  \\nATOM    955  CD2 HIS A 164       2.934  16.454  16.620  1.00 11.81           C  \\nATOM    956  CE1 HIS A 164       2.669  18.473  15.806  1.00 14.28           C  \\nATOM    957  NE2 HIS A 164       3.383  17.753  16.655  1.00 12.06           N  \\nATOM    958  N   SER A 165       0.737  15.142  18.799  1.00 10.55           N  \\nATOM    959  CA  SER A 165       0.351  16.013  19.900  1.00 13.84           C  \\nATOM    960  C   SER A 165      -0.981  15.476  20.427  1.00 14.25           C  \\nATOM    961  O   SER A 165      -1.052  14.302  20.747  1.00 13.30           O  \\nATOM    962  CB  SER A 165       0.287  17.472  19.470  1.00 13.92           C  \\nATOM    963  OG  SER A 165       0.528  18.311  20.600  1.00 16.90           O  \\nATOM    964  N   MET A 166      -2.020  16.280  20.563  1.00 12.97           N  \\nATOM    965  CA  MET A 166      -3.285  15.696  21.024  1.00 13.40           C  \\nATOM    966  C   MET A 166      -3.698  14.467  20.198  1.00 13.52           C  \\nATOM    967  O   MET A 166      -4.370  13.568  20.704  1.00 12.91           O  \\nATOM    968  CB  MET A 166      -4.416  16.726  21.048  1.00 15.71           C  \\nATOM    969  CG  MET A 166      -4.331  17.697  22.204  1.00 12.94           C  \\nATOM    970  SD  MET A 166      -5.884  18.589  22.360  1.00 14.49           S  \\nATOM    971  CE  MET A 166      -5.391  19.863  23.515  1.00 15.06           C  \\nATOM    972  N   GLY A 167      -3.328  14.443  18.919  1.00 12.87           N  \\nATOM    973  CA  GLY A 167      -3.521  13.238  18.110  1.00 14.82           C  \\nATOM    974  C   GLY A 167      -2.684  12.074  18.617  1.00 13.25           C  \\nATOM    975  O   GLY A 167      -3.054  10.903  18.442  1.00 13.52           O  \\nATOM    976  N   GLY A 168      -1.532  12.379  19.204  1.00 13.42           N  \\nATOM    977  CA  GLY A 168      -0.711  11.333  19.810  1.00 13.62           C  \\nATOM    978  C   GLY A 168      -1.322  10.856  21.111  1.00 12.88           C  \\nATOM    979  O   GLY A 168      -1.247   9.665  21.446  1.00 12.75           O  \\nATOM    980  N   GLY A 169      -1.965  11.765  21.845  1.00 13.28           N  \\nATOM    981  CA  GLY A 169      -2.816  11.357  22.974  1.00 13.89           C  \\nATOM    982  C   GLY A 169      -3.947  10.441  22.544  1.00 14.71           C  \\nATOM    983  O   GLY A 169      -4.191   9.408  23.175  1.00 13.74           O  \\nATOM    984  N   GLY A 170      -4.576  10.759  21.420  1.00 13.32           N  \\nATOM    985  CA  GLY A 170      -5.589   9.883  20.829  1.00 12.66           C  \\nATOM    986  C   GLY A 170      -5.075   8.508  20.436  1.00 12.72           C  \\nATOM    987  O   GLY A 170      -5.758   7.504  20.634  1.00 13.85           O  \\nATOM    988  N   THR A 171      -3.846   8.471  19.926  1.00 11.41           N  \\nATOM    989  CA  THR A 171      -3.167   7.225  19.553  1.00 11.98           C  \\nATOM    990  C   THR A 171      -2.972   6.327  20.782  1.00 12.75           C  \\nATOM    991  O   THR A 171      -3.259   5.124  20.735  1.00 12.15           O  \\nATOM    992  CB  THR A 171      -1.803   7.551  18.906  1.00 14.92           C  \\nATOM    993  OG1 THR A 171      -2.026   8.247  17.670  1.00 14.51           O  \\nATOM    994  CG2 THR A 171      -0.999   6.286  18.648  1.00 14.28           C  \\nATOM    995  N   LEU A 172      -2.575   6.919  21.906  1.00 12.16           N  \\nATOM    996  CA  LEU A 172      -2.423   6.148  23.132  1.00 11.87           C  \\nATOM    997  C   LEU A 172      -3.760   5.604  23.638  1.00 15.63           C  \\nATOM    998  O   LEU A 172      -3.849   4.441  24.041  1.00 14.29           O  \\nATOM    999  CB  LEU A 172      -1.737   6.995  24.203  1.00 11.86           C  \\nATOM   1000  CG  LEU A 172      -0.238   7.201  23.973  1.00 13.83           C  \\nATOM   1001  CD1 LEU A 172       0.229   8.468  24.704  1.00 14.02           C  \\nATOM   1002  CD2 LEU A 172       0.509   5.982  24.477  1.00 15.52           C  \\nATOM   1003  N   ARG A 173      -4.812   6.414  23.553  1.00 14.01           N  \\nATOM   1004  CA  ARG A 173      -6.133   5.974  23.996  1.00 13.09           C  \\nATOM   1005  C   ARG A 173      -6.717   4.857  23.121  1.00 14.16           C  \\nATOM   1006  O   ARG A 173      -7.232   3.875  23.645  1.00 14.52           O  \\nATOM   1007  CB  ARG A 173      -7.097   7.164  24.073  1.00 14.07           C  \\nATOM   1008  CG  ARG A 173      -8.504   6.791  24.536  1.00 13.91           C  \\nATOM   1009  CD  ARG A 173      -8.502   6.401  26.010  1.00 16.38           C  \\nATOM   1010  NE  ARG A 173      -8.342   7.593  26.841  1.00 15.22           N  \\nATOM   1011  CZ  ARG A 173      -8.033   7.595  28.131  1.00 19.64           C  \\nATOM   1012  NH1 ARG A 173      -7.816   6.454  28.776  1.00 24.62           N  \\nATOM   1013  NH2 ARG A 173      -8.068   8.736  28.808  1.00 22.51           N  \\nATOM   1014  N   ILE A 174      -6.650   5.001  21.804  1.00 14.15           N  \\nATOM   1015  CA  ILE A 174      -7.203   3.976  20.927  1.00 13.00           C  \\nATOM   1016  C   ILE A 174      -6.419   2.677  21.077  1.00 15.24           C  \\nATOM   1017  O   ILE A 174      -7.001   1.595  21.023  1.00 16.08           O  \\nATOM   1018  CB  ILE A 174      -7.289   4.427  19.457  1.00 12.26           C  \\nATOM   1019  CG1 ILE A 174      -8.173   3.447  18.670  1.00 15.65           C  \\nATOM   1020  CG2 ILE A 174      -5.892   4.507  18.835  1.00 14.88           C  \\nATOM   1021  CD1 ILE A 174      -8.286   3.782  17.199  1.00 16.20           C  \\nATOM   1022  N   ALA A 175      -5.144   2.790  21.439  1.00 13.98           N  \\nATOM   1023  CA  ALA A 175      -4.342   1.593  21.694  1.00 15.21           C  \\nATOM   1024  C   ALA A 175      -4.877   0.822  22.898  1.00 14.90           C  \\nATOM   1025  O   ALA A 175      -4.938  -0.404  22.882  1.00 16.52           O  \\nATOM   1026  CB  ALA A 175      -2.903   1.960  21.890  1.00 15.11           C  \\nATOM   1027  N   GLU A 176      -5.322   1.540  23.922  1.00 16.94           N  \\nATOM   1028  CA  GLU A 176      -6.019   0.911  25.048  1.00 18.07           C  \\nATOM   1029  C   GLU A 176      -7.273   0.206  24.574  1.00 17.50           C  \\nATOM   1030  O   GLU A 176      -7.619  -0.871  25.056  1.00 21.18           O  \\nATOM   1031  CB  GLU A 176      -6.459   1.981  26.028  1.00 24.57           C  \\nATOM   1032  CG  GLU A 176      -5.495   2.309  27.099  1.00 28.65           C  \\nATOM   1033  CD  GLU A 176      -6.181   3.062  28.209  1.00 21.95           C  \\nATOM   1034  OE1 GLU A 176      -7.166   3.786  27.919  1.00 19.39           O  \\nATOM   1035  OE2 GLU A 176      -5.708   2.978  29.349  1.00 19.79           O  \\nATOM   1036  N   GLN A 177      -8.056   0.927  23.783  1.00 18.41           N  \\nATOM   1037  CA  GLN A 177      -9.399   0.501  23.424  1.00 16.95           C  \\nATOM   1038  C   GLN A 177      -9.385  -0.717  22.482  1.00 19.62           C  \\nATOM   1039  O   GLN A 177     -10.361  -1.467  22.412  1.00 21.33           O  \\nATOM   1040  CB  GLN A 177     -10.155   1.672  22.781  1.00 17.76           C  \\nATOM   1041  CG  GLN A 177     -10.467   2.805  23.757  1.00 21.19           C  \\nATOM   1042  CD  GLN A 177     -10.901   4.103  23.078  1.00 23.31           C  \\nATOM   1043  OE1 GLN A 177     -10.624   4.348  21.900  1.00 23.46           O  \\nATOM   1044  NE2 GLN A 177     -11.522   4.975  23.852  1.00 23.78           N  \\nATOM   1045  N   ASN A 178      -8.324  -0.852  21.691  1.00 17.29           N  \\nATOM   1046  CA  ASN A 178      -8.311  -1.795  20.573  1.00 17.91           C  \\nATOM   1047  C   ASN A 178      -7.011  -2.587  20.567  1.00 17.22           C  \\nATOM   1048  O   ASN A 178      -6.061  -2.238  19.868  1.00 16.35           O  \\nATOM   1049  CB  ASN A 178      -8.511  -1.071  19.240  1.00 15.82           C  \\nATOM   1050  CG  ASN A 178      -8.651  -2.031  18.065  1.00 17.24           C  \\nATOM   1051  OD1 ASN A 178      -8.386  -3.234  18.197  1.00 19.78           O  \\nATOM   1052  ND2 ASN A 178      -9.041  -1.499  16.900  1.00 20.95           N  \\nATOM   1053  N   PRO A 179      -6.991  -3.711  21.296  1.00 17.57           N  \\nATOM   1054  CA  PRO A 179      -5.778  -4.512  21.431  1.00 17.25           C  \\nATOM   1055  C   PRO A 179      -5.418  -5.273  20.157  1.00 17.14           C  \\nATOM   1056  O   PRO A 179      -4.368  -5.915  20.101  1.00 18.24           O  \\nATOM   1057  CB  PRO A 179      -6.114  -5.492  22.565  1.00 21.56           C  \\nATOM   1058  CG  PRO A 179      -7.587  -5.488  22.661  1.00 20.33           C  \\nATOM   1059  CD  PRO A 179      -8.033  -4.120  22.253  1.00 18.33           C  \\nATOM   1060  N   SER A 180      -6.282  -5.217  19.146  1.00 16.20           N  \\nATOM   1061  CA  SER A 180      -5.993  -5.875  17.877  1.00 16.56           C  \\nATOM   1062  C   SER A 180      -5.032  -5.060  17.001  1.00 14.08           C  \\nATOM   1063  O   SER A 180      -4.480  -5.578  16.027  1.00 14.61           O  \\nATOM   1064  CB  SER A 180      -7.284  -6.145  17.109  1.00 16.32           C  \\nATOM   1065  OG  SER A 180      -7.769  -4.962  16.495  1.00 23.67           O  \\nATOM   1066  N   LEU A 181      -4.832  -3.789  17.339  1.00 13.92           N  \\nATOM   1067  CA  LEU A 181      -3.915  -2.954  16.564  1.00 12.79           C  \\nATOM   1068  C   LEU A 181      -2.480  -3.483  16.657  1.00 12.45           C  \\nATOM   1069  O   LEU A 181      -2.058  -4.014  17.698  1.00 12.44           O  \\nATOM   1070  CB  LEU A 181      -3.978  -1.507  17.071  1.00 11.77           C  \\nATOM   1071  CG  LEU A 181      -5.292  -0.775  16.785  1.00 13.07           C  \\nATOM   1072  CD1 LEU A 181      -5.323   0.557  17.504  1.00 16.74           C  \\nATOM   1073  CD2 LEU A 181      -5.453  -0.572  15.296  1.00 17.98           C  \\nATOM   1074  N   LYS A 182      -1.740  -3.376  15.556  1.00 11.64           N  \\nATOM   1075  CA  LYS A 182      -0.384  -3.928  15.529  1.00 12.63           C  \\nATOM   1076  C   LYS A 182       0.638  -3.012  16.198  1.00 11.88           C  \\nATOM   1077  O   LYS A 182       1.663  -3.484  16.687  1.00 13.28           O  \\nATOM   1078  CB  LYS A 182       0.078  -4.243  14.099  1.00 12.15           C  \\nATOM   1079  CG  LYS A 182      -0.842  -5.161  13.280  1.00 13.97           C  \\nATOM   1080  CD  LYS A 182      -1.142  -6.447  14.025  1.00 19.87           C  \\nATOM   1081  CE  LYS A 182       0.121  -7.244  14.270  1.00 21.99           C  \\nATOM   1082  NZ  LYS A 182      -0.198  -8.681  14.572  1.00 24.10           N  \\nATOM   1083  N   ALA A 183       0.408  -1.698  16.136  1.00 11.72           N  \\nATOM   1084  CA  ALA A 183       1.396  -0.739  16.607  1.00 12.23           C  \\nATOM   1085  C   ALA A 183       0.725   0.612  16.794  1.00 10.64           C  \\nATOM   1086  O   ALA A 183      -0.293   0.906  16.151  1.00 12.16           O  \\nATOM   1087  CB  ALA A 183       2.554  -0.619  15.610  1.00 12.50           C  \\nATOM   1088  N   ALA A 184       1.364   1.452  17.610  1.00 11.39           N  \\nATOM   1089  CA  ALA A 184       0.948   2.836  17.833  1.00 11.29           C  \\nATOM   1090  C   ALA A 184       2.193   3.704  17.915  1.00 12.54           C  \\nATOM   1091  O   ALA A 184       3.161   3.358  18.596  1.00 12.77           O  \\nATOM   1092  CB  ALA A 184       0.164   2.930  19.136  1.00 13.38           C  \\nATOM   1093  N   VAL A 185       2.172   4.833  17.208  1.00 10.19           N  \\nATOM   1094  CA  VAL A 185       3.272   5.798  17.265  1.00 10.41           C  \\nATOM   1095  C   VAL A 185       2.742   7.190  17.622  1.00 10.58           C  \\nATOM   1096  O   VAL A 185       2.349   7.968  16.743  1.00 11.35           O  \\nATOM   1097  CB  VAL A 185       4.069   5.858  15.944  1.00 10.81           C  \\nATOM   1098  CG1 VAL A 185       5.185   6.902  16.048  1.00 13.31           C  \\nATOM   1099  CG2 VAL A 185       4.651   4.466  15.614  1.00 11.15           C  \\nATOM   1100  N   PRO A 186       2.777   7.520  18.916  1.00 11.18           N  \\nATOM   1101  CA  PRO A 186       2.457   8.886  19.342  1.00 11.18           C  \\nATOM   1102  C   PRO A 186       3.630   9.829  19.112  1.00 11.41           C  \\nATOM   1103  O   PRO A 186       4.726   9.595  19.613  1.00 12.12           O  \\nATOM   1104  CB  PRO A 186       2.164   8.744  20.850  1.00 10.92           C  \\nATOM   1105  CG  PRO A 186       1.940   7.208  21.059  1.00 11.19           C  \\nATOM   1106  CD  PRO A 186       2.875   6.590  20.058  1.00 11.05           C  \\nATOM   1107  N   LEU A 187       3.400  10.857  18.298  1.00 10.93           N  \\nATOM   1108  CA  LEU A 187       4.425  11.848  17.973  1.00 11.20           C  \\nATOM   1109  C   LEU A 187       4.300  13.166  18.715  1.00 11.70           C  \\nATOM   1110  O   LEU A 187       3.387  13.948  18.434  1.00 12.10           O  \\nATOM   1111  CB  LEU A 187       4.340  12.226  16.495  1.00 11.57           C  \\nATOM   1112  CG  LEU A 187       4.468  11.071  15.502  1.00 10.71           C  \\nATOM   1113  CD1 LEU A 187       4.121  11.565  14.107  1.00 15.77           C  \\nATOM   1114  CD2 LEU A 187       5.878  10.507  15.535  1.00 15.13           C  \\nATOM   1115  N   THR A 188       5.209  13.474  19.531  1.00 11.54           N  \\nATOM   1116  CA  THR A 188       5.194  14.570  20.508  1.00 11.47           C  \\nATOM   1117  C   THR A 188       3.824  14.689  21.176  1.00 13.23           C  \\nATOM   1118  O   THR A 188       3.214  15.751  21.198  1.00 13.47           O  \\nATOM   1119  CB  THR A 188       5.605  15.928  19.911  1.00 20.00           C  \\nATOM   1120  OG1 THR A 188       4.702  16.284  18.859  1.00 20.00           O  \\nATOM   1121  CG2 THR A 188       6.945  15.815  19.200  1.00 20.00           C  \\nATOM   1122  N   PRO A 189       3.425  13.604  21.854  1.00 11.17           N  \\nATOM   1123  CA  PRO A 189       2.031  13.446  22.267  1.00 11.81           C  \\nATOM   1124  C   PRO A 189       1.645  14.371  23.423  1.00 12.06           C  \\nATOM   1125  O   PRO A 189       2.501  14.770  24.217  1.00 11.32           O  \\nATOM   1126  CB  PRO A 189       1.979  11.978  22.722  1.00 12.95           C  \\nATOM   1127  CG  PRO A 189       3.374  11.694  23.169  1.00 11.91           C  \\nATOM   1128  CD  PRO A 189       4.239  12.415  22.165  1.00 11.73           C  \\nATOM   1129  N   TRP A 190       0.353  14.660  23.538  1.00 11.97           N  \\nATOM   1130  CA  TRP A 190      -0.192  15.481  24.631  1.00 12.66           C  \\nATOM   1131  C   TRP A 190      -1.428  14.760  25.140  1.00 11.64           C  \\nATOM   1132  O   TRP A 190      -2.287  14.361  24.343  1.00 12.90           O  \\nATOM   1133  CB  TRP A 190      -0.586  16.859  24.070  1.00 12.75           C  \\nATOM   1134  CG  TRP A 190      -1.007  17.893  25.098  1.00 13.85           C  \\nATOM   1135  CD1 TRP A 190      -0.307  19.016  25.455  1.00 14.45           C  \\nATOM   1136  CD2 TRP A 190      -2.301  18.016  25.721  1.00 13.89           C  \\nATOM   1137  NE1 TRP A 190      -1.062  19.797  26.313  1.00 14.91           N  \\nATOM   1138  CE2 TRP A 190      -2.278  19.193  26.504  1.00 14.93           C  \\nATOM   1139  CE3 TRP A 190      -3.472  17.245  25.691  1.00 14.32           C  \\nATOM   1140  CZ2 TRP A 190      -3.369  19.600  27.272  1.00 16.85           C  \\nATOM   1141  CZ3 TRP A 190      -4.557  17.651  26.450  1.00 13.92           C  \\nATOM   1142  CH2 TRP A 190      -4.509  18.840  27.202  1.00 15.47           C  \\nATOM   1143  N   HIS A 191      -1.536  14.646  26.463  1.00 13.40           N  \\nATOM   1144  CA  HIS A 191      -2.751  14.156  27.109  1.00 12.47           C  \\nATOM   1145  C   HIS A 191      -2.669  14.446  28.604  1.00 13.92           C  \\nATOM   1146  O   HIS A 191      -1.576  14.431  29.188  1.00 14.91           O  \\nATOM   1147  CB  HIS A 191      -2.901  12.643  26.891  1.00 12.22           C  \\nATOM   1148  CG  HIS A 191      -4.317  12.167  26.911  1.00 13.31           C  \\nATOM   1149  ND1 HIS A 191      -5.221  12.558  27.877  1.00 20.06           N  \\nATOM   1150  CD2 HIS A 191      -4.967  11.272  26.129  1.00 13.78           C  \\nATOM   1151  CE1 HIS A 191      -6.400  12.026  27.607  1.00 19.67           C  \\nATOM   1152  NE2 HIS A 191      -6.251  11.170  26.609  1.00 13.66           N  \\nATOM   1153  N   THR A 192      -3.817  14.699  29.228  1.00 15.39           N  \\nATOM   1154  CA  THR A 192      -3.835  14.912  30.671  1.00 15.37           C  \\nATOM   1155  C   THR A 192      -3.849  13.603  31.448  1.00 15.35           C  \\nATOM   1156  O   THR A 192      -3.305  13.529  32.545  1.00 17.10           O  \\nATOM   1157  CB  THR A 192      -5.038  15.757  31.099  1.00 16.13           C  \\nATOM   1158  OG1 THR A 192      -6.224  15.181  30.552  1.00 17.05           O  \\nATOM   1159  CG2 THR A 192      -4.889  17.199  30.574  1.00 17.23           C  \\nATOM   1160  N   ASP A 193      -4.307  12.541  30.794  1.00 13.69           N  \\nATOM   1161  CA  ASP A 193      -4.171  11.191  31.340  1.00 15.39           C  \\nATOM   1162  C   ASP A 193      -2.766  10.681  31.036  1.00 16.25           C  \\nATOM   1163  O   ASP A 193      -2.313  10.755  29.895  1.00 16.36           O  \\nATOM   1164  CB  ASP A 193      -5.227  10.275  30.713  1.00 19.17           C  \\nATOM   1165  CG  ASP A 193      -5.075   8.821  31.134  1.00 30.88           C  \\nATOM   1166  OD1 ASP A 193      -4.449   8.549  32.184  1.00 27.86           O  \\nATOM   1167  OD2 ASP A 193      -5.709   7.963  30.481  1.00 41.54           O  \\nATOM   1168  N   LYS A 194      -2.060  10.215  32.063  1.00 15.12           N  \\nATOM   1169  CA  LYS A 194      -0.662   9.831  31.910  1.00 15.82           C  \\nATOM   1170  C   LYS A 194      -0.452   8.314  31.819  1.00 15.90           C  \\nATOM   1171  O   LYS A 194       0.641   7.854  31.485  1.00 15.53           O  \\nATOM   1172  CB  LYS A 194       0.194  10.412  33.036  1.00 13.96           C  \\nATOM   1173  CG  LYS A 194      -0.011  11.896  33.312  1.00 16.61           C  \\nATOM   1174  CD  LYS A 194       0.269  12.756  32.095  1.00 17.47           C  \\nATOM   1175  CE  LYS A 194      -0.009  14.223  32.408  1.00 16.47           C  \\nATOM   1176  NZ  LYS A 194       0.175  15.081  31.186  1.00 16.69           N  \\nATOM   1177  N   THR A 195      -1.508   7.540  32.058  1.00 14.80           N  \\nATOM   1178  CA  THR A 195      -1.369   6.094  32.176  1.00 14.43           C  \\nATOM   1179  C   THR A 195      -2.271   5.360  31.199  1.00 14.45           C  \\nATOM   1180  O   THR A 195      -3.502   5.444  31.287  1.00 17.24           O  \\nATOM   1181  CB  THR A 195      -1.700   5.637  33.601  1.00 15.27           C  \\nATOM   1182  OG1 THR A 195      -0.846   6.325  34.524  1.00 18.45           O  \\nATOM   1183  CG2 THR A 195      -1.503   4.124  33.735  1.00 18.77           C  \\nATOM   1184  N   PHE A 196      -1.666   4.622  30.276  1.00 14.28           N  \\nATOM   1185  CA  PHE A 196      -2.424   3.926  29.245  1.00 14.26           C  \\nATOM   1186  C   PHE A 196      -2.169   2.440  29.324  1.00 15.83           C  \\nATOM   1187  O   PHE A 196      -1.029   1.977  29.257  1.00 18.61           O  \\nATOM   1188  CB  PHE A 196      -2.070   4.468  27.853  1.00 13.48           C  \\nATOM   1189  CG  PHE A 196      -2.450   5.906  27.663  1.00 14.72           C  \\nATOM   1190  CD1 PHE A 196      -3.747   6.241  27.325  1.00 13.32           C  \\nATOM   1191  CD2 PHE A 196      -1.559   6.915  27.979  1.00 15.24           C  \\nATOM   1192  CE1 PHE A 196      -4.148   7.568  27.271  1.00 15.06           C  \\nATOM   1193  CE2 PHE A 196      -1.960   8.235  27.952  1.00 13.18           C  \\nATOM   1194  CZ  PHE A 196      -3.254   8.553  27.610  1.00 14.24           C  \\nATOM   1195  N   ASN A 197      -3.233   1.692  29.534  1.00 14.46           N  \\nATOM   1196  CA  ASN A 197      -3.111   0.279  29.777  1.00 14.55           C  \\nATOM   1197  C   ASN A 197      -3.363  -0.457  28.476  1.00 14.59           C  \\nATOM   1198  O   ASN A 197      -4.501  -0.750  28.123  1.00 18.25           O  \\nATOM   1199  CB  ASN A 197      -4.106  -0.126  30.865  1.00 14.41           C  \\nATOM   1200  CG  ASN A 197      -3.938  -1.548  31.308  1.00 19.85           C  \\nATOM   1201  OD1 ASN A 197      -3.143  -2.297  30.741  1.00 22.89           O  \\nATOM   1202  ND2 ASN A 197      -4.797  -1.976  32.232  1.00 25.51           N  \\nATOM   1203  N   THR A 198      -2.316  -0.539  27.663  1.00 13.16           N  \\nATOM   1204  CA  THR A 198      -2.434  -1.111  26.319  1.00 14.67           C  \\nATOM   1205  C   THR A 198      -1.413  -2.215  26.142  1.00 14.66           C  \\nATOM   1206  O   THR A 198      -0.326  -2.151  26.729  1.00 15.40           O  \\nATOM   1207  CB  THR A 198      -2.241  -0.051  25.192  1.00 15.56           C  \\nATOM   1208  OG1 THR A 198      -2.326  -0.693  23.912  1.00 15.15           O  \\nATOM   1209  CG2 THR A 198      -0.895   0.658  25.313  1.00 17.94           C  \\nATOM   1210  N   SER A 199      -1.777  -3.229  25.360  1.00 13.15           N  \\nATOM   1211  CA  SER A 199      -0.847  -4.281  24.959  1.00 13.30           C  \\nATOM   1212  C   SER A 199      -0.313  -4.079  23.535  1.00 13.92           C  \\nATOM   1213  O   SER A 199       0.398  -4.930  22.994  1.00 15.13           O  \\nATOM   1214  CB  SER A 199      -1.536  -5.646  25.062  1.00 12.71           C  \\nATOM   1215  OG  SER A 199      -2.773  -5.660  24.364  1.00 16.84           O  \\nATOM   1216  N   VAL A 200      -0.679  -2.958  22.914  1.00 12.85           N  \\nATOM   1217  CA  VAL A 200      -0.200  -2.645  21.572  1.00 13.67           C  \\nATOM   1218  C   VAL A 200       1.208  -2.050  21.666  1.00 12.82           C  \\nATOM   1219  O   VAL A 200       1.435  -1.137  22.448  1.00 13.66           O  \\nATOM   1220  CB  VAL A 200      -1.136  -1.608  20.916  1.00 10.23           C  \\nATOM   1221  CG1 VAL A 200      -0.624  -1.194  19.555  1.00 13.33           C  \\nATOM   1222  CG2 VAL A 200      -2.551  -2.164  20.800  1.00 14.72           C  \\nATOM   1223  N   PRO A 201       2.152  -2.544  20.849  1.00 11.02           N  \\nATOM   1224  CA  PRO A 201       3.506  -1.969  20.855  1.00 11.42           C  \\nATOM   1225  C   PRO A 201       3.522  -0.470  20.548  1.00 10.64           C  \\nATOM   1226  O   PRO A 201       2.941  -0.024  19.543  1.00 12.55           O  \\nATOM   1227  CB  PRO A 201       4.208  -2.724  19.732  1.00 11.54           C  \\nATOM   1228  CG  PRO A 201       3.535  -4.096  19.750  1.00 11.49           C  \\nATOM   1229  CD  PRO A 201       2.082  -3.781  20.035  1.00 11.46           C  \\nATOM   1230  N   VAL A 202       4.170   0.296  21.424  1.00 10.67           N  \\nATOM   1231  CA  VAL A 202       4.130   1.755  21.354  1.00 12.12           C  \\nATOM   1232  C   VAL A 202       5.553   2.287  21.193  1.00 12.04           C  \\nATOM   1233  O   VAL A 202       6.418   1.974  21.999  1.00 11.96           O  \\nATOM   1234  CB  VAL A 202       3.558   2.331  22.655  1.00 10.58           C  \\nATOM   1235  CG1 VAL A 202       3.724   3.851  22.675  1.00 13.83           C  \\nATOM   1236  CG2 VAL A 202       2.081   1.979  22.807  1.00 12.92           C  \\nATOM   1237  N   LEU A 203       5.759   3.153  20.196  1.00 11.16           N  \\nATOM   1238  CA  LEU A 203       6.968   3.964  20.122  1.00 10.70           C  \\nATOM   1239  C   LEU A 203       6.588   5.436  20.146  1.00 10.31           C  \\nATOM   1240  O   LEU A 203       5.800   5.893  19.313  1.00 11.65           O  \\nATOM   1241  CB  LEU A 203       7.773   3.650  18.854  1.00 10.89           C  \\nATOM   1242  CG  LEU A 203       9.094   4.417  18.721  1.00 10.45           C  \\nATOM   1243  CD1 LEU A 203      10.105   3.627  17.903  1.00 11.91           C  \\nATOM   1244  CD2 LEU A 203       8.895   5.806  18.101  1.00 13.96           C  \\nATOM   1245  N   ILE A 204       7.019   6.115  21.202  1.00 11.42           N  \\nATOM   1246  CA  ILE A 204       6.737   7.543  21.383  1.00 11.26           C  \\nATOM   1247  C   ILE A 204       7.928   8.358  20.945  1.00 11.89           C  \\nATOM   1248  O   ILE A 204       9.072   8.056  21.314  1.00 12.16           O  \\nATOM   1249  CB  ILE A 204       6.444   7.856  22.863  1.00 10.91           C  \\nATOM   1250  CG1 ILE A 204       5.141   7.189  23.322  1.00 11.23           C  \\nATOM   1251  CG2 ILE A 204       6.386   9.377  23.098  1.00 12.88           C  \\nATOM   1252  CD1 ILE A 204       5.076   6.933  24.833  1.00 15.36           C  \\nATOM   1253  N   VAL A 205       7.672   9.375  20.120  1.00 11.62           N  \\nATOM   1254  CA  VAL A 205       8.694  10.386  19.820  1.00 11.80           C  \\nATOM   1255  C   VAL A 205       8.446  11.602  20.694  1.00 12.82           C  \\nATOM   1256  O   VAL A 205       7.388  12.222  20.614  1.00 14.31           O  \\nATOM   1257  CB  VAL A 205       8.642  10.815  18.345  1.00 12.08           C  \\nATOM   1258  CG1 VAL A 205       9.667  11.911  18.073  1.00 13.30           C  \\nATOM   1259  CG2 VAL A 205       8.890   9.609  17.428  1.00 13.77           C  \\nATOM   1260  N   GLY A 206       9.436  11.959  21.504  1.00 12.29           N  \\nATOM   1261  CA  GLY A 206       9.412  13.245  22.211  1.00 13.24           C  \\nATOM   1262  C   GLY A 206      10.312  14.259  21.519  1.00 13.59           C  \\nATOM   1263  O   GLY A 206      11.221  13.886  20.763  1.00 12.87           O  \\nATOM   1264  N   ALA A 207      10.011  15.543  21.714  1.00 12.77           N  \\nATOM   1265  CA  ALA A 207      10.881  16.621  21.237  1.00 14.48           C  \\nATOM   1266  C   ALA A 207      11.503  17.336  22.433  1.00 13.52           C  \\nATOM   1267  O   ALA A 207      10.792  17.837  23.302  1.00 14.74           O  \\nATOM   1268  CB  ALA A 207      10.081  17.602  20.379  1.00 14.12           C  \\nATOM   1269  N   GLU A 208      12.825  17.266  22.547  1.00 14.92           N  \\nATOM   1270  CA  GLU A 208      13.475  17.691  23.783  1.00 15.58           C  \\nATOM   1271  C   GLU A 208      12.967  19.058  24.231  1.00 14.92           C  \\nATOM   1272  O   GLU A 208      12.730  19.272  25.413  1.00 16.18           O  \\nATOM   1273  CB  GLU A 208      14.997  17.723  23.638  1.00 15.18           C  \\nATOM   1274  CG  GLU A 208      15.701  17.942  24.969  1.00 19.06           C  \\nATOM   1275  CD  GLU A 208      17.217  17.980  24.850  1.00 21.51           C  \\nATOM   1276  OE1 GLU A 208      17.732  18.269  23.756  1.00 20.34           O  \\nATOM   1277  OE2 GLU A 208      17.889  17.797  25.884  1.00 26.51           O  \\nATOM   1278  N   ALA A 209      12.909  20.006  23.295  1.00 16.76           N  \\nATOM   1279  CA  ALA A 209      12.605  21.400  23.630  1.00 17.31           C  \\nATOM   1280  C   ALA A 209      11.128  21.749  23.482  1.00 19.50           C  \\nATOM   1281  O   ALA A 209      10.762  22.920  23.435  1.00 19.62           O  \\nATOM   1282  CB  ALA A 209      13.448  22.338  22.783  1.00 16.74           C  \\nATOM   1283  N   ASP A 210      10.270  20.736  23.476  1.00 16.08           N  \\nATOM   1284  CA  ASP A 210       8.844  20.939  23.241  1.00 14.87           C  \\nATOM   1285  C   ASP A 210       8.283  21.831  24.340  1.00 16.14           C  \\nATOM   1286  O   ASP A 210       8.425  21.526  25.517  1.00 17.30           O  \\nATOM   1287  CB  ASP A 210       8.138  19.573  23.301  1.00 14.52           C  \\nATOM   1288  CG  ASP A 210       6.768  19.580  22.641  1.00 15.16           C  \\nATOM   1289  OD1 ASP A 210       6.052  20.602  22.719  1.00 14.74           O  \\nATOM   1290  OD2 ASP A 210       6.366  18.532  22.092  1.00 14.08           O  \\nATOM   1291  N   THR A 211       7.630  22.914  23.935  1.00 14.27           N  \\nATOM   1292  CA  THR A 211       6.963  23.810  24.889  1.00 15.04           C  \\nATOM   1293  C   THR A 211       5.453  23.625  24.914  1.00 15.44           C  \\nATOM   1294  O   THR A 211       4.777  24.159  25.802  1.00 17.34           O  \\nATOM   1295  CB  THR A 211       7.257  25.296  24.577  1.00 16.01           C  \\nATOM   1296  OG1 THR A 211       6.830  25.592  23.242  1.00 17.13           O  \\nATOM   1297  CG2 THR A 211       8.736  25.612  24.724  1.00 17.95           C  \\nATOM   1298  N   VAL A 212       4.914  22.902  23.936  1.00 13.06           N  \\nATOM   1299  CA  VAL A 212       3.472  22.672  23.840  1.00 13.74           C  \\nATOM   1300  C   VAL A 212       3.086  21.443  24.672  1.00 15.19           C  \\nATOM   1301  O   VAL A 212       2.135  21.474  25.470  1.00 15.25           O  \\nATOM   1302  CB  VAL A 212       3.052  22.446  22.377  1.00 15.64           C  \\nATOM   1303  CG1 VAL A 212       1.594  22.047  22.292  1.00 17.94           C  \\nATOM   1304  CG2 VAL A 212       3.342  23.697  21.544  1.00 16.27           C  \\nATOM   1305  N   ALA A 213       3.867  20.374  24.512  1.00 13.87           N  \\nATOM   1306  CA  ALA A 213       3.661  19.148  25.283  1.00 13.06           C  \\nATOM   1307  C   ALA A 213       4.988  18.744  25.921  1.00 13.86           C  \\nATOM   1308  O   ALA A 213       5.690  17.873  25.413  1.00 13.70           O  \\nATOM   1309  CB  ALA A 213       3.139  18.024  24.361  1.00 12.73           C  \\nATOM   1310  N   PRO A 214       5.430  19.489  26.934  1.00 13.40           N  \\nATOM   1311  CA  PRO A 214       6.774  19.251  27.452  1.00 13.09           C  \\nATOM   1312  C   PRO A 214       6.991  17.785  27.803  1.00 14.09           C  \\nATOM   1313  O   PRO A 214       6.117  17.165  28.402  1.00 14.10           O  \\nATOM   1314  CB  PRO A 214       6.805  20.103  28.724  1.00 16.34           C  \\nATOM   1315  CG  PRO A 214       5.862  21.206  28.443  1.00 15.88           C  \\nATOM   1316  CD  PRO A 214       4.757  20.613  27.612  1.00 16.50           C  \\nATOM   1317  N   VAL A 215       8.156  17.250  27.459  1.00 12.48           N  \\nATOM   1318  CA  VAL A 215       8.360  15.798  27.578  1.00 13.09           C  \\nATOM   1319  C   VAL A 215       8.345  15.321  29.032  1.00 13.90           C  \\nATOM   1320  O   VAL A 215       7.891  14.220  29.308  1.00 15.25           O  \\nATOM   1321  CB  VAL A 215       9.636  15.299  26.853  1.00 12.90           C  \\nATOM   1322  CG1 VAL A 215       9.534  15.555  25.355  1.00 14.15           C  \\nATOM   1323  CG2 VAL A 215      10.890  15.940  27.432  1.00 15.47           C  \\nATOM   1324  N   SER A 216       8.706  16.194  29.974  1.00 15.46           N  \\nATOM   1325  CA  SER A 216       8.652  15.825  31.398  1.00 14.83           C  \\nATOM   1326  C   SER A 216       7.226  15.623  31.903  1.00 16.24           C  \\nATOM   1327  O   SER A 216       7.005  14.893  32.875  1.00 16.57           O  \\nATOM   1328  CB  SER A 216       9.363  16.873  32.263  1.00 19.67           C  \\nATOM   1329  OG  SER A 216       8.633  18.085  32.277  1.00 23.97           O  \\nATOM   1330  N   GLN A 217       6.258  16.219  31.211  1.00 15.78           N  \\nATOM   1331  CA  GLN A 217       4.870  16.188  31.637  1.00 16.56           C  \\nATOM   1332  C   GLN A 217       3.998  15.265  30.800  1.00 15.27           C  \\nATOM   1333  O   GLN A 217       2.906  14.879  31.216  1.00 17.01           O  \\nATOM   1334  CB  GLN A 217       4.287  17.601  31.656  1.00 15.97           C  \\nATOM   1335  CG  GLN A 217       4.871  18.446  32.775  1.00 24.82           C  \\nATOM   1336  CD  GLN A 217       4.599  19.923  32.598  1.00 39.63           C  \\nATOM   1337  OE1 GLN A 217       5.298  20.607  31.850  1.00 37.51           O  \\nATOM   1338  NE2 GLN A 217       3.673  20.450  33.394  1.00 48.52           N  \\nATOM   1339  N   HIS A 218       4.385  14.942  29.608  1.00 13.21           N  \\nATOM   1340  CA  HIS A 218       3.621  14.173  28.628  1.00 13.20           C  \\nATOM   1341  C   HIS A 218       4.418  12.974  28.103  1.00 12.89           C  \\nATOM   1342  O   HIS A 218       4.424  11.916  28.738  1.00 13.69           O  \\nATOM   1343  CB  HIS A 218       3.196  15.068  27.463  1.00 20.00           C  \\nATOM   1344  CG  HIS A 218       2.222  16.137  27.847  1.00 20.00           C  \\nATOM   1345  ND1 HIS A 218       2.613  17.338  28.401  1.00 20.00           N  \\nATOM   1346  CD2 HIS A 218       0.872  16.189  27.755  1.00 20.00           C  \\nATOM   1347  CE1 HIS A 218       1.547  18.081  28.635  1.00 20.00           C  \\nATOM   1348  NE2 HIS A 218       0.477  17.409  28.253  1.00 20.00           N  \\nATOM   1349  N   ALA A 219       5.252  13.059  27.166  1.00 12.26           N  \\nATOM   1350  CA  ALA A 219       5.910  11.895  26.558  1.00 12.73           C  \\nATOM   1351  C   ALA A 219       6.533  10.942  27.564  1.00 14.21           C  \\nATOM   1352  O   ALA A 219       6.317   9.726  27.480  1.00 13.41           O  \\nATOM   1353  CB  ALA A 219       6.942  12.346  25.543  1.00 12.39           C  \\nATOM   1354  N   ILE A 220       7.324  11.478  28.492  1.00 13.27           N  \\nATOM   1355  CA  ILE A 220       8.060  10.633  29.434  1.00 14.36           C  \\nATOM   1356  C   ILE A 220       7.166   9.879  30.425  1.00 15.12           C  \\nATOM   1357  O   ILE A 220       7.208   8.643  30.484  1.00 14.11           O  \\nATOM   1358  CB  ILE A 220       9.192  11.399  30.120  1.00 14.19           C  \\nATOM   1359  CG1 ILE A 220      10.259  11.709  29.071  1.00 15.86           C  \\nATOM   1360  CG2 ILE A 220       9.810  10.548  31.228  1.00 14.80           C  \\nATOM   1361  CD1 ILE A 220      11.324  12.683  29.540  1.00 16.14           C  \\nATOM   1362  N   PRO A 221       6.191  10.578  31.027  1.00 12.13           N  \\nATOM   1363  CA  PRO A 221       5.230   9.836  31.836  1.00 12.42           C  \\nATOM   1364  C   PRO A 221       4.433   8.816  31.052  1.00 11.13           C  \\nATOM   1365  O   PRO A 221       4.239   7.702  31.548  1.00 13.20           O  \\nATOM   1366  CB  PRO A 221       4.324  10.935  32.408  1.00 13.37           C  \\nATOM   1367  CG  PRO A 221       5.230  12.119  32.492  1.00 15.10           C  \\nATOM   1368  CD  PRO A 221       6.106  12.032  31.268  1.00 13.36           C  \\nATOM   1369  N   PHE A 222       4.039   9.123  29.810  1.00 13.07           N  \\nATOM   1370  CA  PHE A 222       3.339   8.107  29.000  1.00 11.14           C  \\nATOM   1371  C   PHE A 222       4.202   6.866  28.842  1.00 12.77           C  \\nATOM   1372  O   PHE A 222       3.738   5.746  29.049  1.00 13.00           O  \\nATOM   1373  CB  PHE A 222       2.955   8.603  27.601  1.00 13.45           C  \\nATOM   1374  CG  PHE A 222       2.170   9.884  27.581  1.00 13.38           C  \\nATOM   1375  CD1 PHE A 222       1.355  10.253  28.651  1.00 14.69           C  \\nATOM   1376  CD2 PHE A 222       2.306  10.760  26.513  1.00 13.37           C  \\nATOM   1377  CE1 PHE A 222       0.609  11.417  28.597  1.00 16.94           C  \\nATOM   1378  CE2 PHE A 222       1.604  11.961  26.471  1.00 13.69           C  \\nATOM   1379  CZ  PHE A 222       0.772  12.306  27.527  1.00 17.25           C  \\nATOM   1380  N   TYR A 223       5.506   7.163  28.413  1.00 11.59           N  \\nATOM   1381  CA  TYR A 223       6.334   5.964  28.321  1.00 12.85           C  \\nATOM   1382  C   TYR A 223       6.337   5.189  29.619  1.00 12.79           C  \\nATOM   1383  O   TYR A 223       6.077   3.981  29.625  1.00 14.46           O  \\nATOM   1384  CB  TYR A 223       7.766   6.331  27.928  1.00 20.00           C  \\nATOM   1385  CG  TYR A 223       8.717   5.155  27.903  1.00 20.00           C  \\nATOM   1386  CD1 TYR A 223       8.706   4.252  26.846  1.00 20.00           C  \\nATOM   1387  CD2 TYR A 223       9.623   4.950  28.933  1.00 20.00           C  \\nATOM   1388  CE1 TYR A 223       9.572   3.177  26.819  1.00 20.00           C  \\nATOM   1389  CE2 TYR A 223      10.495   3.876  28.914  1.00 20.00           C  \\nATOM   1390  CZ  TYR A 223      10.465   2.994  27.855  1.00 20.00           C  \\nATOM   1391  OH  TYR A 223      11.328   1.922  27.831  1.00 20.00           O  \\nATOM   1392  N   GLN A 224       6.727   5.847  30.679  1.00 12.16           N  \\nATOM   1393  CA  GLN A 224       7.019   5.174  31.938  1.00 13.90           C  \\nATOM   1394  C   GLN A 224       5.845   4.333  32.407  1.00 14.27           C  \\nATOM   1395  O   GLN A 224       6.041   3.303  33.040  1.00 17.63           O  \\nATOM   1396  CB  GLN A 224       7.375   6.190  33.015  1.00 14.30           C  \\nATOM   1397  CG  GLN A 224       8.733   6.827  32.816  1.00 14.69           C  \\nATOM   1398  CD  GLN A 224       8.995   7.931  33.797  1.00 16.62           C  \\nATOM   1399  OE1 GLN A 224       8.057   8.544  34.314  1.00 22.84           O  \\nATOM   1400  NE2 GLN A 224      10.264   8.239  34.017  1.00 21.42           N  \\nATOM   1401  N   ASN A 225       4.634   4.810  32.158  1.00 14.71           N  \\nATOM   1402  CA  ASN A 225       3.451   4.203  32.764  1.00 14.26           C  \\nATOM   1403  C   ASN A 225       2.760   3.161  31.902  1.00 15.33           C  \\nATOM   1404  O   ASN A 225       1.788   2.543  32.339  1.00 16.76           O  \\nATOM   1405  CB  ASN A 225       2.435   5.269  33.184  1.00 14.55           C  \\nATOM   1406  CG  ASN A 225       2.978   6.226  34.226  1.00 19.09           C  \\nATOM   1407  OD1 ASN A 225       3.681   5.826  35.147  1.00 22.80           O  \\nATOM   1408  ND2 ASN A 225       2.507   7.467  34.176  1.00 18.49           N  \\nATOM   1409  N   LEU A 226       3.229   2.968  30.674  1.00 14.79           N  \\nATOM   1410  CA  LEU A 226       2.760   1.843  29.865  1.00 15.57           C  \\nATOM   1411  C   LEU A 226       3.143   0.539  30.554  1.00 14.36           C  \\nATOM   1412  O   LEU A 226       4.163   0.474  31.229  1.00 15.75           O  \\nATOM   1413  CB  LEU A 226       3.396   1.877  28.478  1.00 15.47           C  \\nATOM   1414  CG  LEU A 226       2.910   2.955  27.502  1.00 14.92           C  \\nATOM   1415  CD1 LEU A 226       3.921   3.071  26.367  1.00 17.80           C  \\nATOM   1416  CD2 LEU A 226       1.542   2.624  26.939  1.00 18.00           C  \\nATOM   1417  N   PRO A 227       2.339  -0.519  30.368  1.00 14.23           N  \\nATOM   1418  CA  PRO A 227       2.670  -1.777  31.040  1.00 14.15           C  \\nATOM   1419  C   PRO A 227       4.093  -2.259  30.749  1.00 12.90           C  \\nATOM   1420  O   PRO A 227       4.567  -2.171  29.613  1.00 13.39           O  \\nATOM   1421  CB  PRO A 227       1.653  -2.760  30.465  1.00 14.31           C  \\nATOM   1422  CG  PRO A 227       0.492  -1.896  30.003  1.00 16.99           C  \\nATOM   1423  CD  PRO A 227       1.153  -0.639  29.508  1.00 14.29           C  \\nATOM   1424  N   SER A 228       4.727  -2.869  31.743  1.00 13.58           N  \\nATOM   1425  CA  SER A 228       6.090  -3.357  31.583  1.00 13.34           C  \\nATOM   1426  C   SER A 228       6.197  -4.533  30.630  1.00 12.75           C  \\nATOM   1427  O   SER A 228       7.243  -4.733  29.996  1.00 13.76           O  \\nATOM   1428  CB  SER A 228       6.688  -3.718  32.932  1.00 14.32           C  \\nATOM   1429  OG  SER A 228       6.789  -2.542  33.718  1.00 19.93           O  \\nATOM   1430  N   THR A 229       5.058  -5.177  30.382  1.00 12.69           N  \\nATOM   1431  CA  THR A 229       4.993  -6.317  29.488  1.00 12.66           C  \\nATOM   1432  C   THR A 229       4.679  -5.931  28.034  1.00 11.40           C  \\nATOM   1433  O   THR A 229       4.667  -6.791  27.155  1.00 13.06           O  \\nATOM   1434  CB  THR A 229       3.934  -7.327  29.975  1.00 13.83           C  \\nATOM   1435  OG1 THR A 229       2.729  -6.632  30.313  1.00 15.60           O  \\nATOM   1436  CG2 THR A 229       4.437  -8.096  31.211  1.00 16.94           C  \\nATOM   1437  N   THR A 230       4.417  -4.647  27.770  1.00 12.80           N  \\nATOM   1438  CA  THR A 230       4.119  -4.196  26.398  1.00 11.90           C  \\nATOM   1439  C   THR A 230       5.384  -3.632  25.747  1.00 12.65           C  \\nATOM   1440  O   THR A 230       6.066  -2.801  26.346  1.00 12.26           O  \\nATOM   1441  CB  THR A 230       3.082  -3.087  26.430  1.00 10.26           C  \\nATOM   1442  OG1 THR A 230       1.872  -3.591  27.027  1.00 12.31           O  \\nATOM   1443  CG2 THR A 230       2.802  -2.552  25.033  1.00 13.59           C  \\nATOM   1444  N   PRO A 231       5.733  -4.116  24.546  1.00 11.15           N  \\nATOM   1445  CA  PRO A 231       6.909  -3.544  23.878  1.00 10.81           C  \\nATOM   1446  C   PRO A 231       6.773  -2.039  23.751  1.00 12.26           C  \\nATOM   1447  O   PRO A 231       5.749  -1.549  23.269  1.00 12.84           O  \\nATOM   1448  CB  PRO A 231       6.897  -4.211  22.505  1.00 12.75           C  \\nATOM   1449  CG  PRO A 231       6.200  -5.526  22.761  1.00 13.69           C  \\nATOM   1450  CD  PRO A 231       5.105  -5.176  23.731  1.00 10.22           C  \\nATOM   1451  N   LYS A 232       7.772  -1.319  24.254  1.00 11.13           N  \\nATOM   1452  CA  LYS A 232       7.708   0.141  24.255  1.00 12.12           C  \\nATOM   1453  C   LYS A 232       9.075   0.752  24.021  1.00 13.25           C  \\nATOM   1454  O   LYS A 232      10.092   0.221  24.459  1.00 13.28           O  \\nATOM   1455  CB  LYS A 232       7.068   0.674  25.543  1.00 13.29           C  \\nATOM   1456  CG  LYS A 232       7.776   0.288  26.851  1.00 13.46           C  \\nATOM   1457  CD  LYS A 232       7.053   0.951  28.015  1.00 15.98           C  \\nATOM   1458  CE  LYS A 232       7.625   0.573  29.360  1.00 16.25           C  \\nATOM   1459  NZ  LYS A 232       6.825   1.253  30.446  1.00 20.08           N  \\nATOM   1460  N   VAL A 233       9.080   1.852  23.280  1.00 12.53           N  \\nATOM   1461  CA  VAL A 233      10.294   2.632  23.009  1.00 12.23           C  \\nATOM   1462  C   VAL A 233       9.955   4.108  23.145  1.00 12.19           C  \\nATOM   1463  O   VAL A 233       8.849   4.537  22.816  1.00 13.40           O  \\nATOM   1464  CB  VAL A 233      10.807   2.385  21.572  1.00 12.09           C  \\nATOM   1465  CG1 VAL A 233      11.880   3.392  21.204  1.00 11.26           C  \\nATOM   1466  CG2 VAL A 233      11.354   0.958  21.441  1.00 13.52           C  \\nATOM   1467  N   TYR A 234      10.896   4.859  23.703  1.00 12.07           N  \\nATOM   1468  CA  TYR A 234      10.835   6.322  23.674  1.00 10.41           C  \\nATOM   1469  C   TYR A 234      12.080   6.843  22.955  1.00 11.74           C  \\nATOM   1470  O   TYR A 234      13.215   6.494  23.310  1.00 13.17           O  \\nATOM   1471  CB  TYR A 234      10.795   6.874  25.105  1.00 13.52           C  \\nATOM   1472  CG  TYR A 234      10.926   8.382  25.194  1.00 13.81           C  \\nATOM   1473  CD1 TYR A 234       9.966   9.206  24.637  1.00 13.95           C  \\nATOM   1474  CD2 TYR A 234      11.941   8.964  25.935  1.00 13.95           C  \\nATOM   1475  CE1 TYR A 234      10.064  10.591  24.723  1.00 12.49           C  \\nATOM   1476  CE2 TYR A 234      12.039  10.349  26.063  1.00 14.30           C  \\nATOM   1477  CZ  TYR A 234      11.108  11.149  25.430  1.00 11.80           C  \\nATOM   1478  OH  TYR A 234      11.172  12.535  25.555  1.00 15.44           O  \\nATOM   1479  N   VAL A 235      11.868   7.661  21.926  1.00 11.98           N  \\nATOM   1480  CA  VAL A 235      12.967   8.383  21.289  1.00 12.74           C  \\nATOM   1481  C   VAL A 235      12.759   9.883  21.429  1.00 11.48           C  \\nATOM   1482  O   VAL A 235      11.647  10.373  21.227  1.00 13.02           O  \\nATOM   1483  CB  VAL A 235      13.146   8.001  19.810  1.00 13.23           C  \\nATOM   1484  CG1 VAL A 235      13.240   6.482  19.666  1.00 15.06           C  \\nATOM   1485  CG2 VAL A 235      12.024   8.536  18.967  1.00 16.40           C  \\nATOM   1486  N   GLU A 236      13.755  10.556  21.993  1.00 12.87           N  \\nATOM   1487  CA  GLU A 236      13.651  11.988  22.208  1.00 14.22           C  \\nATOM   1488  C   GLU A 236      14.553  12.725  21.243  1.00 14.04           C  \\nATOM   1489  O   GLU A 236      15.777  12.615  21.338  1.00 15.44           O  \\nATOM   1490  CB  GLU A 236      14.039  12.326  23.645  1.00 12.71           C  \\nATOM   1491  CG  GLU A 236      13.776  13.769  24.027  1.00 14.27           C  \\nATOM   1492  CD  GLU A 236      14.184  14.044  25.457  1.00 17.93           C  \\nATOM   1493  OE1 GLU A 236      13.440  13.646  26.372  1.00 15.59           O  \\nATOM   1494  OE2 GLU A 236      15.316  14.533  25.674  1.00 20.69           O  \\nATOM   1495  N   LEU A 237      13.949  13.554  20.396  1.00 13.73           N  \\nATOM   1496  CA  LEU A 237      14.690  14.316  19.386  1.00 13.08           C  \\nATOM   1497  C   LEU A 237      15.473  15.458  20.012  1.00 15.49           C  \\nATOM   1498  O   LEU A 237      14.903  16.343  20.640  1.00 15.37           O  \\nATOM   1499  CB  LEU A 237      13.739  14.885  18.331  1.00 12.20           C  \\nATOM   1500  CG  LEU A 237      12.798  13.894  17.639  1.00 15.27           C  \\nATOM   1501  CD1 LEU A 237      11.886  14.635  16.687  1.00 16.55           C  \\nATOM   1502  CD2 LEU A 237      13.577  12.816  16.885  1.00 18.35           C  \\nATOM   1503  N   ASP A 238      16.790  15.428  19.817  1.00 17.04           N  \\nATOM   1504  CA  ASP A 238      17.702  16.408  20.390  1.00 19.74           C  \\nATOM   1505  C   ASP A 238      17.303  17.826  19.967  1.00 17.63           C  \\nATOM   1506  O   ASP A 238      17.189  18.109  18.777  1.00 18.16           O  \\nATOM   1507  CB  ASP A 238      19.116  16.095  19.880  1.00 19.24           C  \\nATOM   1508  CG  ASP A 238      20.159  17.017  20.442  1.00 24.11           C  \\nATOM   1509  OD1 ASP A 238      20.037  17.427  21.608  1.00 24.89           O  \\nATOM   1510  OD2 ASP A 238      21.174  17.216  19.749  1.00 33.38           O  \\nATOM   1511  N   ASN A 239      17.047  18.675  20.962  1.00 19.41           N  \\nATOM   1512  CA  ASN A 239      16.772  20.103  20.770  1.00 20.66           C  \\nATOM   1513  C   ASN A 239      15.589  20.435  19.872  1.00 21.24           C  \\nATOM   1514  O   ASN A 239      15.459  21.563  19.397  1.00 22.64           O  \\nATOM   1515  CB  ASN A 239      18.011  20.864  20.305  1.00 22.58           C  \\nATOM   1516  CG  ASN A 239      17.990  22.315  20.748  1.00 29.52           C  \\nATOM   1517  OD1 ASN A 239      17.396  22.651  21.776  1.00 36.97           O  \\nATOM   1518  ND2 ASN A 239      18.536  23.193  19.919  1.00 37.81           N  \\nATOM   1519  N   ALA A 240      14.656  19.499  19.745  1.00 16.63           N  \\nATOM   1520  CA  ALA A 240      13.575  19.659  18.785  1.00 16.07           C  \\nATOM   1521  C   ALA A 240      12.357  20.365  19.354  1.00 15.03           C  \\nATOM   1522  O   ALA A 240      12.074  20.310  20.556  1.00 16.12           O  \\nATOM   1523  CB  ALA A 240      13.175  18.312  18.212  1.00 14.17           C  \\nATOM   1524  N   SER A 241      11.587  20.958  18.453  1.00 16.05           N  \\nATOM   1525  CA  SER A 241      10.371  21.637  18.843  1.00 16.86           C  \\nATOM   1526  C   SER A 241       9.132  20.779  18.642  1.00 16.61           C  \\nATOM   1527  O   SER A 241       9.163  19.729  17.968  1.00 15.47           O  \\nATOM   1528  CB  SER A 241      10.230  22.952  18.072  1.00 17.99           C  \\nATOM   1529  OG  SER A 241       9.590  22.743  16.827  1.00 23.43           O  \\nATOM   1530  N   HIS A 242       8.032  21.237  19.219  1.00 14.73           N  \\nATOM   1531  CA  HIS A 242       6.744  20.625  18.985  1.00 15.64           C  \\nATOM   1532  C   HIS A 242       6.402  20.459  17.506  1.00 16.01           C  \\nATOM   1533  O   HIS A 242       5.612  19.582  17.145  1.00 16.98           O  \\nATOM   1534  CB  HIS A 242       5.652  21.420  19.692  1.00 16.33           C  \\nATOM   1535  CG  HIS A 242       4.432  20.616  19.975  1.00 24.08           C  \\nATOM   1536  ND1 HIS A 242       4.476  19.481  20.758  1.00 25.00           N  \\nATOM   1537  CD2 HIS A 242       3.216  20.615  19.382  1.00 12.36           C  \\nATOM   1538  CE1 HIS A 242       3.319  18.856  20.692  1.00 15.95           C  \\nATOM   1539  NE2 HIS A 242       2.518  19.550  19.901  1.00 17.12           N  \\nATOM   1540  N   PHE A 243       6.915  21.355  16.667  1.00 13.57           N  \\nATOM   1541  CA  PHE A 243       6.547  21.361  15.258  1.00 16.72           C  \\nATOM   1542  C   PHE A 243       7.366  20.374  14.440  1.00 15.55           C  \\nATOM   1543  O   PHE A 243       7.219  20.314  13.216  1.00 15.91           O  \\nATOM   1544  CB  PHE A 243       6.720  22.760  14.658  1.00 18.63           C  \\nATOM   1545  CG  PHE A 243       5.804  23.785  15.243  1.00 26.36           C  \\nATOM   1546  CD1 PHE A 243       4.436  23.687  15.070  1.00 32.76           C  \\nATOM   1547  CD2 PHE A 243       6.318  24.893  15.890  1.00 27.79           C  \\nATOM   1548  CE1 PHE A 243       3.584  24.646  15.595  1.00 40.92           C  \\nATOM   1549  CE2 PHE A 243       5.473  25.883  16.379  1.00 41.38           C  \\nATOM   1550  CZ  PHE A 243       4.103  25.731  16.273  1.00 38.04           C  \\nATOM   1551  N   ALA A 244       8.326  19.713  15.081  1.00 15.54           N  \\nATOM   1552  CA  ALA A 244       9.200  18.798  14.342  1.00 14.42           C  \\nATOM   1553  C   ALA A 244       8.450  17.782  13.487  1.00 13.99           C  \\nATOM   1554  O   ALA A 244       8.851  17.535  12.358  1.00 13.78           O  \\nATOM   1555  CB  ALA A 244      10.181  18.109  15.264  1.00 14.62           C  \\nATOM   1556  N   PRO A 245       7.295  17.272  13.965  1.00 14.34           N  \\nATOM   1557  CA  PRO A 245       6.632  16.260  13.146  1.00 14.45           C  \\nATOM   1558  C   PRO A 245       5.901  16.831  11.932  1.00 14.65           C  \\nATOM   1559  O   PRO A 245       5.404  16.066  11.107  1.00 13.90           O  \\nATOM   1560  CB  PRO A 245       5.610  15.631  14.110  1.00 15.13           C  \\nATOM   1561  CG  PRO A 245       6.190  15.859  15.460  1.00 14.45           C  \\nATOM   1562  CD  PRO A 245       6.851  17.220  15.368  1.00 13.80           C  \\nATOM   1563  N   ASN A 246       5.859  18.158  11.797  1.00 13.15           N  \\nATOM   1564  CA  ASN A 246       5.094  18.784  10.721  1.00 13.15           C  \\nATOM   1565  C   ASN A 246       5.918  19.144   9.480  1.00 13.74           C  \\nATOM   1566  O   ASN A 246       5.390  19.725   8.519  1.00 17.00           O  \\nATOM   1567  CB  ASN A 246       4.397  20.053  11.226  1.00 14.77           C  \\nATOM   1568  CG  ASN A 246       3.460  19.791  12.397  1.00 18.01           C  \\nATOM   1569  OD1 ASN A 246       3.455  20.540  13.367  1.00 19.04           O  \\nATOM   1570  ND2 ASN A 246       2.718  18.696  12.340  1.00 14.36           N  \\nATOM   1571  N   SER A 247       7.192  18.767   9.489  1.00 14.73           N  \\nATOM   1572  CA  SER A 247       7.997  18.803   8.279  1.00 15.76           C  \\nATOM   1573  C   SER A 247       8.979  17.649   8.249  1.00 14.89           C  \\nATOM   1574  O   SER A 247       9.230  17.010   9.274  1.00 15.67           O  \\nATOM   1575  CB  SER A 247       8.717  20.150   8.120  1.00 19.00           C  \\nATOM   1576  OG  SER A 247       9.660  20.360   9.136  1.00 22.89           O  \\nATOM   1577  N   ASN A 248       9.507  17.363   7.067  1.00 14.98           N  \\nATOM   1578  CA  ASN A 248      10.438  16.250   6.884  1.00 16.66           C  \\nATOM   1579  C   ASN A 248      11.461  16.201   8.008  1.00 16.80           C  \\nATOM   1580  O   ASN A 248      12.116  17.202   8.329  1.00 18.51           O  \\nATOM   1581  CB  ASN A 248      11.135  16.387   5.526  1.00 16.82           C  \\nATOM   1582  CG  ASN A 248      11.868  15.126   5.095  1.00 22.21           C  \\nATOM   1583  OD1 ASN A 248      11.660  14.039   5.644  1.00 18.87           O  \\nATOM   1584  ND2 ASN A 248      12.609  15.242   3.996  1.00 28.19           N  \\nATOM   1585  N   ASN A 249      11.556  15.039   8.648  1.00 13.85           N  \\nATOM   1586  CA  ASN A 249      12.422  14.870   9.791  1.00 14.06           C  \\nATOM   1587  C   ASN A 249      13.013  13.469   9.754  1.00 14.71           C  \\nATOM   1588  O   ASN A 249      12.280  12.471   9.773  1.00 15.38           O  \\nATOM   1589  CB  ASN A 249      11.619  15.050  11.084  1.00 16.46           C  \\nATOM   1590  CG  ASN A 249      12.499  15.079  12.308  1.00 15.90           C  \\nATOM   1591  OD1 ASN A 249      12.750  16.150  12.885  1.00 23.40           O  \\nATOM   1592  ND2 ASN A 249      12.958  13.910  12.742  1.00 12.98           N  \\nATOM   1593  N   ALA A 250      14.329  13.393   9.608  1.00 13.09           N  \\nATOM   1594  CA  ALA A 250      14.979  12.108   9.378  1.00 13.93           C  \\nATOM   1595  C   ALA A 250      14.713  11.115  10.504  1.00 14.90           C  \\nATOM   1596  O   ALA A 250      14.417   9.954  10.247  1.00 13.15           O  \\nATOM   1597  CB  ALA A 250      16.473  12.307   9.178  1.00 13.69           C  \\nATOM   1598  N   ALA A 251      14.894  11.536  11.749  1.00 12.35           N  \\nATOM   1599  CA  ALA A 251      14.758  10.607  12.854  1.00 14.31           C  \\nATOM   1600  C   ALA A 251      13.313  10.146  13.016  1.00 15.10           C  \\nATOM   1601  O   ALA A 251      13.059   8.953  13.191  1.00 12.54           O  \\nATOM   1602  CB  ALA A 251      15.298  11.197  14.141  1.00 15.15           C  \\nATOM   1603  N   ILE A 252      12.357  11.055  12.822  1.00 11.99           N  \\nATOM   1604  CA  ILE A 252      10.970  10.645  12.864  1.00 13.31           C  \\nATOM   1605  C   ILE A 252      10.669   9.619  11.766  1.00 14.73           C  \\nATOM   1606  O   ILE A 252      10.000   8.606  12.001  1.00 12.93           O  \\nATOM   1607  CB  ILE A 252      10.020  11.846  12.752  1.00 13.20           C  \\nATOM   1608  CG1 ILE A 252      10.138  12.709  14.006  1.00 14.29           C  \\nATOM   1609  CG2 ILE A 252       8.583  11.362  12.565  1.00 13.66           C  \\nATOM   1610  CD1 ILE A 252       9.250  13.966  13.954  1.00 15.11           C  \\nATOM   1611  N   SER A 253      11.269   9.818  10.603  1.00 12.61           N  \\nATOM   1612  CA  SER A 253      11.075   8.909   9.487  1.00 12.16           C  \\nATOM   1613  C   SER A 253      11.663   7.539   9.822  1.00 12.26           C  \\nATOM   1614  O   SER A 253      10.979   6.517   9.700  1.00 12.44           O  \\nATOM   1615  CB  SER A 253      11.761   9.492   8.246  1.00 12.68           C  \\nATOM   1616  OG  SER A 253      11.414   8.777   7.081  1.00 15.57           O  \\nATOM   1617  N   VAL A 254      12.901   7.519  10.306  1.00 11.27           N  \\nATOM   1618  CA  VAL A 254      13.559   6.252  10.644  1.00 12.07           C  \\nATOM   1619  C   VAL A 254      12.741   5.461  11.658  1.00 11.96           C  \\nATOM   1620  O   VAL A 254      12.476   4.265  11.462  1.00 12.40           O  \\nATOM   1621  CB  VAL A 254      14.992   6.464  11.157  1.00 10.86           C  \\nATOM   1622  CG1 VAL A 254      15.581   5.158  11.730  1.00 12.84           C  \\nATOM   1623  CG2 VAL A 254      15.880   7.002  10.040  1.00 13.26           C  \\nATOM   1624  N   TYR A 255      12.381   6.093  12.764  1.00 11.53           N  \\nATOM   1625  CA  TYR A 255      11.747   5.357  13.842  1.00 11.10           C  \\nATOM   1626  C   TYR A 255      10.305   4.974  13.542  1.00 12.18           C  \\nATOM   1627  O   TYR A 255       9.843   3.904  13.965  1.00 12.26           O  \\nATOM   1628  CB  TYR A 255      11.881   6.096  15.172  1.00 12.18           C  \\nATOM   1629  CG  TYR A 255      13.309   6.058  15.671  1.00 12.31           C  \\nATOM   1630  CD1 TYR A 255      13.885   4.863  16.097  1.00 11.99           C  \\nATOM   1631  CD2 TYR A 255      14.093   7.208  15.683  1.00 10.85           C  \\nATOM   1632  CE1 TYR A 255      15.205   4.815  16.519  1.00 15.66           C  \\nATOM   1633  CE2 TYR A 255      15.447   7.145  15.994  1.00 14.01           C  \\nATOM   1634  CZ  TYR A 255      15.977   5.961  16.464  1.00 18.53           C  \\nATOM   1635  OH  TYR A 255      17.307   5.916  16.832  1.00 17.67           O  \\nATOM   1636  N   THR A 256       9.581   5.852  12.871  1.00 11.41           N  \\nATOM   1637  CA  THR A 256       8.188   5.561  12.544  1.00  9.80           C  \\nATOM   1638  C   THR A 256       8.096   4.448  11.500  1.00 10.41           C  \\nATOM   1639  O   THR A 256       7.359   3.465  11.692  1.00 11.30           O  \\nATOM   1640  CB  THR A 256       7.438   6.809  12.057  1.00 10.89           C  \\nATOM   1641  OG1 THR A 256       7.558   7.833  13.052  1.00 14.03           O  \\nATOM   1642  CG2 THR A 256       5.967   6.511  11.887  1.00 11.36           C  \\nATOM   1643  N   ILE A 257       8.893   4.536  10.442  1.00 11.82           N  \\nATOM   1644  CA  ILE A 257       8.934   3.431   9.482  1.00 10.72           C  \\nATOM   1645  C   ILE A 257       9.344   2.148  10.176  1.00 11.64           C  \\nATOM   1646  O   ILE A 257       8.749   1.097   9.933  1.00 11.99           O  \\nATOM   1647  CB  ILE A 257       9.870   3.713   8.293  1.00 10.94           C  \\nATOM   1648  CG1 ILE A 257       9.352   4.894   7.464  1.00 11.67           C  \\nATOM   1649  CG2 ILE A 257      10.008   2.463   7.412  1.00 11.41           C  \\nATOM   1650  CD1 ILE A 257      10.409   5.450   6.493  1.00 13.52           C  \\nATOM   1651  N   SER A 258      10.393   2.206  10.997  1.00 11.08           N  \\nATOM   1652  CA  SER A 258      10.916   0.979  11.582  1.00 12.38           C  \\nATOM   1653  C   SER A 258       9.905   0.302  12.482  1.00 12.71           C  \\nATOM   1654  O   SER A 258       9.760  -0.936  12.457  1.00 11.61           O  \\nATOM   1655  CB  SER A 258      12.205   1.223  12.349  1.00 12.31           C  \\nATOM   1656  OG  SER A 258      13.232   1.589  11.445  1.00 12.48           O  \\nATOM   1657  N   TRP A 259       9.231   1.094  13.318  1.00 10.42           N  \\nATOM   1658  CA  TRP A 259       8.256   0.528  14.244  1.00 10.55           C  \\nATOM   1659  C   TRP A 259       7.067  -0.056  13.479  1.00 11.05           C  \\nATOM   1660  O   TRP A 259       6.619  -1.178  13.752  1.00 11.96           O  \\nATOM   1661  CB  TRP A 259       7.781   1.591  15.235  1.00 10.06           C  \\nATOM   1662  CG  TRP A 259       7.144   1.016  16.478  1.00  9.86           C  \\nATOM   1663  CD1 TRP A 259       5.842   1.161  16.874  1.00  9.92           C  \\nATOM   1664  CD2 TRP A 259       7.816   0.341  17.550  1.00  9.64           C  \\nATOM   1665  NE1 TRP A 259       5.651   0.591  18.115  1.00 11.58           N  \\nATOM   1666  CE2 TRP A 259       6.855   0.111  18.567  1.00 10.94           C  \\nATOM   1667  CE3 TRP A 259       9.138  -0.071  17.761  1.00 11.21           C  \\nATOM   1668  CZ2 TRP A 259       7.168  -0.556  19.748  1.00 12.86           C  \\nATOM   1669  CZ3 TRP A 259       9.442  -0.759  18.931  1.00 12.14           C  \\nATOM   1670  CH2 TRP A 259       8.468  -0.953  19.927  1.00 10.93           C  \\nATOM   1671  N   MET A 260       6.581   0.670  12.480  1.00 11.09           N  \\nATOM   1672  CA  MET A 260       5.481   0.161  11.665  1.00 11.36           C  \\nATOM   1673  C   MET A 260       5.887  -1.108  10.940  1.00 12.08           C  \\nATOM   1674  O   MET A 260       5.135  -2.085  10.936  1.00 11.80           O  \\nATOM   1675  CB  MET A 260       4.977   1.214  10.667  1.00 12.28           C  \\nATOM   1676  CG  MET A 260       4.328   2.411  11.340  1.00 10.30           C  \\nATOM   1677  SD  MET A 260       2.881   1.956  12.315  1.00 12.18           S  \\nATOM   1678  CE  MET A 260       2.325   3.602  12.805  1.00 12.32           C  \\nATOM   1679  N   LYS A 261       7.090  -1.122  10.359  1.00 10.85           N  \\nATOM   1680  CA  LYS A 261       7.564  -2.332   9.685  1.00 11.87           C  \\nATOM   1681  C   LYS A 261       7.678  -3.519  10.634  1.00 12.02           C  \\nATOM   1682  O   LYS A 261       7.237  -4.612  10.315  1.00 11.37           O  \\nATOM   1683  CB  LYS A 261       8.910  -2.080   9.019  1.00 12.14           C  \\nATOM   1684  CG  LYS A 261       8.851  -1.239   7.736  1.00 11.69           C  \\nATOM   1685  CD  LYS A 261       8.422  -2.121   6.561  1.00 13.34           C  \\nATOM   1686  CE  LYS A 261       8.625  -1.414   5.222  1.00 12.11           C  \\nATOM   1687  NZ  LYS A 261       8.301  -2.351   4.099  1.00 12.41           N  \\nATOM   1688  N   LEU A 262       8.299  -3.299  11.788  1.00 12.69           N  \\nATOM   1689  CA  LEU A 262       8.559  -4.407  12.712  1.00 11.45           C  \\nATOM   1690  C   LEU A 262       7.254  -5.087  13.122  1.00 12.76           C  \\nATOM   1691  O   LEU A 262       7.196  -6.317  13.234  1.00 12.42           O  \\nATOM   1692  CB  LEU A 262       9.293  -3.916  13.956  1.00 11.74           C  \\nATOM   1693  CG  LEU A 262       9.666  -4.981  15.001  1.00 10.45           C  \\nATOM   1694  CD1 LEU A 262      10.516  -6.087  14.346  1.00 13.77           C  \\nATOM   1695  CD2 LEU A 262      10.405  -4.359  16.170  1.00 14.47           C  \\nATOM   1696  N   TRP A 263       6.242  -4.289  13.459  1.00 12.47           N  \\nATOM   1697  CA  TRP A 263       5.024  -4.837  14.041  1.00 11.16           C  \\nATOM   1698  C   TRP A 263       3.888  -5.055  13.047  1.00 12.63           C  \\nATOM   1699  O   TRP A 263       3.199  -6.088  13.089  1.00 15.08           O  \\nATOM   1700  CB  TRP A 263       4.558  -3.964  15.200  1.00 11.92           C  \\nATOM   1701  CG  TRP A 263       5.603  -3.822  16.246  1.00 11.48           C  \\nATOM   1702  CD1 TRP A 263       6.341  -2.708  16.505  1.00 11.90           C  \\nATOM   1703  CD2 TRP A 263       6.047  -4.822  17.185  1.00 11.81           C  \\nATOM   1704  NE1 TRP A 263       7.214  -2.948  17.532  1.00 13.10           N  \\nATOM   1705  CE2 TRP A 263       6.997  -4.211  18.021  1.00 12.19           C  \\nATOM   1706  CE3 TRP A 263       5.668  -6.144  17.451  1.00 13.62           C  \\nATOM   1707  CZ2 TRP A 263       7.680  -4.912  19.013  1.00 14.21           C  \\nATOM   1708  CZ3 TRP A 263       6.337  -6.838  18.464  1.00 14.11           C  \\nATOM   1709  CH2 TRP A 263       7.299  -6.206  19.254  1.00 13.99           C  \\nATOM   1710  N   VAL A 264       3.664  -4.104  12.144  1.00 11.14           N  \\nATOM   1711  CA  VAL A 264       2.608  -4.290  11.166  1.00 11.10           C  \\nATOM   1712  C   VAL A 264       2.957  -5.429  10.215  1.00 12.77           C  \\nATOM   1713  O   VAL A 264       2.091  -6.209   9.844  1.00 12.18           O  \\nATOM   1714  CB  VAL A 264       2.288  -2.993  10.383  1.00 11.44           C  \\nATOM   1715  CG1 VAL A 264       1.176  -3.253   9.351  1.00 13.95           C  \\nATOM   1716  CG2 VAL A 264       1.870  -1.883  11.358  1.00 11.45           C  \\nATOM   1717  N   ASP A 265       4.228  -5.506   9.814  1.00 11.40           N  \\nATOM   1718  CA  ASP A 265       4.654  -6.466   8.793  1.00 12.67           C  \\nATOM   1719  C   ASP A 265       5.423  -7.641   9.384  1.00 12.59           C  \\nATOM   1720  O   ASP A 265       6.003  -8.427   8.623  1.00 14.26           O  \\nATOM   1721  CB  ASP A 265       5.562  -5.808   7.747  1.00 12.01           C  \\nATOM   1722  CG  ASP A 265       4.873  -4.731   6.918  1.00 13.70           C  \\nATOM   1723  OD1 ASP A 265       3.640  -4.515   7.043  1.00 12.66           O  \\nATOM   1724  OD2 ASP A 265       5.592  -4.146   6.080  1.00 13.42           O  \\nATOM   1725  N   ASN A 266       5.632  -7.633  10.697  1.00 12.52           N  \\nATOM   1726  CA  ASN A 266       6.521  -8.611  11.341  1.00 11.18           C  \\nATOM   1727  C   ASN A 266       7.944  -8.545  10.784  1.00 13.24           C  \\nATOM   1728  O   ASN A 266       8.655  -9.567  10.712  1.00 15.02           O  \\nATOM   1729  CB  ASN A 266       5.983 -10.054  11.202  1.00 12.09           C  \\nATOM   1730  CG  ASN A 266       4.612 -10.252  11.838  1.00 15.02           C  \\nATOM   1731  OD1 ASN A 266       4.126  -9.405  12.573  1.00 14.55           O  \\nATOM   1732  ND2 ASN A 266       4.036 -11.432  11.630  1.00 18.58           N  \\nATOM   1733  N   ASP A 267       8.328  -7.374  10.276  1.00 11.89           N  \\nATOM   1734  CA  ASP A 267       9.573  -7.276   9.527  1.00 11.27           C  \\nATOM   1735  C   ASP A 267      10.786  -7.080  10.427  1.00 13.18           C  \\nATOM   1736  O   ASP A 267      11.176  -5.950  10.764  1.00 12.99           O  \\nATOM   1737  CB  ASP A 267       9.494  -6.170   8.489  1.00 15.07           C  \\nATOM   1738  CG  ASP A 267      10.526  -6.328   7.390  1.00 12.54           C  \\nATOM   1739  OD1 ASP A 267      11.588  -6.929   7.643  1.00 14.33           O  \\nATOM   1740  OD2 ASP A 267      10.342  -5.731   6.311  1.00 12.28           O  \\nATOM   1741  N   THR A 268      11.427  -8.189  10.757  1.00 12.31           N  \\nATOM   1742  CA  THR A 268      12.497  -8.171  11.736  1.00 12.82           C  \\nATOM   1743  C   THR A 268      13.795  -7.648  11.156  1.00 11.68           C  \\nATOM   1744  O   THR A 268      14.725  -7.371  11.920  1.00 12.95           O  \\nATOM   1745  CB  THR A 268      12.741  -9.554  12.354  1.00 11.24           C  \\nATOM   1746  OG1 THR A 268      12.969 -10.499  11.301  1.00 14.17           O  \\nATOM   1747  CG2 THR A 268      11.541  -9.972  13.202  1.00 15.91           C  \\nATOM   1748  N   ARG A 269      13.789  -7.312   9.858  1.00 12.31           N  \\nATOM   1749  CA  ARG A 269      14.920  -6.607   9.261  1.00 13.14           C  \\nATOM   1750  C   ARG A 269      15.073  -5.216   9.845  1.00 11.75           C  \\nATOM   1751  O   ARG A 269      16.130  -4.600   9.707  1.00 13.84           O  \\nATOM   1752  CB  ARG A 269      14.818  -6.522   7.739  1.00 13.40           C  \\nATOM   1753  CG  ARG A 269      14.826  -7.866   7.052  1.00 12.81           C  \\nATOM   1754  CD  ARG A 269      14.604  -7.722   5.558  1.00 12.45           C  \\nATOM   1755  NE  ARG A 269      13.348  -7.034   5.262  1.00 12.27           N  \\nATOM   1756  CZ  ARG A 269      12.980  -6.668   4.042  1.00 12.37           C  \\nATOM   1757  NH1 ARG A 269      13.759  -6.960   3.006  1.00 15.72           N  \\nATOM   1758  NH2 ARG A 269      11.777  -6.132   3.832  1.00 14.97           N  \\nATOM   1759  N   TYR A 270      14.028  -4.730  10.508  1.00 13.33           N  \\nATOM   1760  CA  TYR A 270      14.066  -3.408  11.101  1.00 12.35           C  \\nATOM   1761  C   TYR A 270      14.439  -3.377  12.579  1.00 11.91           C  \\nATOM   1762  O   TYR A 270      14.615  -2.298  13.157  1.00 13.58           O  \\nATOM   1763  CB  TYR A 270      12.749  -2.684  10.845  1.00 12.29           C  \\nATOM   1764  CG  TYR A 270      12.558  -2.333   9.391  1.00 11.77           C  \\nATOM   1765  CD1 TYR A 270      12.785  -1.049   8.944  1.00 12.58           C  \\nATOM   1766  CD2 TYR A 270      12.173  -3.297   8.459  1.00 13.45           C  \\nATOM   1767  CE1 TYR A 270      12.648  -0.728   7.599  1.00 12.13           C  \\nATOM   1768  CE2 TYR A 270      12.037  -2.979   7.106  1.00 12.44           C  \\nATOM   1769  CZ  TYR A 270      12.323  -1.698   6.687  1.00 11.58           C  \\nATOM   1770  OH  TYR A 270      12.168  -1.323   5.373  1.00 12.39           O  \\nATOM   1771  N   ARG A 271      14.714  -4.554  13.156  1.00 12.19           N  \\nATOM   1772  CA  ARG A 271      15.200  -4.637  14.523  1.00 12.97           C  \\nATOM   1773  C   ARG A 271      16.505  -3.878  14.686  1.00 12.52           C  \\nATOM   1774  O   ARG A 271      16.766  -3.280  15.722  1.00 14.16           O  \\nATOM   1775  CB  ARG A 271      15.405  -6.098  14.951  1.00 16.58           C  \\nATOM   1776  CG  ARG A 271      14.143  -6.825  15.295  1.00 19.34           C  \\nATOM   1777  CD  ARG A 271      14.447  -8.288  15.614  1.00 22.29           C  \\nATOM   1778  NE  ARG A 271      14.729  -8.501  17.026  1.00 31.63           N  \\nATOM   1779  CZ  ARG A 271      15.019  -9.684  17.556  1.00 38.66           C  \\nATOM   1780  NH1 ARG A 271      15.304 -10.708  16.761  1.00 38.16           N  \\nATOM   1781  NH2 ARG A 271      14.952  -9.864  18.872  1.00 34.63           N  \\nATOM   1782  N   GLN A 272      17.325  -3.882  13.646  1.00 13.50           N  \\nATOM   1783  CA  GLN A 272      18.649  -3.283  13.733  1.00 12.42           C  \\nATOM   1784  C   GLN A 272      18.595  -1.798  14.086  1.00 14.05           C  \\nATOM   1785  O   GLN A 272      19.556  -1.269  14.641  1.00 16.37           O  \\nATOM   1786  CB  GLN A 272      19.409  -3.486  12.425  1.00 15.05           C  \\nATOM   1787  CG  GLN A 272      18.705  -2.887  11.218  1.00 13.31           C  \\nATOM   1788  CD  GLN A 272      19.459  -3.162   9.939  1.00 15.55           C  \\nATOM   1789  OE1 GLN A 272      20.625  -2.762   9.795  1.00 15.68           O  \\nATOM   1790  NE2 GLN A 272      18.823  -3.878   9.015  1.00 13.01           N  \\nATOM   1791  N   PHE A 273      17.517  -1.109  13.713  1.00 12.08           N  \\nATOM   1792  CA  PHE A 273      17.402   0.326  13.984  1.00 12.72           C  \\nATOM   1793  C   PHE A 273      16.799   0.588  15.363  1.00 14.19           C  \\nATOM   1794  O   PHE A 273      16.764   1.729  15.832  1.00 14.03           O  \\nATOM   1795  CB  PHE A 273      16.563   1.006  12.898  1.00 11.48           C  \\nATOM   1796  CG  PHE A 273      16.990   0.649  11.505  1.00 12.19           C  \\nATOM   1797  CD1 PHE A 273      18.253   0.981  11.051  1.00 14.52           C  \\nATOM   1798  CD2 PHE A 273      16.158  -0.081  10.682  1.00 12.15           C  \\nATOM   1799  CE1 PHE A 273      18.674   0.595   9.787  1.00 14.45           C  \\nATOM   1800  CE2 PHE A 273      16.581  -0.507   9.440  1.00 12.61           C  \\nATOM   1801  CZ  PHE A 273      17.844  -0.149   8.977  1.00 13.43           C  \\nATOM   1802  N   LEU A 274      16.292  -0.475  15.987  1.00 11.82           N  \\nATOM   1803  CA  LEU A 274      15.478  -0.366  17.202  1.00 12.32           C  \\nATOM   1804  C   LEU A 274      16.104  -1.080  18.389  1.00 12.08           C  \\nATOM   1805  O   LEU A 274      15.568  -1.020  19.494  1.00 12.89           O  \\nATOM   1806  CB  LEU A 274      14.073  -0.937  16.971  1.00 13.57           C  \\nATOM   1807  CG  LEU A 274      13.237  -0.202  15.923  1.00 13.06           C  \\nATOM   1808  CD1 LEU A 274      12.047  -1.051  15.529  1.00 12.74           C  \\nATOM   1809  CD2 LEU A 274      12.778   1.151  16.456  1.00 15.14           C  \\nATOM   1810  N   CYS A 275      17.267  -1.681  18.171  1.00 14.32           N  \\nATOM   1811  CA  CYS A 275      17.969  -2.443  19.196  1.00 12.98           C  \\nATOM   1812  C   CYS A 275      19.408  -1.942  19.246  1.00 14.03           C  \\nATOM   1813  O   CYS A 275      19.965  -1.526  18.223  1.00 14.45           O  \\nATOM   1814  CB  CYS A 275      17.939  -3.932  18.851  1.00 13.90           C  \\nATOM   1815  SG  CYS A 275      16.281  -4.658  18.936  1.00 14.13           S  \\nATOM   1816  N   ASN A 276      20.018  -2.008  20.428  1.00 15.42           N  \\nATOM   1817  CA  ASN A 276      21.405  -1.575  20.607  1.00 14.74           C  \\nATOM   1818  C   ASN A 276      21.614  -0.162  20.096  1.00 16.18           C  \\nATOM   1819  O   ASN A 276      22.522   0.103  19.303  1.00 17.65           O  \\nATOM   1820  CB  ASN A 276      22.395  -2.544  19.955  1.00 17.75           C  \\nATOM   1821  CG  ASN A 276      22.204  -3.982  20.422  1.00 20.60           C  \\nATOM   1822  OD1 ASN A 276      21.820  -4.860  19.642  1.00 22.74           O  \\nATOM   1823  ND2 ASN A 276      22.469  -4.225  21.696  1.00 21.17           N  \\nATOM   1824  N   VAL A 277      20.750   0.740  20.544  1.00 15.01           N  \\nATOM   1825  CA  VAL A 277      20.635   2.054  19.940  1.00 13.99           C  \\nATOM   1826  C   VAL A 277      21.592   3.042  20.614  1.00 16.12           C  \\nATOM   1827  O   VAL A 277      21.612   3.166  21.837  1.00 19.10           O  \\nATOM   1828  CB  VAL A 277      19.186   2.566  19.997  1.00 14.89           C  \\nATOM   1829  CG1 VAL A 277      19.100   3.990  19.466  1.00 15.66           C  \\nATOM   1830  CG2 VAL A 277      18.281   1.647  19.181  1.00 15.63           C  \\nATOM   1831  N   ASN A 278      22.404   3.719  19.801  1.00 16.50           N  \\nATOM   1832  CA  ASN A 278      23.162   4.877  20.255  1.00 16.44           C  \\nATOM   1833  C   ASN A 278      23.144   5.951  19.184  1.00 19.04           C  \\nATOM   1834  O   ASN A 278      23.975   5.948  18.272  1.00 19.03           O  \\nATOM   1835  CB  ASN A 278      24.614   4.493  20.592  1.00 17.33           C  \\nATOM   1836  CG  ASN A 278      25.430   5.684  21.083  1.00 19.48           C  \\nATOM   1837  OD1 ASN A 278      24.867   6.663  21.560  1.00 29.88           O  \\nATOM   1838  ND2 ASN A 278      26.740   5.499  21.183  1.00 30.33           N  \\nATOM   1839  N   ASP A 279      22.099   6.772  19.215  1.00 15.28           N  \\nATOM   1840  CA  ASP A 279      21.730   7.592  18.068  1.00 15.24           C  \\nATOM   1841  C   ASP A 279      21.992   9.058  18.390  1.00 16.97           C  \\nATOM   1842  O   ASP A 279      21.451   9.582  19.364  1.00 17.35           O  \\nATOM   1843  CB  ASP A 279      20.244   7.379  17.763  1.00 17.73           C  \\nATOM   1844  CG  ASP A 279      19.757   8.222  16.606  1.00 16.04           C  \\nATOM   1845  OD1 ASP A 279      20.509   9.116  16.157  1.00 18.33           O  \\nATOM   1846  OD2 ASP A 279      18.648   7.946  16.084  1.00 16.99           O  \\nATOM   1847  N   PRO A 280      22.967   9.671  17.693  1.00 14.38           N  \\nATOM   1848  CA  PRO A 280      23.356  11.029  18.042  1.00 15.32           C  \\nATOM   1849  C   PRO A 280      22.273  12.072  17.778  1.00 17.79           C  \\nATOM   1850  O   PRO A 280      22.365  13.176  18.304  1.00 18.14           O  \\nATOM   1851  CB  PRO A 280      24.567  11.304  17.125  1.00 18.12           C  \\nATOM   1852  CG  PRO A 280      24.451  10.343  16.038  1.00 21.23           C  \\nATOM   1853  CD  PRO A 280      23.786   9.116  16.603  1.00 16.66           C  \\nATOM   1854  N   ALA A 281      21.234  11.705  17.031  1.00 16.98           N  \\nATOM   1855  CA  ALA A 281      20.101  12.599  16.792  1.00 19.03           C  \\nATOM   1856  C   ALA A 281      19.179  12.697  18.004  1.00 18.93           C  \\nATOM   1857  O   ALA A 281      18.263  13.516  18.023  1.00 20.07           O  \\nATOM   1858  CB  ALA A 281      19.327  12.146  15.580  1.00 18.92           C  \\nATOM   1859  N   LEU A 282      19.414  11.850  19.002  1.00 18.83           N  \\nATOM   1860  CA  LEU A 282      18.538  11.749  20.157  1.00 15.60           C  \\nATOM   1861  C   LEU A 282      19.198  12.299  21.419  1.00 19.69           C  \\nATOM   1862  O   LEU A 282      20.342  11.944  21.743  1.00 21.85           O  \\nATOM   1863  CB  LEU A 282      18.142  10.291  20.393  1.00 17.31           C  \\nATOM   1864  CG  LEU A 282      17.433   9.579  19.254  1.00 15.92           C  \\nATOM   1865  CD1 LEU A 282      16.983   8.182  19.697  1.00 16.12           C  \\nATOM   1866  CD2 LEU A 282      16.259  10.399  18.690  1.00 17.02           C  \\nATOM   1867  N   SER A 283      18.421  13.014  22.220  1.00 17.52           N  \\nATOM   1868  CA  SER A 283      18.846  13.373  23.552  1.00 16.40           C  \\nATOM   1869  C   SER A 283      18.529  12.289  24.567  1.00 17.03           C  \\nATOM   1870  O   SER A 283      18.994  12.352  25.695  1.00 20.61           O  \\nATOM   1871  CB  SER A 283      18.247  14.712  23.991  1.00 15.74           C  \\nATOM   1872  OG  SER A 283      16.897  14.824  23.558  1.00 15.31           O  \\nATOM   1873  N   ASP A 284      17.623  11.378  24.209  1.00 16.35           N  \\nATOM   1874  CA  ASP A 284      17.284  10.260  25.093  1.00 15.24           C  \\nATOM   1875  C   ASP A 284      16.714   9.120  24.268  1.00 15.82           C  \\nATOM   1876  O   ASP A 284      16.162   9.335  23.193  1.00 14.91           O  \\nATOM   1877  CB  ASP A 284      16.263  10.692  26.153  1.00 16.30           C  \\nATOM   1878  CG  ASP A 284      16.104   9.670  27.271  1.00 14.35           C  \\nATOM   1879  OD1 ASP A 284      16.985   8.790  27.423  1.00 20.71           O  \\nATOM   1880  OD2 ASP A 284      15.183   9.857  28.091  1.00 18.74           O  \\nATOM   1881  N   PHE A 285      16.958   7.902  24.735  1.00 14.03           N  \\nATOM   1882  CA  PHE A 285      16.357   6.706  24.156  1.00 13.86           C  \\nATOM   1883  C   PHE A 285      16.037   5.748  25.303  1.00 13.50           C  \\nATOM   1884  O   PHE A 285      16.873   5.511  26.176  1.00 14.38           O  \\nATOM   1885  CB  PHE A 285      17.345   6.037  23.195  1.00 14.86           C  \\nATOM   1886  CG  PHE A 285      16.821   4.776  22.570  1.00 13.89           C  \\nATOM   1887  CD1 PHE A 285      16.067   4.836  21.407  1.00 14.72           C  \\nATOM   1888  CD2 PHE A 285      16.951   3.557  23.218  1.00 16.18           C  \\nATOM   1889  CE1 PHE A 285      15.533   3.689  20.839  1.00 15.70           C  \\nATOM   1890  CE2 PHE A 285      16.380   2.413  22.678  1.00 15.98           C  \\nATOM   1891  CZ  PHE A 285      15.662   2.488  21.497  1.00 16.12           C  \\nATOM   1892  N   ARG A 286      14.798   5.261  25.346  1.00 12.39           N  \\nATOM   1893  CA  ARG A 286      14.418   4.300  26.394  1.00 13.37           C  \\nATOM   1894  C   ARG A 286      13.654   3.159  25.752  1.00 13.99           C  \\nATOM   1895  O   ARG A 286      12.837   3.365  24.855  1.00 14.48           O  \\nATOM   1896  CB  ARG A 286      13.516   4.954  27.437  1.00 14.82           C  \\nATOM   1897  CG  ARG A 286      14.049   6.249  28.005  1.00 14.23           C  \\nATOM   1898  CD  ARG A 286      13.145   6.776  29.082  1.00 16.32           C  \\nATOM   1899  NE  ARG A 286      13.674   8.042  29.569  1.00 15.42           N  \\nATOM   1900  CZ  ARG A 286      13.313   8.628  30.700  1.00 18.27           C  \\nATOM   1901  NH1 ARG A 286      12.467   8.032  31.535  1.00 17.87           N  \\nATOM   1902  NH2 ARG A 286      13.754   9.853  30.954  1.00 18.60           N  \\nATOM   1903  N   THR A 287      13.843   1.957  26.281  1.00 13.04           N  \\nATOM   1904  CA  THR A 287      12.997   0.837  25.879  1.00 13.40           C  \\nATOM   1905  C   THR A 287      12.986  -0.217  26.978  1.00 14.27           C  \\nATOM   1906  O   THR A 287      13.945  -0.297  27.769  1.00 14.86           O  \\nATOM   1907  CB  THR A 287      13.504   0.230  24.560  1.00 13.60           C  \\nATOM   1908  OG1 THR A 287      12.617  -0.805  24.119  1.00 12.63           O  \\nATOM   1909  CG2 THR A 287      14.916  -0.320  24.717  1.00 13.40           C  \\nATOM   1910  N   ASN A 288      11.962  -1.077  26.973  1.00 12.99           N  \\nATOM   1911  CA  ASN A 288      12.045  -2.332  27.728  1.00 12.05           C  \\nATOM   1912  C   ASN A 288      12.652  -3.503  26.958  1.00 12.48           C  \\nATOM   1913  O   ASN A 288      12.816  -4.596  27.521  1.00 13.79           O  \\nATOM   1914  CB  ASN A 288      10.699  -2.702  28.364  1.00 13.25           C  \\nATOM   1915  CG  ASN A 288       9.577  -2.813  27.358  1.00 11.91           C  \\nATOM   1916  OD1 ASN A 288       9.761  -2.574  26.174  1.00 12.51           O  \\nATOM   1917  ND2 ASN A 288       8.399  -3.204  27.838  1.00 11.19           N  \\nATOM   1918  N   ASN A 289      13.095  -3.247  25.727  1.00 14.22           N  \\nATOM   1919  CA  ASN A 289      13.905  -4.199  24.956  1.00 12.82           C  \\nATOM   1920  C   ASN A 289      13.230  -5.554  24.749  1.00 13.86           C  \\nATOM   1921  O   ASN A 289      13.911  -6.554  24.515  1.00 14.14           O  \\nATOM   1922  CB  ASN A 289      15.282  -4.429  25.609  1.00 13.03           C  \\nATOM   1923  CG  ASN A 289      16.372  -3.536  25.038  1.00 14.05           C  \\nATOM   1924  OD1 ASN A 289      17.292  -3.109  25.753  1.00 17.24           O  \\nATOM   1925  ND2 ASN A 289      16.316  -3.306  23.750  1.00 11.32           N  \\nATOM   1926  N   ARG A 290      11.906  -5.590  24.784  1.00 12.67           N  \\nATOM   1927  CA  ARG A 290      11.210  -6.853  24.543  1.00 12.75           C  \\nATOM   1928  C   ARG A 290      11.427  -7.369  23.129  1.00 14.87           C  \\nATOM   1929  O   ARG A 290      11.429  -8.583  22.904  1.00 15.36           O  \\nATOM   1930  CB  ARG A 290       9.726  -6.724  24.876  1.00 13.69           C  \\nATOM   1931  CG  ARG A 290       9.476  -6.616  26.370  1.00 15.87           C  \\nATOM   1932  CD  ARG A 290       7.987  -6.671  26.710  1.00 15.41           C  \\nATOM   1933  NE  ARG A 290       7.396  -7.980  26.420  1.00 13.97           N  \\nATOM   1934  CZ  ARG A 290       7.395  -9.010  27.258  1.00 14.09           C  \\nATOM   1935  NH1 ARG A 290       7.984  -8.920  28.437  1.00 17.11           N  \\nATOM   1936  NH2 ARG A 290       6.803 -10.149  26.900  1.00 15.91           N  \\nATOM   1937  N   HIS A 291      11.706  -6.452  22.201  1.00 14.72           N  \\nATOM   1938  CA  HIS A 291      11.853  -6.763  20.779  1.00 13.19           C  \\nATOM   1939  C   HIS A 291      13.326  -6.994  20.403  1.00 14.13           C  \\nATOM   1940  O   HIS A 291      13.643  -7.190  19.229  1.00 16.03           O  \\nATOM   1941  CB  HIS A 291      11.278  -5.610  19.929  1.00 15.31           C  \\nATOM   1942  CG  HIS A 291      11.954  -4.289  20.163  1.00 15.08           C  \\nATOM   1943  ND1 HIS A 291      11.887  -3.618  21.366  1.00 14.95           N  \\nATOM   1944  CD2 HIS A 291      12.788  -3.566  19.377  1.00 14.15           C  \\nATOM   1945  CE1 HIS A 291      12.660  -2.547  21.317  1.00 14.94           C  \\nATOM   1946  NE2 HIS A 291      13.176  -2.463  20.103  1.00 14.50           N  \\nATOM   1947  N   CYS A 292      14.188  -7.143  21.412  1.00 14.33           N  \\nATOM   1948  CA  CYS A 292      15.632  -7.256  21.202  1.00 15.22           C  \\nATOM   1949  C   CYS A 292      16.275  -8.459  21.885  1.00 18.60           C  \\nATOM   1950  O   CYS A 292      17.474  -8.437  22.189  1.00 25.75           O  \\nATOM   1951  CB  CYS A 292      16.340  -5.989  21.695  1.00 13.03           C  \\nATOM   1952  SG  CYS A 292      15.764  -4.445  20.903  1.00 14.24           S  \\nATOM   1953  N   GLN A 293      15.518  -9.531  22.076  1.00 18.53           N  \\nATOM   1954  CA  GLN A 293      16.109 -10.729  22.661  1.00 21.64           C  \\nATOM   1955  C   GLN A 293      17.208 -11.303  21.785  1.00 22.85           C  \\nATOM   1956  O   GLN A 293      17.123 -11.213  20.559  1.00 26.04           O  \\nATOM   1957  CB  GLN A 293      15.054 -11.785  22.882  1.00 21.59           C  \\nATOM   1958  CG  GLN A 293      14.054 -11.408  23.929  1.00 19.13           C  \\nATOM   1959  CD  GLN A 293      12.868 -12.337  23.915  1.00 19.03           C  \\nATOM   1960  OE1 GLN A 293      12.135 -12.377  22.932  1.00 18.39           O  \\nATOM   1961  NE2 GLN A 293      12.803 -13.232  24.901  1.00 16.69           N  \\nATOM   1962  OXT GLN A 293      18.098 -11.998  22.293  1.00 27.75           O  \\nTER    1963      GLN A 293                                                      \\nHETATM 1964  S   SCN A 301      10.134  -3.022  -3.643  1.00 18.08           S  \\nHETATM 1965  C   SCN A 301       8.906  -4.240  -4.067  1.00 17.44           C  \\nHETATM 1966  N   SCN A 301       8.096  -5.052  -4.325  1.00 19.97           N  \\nHETATM 1967  S   SCN A 302      -7.830  26.388  14.724  1.00 16.65           S  \\nHETATM 1968  C   SCN A 302      -6.768  26.643  13.309  1.00 13.89           C  \\nHETATM 1969  N   SCN A 302      -6.089  26.800  12.362  1.00 15.25           N  \\nHETATM 1970  S   SCN A 303     -17.849   4.124   6.314  1.00 21.15           S  \\nHETATM 1971  C   SCN A 303     -17.707   5.737   7.103  1.00 23.48           C  \\nHETATM 1972  N   SCN A 303     -17.633   6.804   7.616  1.00 22.75           N  \\nHETATM 1973  S   SCN A 304      21.308   4.602  16.307  1.00 42.52           S  \\nHETATM 1974  C   SCN A 304      22.292   3.223  16.913  1.00 35.17           C  \\nHETATM 1975  N   SCN A 304      22.922   2.313  17.312  1.00 36.86           N  \\nHETATM 1976  O   HOH A 401       2.501  -3.110   5.115  1.00 13.13           O  \\nHETATM 1977  O   HOH A 402       5.309  15.174  24.789  1.00 12.47           O  \\nHETATM 1978  O   HOH A 403       7.343  15.970  22.731  1.00 13.26           O  \\nHETATM 1979  O   HOH A 404      11.816   0.459   3.095  1.00 12.12           O  \\nHETATM 1980  O   HOH A 405      21.956   1.038   7.613  1.00 15.26           O  \\nHETATM 1981  O   HOH A 406       9.680  -0.971   1.896  1.00 12.61           O  \\nHETATM 1982  O   HOH A 407      11.783  -2.935   3.423  1.00 13.43           O  \\nHETATM 1983  O   HOH A 408       2.886  -5.771   4.136  1.00 13.44           O  \\nHETATM 1984  O   HOH A 409      -2.002  19.349  20.171  1.00 19.40           O  \\nHETATM 1985  O   HOH A 410     -13.166  16.803  22.491  1.00 17.49           O  \\nHETATM 1986  O   HOH A 411      -7.369  18.872  14.627  1.00 14.31           O  \\nHETATM 1987  O   HOH A 412      19.378  -4.342   6.064  1.00 15.50           O  \\nHETATM 1988  O   HOH A 413       3.565  18.229   5.438  1.00 17.71           O  \\nHETATM 1989  O   HOH A 414      -0.909  -8.851   3.314  1.00 15.51           O  \\nHETATM 1990  O   HOH A 415       1.172  -5.979  28.159  1.00 14.95           O  \\nHETATM 1991  O   HOH A 416       2.039  -6.514  24.479  1.00 17.93           O  \\nHETATM 1992  O   HOH A 417      -0.675  17.826  13.670  1.00 15.12           O  \\nHETATM 1993  O   HOH A 418      21.136  -1.500   7.280  1.00 16.25           O  \\nHETATM 1994  O   HOH A 419       5.531  -2.562   3.946  1.00 14.42           O  \\nHETATM 1995  O   HOH A 420      -3.483 -10.997   6.702  1.00 15.99           O  \\nHETATM 1996  O   HOH A 421       1.084   5.569  29.981  1.00 15.75           O  \\nHETATM 1997  O   HOH A 422      16.822   0.259   1.843  1.00 15.92           O  \\nHETATM 1998  O   HOH A 423      10.234  -3.589  23.624  1.00 15.39           O  \\nHETATM 1999  O   HOH A 424       8.074  23.520  21.081  1.00 15.43           O  \\nHETATM 2000  O   HOH A 425     -12.299  14.322  23.874  1.00 18.24           O  \\nHETATM 2001  O   HOH A 426       8.013  -5.029   5.113  1.00 13.34           O  \\nHETATM 2002  O   HOH A 427      17.470  -6.365  11.992  1.00 15.99           O  \\nHETATM 2003  O   HOH A 428      19.105   0.051  22.903  1.00 18.22           O  \\nHETATM 2004  O   HOH A 429       7.366  12.556   2.093  1.00 18.57           O  \\nHETATM 2005  O   HOH A 430      -2.625  14.875   4.636  1.00 16.37           O  \\nHETATM 2006  O   HOH A 431      -4.443  -5.207  13.196  1.00 15.84           O  \\nHETATM 2007  O   HOH A 432       3.382  -9.224  26.962  1.00 16.42           O  \\nHETATM 2008  O   HOH A 433       8.960 -11.815  11.660  1.00 20.02           O  \\nHETATM 2009  O   HOH A 434      18.913  -3.604  22.582  1.00 16.96           O  \\nHETATM 2010  O   HOH A 435      -4.256  -8.297  15.603  1.00 17.89           O  \\nHETATM 2011  O   HOH A 436      -8.202  18.633  11.344  1.00 16.29           O  \\nHETATM 2012  O   HOH A 437      11.202  18.460  11.347  1.00 19.44           O  \\nHETATM 2013  O   HOH A 438      -5.956  19.295   5.381  1.00 18.31           O  \\nHETATM 2014  O   HOH A 439      14.494   5.627   0.349  1.00 16.06           O  \\nHETATM 2015  O   HOH A 440      -2.521  22.481  10.210  1.00 14.86           O  \\nHETATM 2016  O   HOH A 441      10.839 -10.161   8.088  1.00 19.82           O  \\nHETATM 2017  O   HOH A 442       5.316 -13.136   9.538  1.00 15.41           O  \\nHETATM 2018  O   HOH A 443      10.122  18.785  29.368  1.00 20.71           O  \\nHETATM 2019  O   HOH A 444      10.056  19.361  26.534  1.00 18.16           O  \\nHETATM 2020  O   HOH A 445     -16.566   5.093  10.144  1.00 21.26           O  \\nHETATM 2021  O   HOH A 446      19.914  -1.687  25.034  1.00 19.55           O  \\nHETATM 2022  O   HOH A 447      14.204   1.555   2.100  1.00 13.60           O  \\nHETATM 2023  O   HOH A 448      -1.916  20.588  22.734  1.00 19.85           O  \\nHETATM 2024  O   HOH A 449      18.502  -7.294   9.449  1.00 18.65           O  \\nHETATM 2025  O   HOH A 450       1.664  -8.556  11.411  1.00 17.62           O  \\nHETATM 2026  O   HOH A 451      -4.632  -2.766  24.338  1.00 17.64           O  \\nHETATM 2027  O   HOH A 452       5.441  -4.671  -5.467  1.00 19.71           O  \\nHETATM 2028  O   HOH A 453      -9.565  13.770  25.926  1.00 18.96           O  \\nHETATM 2029  O   HOH A 454      18.002  -6.599   6.815  1.00 19.51           O  \\nHETATM 2030  O   HOH A 455       6.183  -2.307  -6.617  1.00 18.31           O  \\nHETATM 2031  O   HOH A 456      -4.402  24.415  10.940  1.00 15.72           O  \\nHETATM 2032  O   HOH A 457       6.440  -2.954  36.171  1.00 16.99           O  \\nHETATM 2033  O   HOH A 458      18.569   3.761  15.800  1.00 20.39           O  \\nHETATM 2034  O   HOH A 459       2.337  24.112  27.120  1.00 18.10           O  \\nHETATM 2035  O   HOH A 460       4.313  26.542  23.430  1.00 18.83           O  \\nHETATM 2036  O   HOH A 461      19.430   4.692  25.898  1.00 18.02           O  \\nHETATM 2037  O   HOH A 462     -17.230  17.780   2.190  1.00 28.37           O  \\nHETATM 2038  O   HOH A 463      -1.344 -14.842   6.471  1.00 22.90           O  \\nHETATM 2039  O   HOH A 464      -4.693  -0.392  -6.433  1.00 23.06           O  \\nHETATM 2040  O   HOH A 465       6.360  25.419  19.828  1.00 21.63           O  \\nHETATM 2041  O   HOH A 466      20.959   5.032  23.577  1.00 23.10           O  \\nHETATM 2042  O   HOH A 467      20.477   6.937  21.530  1.00 18.93           O  \\nHETATM 2043  O   HOH A 468       0.066   5.421  36.865  1.00 24.14           O  \\nHETATM 2044  O   HOH A 469      -3.422  27.137  13.251  1.00 18.59           O  \\nHETATM 2045  O   HOH A 470      16.754  -1.036  22.144  1.00 17.16           O  \\nHETATM 2046  O   HOH A 471      -9.176  -4.337   1.870  1.00 22.05           O  \\nHETATM 2047  O   HOH A 472     -10.524   3.502  -7.053  1.00 23.85           O  \\nHETATM 2048  O   HOH A 473       8.591  -1.062  -5.784  1.00 15.37           O  \\nHETATM 2049  O   HOH A 474      -1.264  -5.904  29.332  1.00 24.09           O  \\nHETATM 2050  O   HOH A 475      -1.610  25.017  13.866  1.00 18.61           O  \\nHETATM 2051  O   HOH A 476      -2.491   1.082  -6.497  1.00 23.02           O  \\nHETATM 2052  O   HOH A 477     -16.014  13.924  19.970  1.00 23.23           O  \\nHETATM 2053  O   HOH A 478       0.306  -9.520  -0.886  1.00 22.20           O  \\nHETATM 2054  O   HOH A 479     -10.847  20.005  11.321  1.00 18.22           O  \\nHETATM 2055  O   HOH A 480      -1.701  15.720   0.413  1.00 21.37           O  \\nHETATM 2056  O   HOH A 481      15.931  15.855   9.805  1.00 21.09           O  \\nHETATM 2057  O   HOH A 482      17.178   6.986  29.390  1.00 24.03           O  \\nHETATM 2058  O   HOH A 483       8.743  18.851   4.573  1.00 23.66           O  \\nHETATM 2059  O   HOH A 484     -17.453  14.527   8.793  1.00 20.97           O  \\nHETATM 2060  O   HOH A 485       0.600  21.082  14.045  1.00 21.89           O  \\nHETATM 2061  O   HOH A 486      10.900  -7.820   1.205  1.00 19.86           O  \\nHETATM 2062  O   HOH A 487      12.268  11.510  -0.858  1.00 26.19           O  \\nHETATM 2063  O   HOH A 488       1.083   8.445  -8.133  1.00 27.32           O  \\nHETATM 2064  O   HOH A 489      16.135  14.277  12.370  1.00 24.27           O  \\nHETATM 2065  O   HOH A 490      -0.663  22.911  12.311  1.00 18.54           O  \\nHETATM 2066  O   HOH A 491       0.990  -7.962  26.406  1.00 21.30           O  \\nHETATM 2067  O   HOH A 492       8.324   6.586  -3.826  1.00 23.36           O  \\nHETATM 2068  O   HOH A 493      20.712   0.557  16.420  1.00 23.91           O  \\nHETATM 2069  O   HOH A 494      23.643  -1.037  17.024  1.00 23.74           O  \\nHETATM 2070  O   HOH A 495      21.220  -4.729  16.862  1.00 27.42           O  \\nHETATM 2071  O   HOH A 496      -1.273  22.591   7.822  1.00 17.52           O  \\nHETATM 2072  O   HOH A 497      -0.367   0.826  32.231  1.00 25.21           O  \\nHETATM 2073  O   HOH A 498       2.493 -10.643  29.119  1.00 21.11           O  \\nHETATM 2074  O   HOH A 499     -10.754  -3.266  -5.058  1.00 24.92           O  \\nHETATM 2075  O   HOH A 500     -12.636  21.290  22.859  1.00 23.87           O  \\nHETATM 2076  O   HOH A 501      -1.294  -9.137  25.736  1.00 26.15           O  \\nHETATM 2077  O   HOH A 502       1.396   1.395  34.892  1.00 26.64           O  \\nHETATM 2078  O   HOH A 503     -12.552   7.632  22.969  1.00 21.60           O  \\nHETATM 2079  O   HOH A 504      12.606  20.986  15.629  1.00 24.03           O  \\nHETATM 2080  O   HOH A 505      -2.853  10.469  -6.927  1.00 25.47           O  \\nHETATM 2081  O   HOH A 506      -8.357  -6.848   3.614  1.00 26.25           O  \\nHETATM 2082  O   HOH A 507       9.143  -6.590  29.925  1.00 21.43           O  \\nHETATM 2083  O   HOH A 508       0.451 -10.861   1.990  1.00 22.95           O  \\nHETATM 2084  O   HOH A 509       5.831   7.701  36.050  1.00 22.50           O  \\nHETATM 2085  O   HOH A 510      -7.752  27.141  21.641  1.00 29.31           O  \\nHETATM 2086  O   HOH A 511      18.987  -6.261  16.350  1.00 27.44           O  \\nHETATM 2087  O   HOH A 512      -4.420  23.292   6.128  1.00 23.72           O  \\nHETATM 2088  O   HOH A 513      25.288   5.604  16.179  1.00 22.84           O  \\nHETATM 2089  O   HOH A 514     -11.086  16.943   5.223  1.00 23.65           O  \\nHETATM 2090  O   HOH A 515       3.451  -7.997  15.203  1.00 24.84           O  \\nHETATM 2091  O   HOH A 516      11.696 -12.740  11.405  1.00 23.03           O  \\nHETATM 2092  O   HOH A 517      -6.488  -2.080  27.440  1.00 22.75           O  \\nHETATM 2093  O   HOH A 518      -0.121  -6.719  21.067  1.00 21.24           O  \\nHETATM 2094  O   HOH A 519      14.136  13.190   2.560  1.00 25.34           O  \\nHETATM 2095  O   HOH A 520       1.867  -6.237  17.068  1.00 21.13           O  \\nHETATM 2096  O   HOH A 521      -3.230  10.384  34.807  1.00 24.69           O  \\nHETATM 2097  O   HOH A 522      11.792  -6.194  29.485  1.00 25.21           O  \\nHETATM 2098  O   HOH A 523     -15.116  16.691  20.592  1.00 24.44           O  \\nHETATM 2099  O   HOH A 524     -16.634  16.454  10.080  1.00 23.10           O  \\nHETATM 2100  O   HOH A 525       3.151  22.266  31.121  1.00 28.48           O  \\nHETATM 2101  O   HOH A 526     -16.330  17.889  17.292  1.00 23.95           O  \\nHETATM 2102  O   HOH A 527       7.954  -6.110  -1.655  1.00 23.49           O  \\nHETATM 2103  O   HOH A 528      14.086  12.796   6.286  1.00 26.52           O  \\nHETATM 2104  O   HOH A 529      -7.493  13.711  32.372  1.00 25.17           O  \\nHETATM 2105  O   HOH A 530     -12.035   0.794  14.003  1.00 23.16           O  \\nHETATM 2106  O   HOH A 531      -3.309  -4.005  28.548  1.00 27.07           O  \\nHETATM 2107  O   HOH A 532      23.636   3.850  12.431  1.00 24.41           O  \\nHETATM 2108  O   HOH A 533      -1.323  17.756  31.679  1.00 25.85           O  \\nHETATM 2109  O   HOH A 534       2.919  10.080  35.948  1.00 30.32           O  \\nHETATM 2110  O   HOH A 535     -13.499   3.190  -6.466  1.00 25.42           O  \\nHETATM 2111  O   HOH A 536      -9.742   3.373  28.208  1.00 27.52           O  \\nHETATM 2112  O   HOH A 537      -8.443 -12.581   5.987  1.00 26.58           O  \\nHETATM 2113  O   HOH A 538      19.675   1.968  24.856  1.00 27.33           O  \\nHETATM 2114  O   HOH A 539     -10.748   1.020  16.643  1.00 28.53           O  \\nHETATM 2115  O   HOH A 540      10.710  24.482  21.216  1.00 24.80           O  \\nHETATM 2116  O   HOH A 541       1.526   6.287  -9.175  1.00 29.12           O  \\nHETATM 2117  O   HOH A 542       1.476  22.563   8.363  1.00 28.80           O  \\nHETATM 2118  O   HOH A 543     -18.869   7.647  13.414  1.00 27.84           O  \\nHETATM 2119  O   HOH A 544      23.030   5.469  14.548  1.00 27.58           O  \\nHETATM 2120  O   HOH A 545     -18.758  -0.963  11.170  1.00 25.32           O  \\nHETATM 2121  O   HOH A 546      14.813 -12.928  17.607  1.00 32.33           O  \\nHETATM 2122  O   HOH A 547      12.802 -10.178  21.211  1.00 22.66           O  \\nHETATM 2123  O   HOH A 548      -0.057  -8.931 -10.914  1.00 27.52           O  \\nHETATM 2124  O   HOH A 549      -1.963   8.552  35.625  1.00 29.30           O  \\nHETATM 2125  O   HOH A 550      -9.030  12.516  -4.703  1.00 27.73           O  \\nHETATM 2126  O   HOH A 551     -18.681  12.548  13.638  1.00 25.24           O  \\nHETATM 2127  O   HOH A 552      -0.243 -10.581  11.152  1.00 25.28           O  \\nHETATM 2128  O   HOH A 553       4.922   8.889  -5.839  1.00 28.28           O  \\nHETATM 2129  O   HOH A 554       7.971  21.802  11.085  1.00 24.35           O  \\nHETATM 2130  O   HOH A 555      -5.424  -5.322  30.533  1.00 27.23           O  \\nHETATM 2131  O   HOH A 556     -10.024  -0.583  -9.006  1.00 28.43           O  \\nHETATM 2132  O   HOH A 557       2.100  -7.448  19.435  1.00 25.60           O  \\nHETATM 2133  O   HOH A 558      -0.784  20.077  16.468  1.00 28.65           O  \\nHETATM 2134  O   HOH A 559      -1.928  -8.578  16.955  1.00 23.70           O  \\nHETATM 2135  O   HOH A 560      14.192 -12.637  27.413  1.00 25.48           O  \\nHETATM 2136  O   HOH A 561      -6.498   0.431  33.347  1.00 27.98           O  \\nHETATM 2137  O   HOH A 562      19.351  -3.046   0.308  1.00 28.79           O  \\nHETATM 2138  O   HOH A 563      -6.013  -7.081  -3.461  1.00 28.32           O  \\nHETATM 2139  O   HOH A 564      -3.567  17.011   3.408  1.00 28.57           O  \\nHETATM 2140  O   HOH A 565      -6.394  -9.681   3.971  1.00 34.38           O  \\nHETATM 2141  O   HOH A 566       8.811  12.931  33.719  1.00 29.23           O  \\nHETATM 2142  O   HOH A 567      15.008 -10.760   9.614  1.00 31.45           O  \\nHETATM 2143  O   HOH A 568      21.883  -2.507  15.470  1.00 24.02           O  \\nHETATM 2144  O   HOH A 569       6.852   0.466  32.955  1.00 29.52           O  \\nHETATM 2145  O   HOH A 570      -4.707   4.910  -8.120  1.00 27.23           O  \\nHETATM 2146  O   HOH A 571      -3.504   2.404  -8.596  1.00 27.75           O  \\nHETATM 2147  O   HOH A 572     -13.398  -4.656   1.365  1.00 28.37           O  \\nHETATM 2148  O   HOH A 573      -0.106 -15.635   4.230  1.00 34.98           O  \\nHETATM 2149  O   HOH A 574      15.285  -2.401  28.819  1.00 24.74           O  \\nHETATM 2150  O   HOH A 575      -5.720  13.739  -2.409  1.00 29.03           O  \\nHETATM 2151  O   HOH A 576      -7.569  16.495  28.679  1.00 25.62           O  \\nHETATM 2152  O   HOH A 577       1.479  23.703  10.810  1.00 26.64           O  \\nHETATM 2153  O   HOH A 578      17.609  15.705  16.723  1.00 31.63           O  \\nHETATM 2154  O   HOH A 579      13.891  18.697  27.718  1.00 28.57           O  \\nHETATM 2155  O   HOH A 580     -14.612   6.083  -1.217  1.00 30.22           O  \\nHETATM 2156  O   HOH A 581      19.825  -6.150  22.871  1.00 34.20           O  \\nHETATM 2157  O   HOH A 582      -9.699  11.555  29.409  1.00 39.22           O  \\nHETATM 2158  O   HOH A 583      -2.479  -6.180  21.862  1.00 28.86           O  \\nHETATM 2159  O   HOH A 584     -11.173  -4.142   3.771  1.00 30.52           O  \\nHETATM 2160  O   HOH A 585      17.649  -9.768   9.689  1.00 28.69           O  \\nHETATM 2161  O   HOH A 586     -10.961  -1.286  12.487  1.00 32.82           O  \\nHETATM 2162  O   HOH A 587     -17.892  -2.726   3.009  1.00 32.10           O  \\nHETATM 2163  O   HOH A 588       9.193  -9.550   0.038  1.00 28.88           O  \\nHETATM 2164  O   HOH A 589      17.898  -6.609  24.861  1.00 26.39           O  \\nHETATM 2165  O   HOH A 590     -17.676  -2.989  13.226  1.00 26.07           O  \\nHETATM 2166  O   HOH A 591       4.103  -7.544  20.887  1.00 29.02           O  \\nHETATM 2167  O   HOH A 592     -14.267  12.342  23.813  1.00 32.73           O  \\nHETATM 2168  O   HOH A 593      11.100 -13.283   4.523  1.00 27.20           O  \\nHETATM 2169  O   HOH A 594       7.962  13.439  -1.003  1.00 30.25           O  \\nHETATM 2170  O   HOH A 595       0.711  11.272  -5.311  1.00 29.23           O  \\nHETATM 2171  O   HOH A 596       1.214  20.430   4.012  1.00 24.79           O  \\nHETATM 2172  O   HOH A 597       2.134 -10.141  15.005  1.00 34.68           O  \\nHETATM 2173  O   HOH A 598      15.100  -8.417  26.239  1.00 29.00           O  \\nHETATM 2174  O   HOH A 599      14.835  12.463  28.616  1.00 28.58           O  \\nHETATM 2175  O   HOH A 600      21.483  -6.407   2.637  1.00 30.79           O  \\nHETATM 2176  O   HOH A 601     -10.082  -7.680   9.068  1.00 33.33           O  \\nHETATM 2177  O   HOH A 602       4.791  10.719  -3.907  1.00 33.33           O  \\nHETATM 2178  O   HOH A 603       3.568  23.167  12.131  1.00 31.18           O  \\nHETATM 2179  O   HOH A 604     -14.844  11.197  -0.637  1.00 30.26           O  \\nHETATM 2180  O   HOH A 605      18.404  -8.909   5.520  1.00 36.71           O  \\nHETATM 2181  O   HOH A 606      18.237  -8.079  18.750  1.00 33.86           O  \\nHETATM 2182  O   HOH A 607      17.892  11.278   2.623  1.00 33.03           O  \\nHETATM 2183  O   HOH A 608      -4.746  -5.126  26.442  1.00 31.95           O  \\nHETATM 2184  O   HOH A 609      11.423  13.535  33.345  1.00 34.37           O  \\nHETATM 2185  O   HOH A 610       8.019  11.105  35.485  1.00 30.19           O  \\nHETATM 2186  O   HOH A 611     -13.051  25.593   9.302  1.00 25.71           O  \\nHETATM 2187  O   HOH A 612       0.455  26.295  14.856  1.00 29.01           O  \\nHETATM 2188  O   HOH A 613      22.425  12.944  22.904  1.00 26.31           O  \\nHETATM 2189  O   HOH A 614       0.668 -10.203  31.075  1.00 27.19           O  \\nHETATM 2190  O   HOH A 615      20.896   3.083   0.992  1.00 30.51           O  \\nHETATM 2191  O   HOH A 616      11.821  20.916  13.026  1.00 30.52           O  \\nHETATM 2192  O   HOH A 617      25.406  -3.335  17.780  1.00 30.97           O  \\nHETATM 2193  O   HOH A 618      20.235  18.958  23.244  1.00 30.81           O  \\nHETATM 2194  O   HOH A 619       0.080  20.806  18.810  1.00 31.80           O  \\nHETATM 2195  O   HOH A 620     -16.417  15.260  17.238  1.00 32.26           O  \\nHETATM 2196  O   HOH A 621       2.669 -11.179  -1.000  1.00 33.81           O  \\nHETATM 2197  O   HOH A 622      17.095 -13.172  18.671  1.00 41.08           O  \\nHETATM 2198  O   HOH A 623     -18.283  13.408  11.025  1.00 39.75           O  \\nHETATM 2199  O   HOH A 624     -10.963  23.384   7.462  1.00 31.27           O  \\nHETATM 2200  O   HOH A 625       3.066 -13.018   0.735  1.00 29.41           O  \\nHETATM 2201  O   HOH A 626      -2.431  14.736  -3.787  1.00 31.91           O  \\nHETATM 2202  O   HOH A 627       8.359 -11.212  30.088  1.00 30.37           O  \\nHETATM 2203  O   HOH A 628      21.199  -6.911   9.685  1.00 32.86           O  \\nHETATM 2204  O   HOH A 629      21.847  15.540  17.463  1.00 36.92           O  \\nHETATM 2205  O   HOH A 630     -11.951  15.001  26.505  1.00 30.61           O  \\nHETATM 2206  O   HOH A 631     -11.851   4.553  26.798  1.00 27.48           O  \\nHETATM 2207  O   HOH A 632      11.620 -12.860   8.761  1.00 31.54           O  \\nHETATM 2208  O   HOH A 633      22.639  -4.493   8.859  1.00 32.32           O  \\nHETATM 2209  O   HOH A 634       2.930  23.060  33.894  1.00 38.31           O  \\nHETATM 2210  O   HOH A 635     -21.436  12.482   7.060  1.00 38.57           O  \\nHETATM 2211  O   HOH A 636      19.009  -7.230  13.877  1.00 29.90           O  \\nHETATM 2212  O   HOH A 637      -3.256  -6.443  -9.515  1.00 37.20           O  \\nHETATM 2213  O   HOH A 638      -8.924  11.048  -6.823  1.00 37.80           O  \\nHETATM 2214  O   HOH A 639      -9.397  -2.731  14.301  1.00 31.73           O  \\nHETATM 2215  O   HOH A 640      -4.774  -5.226  -7.517  1.00 32.79           O  \\nHETATM 2216  O   HOH A 641      -0.495  22.963   3.617  1.00 27.62           O  \\nHETATM 2217  O   HOH A 642      -4.757  -4.432  33.569  1.00 35.12           O  \\nHETATM 2218  O   HOH A 643      -7.531   7.004  -5.586  1.00 31.28           O  \\nHETATM 2219  O   HOH A 644      12.681  15.494  31.199  1.00 35.93           O  \\nHETATM 2220  O   HOH A 645      11.980  24.366  25.647  1.00 35.08           O  \\nHETATM 2221  O   HOH A 646     -12.915  -0.888  23.525  1.00 32.83           O  \\nHETATM 2222  O   HOH A 647      12.605  17.794  30.316  1.00 33.13           O  \\nHETATM 2223  O   HOH A 648     -12.069  -2.223  19.974  1.00 36.20           O  \\nHETATM 2224  O   HOH A 649      13.668   7.953  -2.739  1.00 31.23           O  \\nHETATM 2225  O   HOH A 650      -1.869  -6.215  18.998  1.00 29.49           O  \\nHETATM 2226  O   HOH A 651     -12.349   4.060  19.763  1.00 34.56           O  \\nHETATM 2227  O   HOH A 652       6.876 -11.384   1.592  1.00 30.36           O  \\nHETATM 2228  O   HOH A 653      -3.157  15.529  34.431  1.00 34.43           O  \\nHETATM 2229  O   HOH A 654      -5.776  13.454  -4.908  1.00 31.05           O  \\nHETATM 2230  O   HOH A 655     -22.154  13.467  11.829  1.00 32.13           O  \\nHETATM 2231  O   HOH A 656      16.818   6.255  -0.939  1.00 35.70           O  \\nHETATM 2232  O   HOH A 657       1.575 -12.236  12.905  1.00 28.98           O  \\nHETATM 2233  O   HOH A 658      -1.734  -3.906  32.294  1.00 33.82           O  \\nHETATM 2234  O   HOH A 659      -1.981  -9.511  12.505  1.00 30.75           O  \\nHETATM 2235  O   HOH A 660      -9.982  20.774   7.643  1.00 40.66           O  \\nHETATM 2236  O   HOH A 661      -1.633   3.147 -10.679  1.00 32.56           O  \\nHETATM 2237  O   HOH A 662     -11.218  -4.492  23.129  1.00 33.42           O  \\nHETATM 2238  O   HOH A 663      -6.246  -0.642  -8.555  1.00 33.34           O  \\nHETATM 2239  O   HOH A 664       8.636  26.185  18.066  1.00 39.09           O  \\nHETATM 2240  O   HOH A 665       0.289  19.426   1.506  1.00 33.45           O  \\nHETATM 2241  O   HOH A 666      24.958   0.561  20.504  1.00 38.90           O  \\nHETATM 2242  O   HOH A 667      14.401 -11.408   2.783  1.00 28.78           O  \\nHETATM 2243  O   HOH A 668     -12.040  19.788   5.161  1.00 35.49           O  \\nHETATM 2244  O   HOH A 669       1.190  -6.798  32.445  1.00 33.64           O  \\nHETATM 2245  O   HOH A 670      16.759  15.132  27.840  1.00 35.09           O  \\nHETATM 2246  O   HOH A 671      12.048  19.898   7.782  1.00 34.90           O  \\nHETATM 2247  O   HOH A 672      -8.324  -7.103  -4.872  1.00 37.96           O  \\nHETATM 2248  O   HOH A 673      17.083  21.360  23.947  1.00 40.58           O  \\nHETATM 2249  O   HOH A 674      21.075  10.005  13.881  1.00 31.41           O  \\nHETATM 2250  O   HOH A 675       5.853  19.591   5.392  1.00 42.35           O  \\nHETATM 2251  O   HOH A 676      10.144  24.384  14.904  1.00 37.30           O  \\nHETATM 2252  O   HOH A 677     -14.221  20.062  20.108  1.00 32.96           O  \\nHETATM 2253  O   HOH A 678       2.680  14.556   0.695  1.00 34.21           O  \\nHETATM 2254  O   HOH A 679       9.223  14.918   2.885  1.00 33.96           O  \\nHETATM 2255  O   HOH A 680      13.358  11.418  33.226  1.00 35.89           O  \\nHETATM 2256  O   HOH A 681      23.633  18.090  21.121  1.00 36.63           O  \\nHETATM 2257  O   HOH A 682      15.387 -13.695  10.254  1.00 33.76           O  \\nHETATM 2258  O   HOH A 683      14.328 -11.970  14.778  1.00 33.87           O  \\nHETATM 2259  O   HOH A 684      17.334   8.866  -1.651  1.00 35.14           O  \\nHETATM 2260  O   HOH A 685       4.615  16.131   1.198  1.00 36.92           O  \\nHETATM 2261  O   HOH A 686      -8.578  -5.606  11.346  1.00 30.81           O  \\nHETATM 2262  O   HOH A 687      19.659   5.627   0.375  1.00 31.39           O  \\nHETATM 2263  O   HOH A 688      -9.265  16.184   3.163  1.00 38.30           O  \\nHETATM 2264  O   HOH A 689      13.957 -11.331   6.586  1.00 33.90           O  \\nHETATM 2265  O   HOH A 690       6.051   9.677  -2.343  1.00 31.22           O  \\nHETATM 2266  O   HOH A 691      -9.651   5.785  -7.090  1.00 34.12           O  \\nHETATM 2267  O   HOH A 692      -7.148   9.545  -6.453  1.00 38.88           O  \\nHETATM 2268  O   HOH A 693       5.474  14.596  35.467  1.00 40.30           O  \\nHETATM 2269  O   HOH A 694      20.475  17.556  25.232  1.00 38.54           O  \\nHETATM 2270  O   HOH A 695      16.741  -9.152  13.095  1.00 36.51           O  \\nHETATM 2271  O   HOH A 696       0.082 -13.611   2.271  1.00 34.99           O  \\nHETATM 2272  O   HOH A 697      -1.617  -1.224 -10.462  1.00 38.27           O  \\nHETATM 2273  O   HOH A 698      17.042 -10.212   3.681  1.00 36.40           O  \\nHETATM 2274  O   HOH A 699      12.879  24.932  19.778  1.00 33.80           O  \\nHETATM 2275  O   HOH A 700      21.175   9.391  22.769  1.00 36.00           O  \\nHETATM 2276  O   HOH A 701      -6.919   3.260  31.656  1.00 37.31           O  \\nHETATM 2277  O   HOH A 702      -0.006  -2.257  33.486  1.00 35.12           O  \\nHETATM 2278  O   HOH A 703      21.762  -4.630   4.548  1.00 40.71           O  \\nHETATM 2279  O   HOH A 704      -1.257  -9.744  -3.237  1.00 33.87           O  \\nHETATM 2280  O   HOH A 705       9.496  22.971  27.447  1.00 39.67           O  \\nHETATM 2281  O   HOH A 706     -17.533   0.852  -0.520  1.00 40.92           O  \\nHETATM 2282  O   HOH A 707       7.197 -10.170  32.790  1.00 37.69           O  \\nHETATM 2283  O   HOH A 708      16.513  25.094  22.300  1.00 40.10           O  \\nHETATM 2284  O   HOH A 709       3.069  13.021  -6.476  1.00 35.88           O  \\nHETATM 2285  O   HOH A 710      15.347  15.896  14.729  1.00 32.31           O  \\nHETATM 2286  O   HOH A 711       0.202  22.824  16.213  1.00 31.62           O  \\nHETATM 2287  O   HOH A 712      19.570  21.797  17.267  1.00 40.23           O  \\nHETATM 2288  O   HOH A 713     -13.232  18.517  24.643  1.00 32.87           O  \\nHETATM 2289  O   HOH A 714     -12.326  18.791  26.729  1.00 37.04           O  \\nHETATM 2290  O   HOH A 715      -0.997  -0.871  -7.903  1.00 23.40           O  \\nCONECT 1815 1952                                                                \\nCONECT 1952 1815                                                                \\nCONECT 1964 1965                                                                \\nCONECT 1965 1964 1966                                                           \\nCONECT 1966 1965                                                                \\nCONECT 1967 1968                                                                \\nCONECT 1968 1967 1969                                                           \\nCONECT 1969 1968                                                                \\nCONECT 1970 1971                                                                \\nCONECT 1971 1970 1972                                                           \\nCONECT 1972 1971                                                                \\nCONECT 1973 1974                                                                \\nCONECT 1974 1973 1975                                                           \\nCONECT 1975 1974                                                                \\nMASTER      285    0    4   12    9    0    7    6 2289    1   14   20          \\nEND                                                                             \\n\",\"pdb\");\n",
              "\tviewer_17290926055486805.setStyle({\"chain\": \"A\"},{\"cartoon\": {\"color\": \"#33ff33\"}});\n",
              "\tviewer_17290926055486805.zoomTo();\n",
              "viewer_17290926055486805.render();\n",
              "});\n",
              "</script>"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "Dropdown(description='Sampling methodology:', layout=Layout(height='30px', width='500px'), options=('argmax', "
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "6947b47837da4ccba98baa1385f4d3d7"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "BoundedIntText(value=1, description='Sampling num:', layout=Layout(display='none', height='30px', width='500px"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "bcee82164a8b4a07a197ac405c3eac1f"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "Text(value='predicted_seq', description='Save name:', layout=Layout(height='30px', width='500px'), style=Descr"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "191833433bae4921b9e70bb8c07ef71c"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "HTML(value='<h3>---------------------------------------------------------------------------</h3>')"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "7010fbdcef0b4981a83185a5cdb777a2"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "Button(button_style='info', description='Predict protein sequence', layout=Layout(height='30px', width='500px'"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "2d2fb545e57048d1a8b26d16167531ef"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "HTML(value=\"<p><font color=red>Note: At any time you can click the run button <img src='https://github.com/wes"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "77b44d6bb79d42229f01c2078cb8a58b"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Predicting sequence...\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Some weights of EsmForMaskedLM were not initialized from the model checkpoint at westlake-repl/SaProt_650M_AF2_inverse_folding and are newly initialized: ['esm.embeddings.position_embeddings.weight', 'esm.contact_head.regression.weight', 'esm.contact_head.regression.bias']\n",
            "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "No lr_scheduler_kwargs provided. The default learning rate is 0.\n",
            "No optimizer_kwargs provided. The default optimizer is AdamW.\n",
            "['APPGYPPPALDPATLAKTGPYAVSTQSVPAGAVSGFSGGTVYYPAGTTGKFPVVVLSPGFTGSQSNVAWLGKHLASHGFVVLAINTPSSFDSPSTRAAQLLAALDYLTTGAPAELAARIDTSRVGVAGHSMGGGGTLIAAAQDPRIKAVIPLAPWQGNTSYKFTAPTLIIGAGNDTVAPASQHAEPYYNGLSPSVDKEYVELKGGTHFTPTQPNPDVRLLSTAWMRRYLDGDEAAAAYLDPTTPAGVSRYKDNTDLAP']\n",
            "APPGYPPPALDPATLAKTGPYAVSTQSVPAGAVSGFSGGTVYYPAGTTGKFPVVVLSPGFTGSQSNVAWLGKHLASHGFVVLAINTPSSFDSPSTRAAQLLAALDYLTTGAPAELAARIDTSRVGVAGHSMGGGGTLIAAAQDPRIKAVIPLAPWQGNTSYKFTAPTLIIGAGNDTVAPASQHAEPYYNGLSPSVDKEYVELKGGTHFTPTQPNPDVRLLSTAWMRRYLDGDEAAAAYLDPTTPAGVSRYKDNTDLAP\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "HTML(value='<html>\\n  <head>\\n  <meta name=\"viewport\" content=\"width=device-width, initial-scale=1\">\\n  </head"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "5a8c37150ef54eaabdc6ba0c797907d6"
            }
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "AdclZZ2e3-9W"
      },
      "id": "AdclZZ2e3-9W",
      "execution_count": null,
      "outputs": []
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 2
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython2",
      "version": "2.7.6"
    },
    "colab": {
      "provenance": []
    },
    "widgets": {
      "application/vnd.jupyter.widget-state+json": {
        "e9acbb732b244f549a20473a27945043": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_ded12d17c69841acbb4a9cc94e4281e9",
            "placeholder": "",
            "style": "IPY_MODEL_62ed23d0ef564e5e96ea089c6185b8d7",
            "value": "<h2>Do you want to train your own model or you only want to use existing models to make prediction?</h2>"
          }
        },
        "ded12d17c69841acbb4a9cc94e4281e9": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "62ed23d0ef564e5e96ea089c6185b8d7": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "326e7f7cb2da4f8287e31dacde987c38": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ButtonModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ButtonModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ButtonView",
            "button_style": "",
            "description": "I want to train my own model",
            "disabled": false,
            "icon": "",
            "layout": "IPY_MODEL_517461bb890441549ac514632f85e0e8",
            "style": "IPY_MODEL_6eccf22bf3ce418aafe68803aa15bb54",
            "tooltip": ""
          }
        },
        "517461bb890441549ac514632f85e0e8": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": "30px",
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": "400px"
          }
        },
        "6eccf22bf3ce418aafe68803aa15bb54": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ButtonStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ButtonStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "button_color": null,
            "font_weight": ""
          }
        },
        "31b1215d2aa64e0aaeec12cf5adba750": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ButtonModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ButtonModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ButtonView",
            "button_style": "",
            "description": "I want to use existing models to make prediction",
            "disabled": false,
            "icon": "",
            "layout": "IPY_MODEL_d4b03491384b4b1d854c05a9aa1e427c",
            "style": "IPY_MODEL_9f5c2c18d8b046299f1a59cb5f20e93f",
            "tooltip": ""
          }
        },
        "d4b03491384b4b1d854c05a9aa1e427c": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": "30px",
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": "400px"
          }
        },
        "9f5c2c18d8b046299f1a59cb5f20e93f": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ButtonStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ButtonStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "button_color": null,
            "font_weight": ""
          }
        },
        "77b44d6bb79d42229f01c2078cb8a58b": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_ac513893c04945a59f3338269b07c62d",
            "placeholder": "",
            "style": "IPY_MODEL_5d2a9091f7c24f4b95bdd3c2c2cf0c70",
            "value": "<p><font color=red>Note: At any time you can click the run button <img src='https://github.com/westlake-repl/SaProtHub/blob/dev/Figure/run_button.png?raw=true' height='25px' width='25px' align='center'> again to restart.</font></p>"
          }
        },
        "ac513893c04945a59f3338269b07c62d": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "5d2a9091f7c24f4b95bdd3c2c2cf0c70": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "a9723407cf1f40ad9f9ddc6472cd5725": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_569e239ab6bb40b591d356977d8c4f4b",
            "placeholder": "",
            "style": "IPY_MODEL_1093f8bd744746b2a06238e85741860d",
            "value": "<h2>Protein sequence design</h2>"
          }
        },
        "569e239ab6bb40b591d356977d8c4f4b": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "1093f8bd744746b2a06238e85741860d": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "175b30b1575440b3b59099e84d2e0f13": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_9999c853c5384d24a80347132be628a3",
            "placeholder": "",
            "style": "IPY_MODEL_ffb3964e5dcd4c5dac74bd6e26f55512",
            "value": "<h3>Please upload the structure backbone:</h3>"
          }
        },
        "9999c853c5384d24a80347132be628a3": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "ffb3964e5dcd4c5dac74bd6e26f55512": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "2c1676e71126401fb39d189d9f8ab49e": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_dc38bc3e6cc64324a147ed689ce71cd7",
            "placeholder": "",
            "style": "IPY_MODEL_d99808b256714956bf9d06c177d54ca8",
            "value": "<p>Chain (to be extracted from the structure):</p>"
          }
        },
        "dc38bc3e6cc64324a147ed689ce71cd7": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "d99808b256714956bf9d06c177d54ca8": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "f084fff8fb7b4b5898a2d1634b3a93a9": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "TextModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "TextModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "TextView",
            "continuous_update": true,
            "description": "",
            "description_tooltip": null,
            "disabled": false,
            "layout": "IPY_MODEL_b6752eaaae2941888af318478d2c631d",
            "placeholder": "Enter the name of chain here",
            "style": "IPY_MODEL_d634a2c416694a58b7e80eba8d7bc622",
            "value": "A"
          }
        },
        "b6752eaaae2941888af318478d2c631d": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": "30px",
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": "500px"
          }
        },
        "d634a2c416694a58b7e80eba8d7bc622": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "479015be7f2348f383ee6b89e9d47235": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_47b9e1fd9e91497296fe4ffb9cc4eff2",
            "placeholder": "",
            "style": "IPY_MODEL_20465a9b162b475b8bdae2c41b2afd00",
            "value": "<p>Upload the protein structure:</p>"
          }
        },
        "47b9e1fd9e91497296fe4ffb9cc4eff2": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "20465a9b162b475b8bdae2c41b2afd00": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "df1ef4d191984d08aa998bce741c2b06": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FileUploadModel",
          "model_module_version": "1.5.0",
          "state": {
            "_counter": 1,
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FileUploadModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "FileUploadView",
            "accept": "",
            "button_style": "",
            "data": [
              null
            ],
            "description": "Upload",
            "description_tooltip": null,
            "disabled": false,
            "error": "",
            "icon": "upload",
            "layout": "IPY_MODEL_ae7ec3398a7a409c811deb7cd651e02e",
            "metadata": [
              {
                "name": "LCC.pdb",
                "type": "",
                "size": 217323,
                "lastModified": 1727745238304
              }
            ],
            "multiple": false,
            "style": "IPY_MODEL_0ee820d318304ea0985ac9e4c685a934"
          }
        },
        "ae7ec3398a7a409c811deb7cd651e02e": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "0ee820d318304ea0985ac9e4c685a934": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ButtonStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ButtonStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "button_color": null,
            "font_weight": ""
          }
        },
        "c57b814c8c5f4bcb853f1d0acd08fb76": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_6715e54762c04feb848b8a9e856e5f96",
            "placeholder": "",
            "style": "IPY_MODEL_be6a0ff96fad472297de0c2b74dd3136",
            "value": "<p><strong>3Di sequence:</strong></p>\n<p>dfplddadlddlvqlldqadfdwdkdwfcqvvddlfrtfmkiagppdqeaaeeeeeeeaqpdalllqvvvnrrqrsrqytytythgpdshdalcsllrssvrvlvcqqppddpvsnsrhpsqfyayeyaarrlssqlanqlvdqsyqeyeyecyydqdlagehqhayeyehecpepvnhcvrhvvsnqvrydlnhwykyfyfdngyssqsrdrdsvvslvvnlscccgrrvdcssvvpvppdddprtpdmdtscpspd</p>\n<p><strong>Backbone visualization of LCC.pdb (258 amino acids):</strong></p>"
          }
        },
        "6715e54762c04feb848b8a9e856e5f96": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "be6a0ff96fad472297de0c2b74dd3136": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "6947b47837da4ccba98baa1385f4d3d7": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DropdownModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DropdownModel",
            "_options_labels": [
              "argmax",
              "muiltinomial"
            ],
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "DropdownView",
            "description": "Sampling methodology:",
            "description_tooltip": null,
            "disabled": false,
            "index": 0,
            "layout": "IPY_MODEL_42cf804e522b4cbfb5d08031d98221f9",
            "style": "IPY_MODEL_70397e15a0154fbdb8b6099adeb7f703"
          }
        },
        "42cf804e522b4cbfb5d08031d98221f9": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": "30px",
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": "500px"
          }
        },
        "70397e15a0154fbdb8b6099adeb7f703": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": "initial"
          }
        },
        "bcee82164a8b4a07a197ac405c3eac1f": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "BoundedIntTextModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "BoundedIntTextModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "IntTextView",
            "continuous_update": false,
            "description": "Sampling num:",
            "description_tooltip": null,
            "disabled": false,
            "layout": "IPY_MODEL_60ae6a0b0b384f84992ed3015a88a711",
            "max": 100,
            "min": 1,
            "step": 1,
            "style": "IPY_MODEL_b23e6852167f40d681fe2978573541ec",
            "value": 1
          }
        },
        "60ae6a0b0b384f84992ed3015a88a711": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": "none",
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": "30px",
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": "500px"
          }
        },
        "b23e6852167f40d681fe2978573541ec": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": "initial"
          }
        },
        "191833433bae4921b9e70bb8c07ef71c": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "TextModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "TextModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "TextView",
            "continuous_update": true,
            "description": "Save name:",
            "description_tooltip": null,
            "disabled": false,
            "layout": "IPY_MODEL_8a45ad0aefa74b269a1c219a85fdf815",
            "placeholder": "",
            "style": "IPY_MODEL_92226c261557406fb560cf70e9f4465f",
            "value": "predicted_seq"
          }
        },
        "8a45ad0aefa74b269a1c219a85fdf815": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": "30px",
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": "500px"
          }
        },
        "92226c261557406fb560cf70e9f4465f": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": "initial"
          }
        },
        "7010fbdcef0b4981a83185a5cdb777a2": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_5be7f153467c43a4b3c9f04cdd46abcd",
            "placeholder": "",
            "style": "IPY_MODEL_27bca44291ae48498199a4bec2736665",
            "value": "<h3>---------------------------------------------------------------------------</h3>"
          }
        },
        "5be7f153467c43a4b3c9f04cdd46abcd": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "27bca44291ae48498199a4bec2736665": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "2d2fb545e57048d1a8b26d16167531ef": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ButtonModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ButtonModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ButtonView",
            "button_style": "info",
            "description": "Predict protein sequence",
            "disabled": false,
            "icon": "",
            "layout": "IPY_MODEL_18a9e1774f3147a4a8628625278cebc1",
            "style": "IPY_MODEL_c31819dac666410c90f97b98deeed15f",
            "tooltip": ""
          }
        },
        "18a9e1774f3147a4a8628625278cebc1": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": "30px",
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": "500px"
          }
        },
        "c31819dac666410c90f97b98deeed15f": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ButtonStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ButtonStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "button_color": null,
            "font_weight": ""
          }
        },
        "5a8c37150ef54eaabdc6ba0c797907d6": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_87fb0778cd97466a95bbda0ac26003bb",
            "placeholder": "",
            "style": "IPY_MODEL_ad8f2a36a7e14ecba5b7c670f7e2c3e7",
            "value": "<html>\n  <head>\n  <meta name=\"viewport\" content=\"width=device-width, initial-scale=1\">\n  </head>\n  <body>\n  <a download=\"predicted_seq.fasta\" href=\"data:text/csv;base64,PnByZWRpY3RlZF9zZXFfMApBUFBHWVBQUEFMRFBBVExBS1RHUFlBVlNUUVNWUEFHQVZTR0ZTR0dUVllZUEFHVFRHS0ZQVlZWTFNQR0ZUR1NRU05WQVdMR0tITEFTSEdGVlZMQUlOVFBTU0ZEU1BTVFJBQVFMTEFBTERZTFRUR0FQQUVMQUFSSURUU1JWR1ZBR0hTTUdHR0dUTElBQUFRRFBSSUtBVklQTEFQV1FHTlRTWUtGVEFQVExJSUdBR05EVFZBUEFTUUhBRVBZWU5HTFNQU1ZES0VZVkVMS0dHVEhGVFBUUVBOUERWUkxMU1RBV01SUllMREdERUFBQUFZTERQVFRQQUdWU1JZS0ROVERMQVAKCg==\" download>\n  <button class=\"p-Widget jupyter-widgets jupyter-button widget-button mod-warning\">Download File</button>\n  </a>\n  </body>\n  </html>\n  "
          }
        },
        "87fb0778cd97466a95bbda0ac26003bb": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "ad8f2a36a7e14ecba5b7c670f7e2c3e7": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        }
      }
    }
  },
  "nbformat": 4,
  "nbformat_minor": 5
}